{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Using TensorFlow backend.\n"
     ]
    }
   ],
   "source": [
    "%matplotlib inline\n",
    "import matplotlib.pyplot as plt\n",
    "plt.rcParams['figure.figsize'] = (10.0, 7.0)\n",
    "\n",
    "import collections\n",
    "import nltk\n",
    "from collections import defaultdict\n",
    "from nltk import bigrams\n",
    "from nltk import trigrams\n",
    "import pandas as pd\n",
    "from Bio import SeqIO\n",
    "from sklearn.preprocessing import normalize\n",
    "import numpy as np\n",
    "from sklearn.decomposition import TruncatedSVD\n",
    "from sklearn.manifold import TSNE\n",
    "import gensim, logging\n",
    "logging.basicConfig(format='%(asctime)s : %(levelname)s : %(message)s', level=logging.INFO)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "160 14 105\n",
      "14: 1 18: 1 19: 4 20: 1 21: 4 22: 2 24: 1 30: 1 38: 2 40: 2 41: 1 42: 1 43: 2 44: 1 46: 4 47: 7 48: 10 49: 4 50: 2 51: 8 52: 4 53: 3 55: 5 56: 8 57: 9 58: 4 59: 3 60: 2 61: 3 62: 3 63: 6 64: 6 65: 2 66: 2 67: 4 68: 6 69: 2 70: 1 71: 5 72: 3 73: 1 74: 2 75: 1 76: 1 77: 1 78: 1 80: 1 81: 1 83: 1 86: 1 88: 1 91: 2 92: 1 95: 1 99: 1 102: 1 103: 1 105: 1\n"
     ]
    }
   ],
   "source": [
    "\"\"\"\n",
    "Length distribution of class 1 bacteriocins\n",
    "\"\"\"\n",
    "\n",
    "sizes = [len(rec) for rec in SeqIO.parse(\"class_1_protein.fa\", \"fasta\")]\n",
    "print len(sizes), min(sizes), max(sizes)\n",
    "\n",
    "counter_class_1 = collections.Counter(sizes)\n",
    "#print counter\n",
    "\n",
    "keylist = counter_class_1.keys()\n",
    "keylist.sort()\n",
    "for key in keylist:\n",
    "    print \"%s: %s\" % (key, counter_class_1[key]),"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "229 7 255\n",
      "7: 1 9: 2 10: 1 12: 4 13: 1 15: 1 18: 1 19: 1 20: 2 21: 1 22: 1 23: 1 24: 3 25: 2 26: 2 28: 1 29: 1 30: 2 31: 2 34: 1 35: 2 36: 2 37: 2 38: 2 39: 3 40: 1 41: 3 42: 2 43: 8 44: 10 45: 3 46: 2 47: 3 48: 1 49: 1 50: 4 51: 3 52: 2 53: 4 54: 2 55: 7 56: 4 57: 4 58: 4 59: 4 60: 3 61: 9 62: 9 63: 2 64: 5 65: 5 66: 3 67: 7 68: 2 69: 7 70: 2 71: 5 72: 3 74: 5 75: 4 76: 7 77: 2 78: 1 79: 1 80: 1 81: 2 84: 4 85: 2 87: 1 89: 2 90: 3 91: 2 96: 2 97: 2 101: 1 102: 2 103: 1 110: 1 120: 2 128: 1 131: 1 207: 1 225: 1 255: 1\n"
     ]
    }
   ],
   "source": [
    "\"\"\"\n",
    "Length distribution of class 2 bacteriocins\n",
    "\"\"\"\n",
    "\n",
    "sizes = [len(rec) for rec in SeqIO.parse(\"class_2_protein.fa\", \"fasta\")]\n",
    "print len(sizes), min(sizes), max(sizes)\n",
    "\n",
    "counter_class_2 = collections.Counter(sizes)\n",
    "#print counter\n",
    "\n",
    "keylist = counter_class_2.keys()\n",
    "keylist.sort()\n",
    "for key in keylist:\n",
    "    print \"%s: %s\" % (key, counter_class_2[key]),"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "93\n",
      "389\n",
      "[7, 9, 10, 12, 13, 14, 15, 18, 19, 20, 21, 22, 23, 24, 25, 26, 28, 29, 30, 31, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 95, 96, 97, 99, 101, 102, 103, 105, 110, 120, 128, 131, 207, 225, 255]\n"
     ]
    }
   ],
   "source": [
    "print len(merged_class)\n",
    "6 in merged_class\n",
    "count = 0\n",
    "for key in merged_class:\n",
    "    count += merged_class[key]\n",
    "    \n",
    "print count\n",
    "print keylist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[36, 7, 9, 10, 12, 13, 14, 15, 18, 19, 20, 21, 22, 24, 25, 26]\n"
     ]
    }
   ],
   "source": [
    "extra_in_original_neg = list(set(keylist) - set(re_keylist))\n",
    "print extra_in_original_neg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "383 12 255\n",
      "12: 4 13: 1 14: 1 15: 1 18: 2 19: 4 20: 3 21: 5 22: 3 23: 1 24: 3 25: 2 26: 2 28: 1 29: 3 30: 1 31: 2 34: 1 35: 2 36: 2 37: 3 38: 3 39: 3 40: 3 41: 7 42: 6 43: 7 44: 4 45: 3 46: 12 47: 8 48: 7 49: 5 50: 8 51: 7 52: 7 53: 9 54: 9 55: 3 56: 6 57: 18 58: 9 59: 7 60: 10 61: 12 62: 12 63: 8 64: 11 65: 7 66: 5 67: 11 68: 8 69: 9 70: 3 71: 10 72: 6 73: 1 74: 7 75: 5 76: 8 77: 3 78: 2 79: 1 80: 2 81: 3 83: 1 84: 4 85: 2 86: 1 87: 1 88: 1 89: 2 90: 3 91: 4 92: 1 95: 1 96: 2 97: 2 99: 1 101: 1 102: 3 103: 2 105: 1 110: 1 120: 2 128: 1 131: 1 207: 1 225: 1 255: 1\n"
     ]
    }
   ],
   "source": [
    "\"\"\"\n",
    "Length distribution of redundant cdhit negative bacteriocin dataset\n",
    "\"\"\"\n",
    "\n",
    "sizes = [len(rec) for rec in SeqIO.parse(\"final_negative_bacteriocin_set_cd-hit_0.7.fa\", \"fasta\")]\n",
    "print len(sizes), min(sizes), max(sizes)\n",
    "\n",
    "counter_redundant_f = collections.Counter(sizes)\n",
    "#print counter\n",
    "\n",
    "re_keylist_f = counter_redundant_f.keys()\n",
    "re_keylist_f.sort()\n",
    "for key in re_keylist_f:\n",
    "    print \"%s: %s\" % (key, counter_redundant_f[key]),"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n"
     ]
    }
   ],
   "source": [
    "keys_done = defaultdict(int)\n",
    "keys_done[23]\n",
    "print merged_class[23]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 307,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done\n"
     ]
    }
   ],
   "source": [
    "\"\"\"\n",
    "Making a redundant negative bacteriocin dataset from swissprot reviewed bacteria \n",
    "portion\n",
    "\n",
    "Just take all sequences that have a length which is also a bacteriocin length\n",
    "\"\"\"\n",
    "\n",
    "out_handle = open('redundant_negative_bacteriocin_set', 'w')\n",
    "for record in SeqIO.parse(\"uniprot_sprot_bacteria.dat\", \"swiss\"):\n",
    "    if 'keywords' in record.annotations and 'comment' in record.annotations: \n",
    "        if 'Bacteriocin' not in record.annotations['keywords'] and \\\n",
    "            'Plasmid' not in record.annotations['keywords'] and \\\n",
    "            'Antimicrobial' not in record.annotations['keywords'] and \\\n",
    "            'Antibiotic' not in record.annotations['keywords'] and \\\n",
    "            len(record.annotations['comment']) >= 125 and \\\n",
    "            len(record.seq) in keylist:\n",
    "                out_handle.write('>%s|%s|%s|%s\\n%s\\n' % (record.name, record.id, record.description, \n",
    "                                                     str(len(record.seq)),\n",
    "                                                     record.seq))\n",
    "\n",
    "out_handle.close()\n",
    "print \"Done\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FUNCTION: Catalyzes the oxidation of the 12-alpha-hydroxyl group of bile acids, both in their free and conjugated form. Also acts on bile alcohols.\n",
      "CATALYTIC ACTIVITY: 3-alpha,7-alpha,12-alpha-trihydroxy-5-beta- cholanate + NADP(+) = 3-alpha,7-alpha-dihydroxy-12-oxo-5-beta- cholanate + NADPH.\n",
      "SUBUNIT: Homotetramer.\n",
      "MISCELLANEOUS: The thermostability of the enzyme is greatly increased due to NADP binding.\n",
      "['Bile acid catabolism', 'Direct protein sequencing', 'Lipid degradation', 'Lipid metabolism', 'NADP', 'Oxidoreductase', 'Steroid metabolism']\n",
      "MIFDGKVAIITGGGKAKSIGYGIAVAYAK\n",
      "29\n",
      "1\n",
      "\n",
      "SUBCELLULAR LOCATION: Cell outer membrane {ECO:0000305}; Lipid- anchor {ECO:0000305}.\n",
      "SIMILARITY: Belongs to the rickettsiale 17 kDa surface antigen family. {ECO:0000305}.\n",
      "['Cell outer membrane', 'Lipoprotein', 'Membrane']\n",
      "GSQFGKGKGQLIGVGAGALLGAILGNQIGAGMDEQDRRLAELTSQRALETTPSGTSIEWRNPDNGNYGYVTPSKTYKNST\n",
      "80\n",
      "1\n",
      "\n",
      "INDUCTION: Under acid-stress, this protein is expressed at a higher level in wild-type B.cereus than in the acid-sensitive mutant strain NB1. {ECO:0000269|PubMed:11872115}.\n",
      "['Direct protein sequencing']\n",
      "MKLIAKTRILNTLVFSAAGK\n",
      "20\n",
      "1\n",
      "\n",
      "FUNCTION: Converts 2-aminomuconate to 4-oxalocrotonate, an intermediate step in the biodegradation of nitrobenzene.\n",
      "CATALYTIC ACTIVITY: 2-aminomuconate + H(2)O = (3E)-2-oxohex-3- enedioate + NH(3).\n",
      "BIOPHYSICOCHEMICAL PROPERTIES: Kinetic parameters:   KM=67 uM for 2-aminomuconate;   Vmax=125 umol/min/mg enzyme with glucose as substrate; pH dependence:   Optimum pH is 6.6.;\n",
      "PATHWAY: Xenobiotic degradation; nitrobenzene degradation.\n",
      "SUBUNIT: Homohexamer.\n",
      "['Direct protein sequencing', 'Hydrolase']\n",
      "STLSSNDAKVVDGKATPLGSFPHVKRAGDFLYVSGTSSRRPDNTIAGAELDSTS\n",
      "54\n",
      "1\n",
      "\n",
      "FUNCTION: Involved in fermentation of amino acids (Stickland reaction) such as leucine, isoleucine, valine and phenylalanine.\n",
      "CATALYTIC ACTIVITY: Butanoate + NAD(+) = but-2-enoate + NADH.\n",
      "COFACTOR: Name=iron-sulfur cluster; Xref=ChEBI:CHEBI:30408;\n",
      "COFACTOR: Name=FAD; Xref=ChEBI:CHEBI:57692; Name=FMN; Xref=ChEBI:CHEBI:58210;\n",
      "SUBUNIT: Dodecamer; tetramer of trimers.\n",
      "['Direct protein sequencing', 'Flavoprotein', 'Iron', 'Iron-sulfur', 'Metal-binding', 'NAD', 'Oxidoreductase']\n",
      "MKNKSLFEVIKIGKVEVXXKIXMAVMGAFG\n",
      "30\n",
      "1\n",
      "\n",
      "FUNCTION: Putative antitoxin component of a possible type II toxin-antitoxin (TA) module. May neutralize the effect of cognate toxin Rv3098A/RVBD_3098A. {ECO:0000305}.\n",
      "SUBUNIT: Forms a complex with cognate toxin Rv3098A/RVBD_3098A. {ECO:0000250|UniProtKB:O53451}.\n",
      "['Complete proteome', 'Reference proteome']\n",
      "MTATIGFRPTEKDEQIINAAMRSGERKSDVIRRALQLLEREVWIKQARTDAERLRDEDVSTEPDAW\n",
      "66\n",
      "1\n",
      "\n",
      "FUNCTION: Active against HIV-1 virus in vitro.\n",
      "CAUTION: The isopeptide linked residue 9 is shown as Asn rather than Asp as mentioned in PubMed:8286361, because it is not known whether Asp or Asn is encoded and the isopeptide bonds are almost always formed between the amides Asn or Gln and N6-lysine or alpha amino groups, with the liberation of an ammonia that makes the reaction essentially irreversible. {ECO:0000305}.\n",
      "['3D-structure', 'Direct protein sequencing', 'Disulfide bond', 'Isopeptide bond']\n",
      "CLGIGSCNNFAGCGYAVVCFW\n",
      "21\n",
      "1\n",
      "\n",
      "FUNCTION: Catalyzes both 1,3- and 1,5-keto-enol tautomerization of the diacid 2-hydroxymuconate (2-hydroxy-2,4-hexadienedioate) to produce 2-oxo-4-hexenedioate. This reaction is highly stereoselective and produces a mixture of stereoisomers, where the (3S)-isomer of 2-oxo-4-hexenedioate predominates. Also catalyzes the tautomerization of 2-hydroxymuconate to 2-oxo-3-hexenedioate, however this reaction is slower and occurs after the tautomerization of 2-hydroxymuconate to 2-oxo-4-hexenedioate. Using 2-hydroxy-2,4-pentadienoate, phenylenolpyruvate, (p- hydroxyphenyl)-enolpyruvate and 2-hydroxy-2,4-heptadiene-1,7- dioate, YwhB is a highly efficient 1,3-keto-enol tautomerase, but clearly not a 1,5-keto-enol tautomerase. Tautomerization of the two monoacids 2-hydroxy-2,4-pentadienoate and phenylenolpyruvate produces a mixture of stereoisomers, where the (3R)-isomers predominate. {ECO:0000269|PubMed:17902707}.\n",
      "CATALYTIC ACTIVITY: (2Z,4E)-2-hydroxyhexa-2,4-dienedioate = (3E)- 2-oxohex-3-enedioate. {ECO:0000269|PubMed:17902707}.\n",
      "BIOPHYSICOCHEMICAL PROPERTIES: Kinetic parameters:   KM=76 uM for 2-hydroxymuconate (with 2-oxo-4-hexenedioate as   product) {ECO:0000269|PubMed:17902707};   KM=160 uM for (p-hydroxyphenyl)-enolpyruvate   {ECO:0000269|PubMed:17902707};   KM=243 uM for 2-hydroxy-2,4-pentadienoate   {ECO:0000269|PubMed:17902707};   KM=300 uM for phenylenolpyruvate {ECO:0000269|PubMed:17902707};   KM=330 uM for 2-hydroxy-2,4-heptadiene-1,7-dioate   {ECO:0000269|PubMed:17902707};   KM=940 uM for 2-hydroxymuconate (with 2-oxo-3-hexenedioate as   product) {ECO:0000269|PubMed:17902707};   Note=Kcat is 3 (sec-1) for 2-hydroxy-2,4-heptadiene-1,7-dioate,   4.1 (sec-1) for (p-hydroxyphenyl)-enolpyruvate, 10 (sec-1) for   2-hydroxymuconate, 26 (sec-1) for 2-hydroxymuconate, 112 (sec-1)   for 2-hydroxy-2,4-pentadienoate, 125 (sec-1) for   phenylenolpyruvate.;\n",
      "SUBUNIT: Homohexamer. {ECO:0000305|Ref.4}.\n",
      "MISCELLANEOUS: The mono- and diacids apparently bind in different orientations in the active site of YwhB, but the highly stereoselective nature of the YwhB reaction using a diacid suggests that the biological substrate for YwhB may be a diacid. {ECO:0000305|PubMed:17902707}.\n",
      "SIMILARITY: Belongs to the 4-oxalocrotonate tautomerase family. {ECO:0000305}.\n",
      "['3D-structure', 'Complete proteome', 'Isomerase', 'Reference proteome']\n",
      "MPYVTVKMLEGRTDEQKRNLVEKVTEAVKETTGASEEKIVVFIEEMRKDHYAVAGKRLSDME\n",
      "62\n",
      "1\n",
      "\n",
      "FUNCTION: Catalyzes the ketonization of 2-hydroxymuconate stereoselectively to yield 2-oxo-3-hexenedioate.\n",
      "CATALYTIC ACTIVITY: (2Z,4E)-2-hydroxyhexa-2,4-dienedioate = (3E)- 2-oxohex-3-enedioate.\n",
      "PATHWAY: Aromatic compound metabolism; salicylate degradation.\n",
      "SUBUNIT: Homohexamer. {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the 4-oxalocrotonate tautomerase family. {ECO:0000305}.\n",
      "['Aromatic hydrocarbons catabolism', 'Isomerase']\n",
      "MPFAQIYMLEGRSEEQKKAVIEKVTRALVEAVGAPSANVRVWIHDVPKENWGIAGVSAKELGR\n",
      "63\n",
      "1\n",
      "\n",
      "MISCELLANEOUS: Under acid-stress, this protein is expressed at a higher level in wild-type B.cereus than in the acid-sensitive mutant strain NB1.\n",
      "['Direct protein sequencing']\n",
      "AHPPEXLSIA\n",
      "10\n",
      "1\n",
      "\n",
      "FUNCTION: Aminoacyl carrier protein. Can be charged with L-glycine via the formation of a thioester bond between the amino acid and the 4'-phosphopantetheinyl prosthetic group, catalyzed by the bll0957 ligase. {ECO:0000269|PubMed:20663952}.\n",
      "PTM: 4'-phosphopantetheine is transferred from CoA to a specific serine of the apo-form of this carrier protein. {ECO:0000269|PubMed:20663952}.\n",
      "['3D-structure', 'Complete proteome', 'Phosphopantetheine', 'Phosphoprotein', 'Reference proteome']\n",
      "MQAFNTDVRNRIIKLVKGILEQNALAADVTPQAKLVDVGLTSMDMVNLMLGVEAEFDFTIPQSEITPENFQSVETLERMVMTQLQPATAA\n",
      "90\n",
      "1\n",
      "\n",
      "FUNCTION: Aminoacyl carrier protein. Can be charged with L-glycine via the formation of a thioester bond between the amino acid and the 4'-phosphopantetheinyl prosthetic group, catalyzed by the bll6282 ligase. {ECO:0000269|PubMed:20663952}.\n",
      "PTM: 4'-phosphopantetheine is transferred from CoA to a specific serine of the apo-form of this carrier protein.\n",
      "SEQUENCE CAUTION: Sequence=BAC51549.1; Type=Erroneous initiation; Note=Translation N-terminally shortened.; Evidence={ECO:0000305};\n",
      "['Complete proteome', 'Phosphopantetheine', 'Phosphoprotein', 'Reference proteome']\n",
      "MHNTAINVQNRVLSVVRSVLQQNAISADVHPESRLVDIGLSSMGMVELMLKVEAEFDLILPQFEITPENFRSVKAMERMILNQLGSGSG\n",
      "89\n",
      "1\n",
      "\n",
      "FUNCTION: Aminoacyl carrier protein. Can be charged with L- alanine, L-glycine or L-serine, via the formation of a thioester bond between the amino acid and the 4'-phosphopantetheinyl prosthetic group, catalyzed by the Atu2573 ligase. {ECO:0000269|PubMed:20663952}.\n",
      "PTM: 4'-phosphopantetheine is transferred from CoA to a specific serine of the apo-form of this carrier protein.\n",
      "['3D-structure', 'Complete proteome', 'Phosphopantetheine', 'Phosphoprotein', 'Reference proteome']\n",
      "MNATIREILAKFGQLPTPVDTIADEADLYAAGLSSFASVQLMLGIEEAFDIEFPDNLLNRKSFASIKAIEDTVKLILDGKEAA\n",
      "83\n",
      "1\n",
      "\n",
      "SUBCELLULAR LOCATION: Cell membrane {ECO:0000255|HAMAP- Rule:MF_01546}; Multi-pass membrane protein {ECO:0000255|HAMAP- Rule:MF_01546}.\n",
      "SIMILARITY: Belongs to the AaeX family. {ECO:0000255|HAMAP- Rule:MF_01546}.\n",
      "['Cell membrane', 'Complete proteome', 'Membrane', 'Reference proteome', 'Transmembrane', 'Transmembrane helix']\n",
      "MSLFPVIVVFGLSFPPIFFELLLSLAIFWLVRRMLVPTGIYDFVWHPALFNTALYCCLFYLISRLFV\n",
      "67\n",
      "1\n",
      "\n",
      "SUBCELLULAR LOCATION: Cell membrane {ECO:0000305}; Multi-pass membrane protein {ECO:0000305}.\n",
      "SIMILARITY: Belongs to the AaeX family. {ECO:0000305}.\n",
      "['Cell membrane', 'Complete proteome', 'Membrane', 'Transmembrane', 'Transmembrane helix']\n",
      "MHSLPVMVLFGLSFPPIFFVLLVSLTLFFICIRLLHSSGIYEWVWHPALFNTSLFCCLFYLLFHLGV\n",
      "67\n",
      "2\n",
      "\n",
      "FUNCTION: Ambiactive repressor and activator of the transcription of genes expressed during the transition state between vegetative growth and the onset of stationary phase and sporulation. It controls the expression of genes spovG and tycA. AbrB binds to the tycA promoter region at two A- and T-rich sites, it may be the sole repressor of tycA transcription. {ECO:0000269|PubMed:2504584}.\n",
      "SIMILARITY: To B.subtilis Abh and SpoVT. {ECO:0000305}.\n",
      "CAUTION: It is uncertain whether Met-1 or Met-3 is the initiator. {ECO:0000305}.\n",
      "SEQUENCE CAUTION: Sequence=AAA22195.1; Type=Erroneous initiation; Note=Translation N-terminally extended.; Evidence={ECO:0000305}; Sequence=CAA43955.1; Type=Erroneous initiation; Note=Translation N-terminally extended.; Evidence={ECO:0000305};\n",
      "['3D-structure', 'Activator', 'Complete proteome', 'DNA-binding', 'Reference proteome', 'Repressor', 'Sporulation', 'Transcription', 'Transcription regulation']\n",
      "MFMKSTGIVRKVDELGRVVIPIELRRTLGIAEKDALEIYVDDEKIILKKYKPNMTCQVTGEVSDDNLKLAGGKLVLSKEGAEQIISEIQNQLQNLK\n",
      "96\n",
      "1\n",
      "\n",
      "FUNCTION: Component of the acetyl coenzyme A carboxylase (ACC) complex. First, biotin carboxylase catalyzes the carboxylation of biotin on its carrier protein (BCCP) and then the CO(2) group is transferred by the carboxyltransferase to acetyl-CoA to form malonyl-CoA. {ECO:0000255|HAMAP-Rule:MF_00823}.\n",
      "CATALYTIC ACTIVITY: ATP + acetyl-CoA + HCO(3)(-) = ADP + phosphate + malonyl-CoA. {ECO:0000255|HAMAP-Rule:MF_00823}.\n",
      "PATHWAY: Lipid metabolism; malonyl-CoA biosynthesis; malonyl-CoA from acetyl-CoA: step 1/1. {ECO:0000255|HAMAP-Rule:MF_00823}.\n",
      "SUBUNIT: Acetyl-CoA carboxylase is a heterohexamer composed of biotin carboxyl carrier protein (AccB), biotin carboxylase (AccC) and two subunits each of ACCase subunit alpha (AccA) and ACCase subunit beta (AccD). {ECO:0000255|HAMAP-Rule:MF_00823}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000255|HAMAP-Rule:MF_00823}.\n",
      "SIMILARITY: Belongs to the AccA family. {ECO:0000255|HAMAP- Rule:MF_00823}.\n",
      "['ATP-binding', 'Complete proteome', 'Cytoplasm', 'Fatty acid biosynthesis', 'Fatty acid metabolism', 'Ligase', 'Lipid biosynthesis', 'Lipid metabolism', 'Nucleotide-binding']\n",
      "MNIAKIVREAREQSRLTTLDFATGIFDEFIQLHGDRSFRDDGAVVGGIGWLGDQAVTVVGIQKGKSLQDNLKRNFGQPHPEGYRKALRLMKQAEKFGRPVVTFINTAGAYPGVGAEERGQGEAIARNLMEMSDLKVPIIAIIIGEGGSGGALALAVADRVWMLENSIYAILSPEGFASILWKDGTRAMEAAELMKITSHELLEMDVVDKVISEVGLSSKELIKSVKKELQTELARLSQKPLEELLEERYQRFRKY\n",
      "255\n",
      "1\n",
      "\n",
      "FUNCTION: Involved in the metabolic adaptation in response to environmental changes. Catalyzes the reversible formation of succinate and glyoxylate from isocitrate, a key step of the glyoxylate cycle, which operates as an anaplerotic route for replenishing the tricarboxylic acid cycle during growth on fatty acid substrates. {ECO:0000269|PubMed:1938889}.\n",
      "CATALYTIC ACTIVITY: Isocitrate = succinate + glyoxylate. {ECO:0000269|PubMed:1938889}.\n",
      "COFACTOR: Name=Mg(2+); Xref=ChEBI:CHEBI:18420;   Evidence={ECO:0000269|PubMed:1938889}; Note=Divalent metal cations. Mn(2+), Fe(2+) or Co(2+) can be used. {ECO:0000269|PubMed:1938889};\n",
      "ENZYME REGULATION: Inhibited by hydroxymalonate, oxalate and itaconate. {ECO:0000269|PubMed:1938889}.\n",
      "BIOPHYSICOCHEMICAL PROPERTIES: Kinetic parameters:   KM=40 uM for threo-D-isocitrate {ECO:0000269|PubMed:1938889}; pH dependence:   Optimum pH is 7.5. {ECO:0000269|PubMed:1938889};\n",
      "PATHWAY: Carbohydrate metabolism; glyoxylate cycle; (S)-malate from isocitrate: step 1/2. {ECO:0000305|PubMed:1938889}.\n",
      "SUBUNIT: Homotetramer. {ECO:0000269|PubMed:1938889}.\n",
      "SIMILARITY: Belongs to the isocitrate lyase/PEP mutase superfamily. Isocitrate lyase family. {ECO:0000305}.\n",
      "['Cobalt', 'Direct protein sequencing', 'Glyoxylate bypass', 'Iron', 'Lyase', 'Magnesium', 'Manganese', 'Tricarboxylic acid cycle']\n",
      "MTYQTAIDAVRELKA\n",
      "15\n",
      "1\n",
      "\n",
      "FUNCTION: Increases short-circuit current in rabbit ileal tissue mounted in Ussing chambers, by increasing the potential difference. Cultures of V.cholerae containing the cloned ace gene cause fluid secretion in ligated rabbit ileal loops.\n",
      "SUBCELLULAR LOCATION: Secreted. Host cell membrane {ECO:0000305}.\n",
      "CAUTION: It is uncertain whether Met-1, Met-3 or Met-4 is the initiator. {ECO:0000305}.\n",
      "SEQUENCE CAUTION: Sequence=AAF94616.1; Type=Erroneous initiation; Evidence={ECO:0000305};\n",
      "['Complete proteome', 'Enterotoxin', 'Host cell membrane', 'Host membrane', 'Membrane', 'Reference proteome', 'Secreted', 'Toxin', 'Transmembrane', 'Transmembrane helix']\n",
      "MLMMDTLYDWLIDGFTWLVIKLGIMWIESKIFVIQFFWEMSQKVIDMFTIYPLIQQAIDMLPPQYSGFLFFLGLDQALAIVLQALMTRFALRALNL\n",
      "96\n",
      "2\n",
      "\n",
      "FUNCTION: Carrier of the growing fatty acid chain in fatty acid biosynthesis. {ECO:0000255|HAMAP-Rule:MF_01217}.\n",
      "PATHWAY: Lipid metabolism; fatty acid biosynthesis. {ECO:0000255|HAMAP-Rule:MF_01217}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000255|HAMAP-Rule:MF_01217}.\n",
      "PTM: 4'-phosphopantetheine is transferred from CoA to a specific serine of apo-ACP by AcpS. This modification is essential for activity because fatty acids are bound in thioester linkage to the sulfhydryl of the prosthetic group. {ECO:0000255|HAMAP- Rule:MF_01217}.\n",
      "SIMILARITY: Belongs to the acyl carrier protein (ACP) family. {ECO:0000255|HAMAP-Rule:MF_01217}.\n",
      "['3D-structure', 'Complete proteome', 'Cytoplasm', 'Fatty acid biosynthesis', 'Fatty acid metabolism', 'Lipid biosynthesis', 'Lipid metabolism', 'Phosphopantetheine', 'Phosphoprotein', 'Reference proteome']\n",
      "MSTIEERVKKIVAEQLGVKEEEVTNSASFVEDLGADSLDTVELVMALEEEFETEIPDEKAEKITTVQEAIDYIVAHQQ\n",
      "78\n",
      "1\n",
      "\n",
      "FUNCTION: Carrier of the growing fatty acid chain in fatty acid biosynthesis. {ECO:0000255|HAMAP-Rule:MF_01217}.\n",
      "PATHWAY: Lipid metabolism; fatty acid biosynthesis. {ECO:0000255|HAMAP-Rule:MF_01217}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000255|HAMAP-Rule:MF_01217}.\n",
      "PTM: 4'-phosphopantetheine is transferred from CoA to a specific serine of apo-ACP by AcpS. This modification is essential for activity because fatty acids are bound in thioester linkage to the sulfhydryl of the prosthetic group. {ECO:0000255|HAMAP- Rule:MF_01217}.\n",
      "SIMILARITY: Belongs to the acyl carrier protein (ACP) family. {ECO:0000255|HAMAP-Rule:MF_01217}.\n",
      "['Complete proteome', 'Cytoplasm', 'Fatty acid biosynthesis', 'Fatty acid metabolism', 'Lipid biosynthesis', 'Lipid metabolism', 'Phosphopantetheine', 'Phosphoprotein']\n",
      "MSTIEERVKKIIGEQLGVKQEEVTNNASFVEDLGADSLDTVELVMALEEEFDTEIPDEEAEKITTVQAAIDYINGHQA\n",
      "78\n",
      "2\n",
      "\n",
      "FUNCTION: Carrier of the growing fatty acid chain in fatty acid biosynthesis. {ECO:0000255|HAMAP-Rule:MF_01217}.\n",
      "PATHWAY: Lipid metabolism; fatty acid biosynthesis. {ECO:0000255|HAMAP-Rule:MF_01217}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000255|HAMAP-Rule:MF_01217}.\n",
      "PTM: 4'-phosphopantetheine is transferred from CoA to a specific serine of apo-ACP by AcpS. This modification is essential for activity because fatty acids are bound in thioester linkage to the sulfhydryl of the prosthetic group. {ECO:0000255|HAMAP- Rule:MF_01217}.\n",
      "SIMILARITY: Belongs to the acyl carrier protein (ACP) family. {ECO:0000255|HAMAP-Rule:MF_01217}.\n",
      "['Complete proteome', 'Cytoplasm', 'Fatty acid biosynthesis', 'Fatty acid metabolism', 'Lipid biosynthesis', 'Lipid metabolism', 'Phosphopantetheine', 'Phosphoprotein', 'Reference proteome']\n",
      "MDDIETRVRKLVAARFGVEECDIRLDSDFRNDFGAESLEVVELVMALEAEFGVEIADDDAERIETVRQAIDYLEEAVPT\n",
      "79\n",
      "1\n",
      "\n",
      "FUNCTION: Acyl carrier protein. {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "PTM: 4'-phosphopantetheine is transferred from CoA to a specific serine of the apo-ACP-like protein. {ECO:0000305}.\n",
      "['Complete proteome', 'Cytoplasm', 'Fatty acid biosynthesis', 'Fatty acid metabolism', 'Lipid biosynthesis', 'Lipid metabolism', 'Phosphopantetheine', 'Phosphoprotein', 'Reference proteome']\n",
      "MRELTSEIKKEIRDIVYDFFSEECEVDINELNDNTSVVDDLDGDSLMLIELVDILKKKYSLNIQLQSIGKYLLKNPAETLEKVVQTTYLLYEYENDITNVGEK\n",
      "103\n",
      "1\n",
      "\n",
      "FUNCTION: Acyl carrier protein involved in meromycolate extension.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "PTM: 4'-phosphopantetheine is transferred from CoA to a specific serine of apo-AcpM. {ECO:0000305}.\n",
      "SIMILARITY: Belongs to the acyl carrier protein (ACP) family. {ECO:0000305}.\n",
      "['Cytoplasm', 'Fatty acid biosynthesis', 'Fatty acid metabolism', 'Lipid biosynthesis', 'Lipid metabolism', 'Phosphopantetheine', 'Phosphoprotein']\n",
      "MAASQQEIIAGLAEIIEEVTGIEPSEVTPEKSFVDDLDIDSLSMVEIAVQTEDKYGVKIPDEDLAGLRTVGDVVAYIQKLEEENPEAAAALREKFAE\n",
      "97\n",
      "1\n",
      "\n",
      "FUNCTION: Transfers the 4'-phosphopantetheine moiety from coenzyme A to a Ser of acyl-carrier-protein. {ECO:0000255|HAMAP- Rule:MF_00101}.\n",
      "CATALYTIC ACTIVITY: CoA-(4'-phosphopantetheine) + apo-[acyl- carrier-protein] = adenosine 3',5'-bisphosphate + holo-[acyl- carrier-protein]. {ECO:0000255|HAMAP-Rule:MF_00101}.\n",
      "COFACTOR: Name=Mg(2+); Xref=ChEBI:CHEBI:18420;   Evidence={ECO:0000255|HAMAP-Rule:MF_00101};\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000255|HAMAP-Rule:MF_00101}.\n",
      "SIMILARITY: Belongs to the P-Pant transferase superfamily. AcpS family. {ECO:0000255|HAMAP-Rule:MF_00101}.\n",
      "['Complete proteome', 'Cytoplasm', 'Fatty acid biosynthesis', 'Fatty acid metabolism', 'Lipid biosynthesis', 'Lipid metabolism', 'Magnesium', 'Metal-binding', 'Reference proteome', 'Transferase']\n",
      "MIVGLGTDLVEIARMERLLARHGDRALTRLLHVQERAECPRAPDRAARFLARRFAAKEAAAKALGTGIANGIRFADLQVSHDDRGRPLLNLHGEARQHAQALGATGHHLSISDEQTHALAFVVLESSD\n",
      "128\n",
      "1\n",
      "\n",
      "FUNCTION: Involved in developmentally regulated synthesis of a compound biosynthetically related to polyketide antibiotics which is essential for spore color in Streptomyces coelicolor.\n",
      "PTM: 4'-phosphopantetheine is transferred from CoA to a specific serine of the apo-ACP-like protein. {ECO:0000305}.\n",
      "['Complete proteome', 'Phosphopantetheine', 'Phosphoprotein', 'Reference proteome']\n",
      "MTDQQLDYQVTVEELSALMKRTAGVHVDPVTLRQQADDGFDTFGLDSLGLLGIVAELEKRYGLGLPEQAERCKTPADFLALVNGALKTGV\n",
      "90\n",
      "2\n",
      "\n",
      "FUNCTION: Carrier of the growing fatty acid chain in fatty acid biosynthesis. Is involved in the transfer of long hydroxylated fatty acids to lipid A. Is acylated predominantly with 27- hydroxyoctacosanoic acid (By similarity). {ECO:0000250}.\n",
      "PATHWAY: Glycolipid biosynthesis; KDO(2)-lipid A biosynthesis.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "PTM: 4'-phosphopantetheine is transferred from CoA to a specific serine of apo-ACP by AcpS. This modification is essential for activity because fatty acids are bound in thioester linkage to the sulfhydryl of the prosthetic group (By similarity). {ECO:0000250}.\n",
      "SEQUENCE CAUTION: Sequence=ABC91255.1; Type=Erroneous initiation; Note=Translation N-terminally shortened.; Evidence={ECO:0000305};\n",
      "['Complete proteome', 'Cytoplasm', 'Fatty acid biosynthesis', 'Fatty acid metabolism', 'Lipid A biosynthesis', 'Lipid biosynthesis', 'Lipid metabolism', 'Phosphopantetheine', 'Phosphoprotein', 'Reference proteome']\n",
      "MTATFDKVADIIAETSEIDRATITPESHTIDDLGIDSLDFLDIVFAIDKEFGIKIPLEKWTQEVNEGKVSTEEYFVLKNLCAKIDELKAAKA\n",
      "92\n",
      "1\n",
      "\n",
      "FUNCTION: Carrier of the growing fatty acid chain in fatty acid biosynthesis. Is involved in the transfer of long hydroxylated fatty acids to lipid A (By similarity). {ECO:0000250}.\n",
      "PATHWAY: Glycolipid biosynthesis; KDO(2)-lipid A biosynthesis.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "PTM: 4'-phosphopantetheine is transferred from CoA to a specific serine of apo-ACP by AcpS. This modification is essential for activity because fatty acids are bound in thioester linkage to the sulfhydryl of the prosthetic group (By similarity). {ECO:0000250}.\n",
      "['Complete proteome', 'Cytoplasm', 'Fatty acid biosynthesis', 'Fatty acid metabolism', 'Lipid A biosynthesis', 'Lipid biosynthesis', 'Lipid metabolism', 'Phosphopantetheine', 'Phosphoprotein', 'Reference proteome']\n",
      "MRVTATFDKVADIIAETSEIDRETIKPESHTIDDLGIDSLDFLDIVFAIDKEFGIKIPLEQWTQEVNEGKVSTEEYFVLKNLCAKIDELRAAKAG\n",
      "95\n",
      "1\n",
      "\n",
      "FUNCTION: Acyl carrier protein.\n",
      "PTM: 4'-phosphopantetheine is transferred from CoA to a specific serine of the apo-ACP-like protein. {ECO:0000305}.\n",
      "['Complete proteome', 'Phosphopantetheine', 'Phosphoprotein', 'Reference proteome']\n",
      "MTTGLDAARKQEIKEIVCDILEIDEDEVTETSLFKEQHDADSLRAIEILAALERTQKVTIDQAELSRMVNLEGVYVVVSEAAQN\n",
      "84\n",
      "1\n",
      "\n",
      "FUNCTION: Acyl carrier protein.\n",
      "PATHWAY: Antifungal biosynthesis; monensin biosynthesis.\n",
      "PTM: 4'-phosphopantetheine is transferred from CoA to a specific serine of the apo-ACP-like protein. {ECO:0000305}.\n",
      "['Antibiotic biosynthesis', 'Phosphopantetheine', 'Phosphoprotein']\n",
      "MSDRPFTLADLQRILVEAAGADESAGPDDILDTTFALLGYESLALLETGGCIEREIGISLDDDTLTDALTPRELIDHVNERLAAARVA\n",
      "88\n",
      "1\n",
      "\n",
      "FUNCTION: Acyl carrier protein.\n",
      "PATHWAY: Antibiotic biosynthesis; curamycin biosynthesis.\n",
      "PTM: 4'-phosphopantetheine is transferred from CoA to a specific serine of the apo-ACP-like protein. {ECO:0000305}.\n",
      "['Antibiotic biosynthesis', 'Phosphopantetheine', 'Phosphoprotein']\n",
      "MTTEVTQVTVEELATLMKKAAGVTVDPRDLERRANGFADFGLDSLGLLGIVGELENRYARALHRRGTCKSPRAFLDVVNGALASGA\n",
      "86\n",
      "1\n",
      "\n",
      "FUNCTION: Acyl carrier protein.\n",
      "PATHWAY: Antibiotic biosynthesis; granaticin biosynthesis.\n",
      "PTM: 4'-phosphopantetheine is transferred from CoA to a specific serine of the apo-ACP-like protein. {ECO:0000305}.\n",
      "['Antibiotic biosynthesis', 'Phosphopantetheine', 'Phosphoprotein']\n",
      "MARLTLDGLRTILVACAGEDDGVDLSGDILDITFEELGYDSLALMESASRIERELGVALADGDINEELTPRVLLDLVNGAQAEAA\n",
      "85\n",
      "1\n",
      "\n",
      "FUNCTION: Acyl carrier protein.\n",
      "PTM: 4'-phosphopantetheine is transferred from CoA to a specific serine of the apo-ACP-like protein. {ECO:0000305}.\n",
      "['Complete proteome', 'Phosphopantetheine', 'Phosphoprotein', 'Reference proteome']\n",
      "MPETTTTALDKEQLRELVADVLDLDVAEVTDDADFMEDLDVDSLMALEITVRLEKEYGVRLAEAELTSITSLQGTYELLTSKLGDTR\n",
      "87\n",
      "1\n",
      "\n",
      "FUNCTION: Carrier of the growing fatty acid chain in fatty acid biosynthesis. {ECO:0000255|HAMAP-Rule:MF_01217}.\n",
      "PATHWAY: Lipid metabolism; fatty acid biosynthesis. {ECO:0000255|HAMAP-Rule:MF_01217}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000255|HAMAP-Rule:MF_01217}.\n",
      "PTM: 4'-phosphopantetheine is transferred from CoA to a specific serine of apo-ACP by AcpS. This modification is essential for activity because fatty acids are bound in thioester linkage to the sulfhydryl of the prosthetic group. {ECO:0000255|HAMAP- Rule:MF_01217}.\n",
      "SIMILARITY: Belongs to the acyl carrier protein (ACP) family. {ECO:0000255|HAMAP-Rule:MF_01217}.\n",
      "['Complete proteome', 'Cytoplasm', 'Fatty acid biosynthesis', 'Fatty acid metabolism', 'Lipid biosynthesis', 'Lipid metabolism', 'Phosphopantetheine', 'Phosphoprotein', 'Reference proteome']\n",
      "MSESEIFGKVKDIVAEQLSVDADKVTPEASFQNDLDADSLDVVEMVMALEEEFDIEIPDEAAEEILTVQAAVDFIAGKVKA\n",
      "81\n",
      "1\n",
      "\n",
      "FUNCTION: Carrier of the growing fatty acid chain in fatty acid biosynthesis.\n",
      "PATHWAY: Lipid metabolism; fatty acid biosynthesis.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "PTM: 4'-phosphopantetheine is transferred from CoA to a specific serine of apo-ACP by AcpS. This modification is essential for activity because fatty acids are bound in thioester linkage to the sulfhydryl of the prosthetic group.\n",
      "SIMILARITY: Belongs to the acyl carrier protein (ACP) family. {ECO:0000305}.\n",
      "['Cytoplasm', 'Direct protein sequencing', 'Fatty acid biosynthesis', 'Fatty acid metabolism', 'Lipid biosynthesis', 'Lipid metabolism', 'Phosphopantetheine']\n",
      "MSDIEQRIKQAVAEQLGMRAEEIKNE\n",
      "26\n",
      "1\n",
      "\n",
      "FUNCTION: Carrier of the growing fatty acid chain in fatty acid biosynthesis.\n",
      "PATHWAY: Lipid metabolism; fatty acid biosynthesis. {ECO:0000255|HAMAP-Rule:MF_01217}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000255|HAMAP-Rule:MF_01217}.\n",
      "PTM: 4'-phosphopantetheine is transferred from CoA to a specific serine of apo-ACP by AcpS. This modification is essential for activity because fatty acids are bound in thioester linkage to the sulfhydryl of the prosthetic group.\n",
      "SIMILARITY: Belongs to the acyl carrier protein (ACP) family. {ECO:0000255|HAMAP-Rule:MF_01217}.\n",
      "['Cytoplasm', 'Direct protein sequencing', 'Fatty acid biosynthesis', 'Fatty acid metabolism', 'Lipid biosynthesis', 'Lipid metabolism', 'Phosphopantetheine', 'Phosphoprotein']\n",
      "MSQSETFEKVKKIVIEQLSVENPDTVTPEASFANDLQADSLDTVELVMALEEEFDIEIPDEAAEKITTVQAAVDXIN\n",
      "77\n",
      "1\n",
      "\n",
      "FUNCTION: AcrA-AcrB-AcrZ-TolC is a drug efflux protein complex with a broad substrate specificity. This protein binds to AcrB and is required for efflux of some but not all substrates, suggesting it may influence the specificity of drug export. {ECO:0000255|HAMAP-Rule:MF_01484}.\n",
      "SUBUNIT: Part of the AcrA-AcrB-AcrZ-TolC efflux pump, interacts directly with AcrB. {ECO:0000255|HAMAP-Rule:MF_01484}.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane {ECO:0000255|HAMAP- Rule:MF_01484}; Single-pass membrane protein {ECO:0000255|HAMAP- Rule:MF_01484}.\n",
      "SIMILARITY: Belongs to the AcrZ family. {ECO:0000255|HAMAP- Rule:MF_01484}.\n",
      "['Antibiotic resistance', 'Cell inner membrane', 'Cell membrane', 'Complete proteome', 'Membrane', 'Transmembrane', 'Transmembrane helix', 'Transport']\n",
      "MLELLKSLVFAVIMVPVVMAIILGLIYGLGEVFNIFSGVGKKDQPGQNH\n",
      "49\n",
      "1\n",
      "\n",
      "CATALYTIC ACTIVITY: An acylphosphate + H(2)O = a carboxylate + phosphate.\n",
      "SIMILARITY: Belongs to the acylphosphatase family. {ECO:0000305}.\n",
      "SEQUENCE CAUTION: Sequence=CAD13747.1; Type=Erroneous initiation; Evidence={ECO:0000305};\n",
      "['Complete proteome', 'Hydrolase', 'Reference proteome']\n",
      "MGTASDAAALVTRLVRVRGRVQGIGYREACVRRARALGVTGWVRNRMDASVEAMLQGLPLQLEAMCAWLDEGMPAALVEGMEVIEVPAPFPRFDRFEQLPTL\n",
      "102\n",
      "1\n",
      "\n",
      "CATALYTIC ACTIVITY: An acylphosphate + H(2)O = a carboxylate + phosphate.\n",
      "SIMILARITY: Belongs to the acylphosphatase family. {ECO:0000305}.\n",
      "['Complete proteome', 'Hydrolase', 'Reference proteome']\n",
      "MTDVARPDVQACEQVRLTAWVRGRVQGVGFRWWVRARALELGLTGVARNLPDSRVEVVAEGPRERCAELLELLSGVPRHGRPGFVAGVTAEWSTARGGYSGFTQA\n",
      "105\n",
      "1\n",
      "\n",
      "FUNCTION: Catalyzes the conversion of acetoacetate to acetone and carbon dioxide.\n",
      "CATALYTIC ACTIVITY: Acetoacetate = acetone + CO(2).\n",
      "SIMILARITY: Belongs to the ADC family. {ECO:0000305}.\n",
      "['Decarboxylase', 'Direct protein sequencing', 'Lyase']\n",
      "MLKSEVSKQISMPLTAPAF\n",
      "19\n",
      "1\n",
      "\n",
      "FUNCTION: This is a novel enzyme, catalytically different from common alcohol dehydrogenases. It is effective in oxidizing ethanol, other primary alcohols and benzylalcohol only in the presence of p-nitroso-N,N-dimethylaniline (NDMA) as an electron acceptor. NADH acts as a cofactor here instead of as a coenzyme. {ECO:0000269|PubMed:10784035}.\n",
      "CATALYTIC ACTIVITY: N,N-dimethyl-4-nitrosoaniline + a primary alcohol = 4-(hydroxylamino)-N,N-dimethylaniline + an aldehyde. {ECO:0000269|PubMed:10784035}.\n",
      "CATALYTIC ACTIVITY: Ethanol + acceptor = acetaldehyde + reduced acceptor. {ECO:0000269|PubMed:10784035}.\n",
      "COFACTOR: Name=NADH; Xref=ChEBI:CHEBI:57945;   Evidence={ECO:0000269|PubMed:10784035};\n",
      "BIOPHYSICOCHEMICAL PROPERTIES: Kinetic parameters:   KM=0.1 mM for ethanol (when NDMA is the electron acceptor)   {ECO:0000269|PubMed:10784035};   KM=0.0037 mM for propan-1-ol (when NDMA is the electron   acceptor) {ECO:0000269|PubMed:10784035};   KM=0.01 mM for butan-1-ol (when NDMA is the electron acceptor)   {ECO:0000269|PubMed:10784035};   KM=0.0011 mM for heptan-1-ol (when NDMA is the electron   acceptor) {ECO:0000269|PubMed:10784035};   KM=0.059 mM for octan-1-ol (when NDMA is the electron acceptor)   {ECO:0000269|PubMed:10784035};   KM=0.028 mM for 2-propen-1-ol (when NDMA is the electron   acceptor) {ECO:0000269|PubMed:10784035};   KM=0.0072 mM for isobutanol (when NDMA is the electron acceptor)   {ECO:0000269|PubMed:10784035};   KM=0.0028 mM for 3-methylbutan-1-ol (when NDMA is the electron   acceptor) {ECO:0000269|PubMed:10784035};   KM=0.0018 mM for benzyl alcohol (when NDMA is the electron   acceptor) {ECO:0000269|PubMed:10784035};   KM=0.0047 mM for 4-hydroxybenzyl alcohol (when NDMA is the   electron acceptor) {ECO:0000269|PubMed:10784035};   KM=0.39 mM for vanillyl alcohol (when NDMA is the electron   acceptor) {ECO:0000269|PubMed:10784035};   KM=0.16 mM for veratryl alcohol (when NDMA is the electron   acceptor) {ECO:0000269|PubMed:10784035};   KM=0.007 mM for 2-phenylethanol (when NDMA is the electron   acceptor) {ECO:0000269|PubMed:10784035};   KM=0.056 mM for 2-phenylpropan-1-ol (when NDMA is the electron   acceptor) {ECO:0000269|PubMed:10784035};   KM=0.014 mM for 3-phenylpropan-1-ol (when NDMA is the electron   acceptor) {ECO:0000269|PubMed:10784035};   KM=0.0019 mM for 3-(4-hydroxyphenyl)propan-1-ol (when NDMA is   the electron acceptor) {ECO:0000269|PubMed:10784035};   KM=0.0036 mM for cinnamyl alcohol (when NDMA is the electron   acceptor) {ECO:0000269|PubMed:10784035};   KM=0.0055 mM for coniferyl alcohol (when NDMA is the electron   acceptor) {ECO:0000269|PubMed:10784035};   KM=0.2 mM for 2-phenylbutan-1-ol (when NDMA is the electron   acceptor) {ECO:0000269|PubMed:10784035};   KM=0.0059 mM for 4-(4-methoxyphenyl)butan-1-ol (when NDMA is the   electron acceptor) {ECO:0000269|PubMed:10784035};   KM=6 mM for propan-2-ol (when NDMA is the electron acceptor)   {ECO:0000269|PubMed:10784035};   KM=4.1 mM for butan-2-ol (when NDMA is the electron acceptor)   {ECO:0000269|PubMed:10784035};   KM=1.3 mM for pentan-2-ol (when NDMA is the electron acceptor)   {ECO:0000269|PubMed:10784035};   KM=0.83 mM for hexan-2-ol (when NDMA is the electron acceptor)   {ECO:0000269|PubMed:10784035};   KM=0.07 mM for heptan-2-ol (when NDMA is the electron acceptor)   {ECO:0000269|PubMed:10784035};   KM=0.12 mM for octan-2-ol (when NDMA is the electron acceptor)   {ECO:0000269|PubMed:10784035};   KM=9.1 mM for 3-methylheptan-4-ol (when NDMA is the electron   acceptor) {ECO:0000269|PubMed:10784035};   KM=0.28 mM for 4-methylheptan-3-ol (when NDMA is the electron   acceptor) {ECO:0000269|PubMed:10784035};   KM=2.8 mM for 1-phenylethanol (when NDMA is the electron   acceptor) {ECO:0000269|PubMed:10784035};   KM=0.55 mM for 1-phenylpropan-1-ol (when NDMA is the electron   acceptor) {ECO:0000269|PubMed:10784035};   KM=1.6 mM for 1-phenylpropan-2-ol (when NDMA is the electron   acceptor) {ECO:0000269|PubMed:10784035};   KM=0.55 mM for 1-phenylbutan-2-ol (when NDMA is the electron   acceptor) {ECO:0000269|PubMed:10784035};   KM=0.19 mM for cyclohexanol (when NDMA is the electron acceptor)   {ECO:0000269|PubMed:10784035};   KM=35 mM for propane-1,2-diol (when NDMA is the electron   acceptor) {ECO:0000269|PubMed:10784035};   KM=4.1 mM for propane-1,3-diol (when NDMA is the electron   acceptor) {ECO:0000269|PubMed:10784035};   KM=8.3 mM for 3-butene-1,2-diol (when NDMA is the electron   acceptor) {ECO:0000269|PubMed:10784035};   KM=0.2 mM for pentane-1,2-diol (when NDMA is the electron   acceptor) {ECO:0000269|PubMed:10784035};   KM=3.1 mM for 1-phenylethane-1,2-diol (when NDMA is the electron   acceptor) {ECO:0000269|PubMed:10784035};   KM=0.93 mM for formaldehyde (when NDMA is the electron acceptor)   {ECO:0000269|PubMed:10784035};   KM=4.95 mM for acetaldehyde (when veratryl alcohol is the   electron donor) {ECO:0000269|PubMed:10784035};   KM=0.083 mM for benzaldehyde (when veratryl alcohol is the   electron donor) {ECO:0000269|PubMed:10784035};   KM=0.088 mM for 2-phenylpropionaldehyde (when veratryl alcohol   is the electron donor) {ECO:0000269|PubMed:10784035};   KM=0.14 mM for 3-phenylpropionaldehyde (when veratryl alcohol is   the electron donor) {ECO:0000269|PubMed:10784035};   KM=0.12 mM for 3-phenylbutyraldehyde (when veratryl alcohol is   the electron donor) {ECO:0000269|PubMed:10784035};   KM=242 mM for glyoxal (when veratryl alcohol is the electron   donor) {ECO:0000269|PubMed:10784035};   KM=3.2 mM for methylglyoxal (when veratryl alcohol is the   electron donor) {ECO:0000269|PubMed:10784035};   KM=0.21 mM for phenylglyoxal (when veratryl alcohol is the   electron donor) {ECO:0000269|PubMed:10784035};   KM=38 mM for hydroxyacetone (when veratryl alcohol is the   electron donor) {ECO:0000269|PubMed:10784035};   KM=1.6 mM for cyclohexanone (when veratryl alcohol is the   electron donor) {ECO:0000269|PubMed:10784035};   KM=0.0041 mM for NDMA (when veratryl alcohol is the electron   donor) {ECO:0000269|PubMed:10784035};   Vmax=2.0 umol/min/mg enzyme toward ethanol (when NDMA is the   electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=2.5 umol/min/mg enzyme toward propan-1-ol (when NDMA is the   electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=5.8 umol/min/mg enzyme toward butan-1-ol (when NDMA is the   electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=2.0 umol/min/mg enzyme toward pentan-1-ol (when NDMA is the   electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=2.2 umol/min/mg enzyme toward hexan-1-ol (when NDMA is the   electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=2.7 umol/min/mg enzyme toward heptan-1-ol (when NDMA is the   electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=3.5 umol/min/mg enzyme toward octan-1-ol (when NDMA is the   electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=3.0 umol/min/mg enzyme toward 2-propen-1-ol (when NDMA is   the electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=3.3 umol/min/mg enzyme toward isobutanol (when NDMA is the   electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=3.1 umol/min/mg enzyme toward 3-methylbutan-1-ol (when NDMA   is the electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=1.9 umol/min/mg enzyme toward citronellol (when NDMA is the   electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=4.9 umol/min/mg enzyme toward benzyl alcohol (when NDMA is   the electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=0.79 umol/min/mg enzyme toward 4-hydroxybenzyl alcohol   (when NDMA is the electron acceptor)   {ECO:0000269|PubMed:10784035};   Vmax=2.0 umol/min/mg enzyme toward vanillyl alcohol (when NDMA   is the electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=1.8 umol/min/mg enzyme toward veratryl alcohol (when NDMA   is the electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=2.6 umol/min/mg enzyme toward 2-phenylethanol (when NDMA is   the electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=1.8 umol/min/mg enzyme toward 2-phenylpropan-1-ol (when   NDMA is the electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=2.4 umol/min/mg enzyme toward 3-phenylpropan-1-ol (when   NDMA is the electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=2.7 umol/min/mg enzyme toward 3-(4-hydroxyphenyl)propan-1-   ol (when NDMA is the electron acceptor)   {ECO:0000269|PubMed:10784035};   Vmax=3.4 umol/min/mg enzyme toward cinnamyl alcohol (when NDMA   is the electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=0.92 umol/min/mg enzyme toward coniferyl alcohol (when NDMA   is the electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=1.3 umol/min/mg enzyme toward 2-phenylbutan-1-ol (when NDMA   is the electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=4.0 umol/min/mg enzyme toward 4-(4-methoxyphenyl)butan-1-ol   (when NDMA is the electron acceptor)   {ECO:0000269|PubMed:10784035};   Vmax=0.16 umol/min/mg enzyme toward propan-2-ol (when NDMA is   the electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=0.81 umol/min/mg enzyme toward butan-2-ol (when NDMA is the   electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=2.0 umol/min/mg enzyme toward pentan-2-ol (when NDMA is the   electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=2.6 umol/min/mg enzyme toward hexan-2-ol (when NDMA is the   electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=1.1 umol/min/mg enzyme toward heptan-2-ol (when NDMA is the   electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=0.75 umol/min/mg enzyme toward octan-2-ol (when NDMA is the   electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=6.6 umol/min/mg enzyme toward 3-methylheptan-4-ol (when   NDMA is the electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=0.58 umol/min/mg enzyme toward 4-methylheptan-3-ol (when   NDMA is the electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=1.2 umol/min/mg enzyme toward 1-phenylethanol (when NDMA is   the electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=0.84 umol/min/mg enzyme toward 1-phenylpropan-1-ol (when   NDMA is the electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=1.9 umol/min/mg enzyme toward 1-phenylpropan-2-ol (when   NDMA is the electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=1.5 umol/min/mg enzyme toward 1-phenylbutan-2-ol (when NDMA   is the electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=1.9 umol/min/mg enzyme toward cyclohexanol (when NDMA is   the electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=2.3 umol/min/mg enzyme toward propane-1,2-diol (when NDMA   is the electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=3.0 umol/min/mg enzyme toward propane-1,3-diol (when NDMA   is the electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=1.6 umol/min/mg enzyme toward 3-butene-1,2-diol (when NDMA   is the electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=2.0 umol/min/mg enzyme toward pentane-1,2-diol (when NDMA   is the electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=1.4 umol/min/mg enzyme toward 1-phenylethane-1,2-diol (when   NDMA is the electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=3.4 umol/min/mg enzyme toward formaldehyde (when NDMA is   the electron acceptor) {ECO:0000269|PubMed:10784035};   Vmax=2.1 umol/min/mg enzyme toward acetaldehyde (when veratryl   alcohol is the electron donor) {ECO:0000269|PubMed:10784035};   Vmax=3.7 umol/min/mg enzyme toward benzaldehyde (when veratryl   alcohol is the electron donor) {ECO:0000269|PubMed:10784035};   Vmax=2.38 umol/min/mg enzyme toward 2-phenylpropionaldehyde   (when veratryl alcohol is the electron donor)   {ECO:0000269|PubMed:10784035};   Vmax=8.32 umol/min/mg enzyme toward 3-phenylpropionaldehyde   (when veratryl alcohol is the electron donor)   {ECO:0000269|PubMed:10784035};   Vmax=8.52 umol/min/mg enzyme toward 3-phenylbutyraldehyde (when   veratryl alcohol is the electron donor)   {ECO:0000269|PubMed:10784035};   Vmax=1.0 umol/min/mg enzyme toward glyoxal (when veratryl   alcohol is the electron donor) {ECO:0000269|PubMed:10784035};   Vmax=4.5 umol/min/mg enzyme toward methylglyoxal (when veratryl   alcohol is the electron donor) {ECO:0000269|PubMed:10784035};   Vmax=5.1 umol/min/mg enzyme toward phenylglyoxal (when veratryl   alcohol is the electron donor) {ECO:0000269|PubMed:10784035};   Vmax=0.15 umol/min/mg enzyme toward hydroxyacetone (when   veratryl alcohol is the electron donor)   {ECO:0000269|PubMed:10784035};   Vmax=0.097 umol/min/mg enzyme toward cyclohexanone (when   veratryl alcohol is the electron donor)   {ECO:0000269|PubMed:10784035};   Vmax=1.7 umol/min/mg enzyme toward NDMA (when veratryl alcohol   is the electron donor) {ECO:0000269|PubMed:10784035};   Note=The lower limit of the determination of KM with NDMA as   electron acceptor was 0.001 mM. KM was even lower for hexan-1-   ol, pentan-1-ol, and citronellol.; pH dependence:   Optimum pH is 7.0. {ECO:0000269|PubMed:10784035};\n",
      "SUBUNIT: Homotetramer. {ECO:0000269|PubMed:10784035}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm.\n",
      "SIMILARITY: Belongs to the zinc-containing alcohol dehydrogenase family. {ECO:0000305}.\n",
      "['Cytoplasm', 'Direct protein sequencing', 'NAD', 'Oxidoreductase']\n",
      "MKTKAAVLFETHKPFEIVELEL\n",
      "22\n",
      "1\n",
      "\n",
      "FUNCTION: A psychrophilic enzyme that oxidizes mainly primary and secondary aliphatic alcohols, as well as large bulky alcohols. Also catalyzes the reduction of ketones and aldehydes but in lower rates than the oxidations. The enzyme reduces stereospecifically 2-butanone to produce (s)-2-butanol. {ECO:0000305}.\n",
      "CATALYTIC ACTIVITY: An alcohol + NAD(+) = an aldehyde or ketone + NADH.\n",
      "COFACTOR: Name=Zn(2+); Xref=ChEBI:CHEBI:29105; Note=Binds 1 zinc ion per subunit.;\n",
      "ENZYME REGULATION: The presence of Ca(2+) results in a considerable stabilization of the enzyme.\n",
      "BIOPHYSICOCHEMICAL PROPERTIES: Kinetic parameters:   Note=Highest reaction rate when ethanol is used as the   substrate.; pH dependence:   Optimum pH is 7.5.; Temperature dependence:   Thermolabile.;\n",
      "SUBUNIT: Homotetramer. {ECO:0000305}.\n",
      "SIMILARITY: Belongs to the zinc-containing alcohol dehydrogenase family. {ECO:0000305}.\n",
      "['Direct protein sequencing', 'Metal-binding', 'NAD', 'Oxidoreductase', 'Zinc']\n",
      "MLYANPNTESSPVGFRKKYDNFI\n",
      "23\n",
      "1\n",
      "\n",
      "FUNCTION: Forms part of a two-component regulatory system AfsQ1/AfsQ2 involved in secondary metabolism.\n",
      "SUBCELLULAR LOCATION: Cytoplasm.\n",
      "PTM: Phosphorylated by AfsQ2. {ECO:0000305}.\n",
      "['Complete proteome', 'Cytoplasm', 'DNA-binding', 'Phosphoprotein', 'Reference proteome', 'Transcription', 'Transcription regulation', 'Two-component regulatory system']\n",
      "MPSLLLIEDDDAIRTALELSLTRQGHRVATAASGEDGLKLLREQRPDLIVLDVMLPGIDGFEVCRRIRRTDQLPIILLTARNDDIDVVVGLESGADDYVVKPVQGRVLDARIRAVLRRGERESTDSASFGSLVIDRSAMTVTKNGEDLQLTPTELRLLLELSRRPGQALSRQQLLRLVWEHDYLGDSRLVDACVQRLRAKVEDVPSSPTLIRTVRGVGYRLDPPQ\n",
      "225\n",
      "1\n",
      "\n",
      "FUNCTION: A severely truncated paralog of the AfuB uptake protein, homologous only to the last 20% of the intact protein in Actinobacillus. {ECO:0000305}.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane {ECO:0000269|PubMed:15919996}; Multi-pass membrane protein. Note=Upon overexpression (PubMed:15919996). {ECO:0000269|PubMed:15919996}.\n",
      "SIMILARITY: Belongs to the binding-protein-dependent transport system permease family. FbpB subfamily. {ECO:0000305}.\n",
      "CAUTION: Could be the product of a pseudogene. The original protein is truncated by an IS1 element which is inserted right before residue 6. What was the original N-terminal of this protein is not known. Other genes of the operon were presumably also deleted by this IS1 event. {ECO:0000305}.\n",
      "['Cell inner membrane', 'Cell membrane', 'Complete proteome', 'Membrane', 'Reference proteome', 'Transmembrane', 'Transmembrane helix']\n",
      "MESLPGQIDKSLDEASLSLRAGSLRTITHILLPLLRPAILSALIYSFVRAITTVSAIVFLVTPDTRVATAYILNRVEDGEYGVAIAYGSILIVVMLAIIFIFDWLIGESRTSRSKAKNQA\n",
      "120\n",
      "1\n",
      "\n",
      "FUNCTION: Directly reduces organic hydroperoxides in its reduced dithiol form. {ECO:0000250|UniProtKB:P26427}.\n",
      "CATALYTIC ACTIVITY: 2 R'-SH + ROOH = R'-S-S-R' + H(2)O + ROH.\n",
      "SUBUNIT: Homodimer. {ECO:0000250}.\n",
      "INDUCTION: By oxidative stress, primary alcohols, monocyclic aromatics and heat shock. {ECO:0000269|PubMed:11425483}.\n",
      "SIMILARITY: Belongs to the AhpC/TSA family. {ECO:0000305}.\n",
      "['Direct protein sequencing', 'Oxidoreductase', 'Peroxidase', 'Redox-active center', 'Stress response']\n",
      "SLINTEVKPFQATAYHNGQFVEVNE\n",
      "25\n",
      "1\n",
      "\n",
      "FUNCTION: Required for the synthesis of N-butanoyl-L-homoserine lactone (BHL), an autoinducer molecule which binds to AhyR.\n",
      "CATALYTIC ACTIVITY: An acyl-[acyl-carrier-protein] + S-adenosyl-L- methionine = [acyl-carrier-protein] + S-methyl-5'-thioadenosine + an N-acyl-L-homoserine lactone.\n",
      "SIMILARITY: Belongs to the autoinducer synthase family. {ECO:0000255|PROSITE-ProRule:PRU00533}.\n",
      "['Autoinducer synthesis', 'Quorum sensing', 'S-adenosyl-L-methionine', 'Transferase']\n",
      "MLVFKGKLKEHPRWEVENELYRFRNRVFSDRLGWDVESHRGLEQDSFDTPDTHWVLIEDEEGLCGCIRLLSCAKDYMLPSIFPTALAGEAPPRSNDVWELTRLAIDAERAPRLGNGISELTCIIFREVYAFAKAQGIRELVAVVSLPVERIFRRLGLPIERLGHRQAVDLGAVRGVGIRFHLDERFARAVGQPLQGAYDEARELVTE\n",
      "207\n",
      "1\n",
      "\n",
      "FUNCTION: Involved in the production of the bacteriocin subtilosin. Required for maximal production and for optimal immunity to subtilosin. {ECO:0000269|PubMed:10572140, ECO:0000269|PubMed:10809709}.\n",
      "SUBCELLULAR LOCATION: Cell membrane {ECO:0000305}; Multi-pass membrane protein {ECO:0000305}.\n",
      "INDUCTION: Transcription is highly induced by oxygen limitation and is under dual and independent control of Spo0A-AbrB and ResDE. {ECO:0000269|PubMed:10809710}.\n",
      "['Antibiotic biosynthesis', 'Bacteriocin biosynthesis', 'Bacteriocin immunity', 'Cell membrane', 'Complete proteome', 'Membrane', 'Reference proteome', 'Transmembrane', 'Transmembrane helix']\n",
      "MSPAQRRILLYILSFIFVIGAVVYFVKSDYLFTLIFIAIAILFGMRARKADSR\n",
      "53\n",
      "1\n",
      "\n",
      "CATALYTIC ACTIVITY: An aldehyde + NAD(+) + H(2)O = a carboxylate + NADH.\n",
      "PATHWAY: Alcohol metabolism; ethanol degradation; acetate from ethanol: step 2/2.\n",
      "SIMILARITY: Belongs to the aldehyde dehydrogenase family. {ECO:0000305}.\n",
      "['NAD', 'Oxidoreductase']\n",
      "MMSSIAAPKLKEKVEKFLSGKKKMYINGSFVESASGKTFDTPNPATGERLATVYEGDAEDIDRAVKAAREAFD\n",
      "73\n",
      "1\n",
      "\n",
      "FUNCTION: Positive regulator of the expression of the slpA gene (PubMed:7511582). When overexpressed, leads to suppression of the capsule overproduction and UV sensitivity phenotypes of cells mutant for the Lon ATP-dependent protease (PubMed:7511582). Part of the cryptic P4-like prophage CP4-57 (PubMed:7511583). Overexpression of AlpA leads to excision of the CP4-57 prophage by IntA. This inactivates ssrA (the gene upstream of the prophage) that encodes tmRNA which is required to rescue stalled ribosomes in a process known as trans-translation (PubMed:7511583). {ECO:0000269|PubMed:7511582, ECO:0000269|PubMed:7511583}.\n",
      "['Activator', 'Complete proteome', 'DNA-binding', 'Reference proteome', 'Transcription', 'Transcription regulation']\n",
      "MNNRSAVRILRLPAVIQKTGMARATIYDWLNPKSPRYDATFPKKRMLGVKSVGWIEAEIDEWLSQRCKLI\n",
      "70\n",
      "1\n",
      "\n",
      "FUNCTION: Catalyzes the interconversion of L-alanine and D- alanine. {ECO:0000269|PubMed:7763424}.\n",
      "CATALYTIC ACTIVITY: L-alanine = D-alanine. {ECO:0000269|PubMed:7763424}.\n",
      "COFACTOR: Name=pyridoxal 5'-phosphate; Xref=ChEBI:CHEBI:597326;   Evidence={ECO:0000269|PubMed:7763424};\n",
      "BIOPHYSICOCHEMICAL PROPERTIES: pH dependence:   Optimum pH is 8.3. {ECO:0000269|PubMed:7763424}; Temperature dependence:   Optimum temperature is 30 degrees Celsius. Thermolabile and   psychrotrophic. Shows catalytic activity even at 0 degree   Celsius. {ECO:0000269|PubMed:7763424};\n",
      "PATHWAY: Amino-acid biosynthesis; D-alanine biosynthesis; D- alanine from L-alanine: step 1/1.\n",
      "SUBUNIT: Homodimer. {ECO:0000269|PubMed:7763424}.\n",
      "SIMILARITY: Belongs to the alanine racemase family. {ECO:0000305}.\n",
      "['Direct protein sequencing', 'Isomerase', 'Pyridoxal phosphate']\n",
      "MXPARALIDLQALRHNYQLAAEVIG\n",
      "25\n",
      "2\n",
      "\n",
      "CATALYTIC ACTIVITY: An N-acyl-aliphatic-L-amino acid + H(2)O = an aliphatic L-amino acid + a carboxylate.\n",
      "CATALYTIC ACTIVITY: An N-acetyl-L-cysteine-S-conjugate + H(2)O = an L-cysteine-S-conjugate + acetate.\n",
      "COFACTOR: Name=Co(2+); Xref=ChEBI:CHEBI:48828; Evidence={ECO:0000250};\n",
      "SUBUNIT: Homotetramer.\n",
      "SIMILARITY: Belongs to the peptidase M20 family. {ECO:0000305}.\n",
      "['Cobalt', 'Direct protein sequencing', 'Hydrolase']\n",
      "MTNEEIKRLVDEVKEGVIAXRRHL\n",
      "24\n",
      "1\n",
      "\n",
      "FUNCTION: Primary acceptor of electrons from methylamine dehydrogenase. Passes those electrons on either a soluble cytochrome c or to pseudoazurin.\n",
      "COFACTOR: Name=Cu cation; Xref=ChEBI:CHEBI:23378; Note=Binds 1 copper ion per subunit.;\n",
      "PATHWAY: One-carbon metabolism; methylamine degradation.\n",
      "SUBCELLULAR LOCATION: Periplasm.\n",
      "['3D-structure', 'Copper', 'Direct protein sequencing', 'Electron transport', 'Metal-binding', 'Periplasm', 'Signal', 'Transport']\n",
      "MISATKIRSCLAACVLAAFGATGALADKATIPSESPFAAAEVADGAIVVDIAKMKYETPELHVKVGDTVTWINREAMPHNVHFVAGVLGEAALKGPMMKKEQAYSLTFTEAGTYDYHCTPHPFMRGKVVVE\n",
      "131\n",
      "1\n",
      "\n",
      "FUNCTION: Metalloenzyme of broad specificity, releasing all N- terminal amino acids.\n",
      "COFACTOR: Name=a divalent metal cation; Xref=ChEBI:CHEBI:60240;   Evidence={ECO:0000250}; Note=Binds 2 divalent metal cations per subunit. {ECO:0000250};\n",
      "BIOPHYSICOCHEMICAL PROPERTIES: Temperature dependence:   Highly thermostable.;\n",
      "SUBUNIT: 12 chains of two different but homologous types, alpha and beta, which can combine in various ratios.\n",
      "SIMILARITY: Belongs to the peptidase M42 family. {ECO:0000305}.\n",
      "['Aminopeptidase', 'Direct protein sequencing', 'Hydrolase', 'Metalloprotease', 'Protease']\n",
      "AKLDETLTMLKALTDAKGVPGNEREARDVM\n",
      "30\n",
      "2\n",
      "\n",
      "FUNCTION: Alpha-amylase active towards amylose, starch, amylopectin and maltodextrins. Has lower activity towards glycogen, and is not active towards alpha/beta-cyclodextrin. {ECO:0000269|PubMed:20109486}.\n",
      "CATALYTIC ACTIVITY: Endohydrolysis of (1->4)-alpha-D-glucosidic linkages in polysaccharides containing three or more (1->4)-alpha- linked D-glucose units. {ECO:0000269|PubMed:20109486}.\n",
      "ENZYME REGULATION: Strongly inhibited by Hg (2+). Inhibited by Zn (2+). Activated by Fe (2+), Mg (2+) and Ba (2+). {ECO:0000269|PubMed:20109486}.\n",
      "BIOPHYSICOCHEMICAL PROPERTIES: pH dependence:   Optimum pH is 4.5. Stable from pH 4.0-7.5, activity decreases   drastically above pH 7.5 and below pH 3.5.   {ECO:0000269|PubMed:20109486}; Temperature dependence:   Optimum temperature is 70 degrees Celsius. Active from 30 to 75   degrees Celsius, inactivated following 150 minutes incubation at   85 degrees Celsius. {ECO:0000269|PubMed:20109486};\n",
      "['Carbohydrate metabolism', 'Direct protein sequencing', 'Glycosidase', 'Hydrolase']\n",
      "AHQLPMGTLCNFYEWYRRDD\n",
      "20\n",
      "2\n",
      "\n",
      "FUNCTION: Light-harvesting photosynthetic bile pigment-protein from the phycobiliprotein complex. Allophycocyanin has a maximum absorption at approximately 650 to 653 nanometers. {ECO:0000305}.\n",
      "SUBUNIT: Heterodimer of an alpha and a beta chain. {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cellular thylakoid membrane {ECO:0000250}; Peripheral membrane protein {ECO:0000250}; Cytoplasmic side {ECO:0000250}. Note=Forms the core of the phycobilisome. {ECO:0000250}.\n",
      "PTM: Contains one covalently linked phycocyanobilin chromophore. {ECO:0000250|UniProtKB:P00318}.\n",
      "SIMILARITY: Belongs to the phycobiliprotein family. {ECO:0000255}.\n",
      "['Bile pigment', 'Chromophore', 'Direct protein sequencing', 'Electron transport', 'Membrane', 'Methylation', 'Photosynthesis', 'Phycobilisome', 'Thylakoid', 'Transport']\n",
      "MAQDAITAV\n",
      "9\n",
      "1\n",
      "\n",
      "FUNCTION: Light-harvesting photosynthetic bile pigment-protein from the phycobiliprotein complex. Allophycocyanin has a maximum absorption at approximately 650 nanometers.\n",
      "SUBUNIT: Heterodimer of an alpha and a beta chain.\n",
      "SUBCELLULAR LOCATION: Cellular thylakoid membrane; Peripheral membrane protein; Cytoplasmic side. Note=Forms the core of the phycobilisome.\n",
      "PTM: Contains one covalently linked bilin chromophore.\n",
      "SIMILARITY: Belongs to the phycobiliprotein family. {ECO:0000305}.\n",
      "['Bile pigment', 'Chromophore', 'Direct protein sequencing', 'Electron transport', 'Membrane', 'Photosynthesis', 'Phycobilisome', 'Thylakoid', 'Transport']\n",
      "TVVSQVILKADDELRYPSSGELKSITEFLQTGEQRVRIAQ\n",
      "40\n",
      "1\n",
      "\n",
      "FUNCTION: Effector protein that participates in the suppression of macrophage apoptosis by blocking the extrinsic pathway. Recognizes the host polypyrimidine tract binding protein-associated splicing factor (PSF), which probably leads to its cleavage, diminishing the level of caspase-8 in macrophages. {ECO:0000269|PubMed:20454556}.\n",
      "SUBUNIT: Interacts with human polypyrimidine tract binding protein-associated splicing factor (PSF). {ECO:0000269|PubMed:20454556}.\n",
      "SUBCELLULAR LOCATION: Secreted {ECO:0000269|PubMed:20454556}. Host cytoplasm {ECO:0000269|PubMed:20454556}. Note=Is secreted into the macrophage cytoplasm. Might be secreted via a type IV pili apparatus that seems to be encoded in the same operon. {ECO:0000269|PubMed:20454556}.\n",
      "INDUCTION: Component of the Rv3654c-Rv3660c operon that is highly up-regulated during M.tuberculosis infection of macrophages. {ECO:0000269|PubMed:20454556}.\n",
      "['Complete proteome', 'Host cytoplasm', 'Reference proteome', 'Secreted', 'Signal', 'Virulence']\n",
      "MVARHRAQAAADLASLAAAARLPSGLAAACARATLVARAMRVEHAQCRVVDLDVVVTVEVAVAFAGVATATARAGPAKVPTTPG\n",
      "84\n",
      "2\n",
      "\n",
      "FUNCTION: Might be part of cytochrome bd-II oxidase (appB and appC). Able to restore reductant resistance to a cydX deletion mutant upon overexpression. CyxD and this protein may have some functional overlap. {ECO:0000269|PubMed:23749980}.\n",
      "SUBUNIT: Able to interact with CydA and CydB upon overexpression.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane {ECO:0000305}; Single- pass membrane protein {ECO:0000305}. Note=In sucrose cushions fractionates with the outer membrane, in sarcosyl fractionation with the inner membrane. May have its C-terminus in the cytoplasm (PubMed:21778229). {ECO:0000269|PubMed:21778229}.\n",
      "INDUCTION: Expressed during stationary phase (PubMed:19121005) and in minimal glycerol medium, at 45 degrees Celsius, strongly (30- fold) induced in low oxygen (PubMed:19734316) (at protein level). {ECO:0000269|PubMed:19121005, ECO:0000269|PubMed:19734316}.\n",
      "DISRUPTION PHENOTYPE: Not essential, no change in sensitivity to reductant (beta-mercaptoethanol). {ECO:0000269|PubMed:23749980}.\n",
      "SIMILARITY: Belongs to the cytochrome ubiquinol oxidase subunit X family. {ECO:0000305}.\n",
      "['Cell inner membrane', 'Cell membrane', 'Complete proteome', 'Membrane', 'Reference proteome', 'Stress response', 'Transmembrane', 'Transmembrane helix']\n",
      "MWYLLWFVGILLMCSLSTLVLVWLDPRLKS\n",
      "30\n",
      "3\n",
      "\n",
      "FUNCTION: Catalyzes the conversion of L-arabinose to L-ribulose. {ECO:0000250}.\n",
      "CATALYTIC ACTIVITY: L-arabinose = L-ribulose.\n",
      "PATHWAY: Carbohydrate degradation; L-arabinose degradation via L- ribulose; D-xylulose 5-phosphate from L-arabinose (bacterial route): step 1/3.\n",
      "SUBUNIT: Homohexamer. {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the arabinose isomerase family. {ECO:0000305}.\n",
      "['Arabinose catabolism', 'Carbohydrate metabolism', 'Isomerase']\n",
      "NLDDMRQFAELHDIELTVIDNDTRLPSFKDALRWNEVYYGSKR\n",
      "43\n",
      "1\n",
      "\n",
      "FUNCTION: Rescues ribosomes stalled at the 3' end of non-stop mRNAs (PubMed:21062370, PubMed:21435036). This activity is crucial when the stalled ribosome cannot be rescued by the SsrA(tmRNA)- SmpB quality control system (PubMed:21062370, PubMed:21435036). Binds the 30S subunit, contacting 16S rRNA with the N-terminus near the decoding center and its C-terminus in the mRNA entry channel; contacts change in the presence of release factor 2 (RF2, also named PrfB) (PubMed:25355516, PubMed:27906160, PubMed:27906161, PubMed:27934701, PubMed:28077875). Requires RF2/PrfB to hydrolyze stalled peptidyl-tRNA on the ribosome; recruits and probably helps position RF2/PrfB correctly in the ribosomal A site so RF2's GGQ motif can hydrolyze the peptidyl- tRNA bond (PubMed:22857598, PubMed:22922063, PubMed:25355516, PubMed:27906160, PubMed:27906161, PubMed:27934701, PubMed:28077875). Does not release ribosomes with a programmed pause caused by regulatory chains such as SecM-mediated pausing (PubMed:22857598). Binds tRNA which may stimulate its ribosome rescue activity (PubMed:22857598). {ECO:0000269|PubMed:21062370, ECO:0000269|PubMed:21435036, ECO:0000269|PubMed:22857598, ECO:0000269|PubMed:22922063, ECO:0000269|PubMed:25355516, ECO:0000269|PubMed:27906160, ECO:0000269|PubMed:27906161, ECO:0000269|PubMed:27934701, ECO:0000269|PubMed:28077875}.\n",
      "SUBUNIT: Interacts with the 50S ribosomal subunit (PubMed:21062370, PubMed:25355516, PubMed:27906160, PubMed:27906161, PubMed:27934701, PubMed:28077875). Interacts with RF2/PrfB on the ribosome (PubMed:27906160, PubMed:27906161, PubMed:27934701, PubMed:28077875). Contacts ribosomal protein S12 (rpsL) (PubMed:27906160, PubMed:27906161). {ECO:0000269|PubMed:21062370, ECO:0000269|PubMed:25355516, ECO:0000269|PubMed:27906160, ECO:0000269|PubMed:27906161, ECO:0000269|PubMed:27934701, ECO:0000269|PubMed:28077875}.\n",
      "DISRUPTION PHENOTYPE: Cells lacking this gene are viable, but those lacking both arfA and ssrA are non-viable (PubMed:21062370). {ECO:0000269|PubMed:21062370}.\n",
      "MISCELLANEOUS: Active protein is only produced when SsrA(tmRNA)- SmpB-mediated trans translation is limited. ArfA can be synthesized from truncated mRNA, via ribonuclease 3 and the SsrA tagging system. The arfA mRNA is cleaved by ribonuclease 3 within a hairpin structure, probably immediately after the Gly-55 codon, which produces a truncated non-stop transcript and leads to ribosome arrest. In the presence of SsrA, the stalled ribosome is rescued and the incomplete nascent ArfA is tagged for degradation. In the absence of SsrA, truncated ArfA chains are released from the ribosome and are functional. Truncated ArfA could be more active than full-length protein (PubMed:21435036, PubMed:21952205). {ECO:0000305|PubMed:21435036, ECO:0000305|PubMed:21952205}.\n",
      "SIMILARITY: Belongs to the alternative ribosome-rescue factor A family. {ECO:0000305}.\n",
      "CAUTION: In all 3D-structures with RF2/PrfB and the 70S ribosome this protein is seen to contact 16S rRNA (PubMed:27906160, PubMed:27906161, PubMed:27934701, PubMed:28077875). In most is also seen to contact 23S rRNA (PubMed:27906160, PubMed:27934701, PubMed:28077875). Contact with ribosomal protein S12 is also reported sometimes (PubMed:27906160, PubMed:27906161). These discrepancies may be due to interpretation of results. {ECO:0000269|PubMed:27906160, ECO:0000269|PubMed:27906161, ECO:0000269|PubMed:27934701, ECO:0000269|PubMed:28077875}.\n",
      "['3D-structure', 'Complete proteome', 'Reference proteome', 'RNA-binding', 'rRNA-binding', 'Translation regulation', 'tRNA-binding']\n",
      "MSRYQHTKGQIKDNAIEALLHDPLFRQRVEKNKKGKGSYMRKGKHGNRGNWEASGKKVNHFFTTGLLLSGAC\n",
      "72\n",
      "1\n",
      "\n",
      "FUNCTION: Rescues the ribosomes stalled at the 3' end of non-stop mRNAs. May induce hydrolysis of the ribosome-bound peptidyl-tRNA (By similarity). {ECO:0000250}.\n",
      "SUBUNIT: Interacts with the 50S ribosomal subunit. {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the alternative ribosome-rescue factor A family. {ECO:0000305}.\n",
      "['Complete proteome', 'Reference proteome', 'Translation regulation']\n",
      "MRKKQKSAVENETVYLHTRGVIKDNAVMALLGDKLFRQRVEKKRKGKGSYQRKAKHPGKIFEKPDYKFF\n",
      "69\n",
      "1\n",
      "\n",
      "FUNCTION: Required for wild-type expression of ArfA and ammonia secretion. {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cell membrane {ECO:0000305}; Single-pass membrane protein {ECO:0000305}.\n",
      "SIMILARITY: Belongs to the ArfB membrane protein family. {ECO:0000305}.\n",
      "['Cell membrane', 'Complete proteome', 'Membrane', 'Transmembrane', 'Transmembrane helix']\n",
      "MDFVIQWSCYLLAFLGGSAVAWVVVTLSIKRASRDEGAAEAPSAAETGAQ\n",
      "50\n",
      "1\n",
      "\n",
      "FUNCTION: Translocates 4-amino-4-deoxy-L-arabinose- phosphoundecaprenol (alpha-L-Ara4N-phosphoundecaprenol) from the cytoplasmic to the periplasmic side of the inner membrane. {ECO:0000255|HAMAP-Rule:MF_01869}.\n",
      "PATHWAY: Bacterial outer membrane biogenesis; lipopolysaccharide biosynthesis. {ECO:0000255|HAMAP-Rule:MF_01869}.\n",
      "SUBUNIT: Heterodimer of ArnE and ArnF. {ECO:0000255|HAMAP- Rule:MF_01869}.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane {ECO:0000255|HAMAP- Rule:MF_01869}; Multi-pass membrane protein {ECO:0000255|HAMAP- Rule:MF_01869}.\n",
      "SIMILARITY: Belongs to the ArnE family. {ECO:0000255|HAMAP- Rule:MF_01869}.\n",
      "['Cell inner membrane', 'Cell membrane', 'Complete proteome', 'Lipid A biosynthesis', 'Lipid biosynthesis', 'Lipid metabolism', 'Lipopolysaccharide biosynthesis', 'Membrane', 'Transmembrane', 'Transmembrane helix', 'Transport']\n",
      "MTTFLLFVLSVFSSSMSQYWQKRAAMLFAMQPDLRPLQKLFSRPLILSVFFLGVGAMTWLGVLSVWDVSMAYPLLSINFVIMLLLSHYVFNEPISVRQWIGITLIMLGVIFLAGGEQWLI\n",
      "120\n",
      "2\n",
      "\n",
      "CATALYTIC ACTIVITY: Phosphoenolpyruvate + D-erythrose 4-phosphate + H(2)O = 3-deoxy-D-arabino-hept-2-ulosonate 7-phosphate + phosphate.\n",
      "PATHWAY: Metabolic intermediate biosynthesis; chorismate biosynthesis; chorismate from D-erythrose 4-phosphate and phosphoenolpyruvate: step 1/7.\n",
      "SUBUNIT: Homodimer. {ECO:0000250}.\n",
      "PTM: The N-terminus is blocked.\n",
      "SIMILARITY: Belongs to the class-II DAHP synthase family. {ECO:0000305}.\n",
      "['Amino-acid biosynthesis', 'Aromatic amino acid biosynthesis', 'Direct protein sequencing', 'Transferase']\n",
      "TRRFDDVLDEVKGFFEVH\n",
      "18\n",
      "1\n",
      "\n",
      "FUNCTION: Catalyzes a trans-dehydration via an enolate intermediate. Is involved in both the catabolism of quinate and the biosynthesis of aromatic amino acids.\n",
      "CATALYTIC ACTIVITY: 3-dehydroquinate = 3-dehydroshikimate + H(2)O.\n",
      "BIOPHYSICOCHEMICAL PROPERTIES: pH dependence:   Optimum pH is 9.0.; Temperature dependence:   Optimum temperature is 76 degrees Celsius. Thermostable.;\n",
      "PATHWAY: Metabolic intermediate biosynthesis; chorismate biosynthesis; chorismate from D-erythrose 4-phosphate and phosphoenolpyruvate: step 3/7.\n",
      "SUBUNIT: Homododecamer.\n",
      "SIMILARITY: Belongs to the type-II 3-dehydroquinase family. {ECO:0000305}.\n",
      "['Amino-acid biosynthesis', 'Aromatic amino acid biosynthesis', 'Direct protein sequencing', 'Lyase', 'Quinate metabolism']\n",
      "MKVFVLNGPNLGRLGKREPA\n",
      "20\n",
      "3\n",
      "\n",
      "FUNCTION: Reduction of arsenate [As(V)] to arsenite [As(III)]. {ECO:0000305}.\n",
      "CATALYTIC ACTIVITY: Arsenate + glutaredoxin = arsenite + glutaredoxin disulfide + H(2)O.\n",
      "SIMILARITY: Belongs to the ArsC family. {ECO:0000305}.\n",
      "['Arsenical resistance', 'Oxidoreductase']\n",
      "MSEIKIFHNPRCSKSRAAVSLLEERGIAAEAVKYLDTPPDLSELKDIFNKLGLESARGMMRVKDDLYKELGLDNPDLDNDALLRAIADHPALLERPIVLGKR\n",
      "102\n",
      "2\n",
      "\n",
      "FUNCTION: Shows a strict specificity for N-acetyl arylalkylamines but not acetanilide derivatives.\n",
      "CATALYTIC ACTIVITY: N-acetylarylalkylamine + H(2)O = arylalkylamine + acetate.\n",
      "ENZYME REGULATION: Activated by divalent metal ions. Inhibited by certain thiol reagents.\n",
      "SUBUNIT: Homotetramer.\n",
      "['Direct protein sequencing', 'Hydrolase']\n",
      "TRRTAFLFDELSLWHSASQYALILPVGG\n",
      "28\n",
      "1\n",
      "\n",
      "FUNCTION: Part of the accessory SecA2/SecY2 system specifically required to export GspB, a serine-rich repeat cell wall protein encoded upstream in the same operon. {ECO:0000269|PubMed:15901716}.\n",
      "SUBUNIT: Part of the accessory SecA2/SecY2 protein translocation apparatus required to export cell wall protein GspB.\n",
      "SUBCELLULAR LOCATION: Cell membrane {ECO:0000305}; Single-pass membrane protein {ECO:0000305}.\n",
      "DISRUPTION PHENOTYPE: No export of mature cell wall protein GspB, a small amount of unprocessed protein is exported to the cell wall while the rest accumulates intracellularly, probably in a glycosylated form. Cells bind less well to platelets. {ECO:0000269|PubMed:15901716}.\n",
      "['Cell membrane', 'Membrane', 'Protein transport', 'Translocation', 'Transmembrane', 'Transmembrane helix', 'Transport']\n",
      "MAKKDLFHKDIEGRLDELKHGKPKKEKASLGENLNKIFVIALGLMILIGLIFTLIGALRK\n",
      "60\n",
      "1\n",
      "\n",
      "FUNCTION: Required for growth and/or survival at acidic conditions. {ECO:0000255|HAMAP-Rule:MF_00546}.\n",
      "SUBCELLULAR LOCATION: Periplasm {ECO:0000255|HAMAP-Rule:MF_00546}.\n",
      "PTM: Proteolytic processing gives rise to the active protein. {ECO:0000255|HAMAP-Rule:MF_00546}.\n",
      "SIMILARITY: Belongs to the Asr family. {ECO:0000255|HAMAP- Rule:MF_00546}.\n",
      "['Complete proteome', 'Periplasm', 'Signal']\n",
      "MKKVLALVVAAAMGLSSAAFAAETATTPAPTATTTKAAPAKTTHHKKQHKAAPAQKAQAAKKHHKNAKAEQKAPEQKAQAAKKHARKHSHQQPAKPAAQPAA\n",
      "102\n",
      "3\n",
      "\n",
      "FUNCTION: Required for growth and/or survival at acidic conditions. {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Periplasm {ECO:0000250}.\n",
      "PTM: Proteolytic processing gives rise to the active protein. {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the Asr family. {ECO:0000305}.\n",
      "SEQUENCE CAUTION: Sequence=AAK92015.1; Type=Erroneous initiation; Evidence={ECO:0000305};\n",
      "['Periplasm', 'Signal']\n",
      "MKKVLALVVAAAMGLSSAAFAAETTATAAPAASTAAPAKTVHHKKHHKAAKPAAEQKAQAAKKHHKKAAKPAVEQKAQAAKKHHKKAAKHEAAKPAAQPAA\n",
      "101\n",
      "1\n",
      "\n",
      "FUNCTION: Produces ATP from ADP in the presence of a proton gradient across the membrane. {ECO:0000255|HAMAP-Rule:MF_00530}.\n",
      "SUBUNIT: F-type ATPases have 2 components, CF(1) - the catalytic core - and CF(0) - the membrane proton channel. CF(1) has five subunits: alpha(3), beta(3), gamma(1), delta(1), epsilon(1). CF(0) has three main subunits: a, b and c.\n",
      "SUBCELLULAR LOCATION: Cell membrane {ECO:0000255|HAMAP- Rule:MF_00530}; Peripheral membrane protein {ECO:0000255|HAMAP- Rule:MF_00530}.\n",
      "SIMILARITY: Belongs to the ATPase epsilon chain family. {ECO:0000255|HAMAP-Rule:MF_00530}.\n",
      "['ATP synthesis', 'Cell membrane', 'CF(1)', 'Complete proteome', 'Hydrogen ion transport', 'Ion transport', 'Membrane', 'Reference proteome', 'Transport']\n",
      "MPIRVAIVSPEQEVWSGDADMVVARTTDGDLGVLPGHVPLLGLLAPGGTVRVKTGGREISASVDGGFISVTHQGVSILAETAKLT\n",
      "85\n",
      "2\n",
      "\n",
      "FUNCTION: Produces ATP from ADP in the presence of a proton gradient across the membrane. {ECO:0000250}.\n",
      "SUBUNIT: F-type ATPases have 2 components, CF(1) - the catalytic core - and CF(0) - the membrane proton channel. CF(1) has five subunits: alpha(3), beta(3), gamma(1), delta(1), epsilon(1). CF(0) has three main subunits: a, b and c.\n",
      "SUBCELLULAR LOCATION: Cell membrane {ECO:0000250}; Peripheral membrane protein {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the ATPase epsilon chain family. {ECO:0000305}.\n",
      "['ATP synthesis', 'Cell membrane', 'CF(1)', 'Complete proteome', 'Direct protein sequencing', 'Hydrogen ion transport', 'Ion transport', 'Membrane', 'Reference proteome', 'Transport']\n",
      "MAELNVEIVSEERSIWSGAASAVSARTVNGEIGILPGHTPMLAVLGDGEVVVRTTDGGTVTAQAHGGFFSVDHDRVVIAATSARLGDAAAA\n",
      "91\n",
      "1\n",
      "\n",
      "FUNCTION: F(1)F(0) ATP synthase produces ATP from ADP in the presence of a proton or sodium gradient. F-type ATPases consist of two structural domains, F(1) containing the extramembraneous catalytic core and F(0) containing the membrane proton channel, linked together by a central stalk and a peripheral stalk. During catalysis, ATP synthesis in the catalytic domain of F(1) is coupled via a rotary mechanism of the central stalk subunits to proton translocation. {ECO:0000255|HAMAP-Rule:MF_01396}.\n",
      "FUNCTION: Key component of the F(0) channel; it plays a direct role in translocation across the membrane. A homomeric c-ring of between 10-14 subunits forms the central stalk rotor element with the F(1) delta and epsilon subunits. {ECO:0000255|HAMAP- Rule:MF_01396}.\n",
      "SUBUNIT: F-type ATPases have 2 components, F(1) - the catalytic core - and F(0) - the membrane proton channel. F(1) has five subunits: alpha(3), beta(3), gamma(1), delta(1), epsilon(1). F(0) has three main subunits: a(1), b(2) and c(10-14). The alpha and beta chains form an alternating ring which encloses part of the gamma chain. F(1) is attached to F(0) by a central stalk formed by the gamma and epsilon chains, while a peripheral stalk is formed by the delta and b chains. {ECO:0000255|HAMAP-Rule:MF_01396}.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane {ECO:0000255|HAMAP- Rule:MF_01396}; Multi-pass membrane protein {ECO:0000255|HAMAP- Rule:MF_01396}.\n",
      "SIMILARITY: Belongs to the ATPase C chain family. {ECO:0000255|HAMAP-Rule:MF_01396}.\n",
      "['ATP synthesis', 'Cell inner membrane', 'Cell membrane', 'CF(0)', 'Complete proteome', 'Hydrogen ion transport', 'Ion transport', 'Lipid-binding', 'Membrane', 'Reference proteome', 'Transmembrane', 'Transmembrane helix', 'Transport']\n",
      "MNFFTMCVFGAAIGMAIGTLGTAIGQGMAVKSAVEGVARNPGAASKIMTTMMIGLAMIESLAIYALVVCLIILFANPYKDIALKMVETVAK\n",
      "91\n",
      "2\n",
      "\n",
      "FUNCTION: F(1)F(0) ATP synthase produces ATP from ADP in the presence of a proton or sodium gradient. F-type ATPases consist of two structural domains, F(1) containing the extramembraneous catalytic core and F(0) containing the membrane proton channel, linked together by a central stalk and a peripheral stalk. During catalysis, ATP synthesis in the catalytic domain of F(1) is coupled via a rotary mechanism of the central stalk subunits to proton translocation. {ECO:0000255|HAMAP-Rule:MF_01396}.\n",
      "FUNCTION: Key component of the F(0) channel; it plays a direct role in translocation across the membrane. A homomeric c-ring of between 10-14 subunits forms the central stalk rotor element with the F(1) delta and epsilon subunits. {ECO:0000255|HAMAP- Rule:MF_01396}.\n",
      "SUBUNIT: F-type ATPases have 2 components, F(1) - the catalytic core - and F(0) - the membrane proton channel. F(1) has five subunits: alpha(3), beta(3), gamma(1), delta(1), epsilon(1). F(0) has three main subunits: a(1), b(2) and c(10-14). The alpha and beta chains form an alternating ring which encloses part of the gamma chain. F(1) is attached to F(0) by a central stalk formed by the gamma and epsilon chains, while a peripheral stalk is formed by the delta and b chains. {ECO:0000255|HAMAP-Rule:MF_01396}.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane {ECO:0000255|HAMAP- Rule:MF_01396}; Multi-pass membrane protein {ECO:0000255|HAMAP- Rule:MF_01396}.\n",
      "SIMILARITY: Belongs to the ATPase C chain family. {ECO:0000255|HAMAP-Rule:MF_01396}.\n",
      "['ATP synthesis', 'Cell inner membrane', 'Cell membrane', 'CF(0)', 'Complete proteome', 'Hydrogen ion transport', 'Ion transport', 'Lipid-binding', 'Membrane', 'Reference proteome', 'Transmembrane', 'Transmembrane helix', 'Transport']\n",
      "MSFFSMCVLGAAIGMAIGTLGTGIGQGLAVKSAVEGVSRNPGASGKIMTTMMIGLAMIESLAIYALVICLIILFANPYKDIALKLAETVAK\n",
      "91\n",
      "3\n",
      "\n",
      "FUNCTION: F(1)F(0) ATP synthase produces ATP from ADP in the presence of a proton or sodium gradient. F-type ATPases consist of two structural domains, F(1) containing the extramembraneous catalytic core and F(0) containing the membrane proton channel, linked together by a central stalk and a peripheral stalk. During catalysis, ATP synthesis in the catalytic domain of F(1) is coupled via a rotary mechanism of the central stalk subunits to proton translocation. {ECO:0000255|HAMAP-Rule:MF_01396}.\n",
      "FUNCTION: Key component of the F(0) channel; it plays a direct role in translocation across the membrane. A homomeric c-ring of between 10-14 subunits forms the central stalk rotor element with the F(1) delta and epsilon subunits. {ECO:0000255|HAMAP- Rule:MF_01396}.\n",
      "SUBUNIT: F-type ATPases have 2 components, F(1) - the catalytic core - and F(0) - the membrane proton channel. F(1) has five subunits: alpha(3), beta(3), gamma(1), delta(1), epsilon(1). F(0) has three main subunits: a(1), b(2) and c(10-14). The alpha and beta chains form an alternating ring which encloses part of the gamma chain. F(1) is attached to F(0) by a central stalk formed by the gamma and epsilon chains, while a peripheral stalk is formed by the delta and b chains. {ECO:0000255|HAMAP-Rule:MF_01396}.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane {ECO:0000255|HAMAP- Rule:MF_01396}; Multi-pass membrane protein {ECO:0000255|HAMAP- Rule:MF_01396}.\n",
      "SIMILARITY: Belongs to the ATPase C chain family. {ECO:0000255|HAMAP-Rule:MF_01396}.\n",
      "['ATP synthesis', 'Cell inner membrane', 'Cell membrane', 'CF(0)', 'Complete proteome', 'Hydrogen ion transport', 'Ion transport', 'Lipid-binding', 'Membrane', 'Reference proteome', 'Transmembrane', 'Transmembrane helix', 'Transport']\n",
      "MELTLGLVAIASAILIAFGALGTAIGFGLLGGRFLEAVARQPELAPQLQTRMFLIAGLLDAVPMIGVGIGLFFIFANPFVG\n",
      "81\n",
      "2\n",
      "\n",
      "FUNCTION: F(1)F(0) ATP synthase produces ATP from ADP in the presence of a proton or sodium gradient. F-type ATPases consist of two structural domains, F(1) containing the extramembraneous catalytic core and F(0) containing the membrane proton channel, linked together by a central stalk and a peripheral stalk. During catalysis, ATP synthesis in the catalytic domain of F(1) is coupled via a rotary mechanism of the central stalk subunits to proton translocation.\n",
      "FUNCTION: Key component of the F(0) channel; it plays a direct role in translocation across the membrane. A homomeric c-ring of 10 subunits forms the central stalk rotor element with the F(1) delta and epsilon subunits.\n",
      "SUBUNIT: F-type ATPases have 2 components, F(1) - the catalytic core - and F(0) - the membrane proton channel. F(1) has five subunits: alpha(3), beta(3), gamma(1), delta(1), epsilon(1). F(0) has three main subunits: a(1), b(2) and c(10). The alpha and beta chains form an alternating ring which encloses part of the gamma chain. F(1) is attached to F(0) by a central stalk formed by the gamma and epsilon chains, while a peripheral stalk is formed by the delta and b chains. {ECO:0000269|PubMed:15302927, ECO:0000269|PubMed:2894854}.\n",
      "SUBCELLULAR LOCATION: Cell membrane; Multi-pass membrane protein.\n",
      "MISCELLANEOUS: Dicyclohexylcarbodiimide (DCDD) binding to the active aspartate residue inhibits ATPase in vitro.\n",
      "MISCELLANEOUS: Fusion proteins corresponding to c(2), c(5) and c(10) permit ATP synthesis in reconstitution experiments with F(1) and in the absence of F(1) function as a proton channel; other fusions of c(3) to c(14) do not.\n",
      "SIMILARITY: Belongs to the ATPase C chain family. {ECO:0000305}.\n",
      "['3D-structure', 'ATP synthesis', 'Cell membrane', 'CF(0)', 'Direct protein sequencing', 'Formylation', 'Hydrogen ion transport', 'Ion transport', 'Lipid-binding', 'Membrane', 'Transmembrane', 'Transmembrane helix', 'Transport']\n",
      "MSLGVLAAAIAVGLGALGAGIGNGLIVSRTIEGIARQPELRPVLQTTMFIGVALVEALPIIGVVFSFIYLGR\n",
      "72\n",
      "2\n",
      "\n",
      "FUNCTION: F(1)F(0) ATP synthase produces ATP from ADP in the presence of a sodium gradient. F-type ATPases consist of two structural domains, F(1) containing the extramembraneous catalytic core and F(0) containing the membrane proton channel, linked together by a central stalk and a peripheral stalk. During catalysis, ATP synthesis in the catalytic domain of F(1) is coupled via a rotary mechanism of the central stalk subunits to proton translocation. {ECO:0000269|PubMed:16816177}.\n",
      "FUNCTION: Key component of the F(0) channel; it plays a direct role in translocation across the membrane. A homomeric c-ring of 11 subunits forms the central stalk rotor element with the F(1) delta and epsilon subunits. {ECO:0000269|PubMed:16816177}.\n",
      "FUNCTION: In this organism this enzyme may function as an ATP- driven Na(+) ion pump to generate a Na(+) ion electrochemical gradient rather than as an ATP synthase. {ECO:0000269|PubMed:16816177}.\n",
      "ENZYME REGULATION: Inhibited by dicyclohexylcarbodiimide; the enzyme is protected from inhibition by Na(+) in a pH-dependent manner.\n",
      "SUBUNIT: F-type ATPases have 2 components, F(1) - the catalytic core - and F(0) - the membrane sodium channel. F(1) has five subunits: alpha(3), beta(3), gamma(1), delta(1), epsilon(1). F(0) has three main subunits: a(1), b(2) and c(11). The alpha and beta chains form an alternating ring which encloses part of the gamma chain. F(1) is attached to F(0) by a central stalk formed by the gamma and epsilon chains, while a peripheral stalk is formed by the delta and b chains. {ECO:0000269|PubMed:16816177, ECO:0000269|PubMed:16980459}.\n",
      "SUBCELLULAR LOCATION: Cell membrane {ECO:0000269|PubMed:16816177}; Multi-pass membrane protein {ECO:0000269|PubMed:16816177}.\n",
      "MASS SPECTROMETRY: Mass=8266; Method=MALDI; Range=1-84; Evidence={ECO:0000269|PubMed:16980459};\n",
      "MISCELLANEOUS: The ATPase of C.paradoxum is of special interest because it uses sodium ions instead of protons as the physiological coupling ion.\n",
      "SIMILARITY: Belongs to the ATPase C chain family. {ECO:0000305}.\n",
      "['ATP synthesis', 'Cell membrane', 'CF(0)', 'Hydrogen ion transport', 'Ion transport', 'Lipid-binding', 'Membrane', 'Sodium', 'Sodium transport', 'Transmembrane', 'Transmembrane helix', 'Transport']\n",
      "MERALILAASAIGAGLAMIAGIGPGIGQGFAAGKGAEAVGKQPEAQGDILRTMLLGAAVAESTGIYALVVALILLFANPLLNLL\n",
      "84\n",
      "3\n",
      "\n",
      "FUNCTION: Involved in branched-chain amino acid transport.\n",
      "SUBCELLULAR LOCATION: Cell membrane {ECO:0000305}; Multi-pass membrane protein {ECO:0000305}.\n",
      "SIMILARITY: Belongs to the AzlD/HI_1737/HP1330 family. {ECO:0000305}.\n",
      "['Amino-acid transport', 'Cell membrane', 'Complete proteome', 'Membrane', 'Reference proteome', 'Transmembrane', 'Transmembrane helix', 'Transport']\n",
      "MTMTMTQQMITIAMVVLGTMLTRFLPFMIFPSGKPTPKYVQYLGKVLPSAVIGLLVIYCLKDVSLLSGSHGIPELVGAAAVVLLHLWKKNMLLSIAGGTVVYMVLVQLVF\n",
      "110\n",
      "1\n",
      "\n",
      "FUNCTION: Inhibitor of the ribonuclease barnase. Forms a one-to- one non-covalent complex.\n",
      "SUBCELLULAR LOCATION: Cytoplasm.\n",
      "SIMILARITY: Belongs to the barstar family. {ECO:0000305}.\n",
      "['3D-structure', 'Cytoplasm']\n",
      "MKKAVINGEQIRSISDLHQTLKKELALPEYYGENLDALWDCLTGWVEYPLVLEWRQFEQSKQLTENGAESVLQVFREAKAEGCDITIILS\n",
      "90\n",
      "3\n",
      "\n",
      "SUBCELLULAR LOCATION: Cell membrane {ECO:0000305}; Single-pass membrane protein {ECO:0000305}.\n",
      "MISCELLANEOUS: Cellulose production is abolished in E.coli K12.\n",
      "SEQUENCE CAUTION: Sequence=AAB18515.1; Type=Erroneous initiation; Note=Translation N-terminally extended.; Evidence={ECO:0000305};\n",
      "['Cell membrane', 'Complete proteome', 'Membrane', 'Reference proteome', 'Transmembrane', 'Transmembrane helix']\n",
      "MMTISDIIEIIVVCALIFFPLGYLARHSLRRIRDTLRLFFAKPRYVKPAGTLRRTEKARATKK\n",
      "63\n",
      "2\n",
      "\n",
      "FUNCTION: Required for cellulose biosynthesis. {ECO:0000269|PubMed:11929533}.\n",
      "SUBCELLULAR LOCATION: Cell membrane {ECO:0000305}; Single-pass membrane protein {ECO:0000305}.\n",
      "['Cell membrane', 'Cellulose biosynthesis', 'Complete proteome', 'Membrane', 'Reference proteome', 'Transmembrane', 'Transmembrane helix']\n",
      "MMTISDIVQIILFCALIFFPLGYLARHSLRRISDTTRLLFAKPRYVKPAGTLRRATKVKADKK\n",
      "63\n",
      "3\n",
      "\n",
      "SEQUENCE CAUTION: Sequence=AAG56288.1; Type=Erroneous initiation; Evidence={ECO:0000305}; Sequence=BAB35508.1; Type=Erroneous initiation; Evidence={ECO:0000305};\n",
      "['Complete proteome']\n",
      "MFTYYQAENSTAEPALVNAIEQGLRAEHGVVTEDDILMELTKWVEASDNDILSDIYQQTINYVVSGQHPTL\n",
      "71\n",
      "1\n",
      "\n",
      "FUNCTION: Seems to associate with BFR; could be a general redox and/or regulatory component participating in the iron storage mobilization functions of BFR. Could participate in the release or the delivery of iron from/to bacterioferritin (or other iron complexes) (By similarity). {ECO:0000250}.\n",
      "COFACTOR: Name=[2Fe-2S] cluster; Xref=ChEBI:CHEBI:49601;   Evidence={ECO:0000250}; Note=Binds 1 [2Fe-2S] cluster. {ECO:0000250};\n",
      "SUBUNIT: Monomer. {ECO:0000250}.\n",
      "['2Fe-2S', 'Complete proteome', 'Electron transport', 'Iron', 'Iron-sulfur', 'Metal-binding', 'Transport']\n",
      "MYVCLCNGISDKKIRQAVRQFSPHSFQQLKKFIPVGNQCGKCVRAAREVMEDELMQLPEFKESA\n",
      "64\n",
      "1\n",
      "\n",
      "FUNCTION: Iron-storage protein, whose ferroxidase center binds Fe(2+) ions, oxidizes them by dioxygen to Fe(3+), and participates in the subsequent Fe(3+) oxide mineral core formation within the central cavity of the protein complex. {ECO:0000250}.\n",
      "FUNCTION: May act as one of the electron carriers in the reverse electron-transport system from cytochrome c-552 to NADP(+).\n",
      "CATALYTIC ACTIVITY: 4 Fe(2+) + 4 H(+) + O(2) = 4 Fe(3+) + 2 H(2)O.\n",
      "COFACTOR: Name=heme b; Xref=ChEBI:CHEBI:60344; Evidence={ECO:0000250}; Note=Binds 1 heme b (iron(II)-protoporphyrin IX) group per dimer. {ECO:0000250};\n",
      "COFACTOR: Name=Fe cation; Xref=ChEBI:CHEBI:24875; Evidence={ECO:0000250}; Note=Binds 2 iron ions per subunit. The catalytic dinuclear iron- binding site within each subunit is known as the ferroxidase center. {ECO:0000250};\n",
      "SUBUNIT: Homooligomer of 24 subunits, arranged as 12 dimers, that are packed together to form an approximately spherical molecule with a central cavity, in which large amounts of iron can be deposited. {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the bacterioferritin family. {ECO:0000305}.\n",
      "['Direct protein sequencing', 'Heme', 'Iron', 'Iron storage', 'Metal-binding', 'Oxidoreductase']\n",
      "MKGDPKVIDYLNKALRHELTAINQYWLHYRLLDNWGIKDLAKKWRAESIE\n",
      "50\n",
      "2\n",
      "\n",
      "CATALYTIC ACTIVITY: A beta-lactam + H(2)O = a substituted beta- amino acid. {ECO:0000255|PROSITE-ProRule:PRU10101}.\n",
      "SUBCELLULAR LOCATION: Secreted. Membrane.\n",
      "SIMILARITY: Belongs to the class-A beta-lactamase family. {ECO:0000305}.\n",
      "['Antibiotic resistance', 'Direct protein sequencing', 'Hydrolase', 'Membrane', 'Secreted']\n",
      "AAAPDIPIANVNA\n",
      "13\n",
      "1\n",
      "\n",
      "FUNCTION: Component of the cell division machinery, which is probably involved in the stabilization of the divisome under certain stress conditions. {ECO:0000269|PubMed:22885295}.\n",
      "SUBUNIT: Interacts with FtsL and several other divisomal proteins, including FtsI, FtsK, FtsN, FtsQ, FtsW and YmgF. {ECO:0000269|PubMed:22885295}.\n",
      "INTERACTION: P0AEN4:ftsL; NbExp=3; IntAct=EBI-6419495, EBI-1119082; P29131:ftsN; NbExp=3; IntAct=EBI-6419495, EBI-1134233;\n",
      "SUBCELLULAR LOCATION: Cell inner membrane {ECO:0000269|PubMed:10931331, ECO:0000269|PubMed:22885295}; Single-pass membrane protein {ECO:0000269|PubMed:10931331, ECO:0000269|PubMed:22885295}. Note=Localizes to the division septum. Localization requires FtsQ and FtsN.\n",
      "PTM: The N-terminus is blocked.\n",
      "DISRUPTION PHENOTYPE: Inactivation increases the susceptibility of bacteria to a number of antibiotics that inhibit peptidoglycan biosynthesis. Increases the susceptibility to cycloserine, bacitracin, and a wide spectrum of beta-lactam antibiotics. Does not affect susceptibility to fosfomycin, valinomycin, tetracycline, chloramphenicol, gentamicin or quinolones. {ECO:0000269|PubMed:10931331}.\n",
      "['Antibiotic resistance', 'Cell cycle', 'Cell division', 'Cell inner membrane', 'Cell membrane', 'Complete proteome', 'Membrane', 'Reference proteome', 'Stress response', 'Transmembrane', 'Transmembrane helix']\n",
      "MNRLIELTGWIVLVVSVILLGVASHIDNYQPPEQSASVQHK\n",
      "41\n",
      "1\n",
      "\n",
      "FUNCTION: Transcriptional regulator that plays an important role in general stress response. {ECO:0000250|UniProtKB:P0ABE2}.\n",
      "SIMILARITY: Belongs to the BolA/IbaG family. {ECO:0000305}.\n",
      "['Complete proteome', 'DNA-binding', 'Reference proteome', 'Stress response', 'Transcription', 'Transcription regulation']\n",
      "MSIQQIIEQKIQKEFQPHFLAIENESHLHHSNRGSESHFKCVIVSADFKNIRKVQRHQRIYQLLNEELNHSIHALALHLFTPEEWKAQNETVPHSTKCAGIGR\n",
      "103\n",
      "2\n",
      "\n",
      "SUBCELLULAR LOCATION: Cell membrane {ECO:0000305}; Lipid-anchor {ECO:0000305}.\n",
      "MISCELLANEOUS: Encoded by the cryptic lambdoid prophage DLP12.\n",
      "SIMILARITY: Belongs to the lambda phage bor family. {ECO:0000305}.\n",
      "['Cell membrane', 'Complete proteome', 'Lipoprotein', 'Membrane', 'Palmitate', 'Reference proteome', 'Signal']\n",
      "MKKMLLATALALLITGCAQQTFTVQNKQTAVAPKETITHHFFVSGIGQKKTVDAAKICGGAENVVKTETQQTFVNGLLGFITLGIYTPLEARVYCSK\n",
      "97\n",
      "2\n",
      "\n",
      "FUNCTION: Part of the BsdBCD complex involved in the decarboxylation and detoxification of phenolic derivatives under both aerobic and anaerobic conditions. It is able to catalyze the reversible decarboxylation of 4-hydroxybenzoate, vanillate and salicylate. Is not active on di- and tri-hydroxybenzoate derivatives. {ECO:0000269|PubMed:18388975}.\n",
      "CATALYTIC ACTIVITY: 4-hydroxybenzoate = phenol + CO(2). {ECO:0000269|PubMed:18388975}.\n",
      "SUBUNIT: Component of the decarboxylase complex composed of the subunits B, C and D. {ECO:0000269|PubMed:18388975}.\n",
      "INDUCTION: Up-regulated by salicylate via the transcriptional regulator BsdA. {ECO:0000269|PubMed:17295427}.\n",
      "SEQUENCE CAUTION: Sequence=BAA08998.1; Type=Frameshift; Positions=75; Evidence={ECO:0000305};\n",
      "['Aromatic hydrocarbons catabolism', 'Complete proteome', 'Decarboxylase', 'Detoxification', 'Lyase', 'Reference proteome']\n",
      "MHTCPRCDSKKGEVMSKSPVEGAWEVYQCQTCFFTWRSCEPESITNPEKYNPAFKIDPKETETAIEVPAVPERKA\n",
      "75\n",
      "1\n",
      "\n",
      "FUNCTION: Catalyzes the addition of fumarate to the methyl group of toluene, leading to the formation of benzylsuccinate. {ECO:0000269|PubMed:9632263}.\n",
      "CATALYTIC ACTIVITY: Toluene + fumarate = benzylsuccinate. {ECO:0000269|PubMed:9632263}.\n",
      "ENZYME REGULATION: Activated by the benzylsuccinate synthase activating enzyme BssD. Rapidly inactivated by oxygen. {ECO:0000269|PubMed:9632263}.\n",
      "BIOPHYSICOCHEMICAL PROPERTIES: pH dependence:   Optimum pH is 8.0. {ECO:0000269|PubMed:9632263};\n",
      "PATHWAY: Xenobiotic degradation; toluene degradation. {ECO:0000269|PubMed:9632263}.\n",
      "SUBUNIT: Heterohexamer composed of 2 alpha subunits, 2 beta subunits and 2 gamma subunits. {ECO:0000269|PubMed:9632263}.\n",
      "INDUCTION: Induced by toluene, probably via the TdiR/TdiS two- component regulatory system. {ECO:0000269|PubMed:11807562, ECO:0000269|PubMed:9632263, ECO:0000269|PubMed:9741082}.\n",
      "['Aromatic hydrocarbons catabolism', 'Direct protein sequencing', 'Lyase']\n",
      "MSATPHTQVHWEENTARPCRKCKWQTPDPTDPLRGQCTVNRHAMGGVWKRWIRDVEHMTCSRHEEGELSFRDHV\n",
      "74\n",
      "1\n",
      "\n",
      "FUNCTION: Catalyzes the addition of fumarate to the methyl group of toluene, leading to the formation of benzylsuccinate. {ECO:0000269|PubMed:9632263}.\n",
      "CATALYTIC ACTIVITY: Toluene + fumarate = benzylsuccinate. {ECO:0000269|PubMed:9632263}.\n",
      "ENZYME REGULATION: Activated by the benzylsuccinate synthase activating enzyme BssD. Rapidly inactivated by oxygen. {ECO:0000269|PubMed:9632263}.\n",
      "BIOPHYSICOCHEMICAL PROPERTIES: pH dependence:   Optimum pH is 8.0. {ECO:0000269|PubMed:9632263};\n",
      "PATHWAY: Xenobiotic degradation; toluene degradation. {ECO:0000269|PubMed:9632263}.\n",
      "SUBUNIT: Heterohexamer composed of 2 alpha subunits, 2 beta subunits and 2 gamma subunits. {ECO:0000269|PubMed:9632263}.\n",
      "INDUCTION: Induced by toluene, probably via the TdiR/TdiS two- component regulatory system. {ECO:0000269|PubMed:11807562, ECO:0000269|PubMed:9632263, ECO:0000269|PubMed:9741082}.\n",
      "['Aromatic hydrocarbons catabolism', 'Direct protein sequencing', 'Lyase']\n",
      "MTTCKDCAFFFSIPEDADDFEKSKGDCVTQKDDEKGRYWLSKPVFENDQCCGAFHKR\n",
      "57\n",
      "1\n",
      "\n",
      "FUNCTION: Represses biofilm formation in M9C glu and LB glu media but not in M9C and LB media. Seems to act as a global regulator of several genes involved in catabolite repression and stress response and regulation of the uptake and export of signaling pathways. Could be involved the regulation of indole as well as uptake and export of AI-2 through a cAMP-dependent pathway (By similarity). {ECO:0000250}.\n",
      "SEQUENCE CAUTION: Sequence=AAN79800.1; Type=Erroneous initiation; Evidence={ECO:0000305};\n",
      "['Complete proteome']\n",
      "MEKNNEVIQTHPLVGWDISTVDSYDALMLRLHYQTPNKSEQEGTEVGQTLWLTTDVARQFISILEAGIAKIESGDFQVNEYRRH\n",
      "84\n",
      "4\n",
      "\n",
      "FUNCTION: Catalyzes the conversion of butyryl-CoA through butyryl phosphate to butyrate. {ECO:0000250}.\n",
      "CATALYTIC ACTIVITY: ATP + butanoate = ADP + butanoyl phosphate.\n",
      "PATHWAY: Lipid metabolism; butanoate metabolism.\n",
      "SUBCELLULAR LOCATION: Cytoplasm.\n",
      "SIMILARITY: Belongs to the acetokinase family. {ECO:0000305}.\n",
      "['ATP-binding', 'Cytoplasm', 'Direct protein sequencing', 'Kinase', 'Nucleotide-binding', 'Transferase']\n",
      "MYKLLIINP\n",
      "9\n",
      "2\n",
      "\n",
      "CATALYTIC ACTIVITY: ATP + butanoate = ADP + butanoyl phosphate.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the acetokinase family. {ECO:0000305}.\n",
      "SEQUENCE CAUTION: Sequence=AAA22570.1; Type=Frameshift; Positions=Several; Evidence={ECO:0000305};\n",
      "['ATP-binding', 'Cytoplasm', 'Kinase', 'Nucleotide-binding', 'Transferase']\n",
      "MQEQKFRILTINPGSTSTKIGVFENERAIVEKTIRHEGRCFGNIKR\n",
      "46\n",
      "1\n",
      "\n",
      "FUNCTION: Catalyzes the NADPH- and oxygen-dependent oxidation of the monocyclic monoterpene ketones 1-hydroxy-2-oxolimonene, dihydrocarvone and menthone. {ECO:0000269|PubMed:10769172}.\n",
      "CATALYTIC ACTIVITY: (-)-menthone + NADPH + O(2) = (4R,7S)-7- isopropyl-4-methyloxepan-2-one + NADP(+) + H(2)O. {ECO:0000269|PubMed:10769172}.\n",
      "CATALYTIC ACTIVITY: Dihydrocarvone + NADPH + O(2) = 4-isopropenyl- 7-methyloxepan-2-one + NADP(+) + H(2)O. {ECO:0000269|PubMed:10769172}.\n",
      "CATALYTIC ACTIVITY: (Iso)-dihydrocarvone + NADPH + O(2) = 6- isopropenyl-3-methyloxepan-2-one + NADP(+) + H(2)O. {ECO:0000269|PubMed:10769172}.\n",
      "CATALYTIC ACTIVITY: 1-hydroxymenth-8-en-2-one + NADPH + O(2) = 3- isopropenyl-6-oxoheptanoate + NADP(+) + H(2)O. {ECO:0000269|PubMed:10769172}.\n",
      "COFACTOR: Name=FAD; Xref=ChEBI:CHEBI:57692;\n",
      "BIOPHYSICOCHEMICAL PROPERTIES: pH dependence:   Optimum pH is 9.5.; Temperature dependence:   Optimum temperature is about 36 degrees Celsius.;\n",
      "PATHWAY: Terpene metabolism; monoterpene degradation.\n",
      "SUBUNIT: Monomer.\n",
      "['Direct protein sequencing', 'FAD', 'Flavoprotein', 'NADP', 'Oxidoreductase']\n",
      "MQTIPSTTETAADFDAVIXGAGFXGLYALHR\n",
      "31\n",
      "1\n",
      "\n",
      "FUNCTION: Affects cell viability at low temperatures.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "INDUCTION: In response to low temperature.\n",
      "['Activator', 'Cytoplasm', 'DNA-binding', 'Stress response', 'Transcription', 'Transcription regulation']\n",
      "MSQRQSGTVKWFNDEKGFGFITPQGGGDDLFVHFKAIESDGFKSLKEGQTVSFVAEKGQKGMQA\n",
      "64\n",
      "2\n",
      "\n",
      "FUNCTION: Affects cell viability at low temperatures.\n",
      "SUBCELLULAR LOCATION: Cytoplasm.\n",
      "INDUCTION: In response to low temperature.\n",
      "['Activator', 'Cytoplasm', 'Direct protein sequencing', 'DNA-binding', 'Stress response', 'Transcription', 'Transcription regulation']\n",
      "MSNRQTGTVKWFNDEKGFGFITPQSGDDLFVHFKAIQSDGFKSLKEGQQVSFIATRGQKGMQAEEVQVI\n",
      "69\n",
      "2\n",
      "\n",
      "FUNCTION: CRISPR (clustered regularly interspaced short palindromic repeat), is an adaptive immune system that provides protection against mobile genetic elements (viruses, transposable elements and conjugative plasmids). CRISPR clusters contain sequences complementary to antecedent mobile elements and target invading nucleic acids. CRISPR clusters are transcribed and processed into CRISPR RNA (crRNA). Functions as a ssRNA-specific endoribonuclease. Involved in the integration of spacer DNA into the CRISPR cassette. {ECO:0000255|HAMAP-Rule:MF_01471}.\n",
      "COFACTOR: Name=Mg(2+); Xref=ChEBI:CHEBI:18420;   Evidence={ECO:0000255|HAMAP-Rule:MF_01471};\n",
      "SUBUNIT: Homodimer, forms a heterotetramer with a Cas1 homodimer. {ECO:0000255|HAMAP-Rule:MF_01471}.\n",
      "SIMILARITY: Belongs to the CRISPR-associated endoribonuclease Cas2 protein family. {ECO:0000255|HAMAP-Rule:MF_01471}.\n",
      "['Antiviral defense', 'Complete proteome', 'Endonuclease', 'Hydrolase', 'Magnesium', 'Metal-binding', 'Nuclease', 'Reference proteome']\n",
      "MILIYDINTEDNDGKRRLVKIMKTSRKYLSHVQKSVFEGDITEGQISLLKKEIMAIVNMKKDFVIIYSLRDGVKLNREILTDTPDPTDNFL\n",
      "91\n",
      "4\n",
      "\n",
      "FUNCTION: Decomposes hydrogen peroxide into water and oxygen; serves to protect cells from the toxic effects of hydrogen peroxide. {ECO:0000305}.\n",
      "CATALYTIC ACTIVITY: 2 H(2)O(2) = O(2) + 2 H(2)O. {ECO:0000255|PROSITE-ProRule:PRU10013, ECO:0000269|PubMed:12094731, ECO:0000269|PubMed:15177292}.\n",
      "COFACTOR: Name=heme; Xref=ChEBI:CHEBI:30413;   Evidence={ECO:0000269|PubMed:12094731};\n",
      "BIOPHYSICOCHEMICAL PROPERTIES: pH dependence:   Active from pH 6.0 to 10.0.; Temperature dependence:   Activity decreases at temperatures above 55 degrees Celsius.;\n",
      "SUBUNIT: Homodimer. {ECO:0000269|PubMed:12094731}.\n",
      "INDUCTION: Constitutively expressed. Inducible by oxidative stress in the exponential phase of bacterial growth. {ECO:0000269|PubMed:12094731}.\n",
      "MASS SPECTROMETRY: Mass=55417; Method=MALDI; Range=1-?; Evidence={ECO:0000269|PubMed:15177292};\n",
      "SIMILARITY: Belongs to the catalase family. {ECO:0000305}.\n",
      "['Direct protein sequencing', 'Heme', 'Hydrogen peroxide', 'Iron', 'Metal-binding', 'Oxidoreductase', 'Peroxidase']\n",
      "THCLTTAAGAPVARFSTVAGERGAADAERDIRRLFSYGDAARRLGVNHQHIPVNAPR\n",
      "57\n",
      "2\n",
      "\n",
      "CATALYTIC ACTIVITY: Catechol + O(2) = cis,cis-muconate. {ECO:0000305}.\n",
      "COFACTOR: Name=Fe(3+); Xref=ChEBI:CHEBI:29034; Note=Binds 1 Fe(3+) ion per subunit.;\n",
      "PATHWAY: Aromatic compound metabolism; beta-ketoadipate pathway; 5-oxo-4,5-dihydro-2-furylacetate from catechol: step 1/3.\n",
      "INDUCTION: By phenol. {ECO:0000269|Ref.1, ECO:0000305}.\n",
      "SIMILARITY: Belongs to the intradiol ring-cleavage dioxygenase family. {ECO:0000305}.\n",
      "['Aromatic hydrocarbons catabolism', 'Dioxygenase', 'Direct protein sequencing', 'Iron', 'Oxidoreductase']\n",
      "MNXQQIDMLVKQLN\n",
      "14\n",
      "1\n",
      "\n",
      "FUNCTION: May affect the post-translational activation and/or assembly of the oligomeric structure of RuBisCO.\n",
      "SIMILARITY: Belongs to the CbbQ/NirQ/NorQ/GpvN family. {ECO:0000305}.\n",
      "['ATP-binding', 'Nucleotide-binding']\n",
      "MSDIDRNQFLIDHEPYYRPVSNEVALYEAAYAARMPVMLKGPTGCGKTRFVEYMAWKLGKPLITVACNEDMTAS\n",
      "74\n",
      "2\n",
      "\n",
      "FUNCTION: Required for the export of heme to the periplasm for the biogenesis of c-type cytochromes.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane {ECO:0000305}; Multi- pass membrane protein {ECO:0000305}.\n",
      "SIMILARITY: Belongs to the CcmC/CycZ/HelC family. {ECO:0000305}.\n",
      "['Cell inner membrane', 'Cell membrane', 'Cytochrome c-type biogenesis', 'Membrane', 'Transmembrane', 'Transport']\n",
      "EFLRPLFIMAIAFTLLFFTLHIMAMRNEIWRRRIAAQRRLAARMASREE\n",
      "49\n",
      "2\n",
      "\n",
      "FUNCTION: Required for the export of heme to the periplasm for the biogenesis of c-type cytochromes. {ECO:0000305}.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane {ECO:0000305}; Single- pass membrane protein {ECO:0000305}.\n",
      "SIMILARITY: Belongs to the CcmD/CycX/HelD family. {ECO:0000305}.\n",
      "['Cell inner membrane', 'Cell membrane', 'Complete proteome', 'Cytochrome c-type biogenesis', 'Membrane', 'Reference proteome', 'Transmembrane', 'Transmembrane helix', 'Transport']\n",
      "MIMSLGPYASFIVTSYAAAALVVAILIGWIATDYRSQTRRLRDLDRSGITRRSGRSAMDRP\n",
      "61\n",
      "1\n",
      "\n",
      "FUNCTION: Required for the export of heme to the periplasm for the biogenesis of c-type cytochromes. {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane {ECO:0000305}; Single- pass membrane protein {ECO:0000305}.\n",
      "SIMILARITY: Belongs to the CcmD/CycX/HelD family. {ECO:0000305}.\n",
      "['Cell inner membrane', 'Cell membrane', 'Complete proteome', 'Cytochrome c-type biogenesis', 'Membrane', 'Transmembrane', 'Transmembrane helix', 'Transport']\n",
      "MTPAFASWNEFFAMGGYAFFVWLAVVMTVIPLVVLVVHSVMQHRAILRGVAQQRAREARLRAAQQQEAA\n",
      "69\n",
      "3\n",
      "\n",
      "FUNCTION: Required for the export of heme to the periplasm for the biogenesis of c-type cytochromes. {ECO:0000305}.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane {ECO:0000305}; Single- pass membrane protein {ECO:0000305}.\n",
      "SIMILARITY: Belongs to the CcmD/CycX/HelD family. {ECO:0000305}.\n",
      "SEQUENCE CAUTION: Sequence=CAA45064.1; Type=Erroneous initiation; Evidence={ECO:0000305};\n",
      "['Cell inner membrane', 'Cell membrane', 'Complete proteome', 'Cytochrome c-type biogenesis', 'Membrane', 'Reference proteome', 'Transmembrane', 'Transmembrane helix', 'Transport']\n",
      "MMPEFGKYAVTILASWGATLVLLAGLIAATLIRGAQVKRALKAQEERMKNDG\n",
      "52\n",
      "1\n",
      "\n",
      "FUNCTION: Immunity protein component of a toxin-immunity protein module, which functions as a cellular contact-dependent growth inhibition (CDI) system. CDI modules allow bacteria to communicate with and inhibit the growth of closely related neighboring bacteria in a contact-dependent fashion. Protects cells against the CDI activity of CdiA, its cognate toxin protein, but not against non-cognate CdiA from E.coli strains 563, EC93 or D.dadantii strain 3937. {ECO:0000269|PubMed:21085179}.\n",
      "SUBUNIT: Probably interacts with cognate toxin CdiA. {ECO:0000305|PubMed:21085179}.\n",
      "SUBCELLULAR LOCATION: Membrane {ECO:0000255}; Multi-pass membrane protein {ECO:0000255}.\n",
      "['Complete proteome', 'Membrane', 'Reference proteome', 'Transmembrane', 'Transmembrane helix']\n",
      "MKNLLYVVLLMAVCILGLLIVGTIFYLFLEVFMYFYVNAPISLESFQFTRLLKMSIYGGGILGLGIGLLRIFKIKGF\n",
      "77\n",
      "2\n",
      "\n",
      "FUNCTION: Activates the cell division inhibited by chromosomal DNA over-replication. {ECO:0000255|HAMAP-Rule:MF_01580}.\n",
      "SIMILARITY: Belongs to the CedA family. {ECO:0000255|HAMAP- Rule:MF_01580}.\n",
      "['Cell cycle', 'Cell division', 'Complete proteome', 'DNA-binding', 'Reference proteome']\n",
      "MMKPLRQQNRPVISYVPRVEPAPPEHAIKMDAFRDVWILRGKYVAFILMGEAFQRSPAFTVPESAQRWANQVRQENEIAE\n",
      "80\n",
      "2\n",
      "\n",
      "FUNCTION: Transfers a methylene group from S-adenosyl-L-methionine to the cis double bond of an unsaturated fatty acid chain resulting in the replacement of the double bond with a methylene bridge. {ECO:0000250}.\n",
      "CATALYTIC ACTIVITY: S-adenosyl-L-methionine + phospholipid olefinic fatty acid = S-adenosyl-L-homocysteine + phospholipid cyclopropane fatty acid.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the CFA/CMAS family. {ECO:0000305}.\n",
      "['Cytoplasm', 'Lipid biosynthesis', 'Lipid metabolism', 'Methyltransferase', 'S-adenosyl-L-methionine', 'Transferase']\n",
      "ISHIAEASESRFVMEDWHNFGSDYDKTLMAWHERFNQAWPELSSRYSATFRRMFNYYLCACAGAFRARDIELWQVLFSRGVEGGIRVYR\n",
      "89\n",
      "2\n",
      "\n",
      "FUNCTION: Component of the cytoplasmic filaments that run the length of the organism just underneath the cytoplasmic membrane.\n",
      "SUBCELLULAR LOCATION: Cytoplasm. Note=An array of 4 to 6 filaments lie in close apposition to the inner membrane and are always localized directly underneath the corresponding group of periplasmic flagella.\n",
      "['Cytoplasm', 'Direct protein sequencing']\n",
      "AILELPQSPNVFHPEKPSAVG\n",
      "21\n",
      "2\n",
      "\n",
      "FUNCTION: Binds to Cpn60 in the presence of Mg-ATP and suppresses the ATPase activity of the latter. {ECO:0000250}.\n",
      "SUBUNIT: Heptamer of 7 subunits arranged in a ring. {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the GroES chaperonin family. {ECO:0000305}.\n",
      "['Chaperone', 'Cytoplasm', 'Direct protein sequencing']\n",
      "MKITPLGDNVVIKKLLATA\n",
      "19\n",
      "2\n",
      "\n",
      "FUNCTION: Binds to Cpn60 in the presence of Mg-ATP and suppresses the ATPase activity of the latter. {ECO:0000255|HAMAP- Rule:MF_00580}.\n",
      "SUBUNIT: Heptamer of 7 subunits arranged in a ring. {ECO:0000255|HAMAP-Rule:MF_00580}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000255|HAMAP-Rule:MF_00580}.\n",
      "SIMILARITY: Belongs to the GroES chaperonin family. {ECO:0000255|HAMAP-Rule:MF_00580}.\n",
      "SEQUENCE CAUTION: Sequence=BAC17411.1; Type=Erroneous initiation; Evidence={ECO:0000305};\n",
      "['Chaperone', 'Complete proteome', 'Cytoplasm', 'Reference proteome']\n",
      "MANVNIKPLEDKILVQINEAETTTASGLVIPDSAKEKPQEATVIAVGPGRFDEKGERIPLDIKEDDVVIFSRYGGTEIKFDGVEYLLLSARDILAIVEK\n",
      "99\n",
      "1\n",
      "\n",
      "FUNCTION: Prevents misfolding and promotes the refolding and proper assembly of unfolded polypeptides generated under stress conditions. {ECO:0000250}.\n",
      "SUBUNIT: Oligomer of 14 subunits composed of two stacked rings of 7 subunits. {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm.\n",
      "INDUCTION: By heat shock and ethanol.\n",
      "SIMILARITY: Belongs to the chaperonin (HSP60) family. {ECO:0000305}.\n",
      "['ATP-binding', 'Chaperone', 'Cytoplasm', 'Direct protein sequencing', 'Nucleotide-binding', 'Stress response']\n",
      "SAKDVKFGDSARSMMIAGVNVIAD\n",
      "24\n",
      "2\n",
      "\n",
      "FUNCTION: Might be a regulator of the sodium-potassium/proton antiporter ChaA. {ECO:0000303|PubMed:15306028}.\n",
      "SUBUNIT: Monomer. {ECO:0000269|PubMed:15306028}.\n",
      "SIMILARITY: Belongs to the ChaB family. {ECO:0000305}.\n",
      "['3D-structure', 'Complete proteome', 'Transcription', 'Transcription regulation']\n",
      "MPYKTKSDLPESVKHVLPSHAQDIYKEAFNSAWDQYKDKEDRRDDASREETAHKVAWAAVKHEYAKGDDDKWHKKS\n",
      "76\n",
      "1\n",
      "\n",
      "FUNCTION: Able to cleave chitin oligomers from N=3 to 6.\n",
      "CATALYTIC ACTIVITY: Random hydrolysis of N-acetyl-beta-D- glucosaminide (1->4)-beta-linkages in chitin and chitodextrins.\n",
      "SUBUNIT: Homodimer, but homotrimers and homotetramers could be observed for the 20.5 and 30 kDa chitinases.\n",
      "PTM: The 70 kDa chitinase is probably the precursor protein of the 30 and 20.5 kDa chitinases.\n",
      "SIMILARITY: Belongs to the glycosyl hydrolase 18 family. Chitinase class II subfamily. {ECO:0000305}.\n",
      "['Carbohydrate metabolism', 'Chitin degradation', 'Direct protein sequencing', 'Glycosidase', 'Hydrolase', 'Polysaccharide degradation']\n",
      "XTSATATYAKTQDWGSCFEGKWTIKNTAACSSYPSWVAGRSYAAGDIVYYTDXGYTDLPVSRQKMCQNGMVTNC\n",
      "74\n",
      "3\n",
      "\n",
      "CATALYTIC ACTIVITY: Choline + 2 O(2) + H(2)O = betaine + 2 H(2)O(2).\n",
      "PATHWAY: Amine and polyamine biosynthesis; betaine biosynthesis via choline pathway; betaine from choline: step 1/1.\n",
      "['Direct protein sequencing', 'Oxidoreductase']\n",
      "DNPNHSR\n",
      "7\n",
      "1\n",
      "\n",
      "FUNCTION: Reduces chlorate and bromate. {ECO:0000269|PubMed:12730181}.\n",
      "CATALYTIC ACTIVITY: AH(2) + chlorate = A + H(2)O + chlorite. {ECO:0000269|PubMed:12730181}.\n",
      "COFACTOR: Name=[4Fe-4S] cluster; Xref=ChEBI:CHEBI:49883;   Evidence={ECO:0000250}; Note=Binds 1 [4Fe-4S] cluster. {ECO:0000250};\n",
      "COFACTOR: Name=Mo-bis(molybdopterin guanine dinucleotide);   Xref=ChEBI:CHEBI:60539; Evidence={ECO:0000305|PubMed:12730181}; Note=Binds 1 molybdenum-bis(molybdopterin guanine dinucleotide) (Mo-bis-MGD) cofactor per subunit. {ECO:0000305|PubMed:12730181};\n",
      "BIOPHYSICOCHEMICAL PROPERTIES: Kinetic parameters:   KM=159 uM for chlorate;   Vmax=51 umol/min/mg enzyme; pH dependence:   Optimum pH is 7.5.;\n",
      "SUBUNIT: Heterotrimer of alpha, beta and gamma subunits. {ECO:0000269|PubMed:12730181}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000269|PubMed:12730181}.\n",
      "['4Fe-4S', 'Cytoplasm', 'Direct protein sequencing', 'Iron', 'Iron-sulfur', 'Metal-binding', 'Molybdenum', 'Oxidoreductase']\n",
      "LNMLEPVGETLASEYPYXK\n",
      "19\n",
      "3\n",
      "\n",
      "FUNCTION: Electron transfer subunit of the chlorate reductase. {ECO:0000250}.\n",
      "COFACTOR: Name=[3Fe-4S] cluster; Xref=ChEBI:CHEBI:21137;   Evidence={ECO:0000250}; Note=Binds 1 [3Fe-4S] cluster. {ECO:0000250};\n",
      "COFACTOR: Name=[4Fe-4S] cluster; Xref=ChEBI:CHEBI:49883;   Evidence={ECO:0000250}; Note=Binds 3 [4Fe-4S] clusters. {ECO:0000250};\n",
      "BIOPHYSICOCHEMICAL PROPERTIES: Kinetic parameters:   KM=159 uM for chlorate;   Vmax=51 umol/min/mg enzyme; pH dependence:   Optimum pH is 7.5.;\n",
      "SUBUNIT: Heterotrimer of alpha, beta and gamma subunits. {ECO:0000269|PubMed:12730181}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000269|PubMed:12730181}.\n",
      "['3Fe-4S', '4Fe-4S', 'Cytoplasm', 'Direct protein sequencing', 'Electron transport', 'Iron', 'Iron-sulfur', 'Metal-binding', 'Transport']\n",
      "TVKXQLSMVLDLNKEIGGQTXTAA\n",
      "24\n",
      "3\n",
      "\n",
      "FUNCTION: May transfer electrons to the iron-sulfur centers of the beta subunit of chlorate reductase. {ECO:0000250}.\n",
      "COFACTOR: Name=heme b; Xref=ChEBI:CHEBI:60344; Evidence={ECO:0000305}; Note=Binds 1 heme b (iron(II)-protoporphyrin IX) group per subunit. {ECO:0000305};\n",
      "BIOPHYSICOCHEMICAL PROPERTIES: Kinetic parameters:   KM=159 uM for chlorate;   Vmax=51 umol/min/mg enzyme; pH dependence:   Optimum pH is 7.5.;\n",
      "SUBUNIT: Heterotrimer of alpha, beta and gamma subunits. {ECO:0000269|PubMed:12730181}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000269|PubMed:12730181}.\n",
      "['Cytoplasm', 'Direct protein sequencing', 'Electron transport', 'Heme', 'Iron', 'Metal-binding', 'Transport']\n",
      "EXSEQNPNILEIKPGDTVKVXT\n",
      "22\n",
      "2\n",
      "\n",
      "FUNCTION: Modifies the set of genes regulated by H-NS; Hha and cnu (YdgT) increase the number of genes bound by H-NS/StpA and may also modulate the oligomerization of the H-NS/StpA-complex on DNA. The complex formed with H-NS binds to the specific 26-bp cnb site in the origin of replication oriC. {ECO:0000250|UniProtKB:P64467}.\n",
      "SUBUNIT: Forms complexes with both H-NS and StpA. {ECO:0000250|UniProtKB:P64467}.\n",
      "SIMILARITY: Belongs to the Hha/YmoA/Cnu family. {ECO:0000305}.\n",
      "['Complete proteome', 'DNA-binding', 'Transcription', 'Transcription regulation']\n",
      "MTVQDYLLKFRKISSLESLEKLYDHLNYTLTDDQELINMYRAADHRRAELVSGGRLFDLGQVPKSVWHYVQ\n",
      "71\n",
      "2\n",
      "\n",
      "FUNCTION: Part of a major quorum-sensing system that regulates the development of genetic competence. Acts through the activation of the two-component regulatory system ComP/ComA composed of a sensor histidine kinase, ComP, and a response regulator, ComA, that regulates directly the transcription of over 20 genes. Transport through the membrane may involve Spo0K. Under certain conditions plays a role in sporulation.\n",
      "SUBCELLULAR LOCATION: Secreted.\n",
      "PTM: Isoprenylation of Trp-56 is essential for activity. The type of isoprenyl derivative differs among the different pherotypes and depends on ComX primary sequence. ComQ is required for the production of ComX and may be responsible alone for its complete maturation. {ECO:0000269|PubMed:12067344}.\n",
      "MISCELLANEOUS: The DNA sequences encoding ComQ, ComX and the N- terminal two-thirds of ComP show a striking polymorphism, which determines the specificity of the quorum-sensing system in the different pherotypes of Bacillus. In ComX, the sole conserved residue is the modified tryptophan essential for the activity.\n",
      "['Competence', 'Lipoprotein', 'Pheromone', 'Prenylation', 'Secreted', 'Sporulation']\n",
      "MKQDMIDYLMKNPQVLTKLENGEASLIGIPDKLIPSIVDIFNKKMTLSKKCKGIFWEQ\n",
      "58\n",
      "1\n",
      "\n",
      "FUNCTION: Part of a cation-transporting system which is associated with copper export out of the H.pylori cells.\n",
      "INTERACTION: P56039:rplN; NbExp=3; IntAct=EBI-7497456, EBI-7497678;\n",
      "MISCELLANEOUS: Copper (2+) binding can oxidyze the Cys residues at the metal-binding site, leading to the formation of a disulfide bond.\n",
      "['3D-structure', 'Complete proteome', 'Copper', 'Disulfide bond', 'Metal-binding', 'Reference proteome']\n",
      "MKATFQVPSITCNHCVDKIEKFVGEIEGVSFIDVSVEKKSVVVEFDAPATQDLIKEALLDAGQEVV\n",
      "66\n",
      "2\n",
      "\n",
      "FUNCTION: Chaperone that serves for the intracellular sequestration and transport of Cu(+). Delivers Cu(+) to the copper-transporting ATPase CopA.\n",
      "SUBUNIT: Monomer in the absence of copper. Homodimer in the presence of copper ions. Forms a heterodimer (electrostatic interactions) with CopA during the transfer of Cu(+). {ECO:0000269|PubMed:12238948, ECO:0000269|PubMed:12600214}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm.\n",
      "INDUCTION: By Cu(2+).\n",
      "MISCELLANEOUS: The copZA operon is activated by CueR and indirectly repressed by YfmP.\n",
      "MISCELLANEOUS: Cu(+) ion is always at least three-coordinated. Physiological thiol may be needed to complete the Cu(+) coordination sphere in order to prevent homodimer (dead-end products) formation between 2 CopZ.\n",
      "['3D-structure', 'Chaperone', 'Complete proteome', 'Copper', 'Cytoplasm', 'Metal-binding', 'Reference proteome']\n",
      "MEQKTLQVEGMSCQHCVKAVETSVGELDGVSAVHVNLEAGKVDVSFDADKVSVKDIADAIEDQGYDVAK\n",
      "69\n",
      "4\n",
      "\n",
      "FUNCTION: Acts as a copper chaperone by delivering 2 Cu(+) ions to CopY Zn(2+)-bound form. This transfer results in displacement of zinc and dissociation of CopY from the promoter, allowing transcription of the copYZAB operon. {ECO:0000269|PubMed:10069368, ECO:0000269|PubMed:11980486, ECO:0000269|PubMed:7876197}.\n",
      "SUBUNIT: Monomer in the absence of copper. Homodimer or homooligomer in the presence of copper ions. Interacts with the copper ATPase CopA. Interacts with CopY via a charge-based interaction. {ECO:0000269|PubMed:10428839, ECO:0000269|PubMed:11594769, ECO:0000269|PubMed:11980486}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000305}.\n",
      "INDUCTION: By Cu(+) and Cu(2+). {ECO:0000269|PubMed:10362527}.\n",
      "MISCELLANEOUS: Controlled by copper-induced proteolysis.\n",
      "['3D-structure', 'Chaperone', 'Complete proteome', 'Copper', 'Cytoplasm', 'Metal-binding', 'Reference proteome']\n",
      "MKQEFSVKGMSCNHCVARIEEAVGRISGVKKVKVQLKKEKAVVKFDEANVQATEICQAINELGYQAEVI\n",
      "69\n",
      "5\n",
      "\n",
      "FUNCTION: Chaperone that serves for the intracellular sequestration and transport of Cu(+). Delivers Cu(+) to the copper-exporting P-type ATPase A (CopA) (By similarity). {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "['Chaperone', 'Copper', 'Cytoplasm', 'Metal-binding']\n",
      "MSQEILNVEGMSCGHCKSAVESALNNIDGVTSADVNLENGQVSVQYDDSKVAVSQMKDAIEDQGYDVV\n",
      "68\n",
      "1\n",
      "\n",
      "FUNCTION: Not required for enzymatic activity or proton pumping of the cytochrome c oxidase complex. {ECO:0000269|PubMed:9038156}.\n",
      "CATALYTIC ACTIVITY: 4 ferrocytochrome c + O(2) + 4 H(+) = 4 ferricytochrome c + 2 H(2)O.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass membrane protein.\n",
      "['3D-structure', 'Cell inner membrane', 'Cell membrane', 'Direct protein sequencing', 'Membrane', 'Oxidoreductase', 'Transmembrane', 'Transmembrane helix']\n",
      "MASHHEITDHKHGEMDIRHQQATFAGFIKGATWVSILSIAVLVFLALANS\n",
      "50\n",
      "3\n",
      "\n",
      "CATALYTIC ACTIVITY: 4 ferrocytochrome c + O(2) + 4 H(+) = 4 ferricytochrome c + 2 H(2)O.\n",
      "SUBCELLULAR LOCATION: Cell membrane; Single-pass membrane protein.\n",
      "MASS SPECTROMETRY: Mass=3794.6; Method=MALDI; Range=1-34; Evidence={ECO:0000269|PubMed:11152118};\n",
      "['3D-structure', 'Cell membrane', 'Complete proteome', 'Direct protein sequencing', 'Electron transport', 'Formylation', 'Hydrogen ion transport', 'Ion transport', 'Membrane', 'Oxidoreductase', 'Reference proteome', 'Respiratory chain', 'Transmembrane', 'Transmembrane helix', 'Transport']\n",
      "MEEKPKGALAVILVLTLTILVFWLGVYAVFFARG\n",
      "34\n",
      "1\n",
      "\n",
      "FUNCTION: Involved in cell division. {ECO:0000255|HAMAP- Rule:MF_00631}.\n",
      "SUBCELLULAR LOCATION: Cell membrane {ECO:0000255|HAMAP- Rule:MF_00631}; Multi-pass membrane protein {ECO:0000255|HAMAP- Rule:MF_00631}.\n",
      "SIMILARITY: Belongs to the CrgA family. {ECO:0000255|HAMAP- Rule:MF_00631}.\n",
      "['Cell cycle', 'Cell division', 'Cell membrane', 'Membrane', 'Transmembrane', 'Transmembrane helix']\n",
      "MSRKKHESSENNPVWFPTIMFGLMGTGAVWMVLFYISNGALPLPAVGTWNILIAFGIIMAGFAMMSRWK\n",
      "69\n",
      "6\n",
      "\n",
      "FUNCTION: Methyltransferase that mediates the O-methylation of 1- hydroxy carotenoids. Converts hydroxyneurosporene to methoxyneurosporene or demethylspheroidene to spheroidene. Also able to produce spirilloxanthin. {ECO:0000269|PubMed:12664193, ECO:0000269|PubMed:7358679}.\n",
      "CATALYTIC ACTIVITY: S-adenosyl-L-methionine + demethylspheroidene = S-adenosyl-L-homocysteine + spheroidene.\n",
      "PATHWAY: Carotenoid biosynthesis; spheroidene biosynthesis.\n",
      "DISRUPTION PHENOTYPE: Cells accumulate demethylspheroidene as their major carotenoid during anaerobic growth and demethylspheroidenone when grown in the presence of oxygen. {ECO:0000269|PubMed:7358679}.\n",
      "SIMILARITY: Belongs to the class I-like SAM-binding methyltransferase superfamily. Cation-independent O- methyltransferase family. {ECO:0000255|PROSITE-ProRule:PRU01020}.\n",
      "['Carotenoid biosynthesis', 'Chlorophyll biosynthesis', 'Methyltransferase', 'Photosynthesis', 'S-adenosyl-L-methionine', 'Transferase']\n",
      "ALPPGGRLIISEAMAGGAKPDRACDVYFAFYTMAMSSGRTRSPEEIKQMLEKAGFTKVSKPRTLRPFITSVIEAERG\n",
      "77\n",
      "3\n",
      "\n",
      "CATALYTIC ACTIVITY: A short-chain (3S)-3-hydroxyacyl-CoA = a short-chain trans-2-enoyl-CoA + H(2)O. {ECO:0000250|UniProtKB:P52046}.\n",
      "PATHWAY: Lipid metabolism; butanoate metabolism.\n",
      "MISCELLANEOUS: On the 2D-gel the determined pI of this unknown protein is 5.2, its MW is 30.9 kDa.\n",
      "SIMILARITY: Belongs to the enoyl-CoA hydratase/isomerase family. {ECO:0000305}.\n",
      "['Direct protein sequencing', 'Fatty acid metabolism', 'Lipid metabolism', 'Lyase']\n",
      "MGFNGGILEGEEXKAVVTINXP\n",
      "22\n",
      "3\n",
      "\n",
      "SUBCELLULAR LOCATION: Cell membrane {ECO:0000305}; Multi-pass membrane protein {ECO:0000305}.\n",
      "DEVELOPMENTAL STAGE: Maximally expressed in early stationary phase in medium containing glucose and glutamine.\n",
      "['Cell membrane', 'Complete proteome', 'Membrane', 'Reference proteome', 'Transmembrane', 'Transmembrane helix']\n",
      "MITKAVFALFFPFMLVVLFTRVTFNHYVAIALTAALLFASYLKGYTETYFIVGLDVVSLVAGGLYMAKKAAEKKEE\n",
      "76\n",
      "2\n",
      "\n",
      "INDUCTION: By phosphate starvation, via the alternative sigma factor sigma-B. {ECO:0000269|PubMed:11988534}.\n",
      "SIMILARITY: Belongs to the UPF0337 (CsbD) family. {ECO:0000305}.\n",
      "['Complete proteome', 'Reference proteome', 'Stress response']\n",
      "MGNDSVKDKMKGGFNKAKGEVKDKVGDMADRTDMQAEGKKDKAKGEIQKDIGKAKDKFSDKD\n",
      "62\n",
      "2\n",
      "\n",
      "FUNCTION: Component of the photosynthetic apparatus. The light harvesting B740 complex binds bacteriochlorophyll e.\n",
      "SUBCELLULAR LOCATION: Chlorosome, chlorosome envelope.\n",
      "SIMILARITY: Belongs to the BChl C/E-binding protein family. {ECO:0000305}.\n",
      "['Bacteriochlorophyll', 'Chlorophyll', 'Chlorosome', 'Chromophore', 'Direct protein sequencing', 'Electron transport', 'Magnesium', 'Metal-binding', 'Photosynthesis', 'Transport']\n",
      "MNGGGVFTDILAASGRIFEVMVEGHWATVGYLFDSLGKGVSRINQNAYGNM\n",
      "51\n",
      "1\n",
      "\n",
      "FUNCTION: Component of the photosynthetic apparatus. The light harvesting B740 complex binds bacteriochlorophyll c.\n",
      "SUBCELLULAR LOCATION: Chlorosome, chlorosome envelope.\n",
      "SIMILARITY: Belongs to the BChl C/E-binding protein family. {ECO:0000305}.\n",
      "['Bacteriochlorophyll', 'Chlorophyll', 'Chlorosome', 'Chromophore', 'Direct protein sequencing', 'Electron transport', 'Magnesium', 'Metal-binding', 'Photosynthesis', 'Transport']\n",
      "MSGGGVFTDILAAAGRIFEVMVEGHWETVGMLFDSLGKGTMRINRNAY\n",
      "48\n",
      "1\n",
      "\n",
      "FUNCTION: Component of the photosynthetic apparatus. The light harvesting B740 complex binds bacteriochlorophyll c.\n",
      "SUBCELLULAR LOCATION: Chlorosome, chlorosome envelope.\n",
      "SIMILARITY: Belongs to the BChl C/E-binding protein family. {ECO:0000305}.\n",
      "['Bacteriochlorophyll', 'Chlorophyll', 'Chlorosome', 'Chromophore', 'Direct protein sequencing', 'Electron transport', 'Formylation', 'Magnesium', 'Metal-binding', 'Photosynthesis', 'Transport']\n",
      "MSGGGVFTDILAAAGRIFEVMVEGHWETVGMLFDSLGKGTMRINRNAYGSMGGGTSLRGS\n",
      "60\n",
      "2\n",
      "\n",
      "FUNCTION: Component of the photosynthetic apparatus. The light harvesting B740 complex binds bacteriochlorophyll c.\n",
      "SUBCELLULAR LOCATION: Chlorosome, chlorosome envelope.\n",
      "SIMILARITY: Belongs to the BChl C/E-binding protein family. {ECO:0000305}.\n",
      "['Bacteriochlorophyll', 'Chlorophyll', 'Chlorosome', 'Chromophore', 'Complete proteome', 'Direct protein sequencing', 'Electron transport', 'Magnesium', 'Metal-binding', 'Photosynthesis', 'Reference proteome', 'Transport']\n",
      "MSGGGVFTDILAAAGRIFEVMVEGHWETVGMLFDSLGKGTMRINRNAYGNLGGGGGSLRGSSPEVSGFAVPTKAVESKFAK\n",
      "81\n",
      "3\n",
      "\n",
      "FUNCTION: Component of the photosynthetic apparatus which may bind the chlorosome to the bacteriochlorophyll a protein monolayer.\n",
      "SUBCELLULAR LOCATION: Chlorosome, chlorosome envelope.\n",
      "SIMILARITY: Belongs to the CsmB/CsmF family. {ECO:0000305}.\n",
      "SEQUENCE CAUTION: Sequence=AAB02562.1; Type=Erroneous initiation; Evidence={ECO:0000305}; Sequence=AAM73271.1; Type=Erroneous initiation; Evidence={ECO:0000305};\n",
      "['Bacteriochlorophyll', 'Chlorophyll', 'Chlorosome', 'Chromophore', 'Complete proteome', 'Photosynthesis', 'Reference proteome', 'Repeat']\n",
      "MSNGTNIDVAGAINTLAETFGKLFQMQIDVANTALKALADVAEPLGKTATDLIGSFTGAATQVLQSVSSAIAPKK\n",
      "75\n",
      "2\n",
      "\n",
      "FUNCTION: Component of the photosynthetic apparatus which may bind the chlorosome to the bacteriochlorophyll a protein monolayer.\n",
      "SUBCELLULAR LOCATION: Chlorosome, chlorosome envelope.\n",
      "PTM: A short form lacking first Ser is produced.\n",
      "SIMILARITY: Belongs to the CsmB/CsmF family. {ECO:0000305}.\n",
      "['Bacteriochlorophyll', 'Chlorophyll', 'Chlorosome', 'Chromophore', 'Direct protein sequencing', 'Photosynthesis', 'Repeat']\n",
      "SNGTNIDVAGAINTLTETFGKLFQMQLDVANTALKALADVAEPLGKTATDLVGNFAGAATQILQSVSAAIAPKK\n",
      "74\n",
      "4\n",
      "\n",
      "FUNCTION: Component of the photosynthetic apparatus which may bind the chlorosome to the bacteriochlorophyll a protein monolayer.\n",
      "SUBCELLULAR LOCATION: Chlorosome, chlorosome envelope.\n",
      "SIMILARITY: Belongs to the CsmB/CsmF family. {ECO:0000305}.\n",
      "['Bacteriochlorophyll', 'Chlorophyll', 'Chlorosome', 'Chromophore', 'Direct protein sequencing', 'Photosynthesis']\n",
      "SNGSNIDVAGAVNTLVETAGKLFQ\n",
      "24\n",
      "4\n",
      "\n",
      "FUNCTION: Acts as a pheromone, induces cells to develop competence for genetic transformation.\n",
      "SUBCELLULAR LOCATION: Secreted.\n",
      "SIMILARITY: Belongs to the ComC family. {ECO:0000305}.\n",
      "['Competence', 'Direct protein sequencing', 'Pheromone', 'Secreted']\n",
      "MKNTVKLEQFVALKEKDLQKIKGGEMRLSKFFRDFILQRKK\n",
      "41\n",
      "2\n",
      "\n",
      "FUNCTION: Acts as a pheromone, induces cells to develop competence for genetic transformation.\n",
      "SUBCELLULAR LOCATION: Secreted.\n",
      "SIMILARITY: Belongs to the ComC family. {ECO:0000305}.\n",
      "['Competence', 'Pheromone', 'Secreted']\n",
      "MKNTVKLEQFVALKEKDLQKIKGGESRLPKIRFDFIFPRKK\n",
      "41\n",
      "3\n",
      "\n",
      "FUNCTION: Acts as a pheromone, induces cells to develop competence for genetic transformation.\n",
      "SUBCELLULAR LOCATION: Secreted.\n",
      "SIMILARITY: Belongs to the ComC family. {ECO:0000305}.\n",
      "['Competence', 'Complete proteome', 'Pheromone', 'Secreted']\n",
      "MKNTVKLEQFVALKEKDLQKIKGGEMRISRIILDFLFLRKK\n",
      "41\n",
      "4\n",
      "\n",
      "SUBUNIT: Homodimer. {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "INDUCTION: In response to low temperature. {ECO:0000250}.\n",
      "['Activator', 'Cytoplasm', 'DNA-binding', 'Stress response', 'Transcription', 'Transcription regulation']\n",
      "EKGFGFISPTDGSKDVFVHFSAIQTPGFKTLDEGQRVEFTIEQGQK\n",
      "46\n",
      "2\n",
      "\n",
      "FUNCTION: Can bind to ATTGG and CCAAT motifs (Y-box motifs) of single-stranded oligonucleotides. {ECO:0000250}.\n",
      "SUBUNIT: Homodimer. {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "['Activator', 'Complete proteome', 'Cytoplasm', 'DNA-binding', 'Reference proteome', 'Transcription', 'Transcription regulation']\n",
      "MAVTGQVKWFNNEKGFGFIEVPGENDVFVHFSAIETDGFKSLEEGQKVSFEIEEGNRGPQAKNVIKL\n",
      "67\n",
      "3\n",
      "\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "SEQUENCE CAUTION: Sequence=CAE32745.1; Type=Erroneous initiation; Evidence={ECO:0000305};\n",
      "['Activator', 'Complete proteome', 'Cytoplasm', 'DNA-binding', 'Transcription', 'Transcription regulation']\n",
      "METGVVKWFNAEKGYGFITPEAGGKDLFAHFSEIQANGFKSLEENQRVSFVTAMGPKGPQATKIQIL\n",
      "67\n",
      "4\n",
      "\n",
      "SUBUNIT: Homodimer. {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "INDUCTION: In response to low temperature. {ECO:0000250}.\n",
      "['Activator', 'Cytoplasm', 'DNA-binding', 'Stress response', 'Transcription', 'Transcription regulation']\n",
      "DKGFGFITPDDGSKDVFVHFSAIQNDGYKSLDEGQKVSFTIESGAK\n",
      "46\n",
      "3\n",
      "\n",
      "FUNCTION: Binds to and stimulates the transcription of the CCAAT- containing, cold-shock-inducible promoters of the H-NS and GyrA proteins. Binds also to the inverted repeat 5'-ATTGG-3' (By similarity). {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "INDUCTION: In response to low temperature. {ECO:0000250}.\n",
      "['Activator', 'Complete proteome', 'Cytoplasm', 'DNA-binding', 'Stress response', 'Transcription', 'Transcription regulation']\n",
      "MSGKMTGIVKWFNADKGFGFITPDDGSKDVFVHFSAIQNDGYKSLDEGQKVSFTIESGAKGPAAGNVTSL\n",
      "70\n",
      "2\n",
      "\n",
      "FUNCTION: Binds to and stimulates the transcription of the CCAAT- containing, cold-shock-inducible promoters of the H-NS and GyrA proteins. Binds also to the inverted repeat 5'-ATTGG-3'. {ECO:0000269|PubMed:1961761}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000269|PubMed:2404279}.\n",
      "INDUCTION: In response to low temperature (at protein level) (PubMed:2404279, PubMed:8898389). {ECO:0000269|PubMed:2404279, ECO:0000269|PubMed:8898389}.\n",
      "['3D-structure', 'Activator', 'Complete proteome', 'Cytoplasm', 'Direct protein sequencing', 'DNA-binding', 'Reference proteome', 'Stress response', 'Transcription', 'Transcription regulation']\n",
      "MSGKMTGIVKWFNADKGFGFITPDDGSKDVFVHFSAIQNDGYKSLDEGQKVSFTIESGAKGPAAGNVTSL\n",
      "70\n",
      "3\n",
      "\n",
      "SUBUNIT: Homodimer. {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "INDUCTION: In response to low temperature. {ECO:0000250}.\n",
      "['Activator', 'Complete proteome', 'Cytoplasm', 'DNA-binding', 'Stress response', 'Transcription', 'Transcription regulation']\n",
      "MEQGTVKWFNAEKGFGFIERENGDDVFVHFSAIQGDGFKSLDEGQAVTFDVEEGQRGPQAANVQKA\n",
      "66\n",
      "3\n",
      "\n",
      "FUNCTION: Involved in cold stress response and in the susceptibility to an antimicrobial peptide of human cathepsin G (CG117-136). Regulates yellowish-orange pigment production through a still unclear SigB-dependent mechanism. {ECO:0000269|PubMed:12874306, ECO:0000269|PubMed:16291691}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "INDUCTION: Up-regulated in response to low temperature.\n",
      "['Complete proteome', 'Cytoplasm', 'Reference proteome']\n",
      "MKQGTVKWFNAEKGFGFIEVEGENDVFVHFSAINQDGYKSLEEGQAVEFEVVEGDRGPQAANVVKL\n",
      "66\n",
      "4\n",
      "\n",
      "SUBUNIT: Homodimer. {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "INDUCTION: In response to low temperature. {ECO:0000250}.\n",
      "SEQUENCE CAUTION: Sequence=AAZ52383.1; Type=Erroneous initiation; Evidence={ECO:0000305};\n",
      "['Activator', 'Complete proteome', 'Cytoplasm', 'DNA-binding', 'Reference proteome', 'Stress response', 'Transcription', 'Transcription regulation']\n",
      "MAQGTVKWFNAEKGFGFISTENGQDVFAHFSAIQTNGFKTLEEGQKVAFDVEEGQRGPQAVNITKLA\n",
      "67\n",
      "5\n",
      "\n",
      "FUNCTION: Affects cell viability at low temperatures.\n",
      "SUBUNIT: Homodimer.\n",
      "SUBCELLULAR LOCATION: Cytoplasm.\n",
      "INDUCTION: In response to low temperature. {ECO:0000250}.\n",
      "['3D-structure', 'Activator', 'Cytoplasm', 'DNA-binding', 'Stress response', 'Transcription', 'Transcription regulation']\n",
      "MQRGKVKWFNNEKGYGFIEVEGGSDVFVHFTAIQGEGFKTLEEGQEVSFEIVQGNRGPQAANVVKL\n",
      "66\n",
      "5\n",
      "\n",
      "FUNCTION: Binds to the pentamer sequences ATTGG and CCAAT with highest affinity in single-stranded DNA, and also to other sequences. Has greater affinity for ATTGG than CCAAT. Can act as transcriptional activator of cold shock genes by recognizing putative ATTGG-box elements present in promoter regions of genes induced under cold shock conditions. {ECO:0000269|PubMed:8307174}.\n",
      "SUBUNIT: Homodimer. Interacts with RNA helicase CshB when cells are transcriptionally active. {ECO:0000269|PubMed:16352840}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm, nucleoid {ECO:0000269|PubMed:16352840}. Note=Shows transcription-dependent localization at subcellular sites surrounding the nucleoid.\n",
      "INDUCTION: In response to low temperature.\n",
      "DISRUPTION PHENOTYPE: A double cshB-cspB disruption mutant cannot be made. {ECO:0000269|PubMed:16352840}.\n",
      "['3D-structure', 'Activator', 'Complete proteome', 'Cytoplasm', 'Direct protein sequencing', 'DNA-binding', 'Reference proteome', 'Stress response', 'Transcription', 'Transcription regulation']\n",
      "MLEGKVKWFNSEKGFGFIEVEGQDDVFVHFSAIQGEGFKTLEEGQAVSFEIVEGNRGPQAANVTKEA\n",
      "67\n",
      "6\n",
      "\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "INDUCTION: In response to low temperature (at protein level). {ECO:0000269|PubMed:8898389}.\n",
      "['Activator', 'Complete proteome', 'Cytoplasm', 'DNA-binding', 'Reference proteome', 'Stress response', 'Transcription', 'Transcription regulation']\n",
      "MSNKMTGLVKWFNADKGFGFISPVDGSKDVFVHFSAIQNDNYRTLFEGQKVTFSIESGAKGPAAANVIITD\n",
      "71\n",
      "3\n",
      "\n",
      "FUNCTION: Affects cell viability at low temperatures.\n",
      "SUBUNIT: Homodimer.\n",
      "SUBCELLULAR LOCATION: Cytoplasm.\n",
      "INDUCTION: In response to low temperature. {ECO:0000250}.\n",
      "['Activator', 'Cytoplasm', 'DNA-binding', 'Stress response', 'Transcription', 'Transcription regulation']\n",
      "IKWFNSEKGFGFIEVEGQDDVFVHFSAIQGEGFKCLEEGQAVSFEIVEGNRGPQAAN\n",
      "57\n",
      "3\n",
      "\n",
      "SUBUNIT: Homodimer. {ECO:0000305}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "SEQUENCE CAUTION: Sequence=AAT57329.1; Type=Erroneous initiation; Evidence={ECO:0000305};\n",
      "['Activator', 'Complete proteome', 'Cytoplasm', 'DNA-binding', 'Reference proteome', 'Transcription', 'Transcription regulation']\n",
      "MQGRVKWFNAEKGFGFIEREDGDDVFVHFSAIQQDGYKSLEEGQQVEFDIVDGARGPQAANVVKL\n",
      "65\n",
      "1\n",
      "\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "MASS SPECTROMETRY: Mass=7273; Method=Electrospray; Range=2-69; Evidence={ECO:0000269|PubMed:12071744};\n",
      "['Activator', 'Complete proteome', 'Cytoplasm', 'Direct protein sequencing', 'DNA-binding', 'Reference proteome', 'Transcription', 'Transcription regulation']\n",
      "MAKIKGQVKWFNESKGFGFITPADGSKDVFVHFSAIQGNGFKTLAEGQNVEFEIQDGQKGPAAVNVTAI\n",
      "69\n",
      "7\n",
      "\n",
      "FUNCTION: Inhibits DNA replication at both initiation and elongation steps, most probably by binding to the opened, single- stranded regions at replication forks. Plays a regulatory role in chromosomal replication in nutrient-depleted cells (By similarity). {ECO:0000250}.\n",
      "SUBUNIT: Homodimer. {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "MISCELLANEOUS: Binds single-stranded DNA and RNA, but not double- stranded DNA, through hydrophobic interaction without sequence specificity, resulting in a packed structure. {ECO:0000250}.\n",
      "['Complete proteome', 'Cytoplasm', 'DNA replication inhibitor', 'DNA-binding', 'RNA-binding']\n",
      "MEKGTVKWFNNAKGFGFICPEGGGEDIFAHYSTIQMDGYRTLKAGQSVQFDVHQGPKGNHASVIVPVEVEAAVA\n",
      "74\n",
      "5\n",
      "\n",
      "FUNCTION: Can bind to ATTGG and CCAAT motifs (Y-box motifs) of single-stranded oligonucleotides. {ECO:0000250}.\n",
      "SUBUNIT: Homodimer. {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "['Activator', 'Complete proteome', 'Cytoplasm', 'DNA-binding', 'Reference proteome', 'Transcription', 'Transcription regulation']\n",
      "MTLTGKVKWFNSEKGFGFIEVEGGNDVFVHFSAITGDGFKSLDEGQEVSFEVEDGNRGPQAKNVVKL\n",
      "67\n",
      "7\n",
      "\n",
      "FUNCTION: Can bind to ATTGG and CCAAT motifs (Y-box motifs) of single-stranded oligonucleotides.\n",
      "SUBUNIT: Homodimer.\n",
      "SUBCELLULAR LOCATION: Cytoplasm.\n",
      "['Activator', 'Cytoplasm', 'Direct protein sequencing', 'DNA-binding', 'Transcription', 'Transcription regulation']\n",
      "MTLTGKVKWFNSEKGFGFIEVADGSDVFVHFSAITGDGFKSLDEGQEVSFEVED\n",
      "54\n",
      "2\n",
      "\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "SEQUENCE CAUTION: Sequence=AAN79187.1; Type=Erroneous initiation; Evidence={ECO:0000305};\n",
      "['Activator', 'Complete proteome', 'Cytoplasm', 'DNA-binding', 'Transcription', 'Transcription regulation']\n",
      "MSKIKGNVKWFNESKGFGFITPEDGSKDVFVHFSAIQTNGFKTLAEGQRVEFEITNGAKGPSAANVIAL\n",
      "69\n",
      "8\n",
      "\n",
      "FUNCTION: Could accelerate the degradation of some genes transcripts potentially through selective RNA binding. {ECO:0000255|HAMAP-Rule:MF_00167}.\n",
      "SIMILARITY: Belongs to the CsrA family. {ECO:0000255|HAMAP- Rule:MF_00167}.\n",
      "['Complete proteome', 'Reference proteome', 'RNA-binding']\n",
      "MLVLTRTNGQAIKIGNEGEITVTVLEVRGNQVRMGIDAPKHIAVHREEIYQRIQEEKPKAPPILEEVE\n",
      "68\n",
      "2\n",
      "\n",
      "FUNCTION: Could accelerate the degradation of some genes transcripts potentially through selective RNA binding. {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the CsrA family. {ECO:0000305}.\n",
      "['Complete proteome', 'Reference proteome', 'RNA-binding']\n",
      "MLILTRRCAESLIIGDGEITVTVLGVKGNQVRIGVNAPKEVAVHREEIYLRIKKEKDEEPSH\n",
      "62\n",
      "3\n",
      "\n",
      "FUNCTION: Could accelerate the degradation of some genes transcripts potentially through selective RNA binding. {ECO:0000255|HAMAP-Rule:MF_00167}.\n",
      "SIMILARITY: Belongs to the CsrA family. {ECO:0000255|HAMAP- Rule:MF_00167}.\n",
      "['Complete proteome', 'Reference proteome', 'RNA-binding']\n",
      "MLILTRKVGESINIGDEITVTILGVQGLQVRLGINAPKNVSVHREEIYKRIQAELAPNQDPQ\n",
      "62\n",
      "4\n",
      "\n",
      "FUNCTION: Could accelerate the degradation of some genes transcripts potentially through selective RNA binding. {ECO:0000255|HAMAP-Rule:MF_00167}.\n",
      "SIMILARITY: Belongs to the CsrA family. {ECO:0000255|HAMAP- Rule:MF_00167}.\n",
      "['Complete proteome', 'Reference proteome', 'RNA-binding']\n",
      "MLILTRKIGESLLIGDNVEITVLSVRGNQVKLGVNAPKEVSVHREEIYQRIKALADDVASDTQQ\n",
      "64\n",
      "3\n",
      "\n",
      "FUNCTION: Could accelerate the degradation of some genes transcripts potentially through selective RNA binding. {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the CsrA family. {ECO:0000305}.\n",
      "['3D-structure', 'Complete proteome', 'Reference proteome', 'RNA-binding']\n",
      "MLVLSRKINEAIQIGADIEVKVIAVEGDQVKLGIDAPKHIDIHRKEIYLTIQEENNRAAALSSDVISALSSQKK\n",
      "74\n",
      "6\n",
      "\n",
      "FUNCTION: Affects glycogen biosynthesis, gluconeogenesis, cell size and surface properties. Regulates glycogen synthesis under both aerobic and anaerobic conditions. Seems to accelerate the degradation of glg gene transcripts, potentially through selective RNA binding. Acts to inhibit interaction between the LetD protein and the A subunit of DNA gyrase. Also required for motility and flagellum biosynthesis through the post-transcriptional activation of flhDC expression. This involves binding to and stabilization of the flhDC message by CsrA. {ECO:0000255|HAMAP-Rule:MF_00167}.\n",
      "SUBUNIT: Homodimer. Binds to the csrB RNA which antagonizes the activity of CsrA. {ECO:0000255|HAMAP-Rule:MF_00167}.\n",
      "SIMILARITY: Belongs to the CsrA family. {ECO:0000255|HAMAP- Rule:MF_00167}.\n",
      "['Complete proteome', 'RNA-binding']\n",
      "MLILTRRVGETLMIGDEVTVTVLGVKGNQVRIGVNAPKEVSVHREEIYQRIQAEKSQQSSY\n",
      "61\n",
      "2\n",
      "\n",
      "FUNCTION: Affects glycogen biosynthesis, gluconeogenesis, cell size and surface properties. Regulates glycogen synthesis under both aerobic and anaerobic conditions. Seems to accelerate the degradation of glg gene transcripts, potentially through selective RNA binding. Acts to inhibit interaction between the LetD protein and the A subunit of DNA gyrase. Also required for motility and flagellum biosynthesis through the post-transcriptional activation of flhDC expression. This involves binding to and stabilization of the flhDC message by CsrA (By similarity). {ECO:0000250}.\n",
      "SUBUNIT: Homodimer. Binds to the csrB RNA which antagonizes the activity of CsrA (By similarity). {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the CsrA family. {ECO:0000305}.\n",
      "['Complete proteome', 'RNA-binding']\n",
      "MLILTRRVGETLMIGDEVTVTVLGVKGNQVRIGVNAPKEVSVHREEIYQRIQAEKSQQSSY\n",
      "61\n",
      "3\n",
      "\n",
      "FUNCTION: Controls extracellular enzymes, N-(3-oxohexanoyl)-L- homoserine lactone, and pathogenicity. Repressor of virulence factors. {ECO:0000269|PubMed:7665490}.\n",
      "SIMILARITY: Belongs to the CsrA family. {ECO:0000305}.\n",
      "['RNA-binding']\n",
      "MLILTRRVGETLIIGDEVTVTVLGVKGNQVRIGVNAPKEVSVHREEIYQRIQAEKSQPTSY\n",
      "61\n",
      "4\n",
      "\n",
      "FUNCTION: Could accelerate the degradation of some genes transcripts potentially through selective RNA binding. {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the CsrA family. {ECO:0000305}.\n",
      "['RNA-binding']\n",
      "MLILTRRCAESLIIGDGEITVTVLGVKGNQVRIGVNAPKEVAVHREEIYLRIKKEKDEEPSH\n",
      "62\n",
      "5\n",
      "\n",
      "FUNCTION: Monoheme cytochrome which functions as an electron carrier in the reduction of nitrite by membrane vesicles.\n",
      "PTM: Binds 1 heme group per subunit.\n",
      "['Direct protein sequencing', 'Electron transport', 'Heme', 'Iron', 'Metal-binding', 'Transport']\n",
      "GGSVDSAAAEEVFESNCASCHGADLSGAGPDLTQV\n",
      "35\n",
      "1\n",
      "\n",
      "FUNCTION: Participates in sulfate respiration coupled with phosphorylation by transferring electrons from the enzyme dehydrogenase to ferredoxin. {ECO:0000269|PubMed:8573595}.\n",
      "SUBUNIT: Homodimer. {ECO:0000269|PubMed:8573595}.\n",
      "SUBCELLULAR LOCATION: Periplasm {ECO:0000250}.\n",
      "PTM: Binds 4 heme groups per subunit. {ECO:0000269|PubMed:8573595}.\n",
      "MISCELLANEOUS: Acidic tetraheme c3 is not a subunit of this octaheme cytochrome c3. {ECO:0000269|PubMed:8573595}.\n",
      "['Direct protein sequencing', 'Electron transport', 'Heme', 'Iron', 'Metal-binding', 'Periplasm', 'Sulfate respiration', 'Transport']\n",
      "SETETKAASIEAETVLQFPVSFKFNRPEGRPTALFAEVKFSHYDH\n",
      "45\n",
      "1\n",
      "\n",
      "FUNCTION: Participates in sulfate respiration coupled with phosphorylation by transferring electrons from the enzyme dehydrogenase to ferredoxin. {ECO:0000269|PubMed:12119407}.\n",
      "BIOPHYSICOCHEMICAL PROPERTIES: Redox potential:   E(0) are -140 mV, -210 mV and -240 mV.;\n",
      "PTM: Binds 3 heme groups per subunit.\n",
      "['3D-structure', 'Direct protein sequencing', 'Electron transport', 'Heme', 'Iron', 'Metal-binding', 'Sulfate respiration', 'Transport']\n",
      "ADVVTYENKKGNVTFDHKAHAEKLGCDACHEGTPAKIAIDKKSAHKDACKTCHKSNNGPTKCGGCHIK\n",
      "68\n",
      "3\n",
      "\n",
      "FUNCTION: Functions as an electron carrier between membrane-bound cytochrome b6-f and photosystem I in oxygenic photosynthesis.\n",
      "SUBUNIT: Monomer. {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cellular thylakoid lumen {ECO:0000305}.\n",
      "PTM: Binds 1 heme group per subunit.\n",
      "SIMILARITY: Belongs to the cytochrome c family. PetJ subfamily. {ECO:0000305}.\n",
      "['Direct protein sequencing', 'Electron transport', 'Heme', 'Iron', 'Metal-binding', 'Photosynthesis', 'Thylakoid', 'Transport']\n",
      "AASGASIFSAKCAQCHLGGKNIINPTKTLSLADLQANGKDTVSAI\n",
      "45\n",
      "2\n",
      "\n",
      "FUNCTION: Required for correct functioning of cytochrome bd oxidase. {ECO:0000269|PubMed:22919638}.\n",
      "CATALYTIC ACTIVITY: 2 ubiquinol + O(2) + 4 H(+)(Side 1) = 2 ubiquinone + 2 H(2)O + 4 H(+)(Side 2).\n",
      "PATHWAY: Energy metabolism; oxidative phosphorylation.\n",
      "SUBUNIT: May be a subunit of cytochrome ubiquinol oxidase.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane {ECO:0000305|PubMed:22919638}; Single-pass membrane protein {ECO:0000305|PubMed:22919638}.\n",
      "INDUCTION: Expressed in stationary phase. Part of the cydB-cydX operon. {ECO:0000269|PubMed:22919638}.\n",
      "DISRUPTION PHENOTYPE: Highly attenuated in macrophage-like line J774 and female BALB/c ByJ mouse infections. Loss of viability in stationary phase growth in culture. Increased sensitivity to H(2)O(2), acid pH, highly sensitive to the combination of NaN(3) plus NiSO(4). {ECO:0000269|PubMed:22919638}.\n",
      "SIMILARITY: Belongs to the cytochrome ubiquinol oxidase subunit X family. {ECO:0000305}.\n",
      "SEQUENCE CAUTION: Sequence=CAJ12892.1; Type=Erroneous initiation; Note=Translation N-terminally shortened.; Evidence={ECO:0000305};\n",
      "['Cell inner membrane', 'Cell membrane', 'Complete proteome', 'Electron transport', 'Membrane', 'Oxidoreductase', 'Reference proteome', 'Transmembrane', 'Transmembrane helix', 'Transport', 'Virulence']\n",
      "MWYFSWLLGLPLAAAFAVLNAMWYELMDDRARKRLAADPTAELALEGNKHH\n",
      "51\n",
      "2\n",
      "\n",
      "FUNCTION: Required for correct functioning of cytochrome bd-I oxidase. This protein and AppX may have some functional overlap.\n",
      "CATALYTIC ACTIVITY: 2 ubiquinol + O(2) + 4 H(+)(Side 1) = 2 ubiquinone + 2 H(2)O + 4 H(+)(Side 2).\n",
      "PATHWAY: Energy metabolism; oxidative phosphorylation.\n",
      "SUBUNIT: May be a subunit of cytochrome bd-I ubiquinol oxidase. Probably interacts with CydA and CydB. {ECO:0000269|PubMed:23749980}.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane {ECO:0000305}; Single- pass membrane protein {ECO:0000305}. Note=In sucrose cushions fractionates with both the inner and outer membranes. {ECO:0000269|PubMed:19121005, ECO:0000269|PubMed:21778229}.\n",
      "INDUCTION: Constitutively expressed during aerobic growth (PubMed:19121005), induced in low oxygen, and maybe by SDS/EDTA (envelope stress) (PubMed:19734316) (at protein level). {ECO:0000269|PubMed:19121005, ECO:0000269|PubMed:19734316}.\n",
      "DISRUPTION PHENOTYPE: Slow growth in aerobic liquid culture, forms mixed colonies on aerobic solid media, increased sensitivity to reductant (beta-mercapoethanol). Reduced cytochrome bd-I oxidase activity. {ECO:0000269|PubMed:23749980}.\n",
      "SIMILARITY: Belongs to the cytochrome ubiquinol oxidase subunit X family. {ECO:0000305}.\n",
      "SEQUENCE CAUTION: Sequence=J03939; Type=Frameshift; Positions=22, 27; Evidence={ECO:0000305};\n",
      "['Cell inner membrane', 'Cell membrane', 'Complete proteome', 'Electron transport', 'Membrane', 'Oxidoreductase', 'Reference proteome', 'Stress response', 'Transmembrane', 'Transmembrane helix', 'Transport']\n",
      "MWYFAWILGTLLACSFGVITALALEHVESGKAGQEDI\n",
      "37\n",
      "1\n",
      "\n",
      "CATALYTIC ACTIVITY: Succinyl-CoA + (S)-2,3,4,5-tetrahydropyridine- 2,6-dicarboxylate + H(2)O = CoA + N-succinyl-L-2-amino-6- oxoheptanedioate.\n",
      "PATHWAY: Amino-acid biosynthesis; L-lysine biosynthesis via DAP pathway; LL-2,6-diaminopimelate from (S)-tetrahydrodipicolinate (succinylase route): step 1/3.\n",
      "SUBUNIT: Homotrimer. {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the transferase hexapeptide repeat family. {ECO:0000305}.\n",
      "['Acyltransferase', 'Amino-acid biosynthesis', 'Cytoplasm', 'Diaminopimelate biosynthesis', 'Lysine biosynthesis', 'Repeat', 'Transferase']\n",
      "MQQLQNVIESAFERRADITPANVDTVTREAVNQVISLLDSGALRVAKKIDG\n",
      "51\n",
      "3\n",
      "\n",
      "CATALYTIC ACTIVITY: L-histidine = histamine + CO(2).\n",
      "COFACTOR: Name=pyruvate; Xref=ChEBI:CHEBI:15361; Note=Binds 1 pyruvoyl group covalently per subunit.;\n",
      "SUBUNIT: Heterohexamer of 3 large and 3 small chains.\n",
      "['Decarboxylase', 'Direct protein sequencing', 'Lyase', 'Pyruvate']\n",
      "MKKTDKILKEIGIQRVAIMEGKKYSKGFMEDGDIGVQY\n",
      "38\n",
      "1\n",
      "\n",
      "FUNCTION: Stimulates the phosphotransfer from phospho-DegS to DegU. Affects protease and levansucrose production (By similarity). {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the DegQ family. {ECO:0000305}.\n",
      "['Complete proteome', 'Reference proteome']\n",
      "MEKQQIEELKQLLWRLENEIRETKDSLRKINKSIDQYDKYTYLKTS\n",
      "46\n",
      "4\n",
      "\n",
      "FUNCTION: Stabilizes the phosphorylated form of DegU, leading to enhanced production of levansucrase, alkaline protease, and neutral protease. {ECO:0000269|PubMed:3082853}.\n",
      "['Transcription', 'Transcription regulation']\n",
      "MDDKDLKLILHKTFIEIYSDLEELADIAKKGKPSMEKYVEEIEQRCKQNILAIEIQMKIK\n",
      "60\n",
      "3\n",
      "\n",
      "CATALYTIC ACTIVITY: Purine nucleoside + phosphate = purine + alpha-D-ribose 1-phosphate.\n",
      "CATALYTIC ACTIVITY: Purine deoxynucleoside + phosphate = purine + 2'-deoxy-alpha-D-ribose 1-phosphate.\n",
      "SUBUNIT: Homohexamer. {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the PNP/UDP phosphorylase family. {ECO:0000305}.\n",
      "['Glycosyltransferase', 'Transferase']\n",
      "ALTLLTVSDSLITKESLSSLERQTTFNTMVKLALEMACELQK\n",
      "42\n",
      "1\n",
      "\n",
      "FUNCTION: Nonheme iron protein possibly involved in electron transport.\n",
      "COFACTOR: Name=Fe cation; Xref=ChEBI:CHEBI:24875; Note=Binds 2 irons ions per subunit via 4 cysteine residues per iron.;\n",
      "SUBUNIT: Homodimer.\n",
      "SIMILARITY: To the N-terminal section of desulfoferrodoxin. {ECO:0000305}.\n",
      "['3D-structure', 'Direct protein sequencing', 'Electron transport', 'Iron', 'Metal-binding', 'Transport']\n",
      "MANEGDVYKCELCGQVVKVLEEGGGTLVCCGEDMVKQ\n",
      "37\n",
      "2\n",
      "\n",
      "CATALYTIC ACTIVITY: An aldehyde + a quinone + H(2)O = a carboxylate + a quinol.\n",
      "COFACTOR: Name=FAD; Xref=ChEBI:CHEBI:57692;   Evidence={ECO:0000269|PubMed:8554333}; Note=Binds 1 FAD per subunit. {ECO:0000269|PubMed:8554333};\n",
      "SUBUNIT: Heterotrimer composed of an alpha, a beta and a gamma chain.\n",
      "['Direct protein sequencing', 'FAD', 'Flavoprotein', 'Oxidoreductase']\n",
      "MIPAQFTYRRVSSVDEALAAVAEHGD\n",
      "26\n",
      "2\n",
      "\n",
      "FUNCTION: Catalyzes the oxidation of primary alcohols including methanol. {ECO:0000250}.\n",
      "CATALYTIC ACTIVITY: A primary alcohol + 2 cytochrome c(L) = an aldehyde + 2 reduced cytochrome c(L) + 2 H(+).\n",
      "SUBUNIT: Heterotetramer composed of 2 alpha and 2 beta subunits. {ECO:0000269|PubMed:7685335}.\n",
      "SUBCELLULAR LOCATION: Periplasm {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the methanol dehydrogenase subunit 2 family. {ECO:0000255}.\n",
      "['Direct protein sequencing', 'Disulfide bond', 'Methanol utilization', 'Oxidoreductase', 'Periplasm']\n",
      "YDGTNCKPGVCWEPKPGY\n",
      "18\n",
      "2\n",
      "\n",
      "CATALYTIC ACTIVITY: Hydrogen sulfide + 2 ferricytochrome c = sulfur + 2 ferrocytochrome c.\n",
      "SUBUNIT: Dimer of one cytochrome and one flavoprotein.\n",
      "SUBCELLULAR LOCATION: Periplasm.\n",
      "['Direct protein sequencing', 'FAD', 'Flavoprotein', 'Oxidoreductase', 'Periplasm']\n",
      "GTKRVVVVGGGFGGASTAKYLRKLDPSISVTLVEPKTAFVTCPFSNAV\n",
      "48\n",
      "2\n",
      "\n",
      "FUNCTION: Involved in cell division inhibition; this function can be repressed by DicA and DicC proteins as well as antitoxin CbeA (yeeU). {ECO:0000269|PubMed:22515815}.\n",
      "['Cell cycle', 'Cell division', 'Complete proteome', 'Reference proteome']\n",
      "MKTLLPNVNTSEGCFEIGVTISNPVFTEDAINKRKQERELLNKICIVSMLARLRLMPKGCAQ\n",
      "62\n",
      "6\n",
      "\n",
      "SUBCELLULAR LOCATION: Membrane {ECO:0000305}; Single-pass membrane protein {ECO:0000305}.\n",
      "SIMILARITY: Belongs to the DinQ family. {ECO:0000305}.\n",
      "SEQUENCE CAUTION: Sequence=AAN82736.1; Type=Erroneous initiation; Evidence={ECO:0000305};\n",
      "['Complete proteome', 'Membrane', 'Transmembrane', 'Transmembrane helix']\n",
      "MSKRMHSHSIAWRKRVIDKAIIVLGALIALLELIRFLLQLLN\n",
      "42\n",
      "2\n",
      "\n",
      "FUNCTION: Active with aldehydes and formate esters as substrates. {ECO:0000269|PubMed:8554333}.\n",
      "COFACTOR: Name=[2Fe-2S] cluster; Xref=ChEBI:CHEBI:49601;   Evidence={ECO:0000269|PubMed:8554333}; Note=Binds 2 [2Fe-2S] clusters per subunit. {ECO:0000269|PubMed:8554333};\n",
      "SUBUNIT: Heterotetramer composed of an alpha, a beta and two gamma chains.\n",
      "['2Fe-2S', 'Direct protein sequencing', 'Iron', 'Iron-sulfur', 'Metal-binding', 'Oxidoreductase']\n",
      "MKVSIEINGTTVSXEVXDRTL\n",
      "21\n",
      "3\n",
      "\n",
      "FUNCTION: Binds calcium ions. May play a role in sequestering additional small ligands. {ECO:0000269|PubMed:21370306}.\n",
      "SUBUNIT: Homododecamer; 12 subunits assemble to form a hollow sphere with a diameter of about 75 Angstroms. Calcium ions are bound at the interface between three subunits. {ECO:0000269|PubMed:21370306}.\n",
      "SIMILARITY: Belongs to the dodecin family. {ECO:0000305}.\n",
      "CAUTION: Dodecin family members from different organisms have non- identical ligand binding specificity. {ECO:0000305}.\n",
      "['3D-structure', 'Calcium', 'Complete proteome', 'Metal-binding', 'Reference proteome']\n",
      "MSVYKVIDIIGTSPTSWEQAAAEAVQRARDSVDDIRVARVIEQDMAVDSAGKITYRIKLEVSFKMRPAQPR\n",
      "71\n",
      "4\n",
      "\n",
      "FUNCTION: May function as storage protein that sequesters various flavins and other cofactors, thereby protecting the cell against undesirable reactions mediated by the free cofactors. Binds and sequesters FMN, FAD, lumiflavin and lumichrome, and can also bind coenzyme A. {ECO:0000269|PubMed:17855371}.\n",
      "SUBUNIT: Homododecamer; four homotrimers assemble to form a dodecameric hollow sphere with an outer diameter of about 60 Angstroms. Flavin dimers are bound between subunits with a stoichiometry of 6 flavin dimers per dodecamer. Besides, trimeric coenzyme A molecules can be bound between subunits. A dodecamer can bind simultaneously 12 flavin and 12 coenzyme A molecules. {ECO:0000269|PubMed:17855371, ECO:0000269|Ref.2, ECO:0000269|Ref.4}.\n",
      "SIMILARITY: Belongs to the dodecin family. {ECO:0000305}.\n",
      "CAUTION: Dodecin family members from different organisms have non- identical ligand binding specificity. {ECO:0000305}.\n",
      "['3D-structure', 'Complete proteome', 'FAD', 'Flavoprotein', 'FMN', 'Nucleotide-binding', 'Reference proteome']\n",
      "MGKVYKKVELVGTSEEGLEAAIQAALARARKTLRHLDWFEVKEIRGTIGEAGVKEYQVVLEVGFRLEET\n",
      "69\n",
      "9\n",
      "\n",
      "FUNCTION: DNA polymerase III is a complex, multichain enzyme responsible for most of the replicative synthesis in bacteria. This DNA polymerase also exhibits 3' to 5' exonuclease activity. The beta chain is required for initiation of replication once it is clamped onto DNA, it slides freely (bidirectional and ATP- independent) along duplex DNA (By similarity). {ECO:0000250}.\n",
      "CATALYTIC ACTIVITY: Deoxynucleoside triphosphate + DNA(n) = diphosphate + DNA(n+1).\n",
      "SUBUNIT: Contains a core (composed of alpha, epsilon, and theta chains) that can repair short gaps created by nuclease in duplex DNA. For efficient replication of the long, single-stranded templates, pol III requires the auxiliary chains beta, gamma, and delta (By similarity). {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "['Cytoplasm', 'DNA replication', 'DNA-directed DNA polymerase', 'Nucleotidyltransferase', 'Transferase']\n",
      "EEIIDVAYQSPEMEVGFNVSYLLDVLNTLKCERVRFNLVDASSSCLIEDCDNSTAEYVIMPMRL\n",
      "64\n",
      "4\n",
      "\n",
      "SIMILARITY: Belongs to the DsrB family. {ECO:0000255|HAMAP- Rule:MF_01549}.\n",
      "SEQUENCE CAUTION: Sequence=AAV76867.1; Type=Erroneous initiation; Evidence={ECO:0000305};\n",
      "['Complete proteome']\n",
      "MKVNDRVTVKTDGGPRRPGVVLAVEEFSEGTMYLVSLEDYPLGIWFFNESGHQDGIFVEKAEQD\n",
      "64\n",
      "5\n",
      "\n",
      "FUNCTION: Acts as antidote to the effect of entericidin B.\n",
      "SUBCELLULAR LOCATION: Cell membrane; Lipid-anchor.\n",
      "SIMILARITY: Belongs to the EcnA/EcnB lipoprotein family. {ECO:0000305}.\n",
      "['Cell membrane', 'Lipoprotein', 'Membrane', 'Palmitate', 'Signal']\n",
      "MMKRLLGLVMLLLFTCTLLTGCNTARGFGEDIKHLGNSISHAAS\n",
      "44\n",
      "1\n",
      "\n",
      "FUNCTION: Plays a role in the bacteriolysis. Is activated under conditions of high osmolarity by the factor sigma S. Entericidin A functions as an antidote.\n",
      "SUBCELLULAR LOCATION: Cell membrane; Lipid-anchor.\n",
      "SIMILARITY: Belongs to the EcnA/EcnB lipoprotein family. {ECO:0000305}.\n",
      "['Cell membrane', 'Lipoprotein', 'Membrane', 'Palmitate', 'Signal']\n",
      "MVKKTIAAIFSVLVLSSVLTACNTTRGVGQDISEGGSAISGAGTKAQQ\n",
      "48\n",
      "3\n",
      "\n",
      "SIMILARITY: Belongs to the protease inhibitor I11 (ecotin) family. {ECO:0000305}.\n",
      "CAUTION: Could be the product of a pseudogene. It is N- and C- terminally truncated compared to othologs. {ECO:0000305}.\n",
      "['Complete proteome']\n",
      "MQATKNAMQDCNRVWFGGKLETKTLEGWGYNYYIIDQVSDHPASTMMACPNVKATIQTVSVFLGDETF\n",
      "68\n",
      "4\n",
      "\n",
      "FUNCTION: This protein promotes the GTP-dependent binding of aminoacyl-tRNA to the A-site of ribosomes during protein biosynthesis. {ECO:0000250}.\n",
      "SUBUNIT: Monomer. {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the GTP-binding elongation factor family. EF-Tu/EF-1A subfamily. {ECO:0000305}.\n",
      "['Cytoplasm', 'Elongation factor', 'GTP-binding', 'Nucleotide-binding', 'Protein biosynthesis']\n",
      "MAKGKFERSKPHVNVGTIGHVDHGKTTLTAALTTILAEKFGGQAKA\n",
      "46\n",
      "5\n",
      "\n",
      "FUNCTION: Endonuclease IV plays a role in DNA repair. It cleaves phosphodiester bonds at apurinic or apyrimidinic sites (AP sites) to produce new 5'-ends that are base-free deoxyribose 5-phosphate residues. It preferentially attacks modified AP sites created by bleomycin and neocarzinostatin (By similarity). {ECO:0000250}.\n",
      "CATALYTIC ACTIVITY: Endonucleolytic cleavage to 5'- phosphooligonucleotide end-products.\n",
      "COFACTOR: Name=Zn(2+); Xref=ChEBI:CHEBI:29105; Evidence={ECO:0000250}; Note=Binds 3 Zn(2+) ions. {ECO:0000250};\n",
      "SIMILARITY: Belongs to the AP endonuclease 2 family. {ECO:0000305}.\n",
      "['DNA damage', 'DNA repair', 'Endonuclease', 'Hydrolase', 'Metal-binding', 'Nuclease', 'Zinc']\n",
      "IMRDPRFDNIPLILETVNPDIWAEEIAWLKSQAEI\n",
      "35\n",
      "2\n",
      "\n",
      "FUNCTION: Catalyzes the reversible conversion of 2- phosphoglycerate into phosphoenolpyruvate. It is essential for the degradation of carbohydrates via glycolysis (By similarity). {ECO:0000250|UniProtKB:P0A6P9}.\n",
      "CATALYTIC ACTIVITY: 2-phospho-D-glycerate = phosphoenolpyruvate + H(2)O. {ECO:0000250|UniProtKB:P0A6P9}.\n",
      "COFACTOR: Name=Mg(2+); Xref=ChEBI:CHEBI:18420;   Evidence={ECO:0000250|UniProtKB:P0A6P9}; Note=Mg(2+) is required for catalysis and for stabilizing the dimer. {ECO:0000250|UniProtKB:P0A6P9};\n",
      "ENZYME REGULATION: The covalent binding to the substrate causes inactivation of the enzyme, and possibly serves as a signal for the export of the protein. {ECO:0000250|UniProtKB:P0A6P9}.\n",
      "PATHWAY: Carbohydrate degradation; glycolysis; pyruvate from D- glyceraldehyde 3-phosphate: step 4/5.\n",
      "SUBCELLULAR LOCATION: Cytoplasm. Secreted. Cell surface. Note=Fractions of enolase are present in both the cytoplasm and on the cell surface. The export of enolase possibly depends on the covalent binding to the substrate; once secreted, it remains attached to the bacterial cell surface (By similarity). {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the enolase family. {ECO:0000255}.\n",
      "['Cytoplasm', 'Direct protein sequencing', 'Glycolysis', 'Lyase', 'Magnesium', 'Metal-binding', 'Secreted']\n",
      "MGAEVFHSLKKVLGEKGLASGVGDEGGFAPNLGSNIRRAGYTAVISHRVAKYNQLLR\n",
      "57\n",
      "4\n",
      "\n",
      "MISCELLANEOUS: Encoded by the cryptic lambdoid prophage DLP12.\n",
      "SIMILARITY: Belongs to the lambda phage S protein family. {ECO:0000305}.\n",
      "['Complete proteome', 'Cytolysis', 'Host cell lysis by virus', 'Reference proteome', 'Virus exit from host cell']\n",
      "MKSMDKLTTGVAYGTSAGSAGYWFLQLLDKVTPSQWAAIGVLGSLVFGLLTYLTNLYFKIKEDKRKAARGE\n",
      "71\n",
      "5\n",
      "\n",
      "FUNCTION: The electron transfer flavoprotein serves as a specific electron acceptor for other dehydrogenases. It transfers the electrons to the main respiratory chain via ETF-ubiquinone oxidoreductase (ETF dehydrogenase) (By similarity). {ECO:0000250}.\n",
      "COFACTOR: Name=FAD; Xref=ChEBI:CHEBI:57692; Evidence={ECO:0000250}; Note=Binds 1 FAD per dimer. {ECO:0000250};\n",
      "SUBUNIT: Heterodimer of an alpha and a beta subunit.\n",
      "SIMILARITY: Belongs to the ETF alpha-subunit/FixB family. {ECO:0000305}.\n",
      "['Direct protein sequencing', 'Electron transport', 'FAD', 'Flavoprotein', 'Transport']\n",
      "MNIADYKGGVWVFAEQXDG\n",
      "19\n",
      "4\n",
      "\n",
      "FUNCTION: Accepts electrons from ETF and reduces ubiquinone.\n",
      "CATALYTIC ACTIVITY: Reduced electron-transferring flavoprotein + ubiquinone = electron-transferring flavoprotein + ubiquinol.\n",
      "COFACTOR: Name=[4Fe-4S] cluster; Xref=ChEBI:CHEBI:49883; Note=Binds 1 [4Fe-4S] cluster.;\n",
      "COFACTOR: Name=FAD; Xref=ChEBI:CHEBI:57692;\n",
      "SUBUNIT: Monomer. {ECO:0000250}.\n",
      "['4Fe-4S', 'Direct protein sequencing', 'Electron transport', 'FAD', 'Flavoprotein', 'Iron', 'Iron-sulfur', 'Metal-binding', 'Oxidoreductase', 'Transport', 'Ubiquinone']\n",
      "SDIGGSMDYDVVIVGAGGAGLSAAILKQVNP\n",
      "31\n",
      "2\n",
      "\n",
      "FUNCTION: Bidirectionally degrades single-stranded DNA into large acid-insoluble oligonucleotides, which are then degraded further into small acid-soluble oligonucleotides. {ECO:0000255|HAMAP- Rule:MF_00337}.\n",
      "CATALYTIC ACTIVITY: Exonucleolytic cleavage in either 5'- to 3'- or 3'- to 5'-direction to yield nucleoside 5'-phosphates. {ECO:0000255|HAMAP-Rule:MF_00337}.\n",
      "SUBUNIT: Heterooligomer composed of large and small subunits. {ECO:0000255|HAMAP-Rule:MF_00337}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000255|HAMAP-Rule:MF_00337}.\n",
      "SIMILARITY: Belongs to the XseB family. {ECO:0000255|HAMAP- Rule:MF_00337}.\n",
      "['Complete proteome', 'Cytoplasm', 'Exonuclease', 'Hydrolase', 'Nuclease', 'Reference proteome']\n",
      "MAKKPTQDFESTLKELEAIVSHLETGELPLEEALNEFETAVKLVQQGQERLQKAEQRIQILLNKNDQAELSDYE\n",
      "74\n",
      "7\n",
      "\n",
      "FUNCTION: Bidirectionally degrades single-stranded DNA into large acid-insoluble oligonucleotides, which are then degraded further into small acid-soluble oligonucleotides. {ECO:0000250}.\n",
      "CATALYTIC ACTIVITY: Exonucleolytic cleavage in either 5'- to 3'- or 3'- to 5'-direction to yield nucleoside 5'-phosphates.\n",
      "SUBUNIT: Heterooligomer composed of large and small subunits. {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the XseB family. {ECO:0000305}.\n",
      "['Complete proteome', 'Cytoplasm', 'Exonuclease', 'Hydrolase', 'Nuclease']\n",
      "MTKKAKNVEKVPFEDAMKRLEEIIDLMNQPTTALEASLALYEEADQLMRICESRIQEVEARIKQLSDQRSES\n",
      "72\n",
      "3\n",
      "\n",
      "FUNCTION: Catalyzes the reduction of a carbon-carbon double bond in an enoyl moiety that is covalently linked to an acyl carrier protein (ACP). Involved in the elongation cycle of fatty acid which are used in the lipid metabolism (By similarity). {ECO:0000250}.\n",
      "CATALYTIC ACTIVITY: An acyl-[acyl-carrier protein] + NAD(+) = a trans-2,3-dehydroacyl-[acyl-carrier protein] + NADH.\n",
      "PATHWAY: Lipid metabolism; fatty acid biosynthesis.\n",
      "SUBUNIT: Homotetramer. {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the short-chain dehydrogenases/reductases (SDR) family. FabI subfamily. {ECO:0000305}.\n",
      "['Fatty acid biosynthesis', 'Fatty acid metabolism', 'Lipid biosynthesis', 'Lipid metabolism', 'NAD', 'Oxidoreductase']\n",
      "MSIPTVKAKLLQGRKGLIVGIANDRSIAWGRARAFRALGAEIAVTYLNDKALPLV\n",
      "55\n",
      "1\n",
      "\n",
      "COFACTOR: Name=[4Fe-4S] cluster; Xref=ChEBI:CHEBI:49883;   Evidence={ECO:0000305}; Note=Binds 2 [4Fe-4S] clusters. {ECO:0000305};\n",
      "['4Fe-4S', 'Electron transport', 'Iron', 'Iron-sulfur', 'Metal-binding', 'Nitrogen fixation', 'Repeat', 'Transport']\n",
      "MAMAIDGYECTVCGDCKPVCPTGSIVLQGGIYVIDADSCNECADLGEPRCLGVCPVDFCIQPLDD\n",
      "65\n",
      "2\n",
      "\n",
      "FUNCTION: Ferredoxins are iron-sulfur proteins that transfer electrons in a wide variety of metabolic reactions. This ferredoxin probably participates in nitrogen fixation.\n",
      "COFACTOR: Name=[4Fe-4S] cluster; Xref=ChEBI:CHEBI:49883;   Evidence={ECO:0000305}; Note=Binds 2 [4Fe-4S] clusters. {ECO:0000305};\n",
      "['4Fe-4S', 'Electron transport', 'Iron', 'Iron-sulfur', 'Metal-binding', 'Nitrogen fixation', 'Repeat', 'Transport']\n",
      "MAMKIDPELCTSCGDCEPVCPTNAIAPKKGVYVINADTCTECEGEHDLPQCVNACMTDNCINPAA\n",
      "65\n",
      "3\n",
      "\n",
      "FUNCTION: Involved in Fe(2+) ion uptake (By similarity). {ECO:0000250|UniProtKB:P0AEL3}.\n",
      "SIMILARITY: Belongs to the FeoA family. {ECO:0000305}.\n",
      "['Complete proteome', 'Ion transport', 'Iron', 'Iron transport', 'Transport']\n",
      "MQYTPDTAWKITGFSREISPAYRQKLLSLGMLPGSSFNVVRVAPLGDPIHIETRRVSLVLRKKDLALLEVEAVSC\n",
      "75\n",
      "3\n",
      "\n",
      "FUNCTION: Might be involved in Fe(2+) ion uptake (By similarity). {ECO:0000250|UniProtKB:P0AEL3}.\n",
      "SIMILARITY: Belongs to the FeoA family. {ECO:0000305}.\n",
      "['Complete proteome', 'Ion transport', 'Iron', 'Iron transport', 'Reference proteome', 'Transport']\n",
      "MTLNEAIKDKVYEIVEIANCDEALKKRFLSFGIHEGVQCILLHYSMKKATLSVKINRIQVALRSHEAQYLVIKESV\n",
      "76\n",
      "3\n",
      "\n",
      "FUNCTION: May function as a transcriptional regulator that controls feoABC expression. {ECO:0000255|HAMAP-Rule:MF_01586}.\n",
      "SIMILARITY: Belongs to the FeoC family. {ECO:0000255|HAMAP- Rule:MF_01586}.\n",
      "['DNA-binding', 'Iron', 'Iron-sulfur', 'Metal-binding', 'Repressor', 'Transcription', 'Transcription regulation']\n",
      "MAGLLQVRDALALRATSGAQLSQSLATPLPLVQAMLDRLTAMGKVERIEQDDGACLSGGCKSCPQPRGAARCSTG\n",
      "75\n",
      "4\n",
      "\n",
      "FUNCTION: Ferredoxins are iron-sulfur proteins that transfer electrons in a wide variety of metabolic reactions. This ferredoxin serves as a carrier for pyruvate dehydrogenase.\n",
      "COFACTOR: Note=Binds 2 4Fe-4S clusters, or 1 4Fe-4S cluster and 1 3Fe-4S cluster.;\n",
      "SUBUNIT: Homodimer.\n",
      "['4Fe-4S', 'Complete proteome', 'Direct protein sequencing', 'Electron transport', 'Iron', 'Iron-sulfur', 'Metal-binding', 'Reference proteome', 'Repeat', 'Transport']\n",
      "MGWTVTVDTDKCTGDGECVDVCPVEVYELQDGKAVPVNEEECLGCESCVEVCEAGAITVEEN\n",
      "62\n",
      "7\n",
      "\n",
      "FUNCTION: Ferredoxins are iron-sulfur proteins that transfer electrons in a wide variety of metabolic reactions.\n",
      "COFACTOR: Name=[4Fe-4S] cluster; Xref=ChEBI:CHEBI:49883; Note=Binds 2 [4Fe-4S] clusters.;\n",
      "['4Fe-4S', 'Direct protein sequencing', 'Electron transport', 'Iron', 'Iron-sulfur', 'Metal-binding', 'Repeat', 'Transport']\n",
      "AHRITEECTYCAACEPECPVNAISAGDEIYIVDESVCTDCEGYYDEPACVAVCPVDCIIKV\n",
      "61\n",
      "5\n",
      "\n",
      "FUNCTION: Ferredoxins are iron-sulfur proteins that transfer electrons in a wide variety of metabolic reactions.\n",
      "COFACTOR: Name=[4Fe-4S] cluster; Xref=ChEBI:CHEBI:49883; Note=Binds 2 [4Fe-4S] clusters.;\n",
      "SUBUNIT: Homodimer.\n",
      "SUBCELLULAR LOCATION: Periplasm.\n",
      "['4Fe-4S', 'Direct protein sequencing', 'Electron transport', 'Iron', 'Iron-sulfur', 'Metal-binding', 'Periplasm', 'Repeat', 'Transport']\n",
      "MGYSVIVDSDKCIGCGECVDVCPVEVYELQNGKAVPVNEEECLGCESCIEVCPQNAIVE\n",
      "59\n",
      "1\n",
      "\n",
      "FUNCTION: Electron transport protein for the cytochrome P-450-SU2 system.\n",
      "COFACTOR: Name=[3Fe-4S] cluster; Xref=ChEBI:CHEBI:21137; Note=Binds 1 [3Fe-4S] cluster.;\n",
      "INDUCTION: By herbicides.\n",
      "['3Fe-4S', 'Direct protein sequencing', 'Electron transport', 'Iron', 'Iron-sulfur', 'Metal-binding', 'Transport']\n",
      "MRIHVDQDKCCGAGSCVLAAPDVFDQREEDGIVVLLDTAPPAALHDAVREAATICPAAAITVTD\n",
      "64\n",
      "6\n",
      "\n",
      "FUNCTION: Ferredoxins are iron-sulfur proteins that transfer electrons in a wide variety of metabolic reactions.\n",
      "COFACTOR: Name=[4Fe-4S] cluster; Xref=ChEBI:CHEBI:49883; Note=Binds 2 [4Fe-4S] clusters.;\n",
      "['4Fe-4S', 'Complete proteome', 'Electron transport', 'Iron', 'Iron-sulfur', 'Metal-binding', 'Reference proteome', 'Repeat', 'Transport']\n",
      "MSLKITEECTFCAACEPECPVNAISAGSDIYVIDESACTECEGYADSPACVAVCPAECIVKA\n",
      "62\n",
      "8\n",
      "\n",
      "FUNCTION: Ferredoxins are iron-sulfur proteins that transfer electrons in a wide variety of metabolic reactions.\n",
      "COFACTOR: Note=Binds 2 4Fe-4S clusters, or 1 4Fe-4S cluster and 1 3Fe-4S cluster.;\n",
      "['4Fe-4S', 'Direct protein sequencing', 'Electron transport', 'Iron', 'Iron-sulfur', 'Metal-binding', 'Repeat', 'Transport']\n",
      "GYKITIDTDKCTGDGECVDVCPVEVYELQDGKAVAVNEDECLGCESCVEVCEQDALTVEEN\n",
      "61\n",
      "6\n",
      "\n",
      "FUNCTION: Electron transport protein for the cytochrome P-450-SOY system.\n",
      "COFACTOR: Name=[3Fe-4S] cluster; Xref=ChEBI:CHEBI:21137;   Evidence={ECO:0000250}; Note=Binds 1 [3Fe-4S] cluster. {ECO:0000250};\n",
      "['3Fe-4S', 'Electron transport', 'Iron', 'Iron-sulfur', 'Metal-binding', 'Transport']\n",
      "MGVQVDKERCVGAGMCALTAPDVFTQDDDGLSEVLPGREATSGTHPLVGEAVRACPVGAVVLSSD\n",
      "65\n",
      "4\n",
      "\n",
      "FUNCTION: Ferredoxins are iron-sulfur proteins that transfer electrons in a wide variety of metabolic reactions.\n",
      "COFACTOR: Name=[4Fe-4S] cluster; Xref=ChEBI:CHEBI:49883; Note=Binds 2 [4Fe-4S] clusters.;\n",
      "['4Fe-4S', 'Direct protein sequencing', 'Electron transport', 'Iron', 'Iron-sulfur', 'Metal-binding', 'Repeat', 'Transport']\n",
      "AYKITDECIACGSCADQCPVEAISEGSIYEIDEALCTDCGACADQCPVEAIVPED\n",
      "55\n",
      "2\n",
      "\n",
      "FUNCTION: An anti-sigma factor for sporulation specific sigma-F factor, by antagonizing sigma-F it allows the switch to sigma-G factor and progression to the late sporulation development stages (PubMed:21037003). Might stabilize or process Holliday junction intermediates, although this may be due to polar effects on the downstream mfd gene (PubMed:15317759). {ECO:0000269|PubMed:15317759, ECO:0000269|PubMed:21037003}.\n",
      "COFACTOR: Name=Zn(2+); Xref=ChEBI:CHEBI:29105;   Evidence={ECO:0000305|PubMed:21037003};\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000269|PubMed:21037003}.\n",
      "DEVELOPMENTAL STAGE: Diffusely expressed in the forespore throughout spore development. {ECO:0000269|PubMed:21037003}.\n",
      "INDUCTION: Induced at 2 hours after sporulation onset as part of the sigma-F regulon, followed by further induction after 3 hours as part of the sigma-G regulon; its concentration rises throughout sporulation (at protein level) (PubMed:21037003). {ECO:0000269|PubMed:21037003}.\n",
      "DISRUPTION PHENOTYPE: Cells missing this gene prolong the activity of sigma-F factor and do not fully induce the activity of sigma-G factor; sporulation efficiency is reduced 50-fold with most spores arresting in the engulfment stage (III) (PubMed:21037003). About 10-fold increased sensitivity to DNA damaging agents, 60% reduced chromosomal DNA transformation; plasmid transformation is unaffected. Partially suppresses DNA damaging agent sensitivity of recU deletions (PubMed:11810266). Partially suppresses DNA damaging agent sensitivity of recU deletions (PubMed:11810266). Partially suppresses DNA repair and chromosome segregation defects in ruvA and recG mutants cells, but not in recD or recU mutant cells (PubMed:15317759). {ECO:0000269|PubMed:11810266, ECO:0000269|PubMed:15317759, ECO:0000269|PubMed:21037003}.\n",
      "CAUTION: Recombination phenotypes may be due to polar effects on the downstream gene mfd rather than due to loss of the anti-sigma factor (PubMed:21037003). {ECO:0000305|PubMed:21037003}.\n",
      "['Complete proteome', 'Cytoplasm', 'Metal-binding', 'Reference proteome', 'Sporulation', 'Transcription', 'Transcription regulation', 'Zinc']\n",
      "MALHYYCRHCGVKVGSLESSMVSTDSLGFQHLTNEERNDMISYKENGDVHVLTICEDCQEALDRNPHYHEYHTFIQ\n",
      "76\n",
      "4\n",
      "\n",
      "FUNCTION: Component of the core of the flagella.\n",
      "SUBUNIT: The flagellum consists of an outer layer composed of repeating units of FlaA around a core that contains one or all of five antigenically related polypeptides.\n",
      "SUBCELLULAR LOCATION: Periplasmic flagellum. Periplasm.\n",
      "SIMILARITY: Belongs to the bacterial flagellin family. {ECO:0000305}.\n",
      "['Bacterial flagellum', 'Direct protein sequencing', 'Periplasm']\n",
      "MIINHNMSAINANRVLGBT\n",
      "19\n",
      "5\n",
      "\n",
      "FUNCTION: Low-potential electron donor to a number of redox enzymes. NifF is the electron donor to nitrogenase.\n",
      "COFACTOR: Name=FMN; Xref=ChEBI:CHEBI:58210;\n",
      "SIMILARITY: Belongs to the flavodoxin family. {ECO:0000305}.\n",
      "['Electron transport', 'Flavoprotein', 'FMN', 'Nitrogen fixation', 'Transport']\n",
      "MANIGIFFGTDTGKTRKIAKMIHKQLGELADAPVNINRTTLDDFMAYPVLLLGTPTLGDGQL\n",
      "62\n",
      "9\n",
      "\n",
      "SUBUNIT: The basal body constitutes a major portion of the flagellar organelle and consists of four rings (L,P,S, and M) mounted on a central rod. The rod consists of about 26 subunits of FlgG in the distal portion, and FlgB, FlgC and FlgF are thought to build up the proximal portion of the rod with about 6 subunits each (By similarity). {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Bacterial flagellum basal body {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the flagella basal body rod proteins family. {ECO:0000305}.\n",
      "['Bacterial flagellum']\n",
      "MGLFSSINTASTGLTAQRLRLDVIANNIANVETTRTSEGGSYRRQRVIFAPRVVSPYWK\n",
      "59\n",
      "2\n",
      "\n",
      "FUNCTION: Required for flagellar hook formation. May act as a scaffolding protein.\n",
      "SIMILARITY: Belongs to the FlgD family. {ECO:0000305}.\n",
      "['Bacterial flagellum biogenesis']\n",
      "MAITNSLTDSDYGINSTTGTSSTTGSSSQDLQNSFLTLLVAQLKNQD\n",
      "47\n",
      "1\n",
      "\n",
      "FUNCTION: Variable subunit of the ferredoxin-thioredoxin reductase (FTR), which catalyzes the two-electron reduction of thioredoxins by the electrons provided by reduced ferredoxin. {ECO:0000269|PubMed:17611542}.\n",
      "SUBUNIT: Heterodimer of subunit A (variable subunit) and subunit B (catalytic subunit). Heterodimeric FTR forms a complex with ferredoxin and thioredoxin. {ECO:0000269|PubMed:10649999, ECO:0000269|PubMed:17611542}.\n",
      "INTERACTION: P27320:petF; NbExp=2; IntAct=EBI-863219, EBI-863421;\n",
      "SIMILARITY: Belongs to the ferredoxin thioredoxin reductase alpha subunit family. {ECO:0000305}.\n",
      "['3D-structure', 'Complete proteome', 'Oxidoreductase', 'Reference proteome']\n",
      "MNVGDRVRVTSSVVVYHHPEHKKTAFDLQGMEGEVAAVLTEWQGRPISANLPVLVKFEQRFKAHFRPDEVTLIED\n",
      "75\n",
      "5\n",
      "\n",
      "FUNCTION: Essential cell division protein. May link together the upstream cell division proteins, which are predominantly cytoplasmic, with the downstream cell division proteins, which are predominantly extracellular. {ECO:0000255|HAMAP-Rule:MF_00599}.\n",
      "SUBCELLULAR LOCATION: Cell membrane; Single-pass type II membrane protein. Note=Localizes to the division septum. {ECO:0000255|HAMAP-Rule:MF_00599}.\n",
      "SIMILARITY: Belongs to the FtsB family. {ECO:0000255|HAMAP- Rule:MF_00599}.\n",
      "['Cell cycle', 'Cell division', 'Cell membrane', 'Complete proteome', 'Membrane', 'Reference proteome', 'Transmembrane', 'Transmembrane helix']\n",
      "MKILKIFLLSLLFWLQYSLWFGKNGVLDFIKIYRRVTIEKKNNEYLDMRNNQIILEIENFNNHINKDKKKT\n",
      "71\n",
      "6\n",
      "\n",
      "FUNCTION: Essential cell division protein. {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cell membrane {ECO:0000250}; Single-pass type II membrane protein {ECO:0000250}. Note=Localizes to the division septum where it forms a ring structure. {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the FtsL family. {ECO:0000305}.\n",
      "['Cell cycle', 'Cell division', 'Cell membrane', 'Complete proteome', 'Membrane', 'Reference proteome', 'Transmembrane', 'Transmembrane helix']\n",
      "MKIQRYDLPKIIKNDFLLYGKIHLTLLIAIILSASLVIITVYNTRLLITEEEDVIQKKKKKMMNGEI\n",
      "67\n",
      "8\n",
      "\n",
      "CATALYTIC ACTIVITY: GTP + H(2)O = formate + 2-amino-4-hydroxy-6- (erythro-1,2,3-trihydroxypropyl)-dihydropteridine triphosphate.\n",
      "PATHWAY: Cofactor biosynthesis; 7,8-dihydroneopterin triphosphate biosynthesis; 7,8-dihydroneopterin triphosphate from GTP: step 1/1.\n",
      "SUBUNIT: Homopolymer.\n",
      "SIMILARITY: Belongs to the GTP cyclohydrolase I family. {ECO:0000305}.\n",
      "['GTP-binding', 'Hydrolase', 'Metal-binding', 'Nucleotide-binding', 'One-carbon metabolism', 'Zinc']\n",
      "VEAEHMCMTMRGVKKPGAKTVTSAVRGTFANVAAARAEVLSFIKND\n",
      "46\n",
      "6\n",
      "\n",
      "FUNCTION: Required for the formation of functionally normal spores. Could be involved in the establishment of normal spore coat structure and/or permeability, which allows the access of germinants to their receptor.\n",
      "DEVELOPMENTAL STAGE: Expressed during sporulation, around the time of spore coat synthesis and assembly, in mother cell compartment.\n",
      "INDUCTION: Expression is sigma K-dependent and negatively regulated by GerE.\n",
      "['Germination', 'Sporulation']\n",
      "MNFYVNQSIIINSIKIDSITTSSVFQIGTAGSIKALSKFSNTGGFTEPLRPLQAKGQIISIKPSTSSS\n",
      "68\n",
      "5\n",
      "\n",
      "FUNCTION: Required for the formation of functionally normal spores. Could be involved in the establishment of normal spore coat structure and/or permeability, which allows the access of germinants to their receptor.\n",
      "DEVELOPMENTAL STAGE: Expressed during sporulation, around the time of spore coat synthesis and assembly, in mother cell compartment.\n",
      "INDUCTION: Expression is sigma K-dependent and negatively regulated by GerE.\n",
      "['Germination', 'Sporulation']\n",
      "MNLNVVNRELKVGQIKMNGVSSSALFLIGDANLLILSSILDTPFETVTEGPFVPLVTDVPPTPG\n",
      "64\n",
      "7\n",
      "\n",
      "FUNCTION: Required for the formation of functionally normal spores. Could be involved in the establishment of normal spore coat structure and/or permeability, which allows the access of germinants to their receptor.\n",
      "SIMILARITY: Belongs to the GerPA/GerPF family. {ECO:0000305}.\n",
      "['Complete proteome', 'Germination', 'Reference proteome', 'Sporulation']\n",
      "MPSVVGNLVVQNSNGSFNLGDFYNVSPKENTKAYNGSGASNVGFVVNTFNGVSATNTFDSDVADQDQIGTA\n",
      "71\n",
      "7\n",
      "\n",
      "FUNCTION: Toxic component of a type V toxin-antitoxin (TA) module. Causes membrane damage when induced by MqsR, slowing cell growth and leading to the formation of dormant persister cells; involved with GhoS, its antitoxin, in reducing cell growth during antibacterial stress. Its toxic effects are neutralized by GhoS, which digests ghoT transcripts in a sequence-specific manner. {ECO:0000250|UniProtKB:P64646}.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane {ECO:0000250|UniProtKB:P64646}; Multi-pass membrane protein {ECO:0000305}.\n",
      "SIMILARITY: Belongs to the GhoT/OrtT toxin family. {ECO:0000305}.\n",
      "['Cell inner membrane', 'Cell membrane', 'Complete proteome', 'Membrane', 'Toxin', 'Transmembrane', 'Transmembrane helix']\n",
      "MALFSKILIFYVIGVNISFVIIWFISHEKTHIRLLSAFLVGITWPMSLPVALLFSLF\n",
      "57\n",
      "5\n",
      "\n",
      "FUNCTION: Toxic component of a type V toxin-antitoxin (TA) module. Causes membrane damage when induced by MqsR, slowing cell growth and increasing the formation of dormant persister cells; involved with GhoS, its antitoxin, in reducing cell growth during antibacterial stress (PubMed:24373067). Overexpression causes cell lysis, forming ghost cells; both effects are neutralized by expression of GhoS. Overexpression in the presence of ampicillin increases persister cell formation (persister cells exhibit antibiotic tolerance without genetic change) (PubMed:22941047). Overexpression causes about 90-fold reduction in cellular ATP levels and dissipates the membrane potential (PubMed:24373067). {ECO:0000269|PubMed:22941047, ECO:0000269|PubMed:24373067}.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane {ECO:0000305|PubMed:24373067}; Multi-pass membrane protein {ECO:0000305}. Note=Localizes to the cell pole. {ECO:0000269|PubMed:24373067}.\n",
      "INDUCTION: Expression controlled by its antitoxin GhoS, which digests ghoT transcripts in a sequence-specific manner (PubMed:22941047). Post-transcriptionally regulated by MqsR which acts on the ghoST transcript selectively, degrading the ghoS segment while leaving ghoT intact; conditions which induce MqsR (e.g. overexpression, nalidixic acid, azolocillin or H(2)O(2)) decrease ghoS expression and thus increase ghoT transcripts (PubMed:23289863). {ECO:0000269|PubMed:22941047, ECO:0000269|PubMed:23289863}.\n",
      "DISRUPTION PHENOTYPE: No visible effect in the absence of stress (PubMed:24373067). Reduces MqsR-mediated persistence (overexpression of MqsR increases persister cell formation). Decreased biofilm formation after 8 hours at 30 and 37 degrees Celsius, biofilm production has risen by 24 hours. Slight decrease in swimming motility (PubMed:22941047). Cells grown in 20 ug/ml ampicillin in the absence of ghoT and which overexpress MqsR elongate, suggesting MqsR inhibits cell elongation via ghoT (PubMed:23289863). When single mutant is grown in the presence of antibiotics carbenicillin or cefoxitin initial metabolism is significantly increased over that of wild-type, after 14 hours wild-type is slighlty less active. In a double ghoS-ghoT mutant in presence of the 2 antibiotics metabolism is significantly increased over that of wild-type, but by 9 hours wild-type has caught up and eventually has slightly greater metablic rates (PubMed:24373067). {ECO:0000269|PubMed:22941047, ECO:0000269|PubMed:23289863, ECO:0000269|PubMed:24373067}.\n",
      "SIMILARITY: Belongs to the GhoT/OrtT toxin family. {ECO:0000305}.\n",
      "['Cell inner membrane', 'Cell membrane', 'Complete proteome', 'Membrane', 'Reference proteome', 'Toxin', 'Transmembrane', 'Transmembrane helix']\n",
      "MALFSKILIFYVIGVNISFVIIWFISHEKTHIRLLSAFLVGITWPMSLPVALLFSLF\n",
      "57\n",
      "6\n",
      "\n",
      "MASS SPECTROMETRY: Mass=8970.3; Method=MALDI; Range=1-72; Evidence={ECO:0000269|PubMed:16817906};\n",
      "CAUTION: Oxidation of Cys-43 to form cysteic acid is most probably an artifact generated during sample preparation. {ECO:0000305|PubMed:16817906}.\n",
      "['Direct protein sequencing', 'Heme', 'Iron', 'Metal-binding']\n",
      "EEAIDAQALVDQNCTGCHGSEVYTRDERRVESLDALHGQVRMCEQNLELTWFDDQVDAVTTLLNREYYNFEP\n",
      "72\n",
      "4\n",
      "\n",
      "FUNCTION: An anti-sigma-G factor, prevents premature activation of sigma-G factor in the forespore; overexpression leads to 1000-fold reduction in spore formation, spore formation stops after engulfment (PubMed:17921305, PubMed:19497328). Overexpression also inhibits sigma-G transcription activation activity (PubMed:18208527). When both Gin and sigma-G are expressed in E.coli Gin inhibits sigma-G, strongly suggesting Gin inhibits by direct physical interaction (PubMed:19497328). {ECO:0000269|PubMed:17921305, ECO:0000269|PubMed:18208527, ECO:0000269|PubMed:19497328}.\n",
      "COFACTOR: Name=Zn(2+); Xref=ChEBI:CHEBI:29105;   Evidence={ECO:0000269|PubMed:19497328}; Note=Binds 0.5 mol of zinc/mol protein, probably 1 zinc per dimer. {ECO:0000269|PubMed:19497328};\n",
      "SUBUNIT: Probably functions as a homodimer (PubMed:19497328). Interacts with sigma-G factor, recognition occurs via the first 71 residues of sigma-G (PubMed:18208527, PubMed:19497328). {ECO:0000269|PubMed:18208527, ECO:0000269|PubMed:19497328}.\n",
      "DEVELOPMENTAL STAGE: Expressed starting 2 hours after sporulation onset for at least 7 hours. {ECO:0000269|PubMed:8759874}.\n",
      "INDUCTION: During sporulation under control of sigma-F factor. {ECO:0000269|PubMed:8759874}.\n",
      "DISRUPTION PHENOTYPE: Premature expression of sigma-G factor (sigG or spoIIIG) activity during the early stages of forespore development (PubMed:17921305, PubMed:18208527). Spore formation continues normally (PubMed:17921305, PubMed:8759874). However its absence has deleterious effects on strain robustness and is strongly selected against in competitions experiments (PubMed:18208527). {ECO:0000269|PubMed:17921305, ECO:0000269|PubMed:18208527, ECO:0000269|PubMed:8759874}.\n",
      "['Complete proteome', 'Metal-binding', 'Reference proteome', 'Sporulation', 'Transcription', 'Transcription regulation', 'Zinc']\n",
      "MDETVKLNHTCVICDQEKNRGIHLYTKFICLDCERKVISTSTSDPDYAFYVKKLKSIHTPPLYS\n",
      "64\n",
      "8\n",
      "\n",
      "FUNCTION: Major determinant of cell surface composition. Negatively regulates motility, adhesion and synthesis of biofilm exopolysaccharides. {ECO:0000255|HAMAP-Rule:MF_00525}.\n",
      "SIMILARITY: Belongs to the GlgS family. {ECO:0000255|HAMAP- Rule:MF_00525}.\n",
      "['Complete proteome']\n",
      "MNNNNVYSLNNFDFLARSFARMQAEGRPVDIQAVTGNMDEEHRDWFCKRYALYCQQATQAKRLELEH\n",
      "67\n",
      "9\n",
      "\n",
      "FUNCTION: Catalyzes the last two sequential reactions in the de novo biosynthetic pathway for UDP-N-acetylglucosamine (UDP- GlcNAc). The C-terminal domain catalyzes the transfer of acetyl group from acetyl coenzyme A to glucosamine-1-phosphate (GlcN-1-P) to produce N-acetylglucosamine-1-phosphate (GlcNAc-1-P), which is converted into UDP-GlcNAc by the transfer of uridine 5- monophosphate (from uridine 5-triphosphate), a reaction catalyzed by the N-terminal domain (By similarity). {ECO:0000250}.\n",
      "CATALYTIC ACTIVITY: Acetyl-CoA + alpha-D-glucosamine 1-phosphate = CoA + N-acetyl-alpha-D-glucosamine 1-phosphate.\n",
      "CATALYTIC ACTIVITY: UTP + N-acetyl-alpha-D-glucosamine 1-phosphate = diphosphate + UDP-N-acetyl-alpha-D-glucosamine.\n",
      "COFACTOR: Name=Mg(2+); Xref=ChEBI:CHEBI:18420; Evidence={ECO:0000250}; Note=Binds 1 Mg(2+) ion per subunit. {ECO:0000250};\n",
      "PATHWAY: Nucleotide-sugar biosynthesis; UDP-N-acetyl-alpha-D- glucosamine biosynthesis; N-acetyl-alpha-D-glucosamine 1-phosphate from alpha-D-glucosamine 6-phosphate (route II): step 2/2.\n",
      "PATHWAY: Nucleotide-sugar biosynthesis; UDP-N-acetyl-alpha-D- glucosamine biosynthesis; UDP-N-acetyl-alpha-D-glucosamine from N- acetyl-alpha-D-glucosamine 1-phosphate: step 1/1.\n",
      "PATHWAY: Bacterial outer membrane biogenesis; LPS lipid A biosynthesis.\n",
      "SUBUNIT: Homotrimer. {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "SIMILARITY: In the N-terminal section; belongs to the N- acetylglucosamine-1-phosphate uridyltransferase family. {ECO:0000305}.\n",
      "SIMILARITY: In the C-terminal section; belongs to the transferase hexapeptide repeat family. {ECO:0000305}.\n",
      "['Acyltransferase', 'Cell shape', 'Cell wall biogenesis/degradation', 'Cytoplasm', 'Magnesium', 'Metal-binding', 'Multifunctional enzyme', 'Nucleotidyltransferase', 'Peptidoglycan synthesis', 'Repeat', 'Transferase']\n",
      "MSKRYAVILAAGQGTRMKSSLYKVLHPVCGKPMVQHVIDQLSRLDLTNLITVVGHGAEKVKSHVGDKS\n",
      "68\n",
      "6\n",
      "\n",
      "FUNCTION: Overexpression increases levels of unsaturated fatty acids and suppresses both the temperature-sensitive fabA6 mutation and cold-sensitive secG null mutation. {ECO:0000269|PubMed:11544213}.\n",
      "INTERACTION: Self; NbExp=2; IntAct=EBI-1121356, EBI-1121356;\n",
      "SIMILARITY: Belongs to the gns family. {ECO:0000305}.\n",
      "['3D-structure', 'Complete proteome', 'Reference proteome']\n",
      "MNIEELKKQAETEIADFIAQKIAELNKNTGKEVSEIRFTAREKMTGLESYDVKIKIM\n",
      "57\n",
      "7\n",
      "\n",
      "FUNCTION: Overexpression increases levels of unsaturated fatty acids and suppresses both the temperature-sensitive fabA6 mutation and cold-sensitive secG null mutation. {ECO:0000269|PubMed:11544213}.\n",
      "SIMILARITY: Belongs to the gns family. {ECO:0000305}.\n",
      "['Complete proteome', 'Reference proteome']\n",
      "MNIENLKTKAEADISEYITKKIIELKKKTGKEVTSIQFTAREKMTGLESYDVKINLI\n",
      "57\n",
      "8\n",
      "\n",
      "FUNCTION: Maintains high levels of reduced glutathione in the cytosol.\n",
      "CATALYTIC ACTIVITY: 2 glutathione + NADP(+) = glutathione disulfide + NADPH.\n",
      "COFACTOR: Name=FAD; Xref=ChEBI:CHEBI:57692; Evidence={ECO:0000250}; Note=Binds 1 FAD per subunit. {ECO:0000250};\n",
      "SUBCELLULAR LOCATION: Cytoplasm.\n",
      "MISCELLANEOUS: The active site is a redox-active disulfide bond.\n",
      "SIMILARITY: Belongs to the class-I pyridine nucleotide-disulfide oxidoreductase family. {ECO:0000305}.\n",
      "['Cytoplasm', 'Direct protein sequencing', 'Disulfide bond', 'FAD', 'Flavoprotein', 'NADP', 'Oxidoreductase', 'Redox-active center']\n",
      "VAIAENSVVGGTCVIRGCVPKLTLTGDDTEPLIVDALLCATGRTTQSNIFAVGDCTDR\n",
      "58\n",
      "2\n",
      "\n",
      "FUNCTION: Involved in a type II secretion system (T2SS, formerly general secretion pathway, GSP) for the export of proteins. {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane {ECO:0000305}.\n",
      "SIMILARITY: Belongs to the GSP N family. {ECO:0000305}.\n",
      "['Cell inner membrane', 'Cell membrane', 'Membrane', 'Protein transport', 'Transmembrane', 'Transport']\n",
      "LQANRQYLFQGSLKPGPELPEEMKQGLPFLGQPDGQGRFPLRYQGRI\n",
      "47\n",
      "2\n",
      "\n",
      "FUNCTION: May confer stability to the gas vesicle membranes. Gas vesicles are small, hollow, gas filled protein structures that are found in several microbial planktonic microorganisms. They allow the positioning of the organism at the favorable depth for growth. {ECO:0000303|PubMed:1527496}.\n",
      "SUBCELLULAR LOCATION: Vacuole, gas vesicle membrane {ECO:0000269|PubMed:1527496}.\n",
      "SIMILARITY: Belongs to the gas vesicle protein type C family. {ECO:0000303|PubMed:1527496}.\n",
      "['Direct protein sequencing', 'Gas vesicle', 'Membrane', 'Vacuole']\n",
      "VSLREQWNEQARERQREISARKTETVALLQEANQERVLVAQQQKALAIELKNQLAQFHEQLETLVGN\n",
      "67\n",
      "10\n",
      "\n",
      "FUNCTION: May confer stability to the gas vesicle membranes. Gas vesicles are small, hollow, gas filled protein structures that are found in several microbial planktonic microorganisms. They allow the positioning of the organism at the favorable depth for growth. {ECO:0000303|PubMed:1527496}.\n",
      "SUBCELLULAR LOCATION: Vacuole, gas vesicle membrane {ECO:0000269|PubMed:1527496}.\n",
      "SIMILARITY: Belongs to the gas vesicle protein type C family. {ECO:0000303|PubMed:1527496}.\n",
      "['Direct protein sequencing', 'Gas vesicle', 'Membrane', 'Vacuole']\n",
      "VSLREQWNEQARERQREISARKTEIEEARSSIQKELQAQYEAALAATQQLQSLXXRFXRFY\n",
      "61\n",
      "7\n",
      "\n",
      "FUNCTION: Gas vesicles are small, hollow, gas filled protein structures that are found in several microbial planktonic microorganisms. They allow the positioning of the organism at the favorable depth for growth. GvpA type proteins form the essential core of the structure.\n",
      "SUBCELLULAR LOCATION: Vacuole, gas vesicle membrane.\n",
      "SIMILARITY: Belongs to the gas vesicle protein type A family. {ECO:0000305}.\n",
      "['Direct protein sequencing', 'Gas vesicle', 'Membrane', 'Vacuole']\n",
      "AVEKTNSSSSLGEVVDRILDKGVVVDLWVRVSLVGIELLALEARVVI\n",
      "47\n",
      "3\n",
      "\n",
      "FUNCTION: Gas vesicles are small, hollow, gas filled protein structures that are found in several microbial planktonic microorganisms. They allow the positioning of the organism at the favorable depth for growth. GvpA type proteins form the essential core of the structure.\n",
      "SUBCELLULAR LOCATION: Vacuole, gas vesicle membrane.\n",
      "SIMILARITY: Belongs to the gas vesicle protein type A family. {ECO:0000305}.\n",
      "['Direct protein sequencing', 'Gas vesicle', 'Membrane', 'Vacuole']\n",
      "MAVEKTNSSSSLAEVIDRILDKGIVIDAWVRVSLVGIELLAIEARIVIASVETYLKYAEAVGLTQSAAVPA\n",
      "71\n",
      "8\n",
      "\n",
      "FUNCTION: Gas vesicles are small, hollow, gas filled protein structures that are found in several microbial planktonic microorganisms. They allow the positioning of the organism at the favorable depth for growth. GvpA type proteins form the essential core of the structure. {ECO:0000255|HAMAP-Rule:MF_00576}.\n",
      "SUBCELLULAR LOCATION: Vacuole, gas vesicle membrane {ECO:0000255|HAMAP-Rule:MF_00576}.\n",
      "SIMILARITY: Belongs to the gas vesicle protein type A family. {ECO:0000255|HAMAP-Rule:MF_00576}.\n",
      "['Complete proteome', 'Gas vesicle', 'Membrane', 'Reference proteome', 'Vacuole']\n",
      "MAVEKTNASSSLVEVIDRILDKGVCIDAWARVSLVGIELLAIEARVVVASVDTFLKYAEAVGLTTTATAMHA\n",
      "72\n",
      "5\n",
      "\n",
      "FUNCTION: May confer stability to the gas vesicle membranes. Gas vesicles are small, hollow, gas filled protein structures that are found in several microbial planktonic microorganisms. They allow the positioning of the organism at the favorable depth for growth.\n",
      "SUBCELLULAR LOCATION: Vacuole, gas vesicle membrane; Peripheral membrane protein; Cytoplasmic side. Note=Binds to the external surface of the gas vesicle membrane.\n",
      "SIMILARITY: Belongs to the gas vesicle protein type C family. {ECO:0000305}.\n",
      "['Direct protein sequencing', 'Gas vesicle', 'Membrane', 'Vacuole']\n",
      "ALQDDWQQSHLERRQVLAESQKQIQATIGALXXERQXXA\n",
      "39\n",
      "1\n",
      "\n",
      "FUNCTION: A type II topoisomerase that negatively supercoils closed circular double-stranded (ds) DNA in an ATP-dependent manner to modulate DNA topology and maintain chromosomes in an underwound state. Negative supercoiling favors strand separation, and DNA replication, transcription, recombination and repair, all of which involve strand separation. Also able to catalyze the interconversion of other topological isomers of dsDNA rings, including catenanes and knotted rings. Type II topoisomerases break and join 2 DNA strands simultaneously in an ATP-dependent manner. {ECO:0000255|HAMAP-Rule:MF_01897}.\n",
      "CATALYTIC ACTIVITY: ATP-dependent breakage, passage and rejoining of double-stranded DNA. {ECO:0000255|HAMAP-Rule:MF_01897}.\n",
      "SUBUNIT: Heterotetramer, composed of two GyrA and two GyrB chains. In the heterotetramer, GyrA contains the active site tyrosine that forms a transient covalent intermediate with DNA, while GyrB binds cofactors and catalyzes ATP hydrolysis. {ECO:0000255|HAMAP- Rule:MF_01897}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000255|HAMAP-Rule:MF_01897, ECO:0000305}.\n",
      "MISCELLANEOUS: Few gyrases are as efficient as E.coli at forming negative supercoils. Not all organisms have 2 type II topoisomerases; in organisms with a single type II topoisomerase this enzyme also has to decatenate newly replicated chromosomes. {ECO:0000255|HAMAP-Rule:MF_01897}.\n",
      "SIMILARITY: Belongs to the type II topoisomerase GyrA/ParC subunit family. {ECO:0000255|HAMAP-Rule:MF_01897}.\n",
      "['ATP-binding', 'Cytoplasm', 'DNA-binding', 'Isomerase', 'Nucleotide-binding', 'Topoisomerase']\n",
      "MADENTPVMPEEVPAVEGVGM\n",
      "21\n",
      "4\n",
      "\n",
      "FUNCTION: Histone acetyltransferase, which by binding to the host chromatin, may manipulate the expression of host genes involved in anti-inflammatory responses to evade clearance and to survive in the intracellular milieu. Acetylates histone H3 at the 'Lys-9' and 'Lys-14' positions. {ECO:0000250|UniProtKB:P9WKY5}.\n",
      "CATALYTIC ACTIVITY: Acetyl-CoA + [histone] = CoA + acetyl- [histone]. {ECO:0000250|UniProtKB:P9WKY5}.\n",
      "ENZYME REGULATION: Is completely inhibited by anacardic acid, an inhibitor of HAT activity. {ECO:0000250|UniProtKB:P9WKY5}.\n",
      "SUBUNIT: Physically interacts with histone H3 in infected macrophages. {ECO:0000250|UniProtKB:P9WKY5}.\n",
      "SUBCELLULAR LOCATION: Secreted {ECO:0000250|UniProtKB:P9WKY5}. Host cytoplasm {ECO:0000250|UniProtKB:P9WKY5}. Host nucleus {ECO:0000250|UniProtKB:P9WKY5}. Note=Colocalizes with the chromatin in the nucleus of infected human macrophages. {ECO:0000250|UniProtKB:P9WKY5}.\n",
      "['Acyltransferase', 'Complete proteome', 'Host cytoplasm', 'Host nucleus', 'Secreted', 'Transferase']\n",
      "MSFGFPTFSQNRFTEQYSGLCPIAPGRGAGLQPCRRDCPVARWLVADHPVFGSDCRCRMMVGVNRVRIGRHELTGA\n",
      "76\n",
      "5\n",
      "\n",
      "FUNCTION: Histone acetyltransferase, which by binding to the host chromatin, may manipulate the expression of host genes involved in anti-inflammatory responses to evade clearance and to survive in the intracellular milieu. Acetylates histone H3 at the 'Lys-9' and 'Lys-14' positions. {ECO:0000269|PubMed:26476134}.\n",
      "CATALYTIC ACTIVITY: Acetyl-CoA + [histone] = CoA + acetyl- [histone]. {ECO:0000269|PubMed:26476134}.\n",
      "ENZYME REGULATION: Is completely inhibited by anacardic acid, an inhibitor of HAT activity. {ECO:0000269|PubMed:26476134}.\n",
      "SUBUNIT: Physically interacts with histone H3 in infected macrophages. {ECO:0000269|PubMed:26476134}.\n",
      "SUBCELLULAR LOCATION: Secreted {ECO:0000269|PubMed:26476134}. Host cytoplasm {ECO:0000269|PubMed:26476134}. Host nucleus {ECO:0000269|PubMed:26476134}. Note=Colocalizes with the chromatin in the nucleus of infected human macrophages. {ECO:0000269|PubMed:26476134}.\n",
      "INDUCTION: Expression of this gene marginally increases (1.5-fold) in macrophages from 12 to 24 h post-infection, and does not decrease up to 60 h. {ECO:0000269|PubMed:26476134}.\n",
      "MASS SPECTROMETRY: Mass=8436.1; Method=MALDI; Range=1-76; Evidence={ECO:0000269|Ref.2};\n",
      "MISCELLANEOUS: On the 2D-gel the determined pI of this protein is: 9.47.\n",
      "['Acyltransferase', 'Complete proteome', 'Direct protein sequencing', 'Host cytoplasm', 'Host nucleus', 'Reference proteome', 'Secreted', 'Transferase']\n",
      "MSFGFPTFSQNRFTEQYSGLCPIAPGRGAGLQPCRRDCPVARWLVADHPVFGSDCRCRMMVGVNRVRIGRHELTGA\n",
      "76\n",
      "6\n",
      "\n",
      "CATALYTIC ACTIVITY: H(2) + 2 ferricytochrome c3 = 2 H(+) + 2 ferrocytochrome c3.\n",
      "COFACTOR: Name=Fe cation; Xref=ChEBI:CHEBI:24875;\n",
      "['Direct protein sequencing', 'Iron', 'Oxidoreductase']\n",
      "ANVLWLQGGAXSGNTMSFLNAEEPSAXDLVTDFGINVLXQ\n",
      "40\n",
      "2\n",
      "\n",
      "FUNCTION: Degrades heparin and heparan sulfate. {ECO:0000250|UniProtKB:Q05819}.\n",
      "CATALYTIC ACTIVITY: Eliminative cleavage of polysaccharides containing (1->4)-linked D-glucuronate or L-iduronate residues and (1->4)-alpha-linked 2-sulfoamino-2-deoxy-6-sulfo-D-glucose residues to give oligosaccharides with terminal 4-deoxy-alpha-D- gluc-4-enuronosyl groups at their non-reducing ends.\n",
      "ENZYME REGULATION: Inhibited by Cu(2+) ion, lead and some agents that modify histidine and cysteine residues. Activated by KCl and by reducing agents, such as dithiothreitol and 2-mercaptoethanol.\n",
      "BIOPHYSICOCHEMICAL PROPERTIES: pH dependence:   Optimum pH is 7.2.; Temperature dependence:   Optimum temperature is 45 degrees Celsius.;\n",
      "PTM: The N-terminus is blocked. {ECO:0000269|PubMed:10920269}.\n",
      "['Direct protein sequencing', 'Heparin-binding', 'Lyase']\n",
      "MADEALQHTFFA\n",
      "12\n",
      "1\n",
      "\n",
      "FUNCTION: Down-regulates hemolysin expression, in complex with H- NS, and can also stimulate transposition events in vivo. Modifies the set of genes regulated by H-NS; Hha and Cnu (YdgT) increase the number of genes DNA bound by H-NS/StpA and may also modulate the oligomerization of the H-NS/StpA-complex. Binds DNA and influences DNA topology in response to environmental stimuli. Decreases biofilm formation (By similarity). {ECO:0000250|UniProtKB:P0ACE3}.\n",
      "SUBUNIT: Forms a heterotrimeric complex with the H-NS dimer. {ECO:0000250|UniProtKB:P0ACE3}.\n",
      "SIMILARITY: Belongs to the Hha/YmoA/Cnu family. {ECO:0000305}.\n",
      "['Complete proteome', 'DNA-binding', 'Repressor', 'Transcription', 'Transcription regulation']\n",
      "MSEKPLTKTDYLMRLRRCQTIDTLERVIEKNKYELSDNELAVFYSAADHRLAELTMNKLYDKIPSSVWKFIR\n",
      "72\n",
      "6\n",
      "\n",
      "FUNCTION: Toxic component of a toxin-antitoxin (TA) module. A probable translation-independent mRNA interferase (By similarity). {ECO:0000250}.\n",
      "SUBUNIT: Probably forms a complex with the cognate antitoxin HicB 1 which inhibits the mRNA interferase activity. {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the HicA mRNA interferase family. {ECO:0000305}.\n",
      "['Complete proteome', 'Endonuclease', 'Hydrolase', 'Nuclease', 'Reference proteome', 'RNA-binding', 'Stress response', 'Toxin']\n",
      "MKQSEFRRWLEAQGVEVSNGTNHLKLRYQGKRSIMPRHPSQEIKEPLRKAIIKQLGL\n",
      "57\n",
      "9\n",
      "\n",
      "FUNCTION: Toxic component of a toxin-antitoxin (TA) module. A probable translation-independent mRNA interferase (By similarity). {ECO:0000250}.\n",
      "SUBUNIT: Probably forms a complex with the cognate antitoxin HicB 2 which inhibits the mRNA interferase activity. {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the HicA mRNA interferase family. {ECO:0000305}.\n",
      "SEQUENCE CAUTION: Sequence=CAE16725.1; Type=Erroneous initiation; Note=Translation N-terminally shortened.; Evidence={ECO:0000305};\n",
      "['Complete proteome', 'Endonuclease', 'Hydrolase', 'Nuclease', 'Reference proteome', 'RNA-binding', 'Stress response', 'Toxin']\n",
      "MKQSEFRRWLKAQGAEFKDGTNHLKVFLNGKQTTMPRHPSKEIPEPLRKLILKQPGIK\n",
      "58\n",
      "3\n",
      "\n",
      "FUNCTION: Toxic component of a toxin-antitoxin (TA) module. A probable translation-independent mRNA interferase (By similarity). {ECO:0000250}.\n",
      "SUBUNIT: Probably forms a complex with the antitoxin HicB which inhibits the mRNA interferase activity. {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the HicA mRNA interferase family. {ECO:0000305}.\n",
      "['Complete proteome', 'Endonuclease', 'Hydrolase', 'Nuclease', 'RNA-binding', 'Stress response', 'Toxin']\n",
      "MKQSEFRRWLESQGVDVANGSNHLKLRFHGRRSVMPRHPCDEIKEPLRKAILKQLGLS\n",
      "58\n",
      "4\n",
      "\n",
      "FUNCTION: Specific class of high-redox-potential 4Fe-4S ferredoxins. Functions in anaerobic electron transport in most purple and in some other photosynthetic bacteria and in at least one genus (Paracoccus) of halophilic, denitrifying bacteria.\n",
      "BIOPHYSICOCHEMICAL PROPERTIES: Redox potential:   E(0) is +120 mV.;\n",
      "SUBUNIT: Homodimer. {ECO:0000305}.\n",
      "MISCELLANEOUS: In E.halophila, two HiPIP isozymes are found.\n",
      "SIMILARITY: Belongs to the high-potential iron-sulfur protein (HiPIP) family. {ECO:0000255|PROSITE-ProRule:PRU00705}.\n",
      "['3D-structure', '4Fe-4S', 'Direct protein sequencing', 'Electron transport', 'Iron', 'Iron-sulfur', 'Metal-binding', 'Transport']\n",
      "EPRAEDGHAHDYVNEAADASGHPRYQEGQLCENCAFWGEAVQDGWGRCTHPDFDEVLVKAEGWCSVYAPAS\n",
      "71\n",
      "9\n",
      "\n",
      "FUNCTION: Specific class of high-redox-potential 4Fe-4S ferredoxins. Functions in anaerobic electron transport in most purple and in some other photosynthetic bacteria and in at least one genus (Paracoccus) of halophilic, denitrifying bacteria.\n",
      "BIOPHYSICOCHEMICAL PROPERTIES: Redox potential:   E(0) is +330 mV.;\n",
      "SUBUNIT: Homodimer. {ECO:0000305}.\n",
      "SIMILARITY: Belongs to the high-potential iron-sulfur protein (HiPIP) family. {ECO:0000255|PROSITE-ProRule:PRU00705}.\n",
      "['4Fe-4S', 'Direct protein sequencing', 'Electron transport', 'Iron', 'Iron-sulfur', 'Metal-binding', 'Transport']\n",
      "GTNASMRKAFNYQEVSKTAGKNCANCAQFIPGASASAAGACKVIPGDSQIQPTGYCDAYIVKK\n",
      "63\n",
      "4\n",
      "\n",
      "FUNCTION: Specific class of high-redox-potential 4Fe-4S ferredoxins. Functions in anaerobic electron transport in most purple and in some other photosynthetic bacteria and in at least one genus (Paracoccus) of halophilic, denitrifying bacteria.\n",
      "BIOPHYSICOCHEMICAL PROPERTIES: Redox potential:   E(0) is +150 mV.;\n",
      "SUBUNIT: Homodimer.\n",
      "SIMILARITY: Belongs to the high-potential iron-sulfur protein (HiPIP) family. {ECO:0000255|PROSITE-ProRule:PRU00705}.\n",
      "['3D-structure', '4Fe-4S', 'Direct protein sequencing', 'Electron transport', 'Iron', 'Iron-sulfur', 'Metal-binding', 'Transport']\n",
      "MERLSEDDPAAQALEYRHDASSVQHPAYEEGQTCLNCLLYTDASAQDWGPCSVFPGKLVSANGWCTAWVAR\n",
      "71\n",
      "10\n",
      "\n",
      "FUNCTION: Specific class of high-redox-potential 4Fe-4S ferredoxins. Functions in anaerobic electron transport in most purple and in some other photosynthetic bacteria and in at least one genus (Paracoccus) of halophilic, denitrifying bacteria.\n",
      "BIOPHYSICOCHEMICAL PROPERTIES: Redox potential:   E(0) is +50 mV.;\n",
      "SUBUNIT: Homodimer. {ECO:0000305}.\n",
      "MISCELLANEOUS: In E.halophila, two HiPIP isozymes are found.\n",
      "SIMILARITY: Belongs to the high-potential iron-sulfur protein (HiPIP) family. {ECO:0000255|PROSITE-ProRule:PRU00705}.\n",
      "['4Fe-4S', 'Direct protein sequencing', 'Electron transport', 'Iron', 'Iron-sulfur', 'Metal-binding', 'Transport']\n",
      "GLPDGVEDLPKAEDDHAHDYVNDAADTDHARFQEGQLCENCQFWVDYVNGWGYCQHPDFTDVLVRGEGWCSVYAPA\n",
      "76\n",
      "7\n",
      "\n",
      "FUNCTION: Virulence factor. Causes hemolysis of erythrocytes from sheep (HD(50)=7.53 mM), rabbit (HD(50)=28.07 mM), guinea pig (HD(50)=4.23 mM), dog (HD(50)=4.37 mM) and human (HD(50)=3.32 mM). Acts synergistically with beta-hemolysins from S.aureus ATCC 25923. Cytotoxic towards human dermal fibroblasts. {ECO:0000269|PubMed:18752624}.\n",
      "SUBCELLULAR LOCATION: Secreted {ECO:0000269|PubMed:18752624}.\n",
      "SIMILARITY: Belongs to the staphylococcal hemolytic protein family. {ECO:0000255}.\n",
      "['Cytolysis', 'Direct protein sequencing', 'Formylation', 'Hemolysis', 'Secreted', 'Toxin', 'Virulence']\n",
      "MSGIVEAISNAVKSGLDHDWVNMGTSIADVVAKGADFIAGFFS\n",
      "43\n",
      "2\n",
      "\n",
      "FUNCTION: Virulence factor. Causes hemolysis of erythrocytes. Acts synergistically with beta-hemolysins from S.aureus ATCC 25923. Cytotoxic towards human dermal fibroblasts. {ECO:0000269|PubMed:18752624}.\n",
      "SUBCELLULAR LOCATION: Secreted {ECO:0000269|PubMed:18752624}.\n",
      "SIMILARITY: Belongs to the staphylococcal hemolytic protein family. {ECO:0000255}.\n",
      "['Cytolysis', 'Direct protein sequencing', 'Formylation', 'Hemolysis', 'Secreted', 'Toxin', 'Virulence']\n",
      "MSGIVEAISNAVKSGLDHDWVNMGTSIADVVAKGADFIAGFFS\n",
      "43\n",
      "3\n",
      "\n",
      "FUNCTION: Virulence factor. Causes hemolysis of erythrocytes from sheep (HD(50)=2.63 mM), rabbit (HD(50)=2.37 mM), guinea pig (HD(50)=1.98 mM), dog (HD(50)=1.02 mM) and human (HD(50)=2.07 mM). Acts synergistically with beta-hemolysins from S.aureus ATCC 25923. Cytotoxic towards human dermal fibroblasts. {ECO:0000269|PubMed:18752624}.\n",
      "SUBCELLULAR LOCATION: Secreted {ECO:0000269|PubMed:18752624}.\n",
      "SIMILARITY: Belongs to the staphylococcal hemolytic protein family. {ECO:0000255}.\n",
      "['Cytolysis', 'Direct protein sequencing', 'Formylation', 'Hemolysis', 'Secreted', 'Toxin', 'Virulence']\n",
      "MSDFVNAISEAVKAGLSADWVTMGTSIADALAKGADFILGFFN\n",
      "43\n",
      "4\n",
      "\n",
      "FUNCTION: Virulence factor. Causes hemolysis of erythrocytes. Acts synergistically with beta-hemolysins from S.aureus ATCC 25923. Cytotoxic towards human dermal fibroblasts. {ECO:0000269|PubMed:18752624}.\n",
      "SUBCELLULAR LOCATION: Secreted {ECO:0000269|PubMed:18752624}.\n",
      "SIMILARITY: Belongs to the staphylococcal hemolytic protein family. {ECO:0000255}.\n",
      "['Cytolysis', 'Direct protein sequencing', 'Formylation', 'Hemolysis', 'Secreted', 'Toxin', 'Virulence']\n",
      "MSDFVNAISEAVKAGLSADWVTMGTSIADALAKGADFILGFFN\n",
      "43\n",
      "5\n",
      "\n",
      "FUNCTION: HMWC (high-molecular-weight cytochrome c), ORF2, ORF3, ORF4, ORF5 and ORF6 in the HMC operon form a transmembrane protein complex that allows electron flow from the periplasmic hydrogenase to the cytoplasmic enzymes that catalyze reduction of sulfates.\n",
      "SUBCELLULAR LOCATION: Cell membrane; Single-pass membrane protein.\n",
      "['Cell membrane', 'Complete proteome', 'Membrane', 'Reference proteome', 'Transmembrane', 'Transmembrane helix']\n",
      "MDQAIYTLHEFMLHTKNWTYILMGVTLLVYVGYWLFLTGRDEKIRKY\n",
      "47\n",
      "4\n",
      "\n",
      "FUNCTION: Overproduction is toxic. {ECO:0000269|PubMed:20105222}.\n",
      "SUBCELLULAR LOCATION: Membrane {ECO:0000305}; Single-pass membrane protein {ECO:0000305}.\n",
      "INDUCTION: May be repressed by MqsA. {ECO:0000269|PubMed:20105222}.\n",
      "SIMILARITY: Belongs to the hok/gef family. {ECO:0000305}.\n",
      "['Complete proteome', 'Membrane', 'Reference proteome', 'Transmembrane', 'Transmembrane helix']\n",
      "MPQKYRLLSLIVICFTLLFFTWMIRDSLCELHIKQESYELAAFLACKLKE\n",
      "50\n",
      "4\n",
      "\n",
      "FUNCTION: Toxic component of a type I toxin-antitoxin (TA) module. Its toxic effects are probably neutralized by sRNA antitoxin sokB, which is encoded in trans on the opposite DNA strand (Probable). Expression leads to membrane depolarization; when protein levels are high enough depolarization probably leads to lowered metabolic activity which in turn induces some cells to enter the persistent state in which they transiently survive antibiotic exposure. {ECO:0000269|PubMed:26051177, ECO:0000305}.\n",
      "SUBCELLULAR LOCATION: Membrane {ECO:0000305|PubMed:26051177}; Single-pass membrane protein {ECO:0000255}.\n",
      "INDUCTION: Induced by GTPase Obg expression, which requires alarmone (p)ppGpp. {ECO:0000269|PubMed:26051177}.\n",
      "DISRUPTION PHENOTYPE: Partial loss of Obg-controlled persistence. {ECO:0000269|PubMed:26051177}.\n",
      "SIMILARITY: Belongs to the hok/gef family. {ECO:0000305}.\n",
      "['Complete proteome', 'Membrane', 'Reference proteome', 'Toxin', 'Transmembrane', 'Transmembrane helix']\n",
      "MKHNPLVVCLLIICITILTFTLLTRQTLYELRFRDGDKEVAALMACTSR\n",
      "49\n",
      "3\n",
      "\n",
      "FUNCTION: When overexpressed, kills the cells from the inside by interfering with a vital function in the cell membrane.\n",
      "SUBUNIT: Homodimer; disulfide-linked.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the hok/gef family. {ECO:0000305}.\n",
      "['Cell inner membrane', 'Cell membrane', 'Complete proteome', 'Disulfide bond', 'Membrane', 'Reference proteome', 'Signal-anchor', 'Transmembrane', 'Transmembrane helix']\n",
      "MKQHKAMIVALIVICITAVVAALVTRKDLCEVHIRTGQTEVAVFTAYESE\n",
      "50\n",
      "5\n",
      "\n",
      "SUBUNIT: Homodimer; disulfide-linked. {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane {ECO:0000250}; Single- pass type II membrane protein {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the hok/gef family. {ECO:0000305}.\n",
      "SEQUENCE CAUTION: Sequence=AAP15562.1; Type=Erroneous initiation; Evidence={ECO:0000305};\n",
      "['Cell inner membrane', 'Cell membrane', 'Complete proteome', 'Disulfide bond', 'Membrane', 'Signal-anchor', 'Transmembrane', 'Transmembrane helix']\n",
      "MKQHKAMIVALIVICITAVVAALVTRKDLCEVHIRTGQTEVAVFTAYESE\n",
      "50\n",
      "6\n",
      "\n",
      "SUBCELLULAR LOCATION: Membrane {ECO:0000305}; Single-pass membrane protein {ECO:0000305}.\n",
      "SIMILARITY: Belongs to the hok/gef family. {ECO:0000305}.\n",
      "SEQUENCE CAUTION: Sequence=BAB36177.1; Type=Erroneous initiation; Evidence={ECO:0000305};\n",
      "['Complete proteome', 'Membrane', 'Transmembrane', 'Transmembrane helix']\n",
      "MKQQKAMLIALIVICLTVIVTALVTRKDLCEVRIRTGQTEVAVFTAYEPEE\n",
      "51\n",
      "4\n",
      "\n",
      "FUNCTION: When overexpressed kills the cells from the inside by interfering with a vital function in the cell membrane.\n",
      "FUNCTION: Overexpression of relF results in collapse of the transmembrane potential and concomitant arrest of oxygen consumption.\n",
      "SUBCELLULAR LOCATION: Cell membrane {ECO:0000305}; Single-pass membrane protein {ECO:0000305}.\n",
      "SIMILARITY: Belongs to the hok/gef family. {ECO:0000305}.\n",
      "['Cell membrane', 'Complete proteome', 'Membrane', 'Reference proteome', 'Transmembrane', 'Transmembrane helix']\n",
      "MKQQKAMLIALIVICLTVIVTALVTRKDLCEVRIRTGQTEVAVFTAYEPEE\n",
      "51\n",
      "5\n",
      "\n",
      "FUNCTION: DNA polymerase III is a complex, multichain enzyme responsible for most of the replicative synthesis in bacteria. This DNA polymerase also exhibits 3' to 5' exonuclease activity (By similarity). {ECO:0000250}.\n",
      "FUNCTION: The exact function of the theta subunit is unknown. {ECO:0000250}.\n",
      "CATALYTIC ACTIVITY: Deoxynucleoside triphosphate + DNA(n) = diphosphate + DNA(n+1).\n",
      "SUBUNIT: The DNA polymerase holoenzyme is a complex that contains 10 different types of subunits. These subunits are organized into 3 functionally essential subassemblies: the pol III core, the beta sliding clamp processivity factor and the clamp-loading complex. The pol III core (subunits alpha,epsilon and theta) contains the polymerase and the 3'-5' exonuclease proofreading activities. The polymerase is tethered to the template via the sliding clamp processivity factor. The clamp-loading complex assembles the beta processivity factor onto the primer template and plays a central role in the organization and communication at the replication fork. This complex contains delta, delta', psi and chi, and copies of either or both of two different DnaX proteins, gamma and tau. The composition of the holoenzyme is, therefore: (alpha,epsilon,theta)[2]-(gamma/tau)[3]-delta,delta', psi,chi- beta[4] (By similarity). {ECO:0000250}.\n",
      "['Complete proteome', 'DNA replication', 'DNA-directed DNA polymerase', 'Nucleotidyltransferase', 'Transferase']\n",
      "MLKNLAKLDQTEMDKVNVDLAAAGVAFKERYNMPVIAEAVEREQPEHLRSWFRERLIAHRLASVNLSRLPYEPKLK\n",
      "76\n",
      "8\n",
      "\n",
      "FUNCTION: Strongly binds nickel and zinc. Binds other metals less strongly: cobalt > copper > cadmium > manganese. May act to increase, or at least to preserve, urease activity. Exact function is still unknown (By similarity). {ECO:0000250}.\n",
      "['Complete proteome', 'Metal-binding', 'Nickel', 'Repeat', 'Zinc']\n",
      "MAHHEEQHGGHHHHHHHTHHHHYHGGEHHHHHHSSHHEEGCCSTSDSHHQEEGCCHGHHE\n",
      "60\n",
      "4\n",
      "\n",
      "FUNCTION: Major structure protein of the hrp pilus, which is a component of the type III secretion system (TTSS, Hrp secretion system) required for effector protein delivery, parasitism, and pathogenicity. The hrp pilus functions as a conduit for protein delivery into the host cell. HrpA is required for the secretion of HrpN and DspA/E. {ECO:0000269|PubMed:11401717}.\n",
      "SUBCELLULAR LOCATION: Secreted, extracellular space {ECO:0000250}. Fimbrium {ECO:0000250}. Note=Extracellular and secreted via type III secretion system. {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the HrpA type 2 family. {ECO:0000305}.\n",
      "SEQUENCE CAUTION: Sequence=AAB49172.1; Type=Erroneous initiation; Evidence={ECO:0000305};\n",
      "['Fimbrium', 'Secreted', 'Virulence']\n",
      "MAGSSLTSASTSASKTLESAMGQSLTESANAQASKMKMDTQNSILDGKMDSASKSINSGHNAAKAIQF\n",
      "68\n",
      "7\n",
      "\n",
      "FUNCTION: Inhibits mammalian alpha-amylases specifically but has no action on plant and microbial alpha-amylases.\n",
      "MISCELLANEOUS: This protein is an effective inhibitor of pancreatic alpha-amylase. It is 5 times less effective against human salivary amylase.\n",
      "['Alpha-amylase inhibitor', 'Direct protein sequencing', 'Disulfide bond']\n",
      "ATGSPAPDCVESFQSWRYTDVRNGCSDAVTVVVQYE\n",
      "36\n",
      "1\n",
      "\n",
      "FUNCTION: Possible increased expression of this protein (due to mutations upstream of the start codon) is proposed to be responsible for resistance to 3-hydroxypropionic acid (3-HP). {ECO:0000269|PubMed:22161628}.\n",
      "MISCELLANEOUS: Mutagenesis of the start codon from ATG to TTG, which should decrease protein expression, results in loss of 3-HP resistance. {ECO:0000305|PubMed:22161628}.\n",
      "CAUTION: Encoded by the 21 C-terminal amino acids of the HcaR transcriptional activator, it is suggested to be produced by an independent promoter within the hcaR gene. However the peptide's existence has not been proven, nor has mutagenesis of HcaR itself rather than overexpression of IroK been ruled out as the cause of 3-HP resistance. {ECO:0000305}.\n",
      "['Complete proteome', 'Reference proteome']\n",
      "MKPALRDFIAIVQERLASVTA\n",
      "21\n",
      "5\n",
      "\n",
      "FUNCTION: May function as iron donor in the assembly of iron- sulfur clusters. {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the IscX family. {ECO:0000305}.\n",
      "['Complete proteome', 'Reference proteome']\n",
      "MKWTDAQLIAEELYDRNPDLDPKTVRFTDLHKWICELENFDDDPNKSNESILEAILLKWLDEFE\n",
      "64\n",
      "9\n",
      "\n",
      "FUNCTION: Catalyzes the reversible transfer of the terminal phosphate group between ATP and AMP. Plays an important role in cellular energy homeostasis and in adenine nucleotide metabolism. {ECO:0000250|UniProtKB:P69441}.\n",
      "CATALYTIC ACTIVITY: ATP + AMP = 2 ADP. {ECO:0000250|UniProtKB:P69441}.\n",
      "PATHWAY: Purine metabolism; AMP biosynthesis via salvage pathway; AMP from ADP: step 1/1. {ECO:0000250|UniProtKB:P69441}.\n",
      "SUBUNIT: Monomer. {ECO:0000250|UniProtKB:P69441}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250|UniProtKB:P69441}.\n",
      "DOMAIN: Consists of three domains, a large central CORE domain and two small peripheral domains, NMPbind and LID, which undergo movements during catalysis. The LID domain closes over the site of phosphoryl transfer upon ATP binding. Assembling and dissambling the active center during each catalytic cycle provides an effective means to prevent ATP hydrolysis. {ECO:0000250|UniProtKB:P69441}.\n",
      "SIMILARITY: Belongs to the adenylate kinase family. {ECO:0000305}.\n",
      "['ATP-binding', 'Cytoplasm', 'Kinase', 'Nucleotide biosynthesis', 'Nucleotide-binding', 'Transferase']\n",
      "MNIILMGLPGAGKGTQASEIVKKFPIPHISTGDMFIKAIK\n",
      "40\n",
      "3\n",
      "\n",
      "FUNCTION: Lanthionine-containing peptide devoid of antibiotic properties, involved in the formation of aerial mycelium. Suggested to self-assemble at air-water interfaces, thus providing a film of surfactant through which nascent aerial hyphae can emerge. The aerial hyphae differentiate further into spores. {ECO:0000269|PubMed:2032288}.\n",
      "PTM: Maturation involves the enzymic conversion of Ser into dehydrated AA and the formation of thioether bonds with cysteine. This is followed by membrane translocation and cleavage of the modified precursor.\n",
      "MASS SPECTROMETRY: Mass=2027; Method=FAB; Range=22-42; Evidence={ECO:0000269|PubMed:2032288};\n",
      "SIMILARITY: Belongs to the lanthionine-containing morphogen family. {ECO:0000305}.\n",
      "['Complete proteome', 'Direct protein sequencing', 'Reference proteome', 'Signal', 'Thioether bond']\n",
      "MNLFDLQSMETPKEEAMGDVETGSRASLLLCGDSSLSITTCN\n",
      "42\n",
      "3\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex alpha subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Direct protein sequencing', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Transmembrane', 'Transmembrane helix']\n",
      "MWRIWKVFDPRRILIATAIWLIIIALTIHVILMTTERFNWLEGAPAAEYYSS\n",
      "52\n",
      "2\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex alpha subunit family. {ECO:0000305}.\n",
      "['3D-structure', 'Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Direct protein sequencing', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Transmembrane', 'Transmembrane helix']\n",
      "MNQGKIWTVVPPAFGLPLMLGAVAITALLVHAAVLTHTTWYAAFLQGGVKKAA\n",
      "53\n",
      "2\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex alpha subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Direct protein sequencing', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Transmembrane', 'Transmembrane helix']\n",
      "MSKFYKIWLVFDPRRVFVAQGVFLFLLAVLIHLILLSTPAFNWLTVATAKHGYVAAAQ\n",
      "58\n",
      "5\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex alpha subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Complete proteome', 'Direct protein sequencing', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Reference proteome', 'Transmembrane', 'Transmembrane helix']\n",
      "MNQARIWTVVKPTVGLPLLLGSVTVIAILVHFAVLSHTTWFSKYWNGKAAAIESSVNVG\n",
      "59\n",
      "3\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex alpha subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Complete proteome', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Reference proteome', 'Transmembrane', 'Transmembrane helix']\n",
      "MSKFYKIWMIFDPRRVFVAQGVFLFLLAVMIHLILLSTPSYNWLEISAAKYNRVAVAE\n",
      "58\n",
      "6\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: An alpha/beta heterodimer. The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components (By similarity). {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "PTM: The N-terminus is blocked.\n",
      "SIMILARITY: Belongs to the antenna complex alpha subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Direct protein sequencing', 'Formylation', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Transmembrane', 'Transmembrane helix']\n",
      "MWRIWRLFDPMRAMVAQAVFLLGLAVLIHLMLLGTNKYNWLDGAKKAPAATAVAPVPAEVTSLAQAK\n",
      "67\n",
      "11\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex alpha subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Direct protein sequencing', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Transmembrane', 'Transmembrane helix']\n",
      "MWRMWKILDYRRTVVLAHVGMAVLALLIHFILLSTESFNWLEGNPYG\n",
      "47\n",
      "5\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex alpha subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Direct protein sequencing', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Transmembrane', 'Transmembrane helix']\n",
      "MWKLWKFVDFRMTAVGFHIFFALIAFAVHFACISSERFNWLEGAPAAEYYMDENPGIWKRTSYDG\n",
      "65\n",
      "5\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex alpha subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Direct protein sequencing', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Transmembrane', 'Transmembrane helix']\n",
      "MNQGKIWTVVNPAVGLPLLLGSVAITALLVHLAVLTHTTWFPAFTQGGLKKAA\n",
      "53\n",
      "3\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex alpha subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Direct protein sequencing', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Transmembrane', 'Transmembrane helix']\n",
      "MNNAKIWTVVKPSTGIPLILGAVAVAALIVHAGLLTNTTWFANYWNGNPMATVVAVAPAQ\n",
      "60\n",
      "5\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex alpha subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Direct protein sequencing', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Transmembrane', 'Transmembrane helix']\n",
      "MNQARIWLVVKPSVGLPLLLGVVLLIALLVHGAILTNTSWYPTYFEGNW\n",
      "49\n",
      "4\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex alpha subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Direct protein sequencing', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Transmembrane', 'Transmembrane helix']\n",
      "MNQGKIWTVVNPSVGLPLLLGSVTVIAILVHAAVLSHTTWFPAYWQGGLKKAA\n",
      "53\n",
      "4\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex alpha subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Direct protein sequencing', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Transmembrane', 'Transmembrane helix']\n",
      "MNQGRIWTVVSPTVGLPLLLGSVAAIAFAVHFAVLENTSWVAAFMNGKSVAAAPAPAAPAAPAKK\n",
      "65\n",
      "6\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex alpha subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Complete proteome', 'Direct protein sequencing', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Reference proteome', 'Transmembrane', 'Transmembrane helix']\n",
      "MNQGRIWTVVKPTVGLPLLLGSVAIMVFLVHFAVLTHTTWVAKFMNGKAAAIESSIKAV\n",
      "59\n",
      "4\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex alpha subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Complete proteome', 'Direct protein sequencing', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Reference proteome', 'Transmembrane', 'Transmembrane helix']\n",
      "MNQGRIWTVVKPTVGLPLLLGSVTVIAILVHFAVLSNTTWFSKYWNGKAAAIESSVSIG\n",
      "59\n",
      "5\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex alpha subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Direct protein sequencing', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Transmembrane', 'Transmembrane helix']\n",
      "MYKLWLLFDPRRTLVALSAFLFVLGLIIHFISLSTDRFNWLEGKPAVRA\n",
      "49\n",
      "5\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex alpha subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Complete proteome', 'Direct protein sequencing', 'Formylation', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Reference proteome', 'Transmembrane', 'Transmembrane helix']\n",
      "MQPRSPVRTNIVIFTILGFVVALLIHFIVLSSPEYNWLSNAEGGALLLSAARALFGI\n",
      "57\n",
      "10\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex alpha subunit family. {ECO:0000305}.\n",
      "['3D-structure', 'Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Direct protein sequencing', 'Formylation', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Transmembrane', 'Transmembrane helix']\n",
      "MWRIWQLFDPRQALVGLATFLFVLALLIHFILLSTERFNWLEGASTKPVQTSMVMPSSDLAV\n",
      "62\n",
      "10\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex alpha subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Complete proteome', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Reference proteome', 'Transmembrane', 'Transmembrane helix']\n",
      "MAKFYKIWLIFDPRRVFVAQGVFLFLLAAMIHLVVLSSGLNWFEAAAAVGGQ\n",
      "52\n",
      "3\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex beta subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Direct protein sequencing', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Transmembrane', 'Transmembrane helix']\n",
      "ADNMSLTGLSDEEAKEFHSIFMQSFLIFTAVAVVAHFLAWAWRPWIPGAEGYG\n",
      "53\n",
      "5\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex beta subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Complete proteome', 'Direct protein sequencing', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Reference proteome', 'Transmembrane', 'Transmembrane helix']\n",
      "MADKTLTGLTVEESEELHKHVIDGTRIFGAIAIVAHFLAYVYSPWLH\n",
      "47\n",
      "6\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex beta subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Direct protein sequencing', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Transmembrane', 'Transmembrane helix']\n",
      "AEDRKSLSGLTEQEAQEFGTLYTQGVAFVAVIAVVAHALVWAWRPWLQ\n",
      "48\n",
      "4\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: An alpha/beta heterodimer. The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components (By similarity). {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "MASS SPECTROMETRY: Mass=5421; Method=Electrospray; Range=2-48; Evidence={ECO:0000269|PubMed:8020505};\n",
      "SIMILARITY: Belongs to the antenna complex beta subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Direct protein sequencing', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Transmembrane', 'Transmembrane helix']\n",
      "MAERKGSISGLTDDEAQEFHKFWVQGFVGFTAVAVVAHFLVWVWRPWL\n",
      "48\n",
      "5\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex beta subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Direct protein sequencing', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Transmembrane', 'Transmembrane helix']\n",
      "ADEMRNVSDEEAKEFHAMFSQAFTVYVGVAVVAHILAWAWRPWIPGDEGFG\n",
      "51\n",
      "6\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex beta subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Direct protein sequencing', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Transmembrane', 'Transmembrane helix']\n",
      "TDIRTGLTDEECQEIHEMNMLGMHAYWSIGLIANALAYAWRPFHQGRAGNRLEDHAPDYVRSALT\n",
      "65\n",
      "7\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex beta subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Direct protein sequencing', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Transmembrane', 'Transmembrane helix']\n",
      "MADKPLTADQAEELHKYVIDGARAFVAIAAFAHVLAYSLTPWLH\n",
      "44\n",
      "2\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex beta subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Complete proteome', 'Direct protein sequencing', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Reference proteome', 'Transmembrane', 'Transmembrane helix']\n",
      "MADDPNKVWPTGLTIAESEELHKHVIDGTRIFGAIAIVAHFLAYVYSPWLH\n",
      "51\n",
      "7\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex beta subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Complete proteome', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Reference proteome', 'Transmembrane', 'Transmembrane helix']\n",
      "MTDDLNKVWPSGLTVAEAEEVHKQLILGTRVFGGMALIAHFLAAAATPWLG\n",
      "51\n",
      "8\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex beta subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Direct protein sequencing', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Transmembrane', 'Transmembrane helix']\n",
      "AEDRKSLSGLTEQEAQEFGTLYTQGVAFVAVIAIVAHALVWAWRPWLQ\n",
      "48\n",
      "6\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex beta subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Complete proteome', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Reference proteome', 'Transmembrane', 'Transmembrane helix']\n",
      "MVDDSKKVWPTGLTIAESEEIHKHVIDGARIFVAIAIVAHFLAYVYSPWLH\n",
      "51\n",
      "9\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: An alpha/beta heterodimer conjugated to 3 bacteriochlorophyll molecules. The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components (By similarity). {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex beta subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Direct protein sequencing', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Transmembrane', 'Transmembrane helix']\n",
      "MADDANKVWPSGLTTAEAEELQKGLVDGTRVFGVIAVLAHILAYAYTPWLH\n",
      "51\n",
      "10\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex beta subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Direct protein sequencing', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Transmembrane', 'Transmembrane helix']\n",
      "ADDVKGLTGLTAAESEELHKHVIDGTRVFFVIAIFAHVLAFAFSPWLH\n",
      "48\n",
      "7\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex beta subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Complete proteome', 'Direct protein sequencing', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Reference proteome', 'Transmembrane', 'Transmembrane helix']\n",
      "MVDDPNKVWPTGLTIAESEELHKHVIDGSRIFVAIAIVAHFLAYVYSPWLH\n",
      "51\n",
      "11\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex beta subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Direct protein sequencing', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Transmembrane', 'Transmembrane helix']\n",
      "AEIDRPVSLSGLTEGEAREFHGVFMTSFMVFIAVAIVAHILAWMWRPWIPGPEGYA\n",
      "56\n",
      "1\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex beta subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Complete proteome', 'Direct protein sequencing', 'Formylation', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Reference proteome', 'Transmembrane', 'Transmembrane helix']\n",
      "MRDDDDLVPPKWRPLFNNQDWLLHDIVVKSFYGFGVIAAIAHLLVYLWKPWLP\n",
      "53\n",
      "6\n",
      "\n",
      "FUNCTION: Antenna complexes are light-harvesting systems, which transfer the excitation energy to the reaction centers.\n",
      "SUBUNIT: The core complex is formed by different alpha and beta chains, binding bacteriochlorophyll molecules, and arranged most probably in tetrameric structures disposed around the reaction center. The non-pigmented gamma chains may constitute additional components.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass type II membrane protein.\n",
      "SIMILARITY: Belongs to the antenna complex beta subunit family. {ECO:0000305}.\n",
      "['Antenna complex', 'Bacteriochlorophyll', 'Cell inner membrane', 'Cell membrane', 'Chlorophyll', 'Chromophore', 'Direct protein sequencing', 'Light-harvesting polypeptide', 'Magnesium', 'Membrane', 'Metal-binding', 'Transmembrane', 'Transmembrane helix']\n",
      "ADANKVWPTGLTVAEAEELHTYVTNGFRVFVGIAVVAHVLVFAAHPWGRGGALVA\n",
      "55\n",
      "3\n",
      "\n",
      "FUNCTION: This thermostable and solvent-tolerant lipase is rather nonspecific. It can synthesize both primary and secondary alcohol esters. Simple triglycerides of short and middle chain fatty acids (C<13) are the preferred substrates.\n",
      "CATALYTIC ACTIVITY: Triacylglycerol + H(2)O = diacylglycerol + a carboxylate.\n",
      "BIOPHYSICOCHEMICAL PROPERTIES: Temperature dependence:   Thermostable.;\n",
      "SIMILARITY: Belongs to the AB hydrolase superfamily. Pseudomonas lipase family. {ECO:0000305}.\n",
      "['Direct protein sequencing', 'Hydrolase', 'Lipid degradation', 'Lipid metabolism']\n",
      "AVDDYAATRYPIILVHGLTTDSKYGGVVEYXYRNPNDLTSHXXAAYVYELRSDPLD\n",
      "56\n",
      "2\n",
      "\n",
      "FUNCTION: Part of the ABC transporter complex LptBFG involved in the translocation of lipopolysaccharide (LPS) from the inner membrane to the outer membrane. Probably responsible for energy coupling to the transport system (By similarity). {ECO:0000250}.\n",
      "SUBUNIT: Component of the lipopolysaccharide transport and assembly complex. The LptBFG transporter is composed of two ATP- binding proteins (LptB) and two transmembrane proteins (LptF and LptG) (By similarity). {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm. Cell inner membrane {ECO:0000250}; Peripheral membrane protein {ECO:0000250}; Cytoplasmic side {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the ABC transporter superfamily. Outer membrane lipopolysaccharide export (TC 1.B.42) family. {ECO:0000305}.\n",
      "['ATP-binding', 'Cell inner membrane', 'Cell membrane', 'Cytoplasm', 'Hydrolase', 'Membrane', 'Nucleotide-binding', 'Transport']\n",
      "IDIKRIIEHLPRQRLGVLITDHNVRETLAVCERAYIVSQGHLIAHGTPQQILEDEQVKRVYLGEDFRL\n",
      "68\n",
      "8\n",
      "\n",
      "SUBUNIT: Homodimer. {ECO:0000305}.\n",
      "SUBCELLULAR LOCATION: Secreted.\n",
      "MISCELLANEOUS: First appears in the mid-log phase. Present at highest concentration in the late-log early-stationary phase, before gradually diminishing. Significant amounts are still present after 30 hours.\n",
      "SIMILARITY: Belongs to the hcp1 family. {ECO:0000305}.\n",
      "['Direct protein sequencing', 'Secreted']\n",
      "ATPAYMSITGTKQGLITAGAFTADSVGNTYQEGFEDQVMVQGFDPAVIIPTGPVYGQ\n",
      "57\n",
      "11\n",
      "\n",
      "FUNCTION: Removes the N-terminal methionine from nascent proteins. The N-terminal methionine is often cleaved when the second residue in the primary sequence is small and uncharged (Met-Ala-, Cys, Gly, Pro, Ser, Thr, or Val). Requires deformylation of the N(alpha)-formylated initiator methionine before it can be hydrolyzed (By similarity). {ECO:0000250}.\n",
      "CATALYTIC ACTIVITY: Release of N-terminal amino acids, preferentially methionine, from peptides and arylamides.\n",
      "COFACTOR: Name=Co(2+); Xref=ChEBI:CHEBI:48828; Evidence={ECO:0000250}; Name=Zn(2+); Xref=ChEBI:CHEBI:29105; Evidence={ECO:0000250}; Name=Mn(2+); Xref=ChEBI:CHEBI:29035; Evidence={ECO:0000250}; Name=Fe(2+); Xref=ChEBI:CHEBI:29033; Evidence={ECO:0000250}; Note=Binds 2 divalent metal cations per subunit. Has a high- affinity and a low affinity metal-binding site. The true nature of the physiological cofactor is under debate. The enzyme is active with cobalt, zinc, manganese or divalent iron ions. Most likely, methionine aminopeptidases function as mononuclear Fe(2+)- metalloproteases under physiological conditions, and the catalytically relevant metal-binding site has been assigned to the histidine-containing high-affinity site. {ECO:0000250};\n",
      "SUBUNIT: Monomer. {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the peptidase M24A family. Methionine aminopeptidase type 1 subfamily. {ECO:0000305}.\n",
      "['Aminopeptidase', 'Hydrolase', 'Metal-binding', 'Protease']\n",
      "TVKTKDRSLSAQYEHTIVVTDNGCEILTLRKDDTIPAIISHNE\n",
      "43\n",
      "6\n",
      "\n",
      "FUNCTION: Inhibits subtilisin, chymotrypsin, and elastase, but not trypsin.\n",
      "SIMILARITY: Belongs to the protease inhibitor I10 family. {ECO:0000305}.\n",
      "['3D-structure', 'Direct protein sequencing', 'Protease inhibitor', 'Pyrrolidone carboxylic acid', 'Serine protease inhibitor', 'Signal']\n",
      "MKTTPFFANLLASQTRELTENELEMTAGGTASQQSPVQEVPEQPFATMRYPSDSDEDGFNFPV\n",
      "63\n",
      "5\n",
      "\n",
      "FUNCTION: Antitoxin component of a type II toxin-antitoxin (TA) module. {ECO:0000305}.\n",
      "SUBUNIT: Forms a complex with cognate toxin MazF1. {ECO:0000250|UniProtKB:O53451}.\n",
      "['Complete proteome', 'Reference proteome']\n",
      "MTTYYYVLLSVTTWVGLRHEAKRELVYRGRRSIGRMPREWACRRSRRFAANGVDAAR\n",
      "57\n",
      "12\n",
      "\n",
      "FUNCTION: Antitoxin component of a type II toxin-antitoxin (TA) module. Labile antitoxin that binds to cognate MazF toxin and counteracts its endoribonuclease activity. {ECO:0000250|UniProtKB:P0C7B4}.\n",
      "SUBUNIT: Forms a complex with cognate toxin MazF which inhibits the endoribonuclease activity of MazF. {ECO:0000250|UniProtKB:P0C7B4}.\n",
      "SIMILARITY: Belongs to the MazE/EndoAI family. {ECO:0000305}.\n",
      "['Complete proteome']\n",
      "MLSFSQNRSHSLEQSLKEGYSQMADLNLSLANEAFPIECEACDCNETYLSSNSTNE\n",
      "56\n",
      "3\n",
      "\n",
      "FUNCTION: Required for microcin H47 production. Possibly involved in a regulatory loop modulating its own expression and that of MchI and MchB.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane {ECO:0000305}; Single- pass membrane protein {ECO:0000305}.\n",
      "['Cell inner membrane', 'Cell membrane', 'Membrane', 'Transmembrane', 'Transmembrane helix']\n",
      "MEFATNRVTVNDSRSALSSTLLLSLIMSATLLEYSLSMT\n",
      "39\n",
      "2\n",
      "\n",
      "FUNCTION: A putative polypeptide that may function as an electron carrier; the C-terminal 8 residues are highly conserved. The genes for mycofactocin and other proteins proposed to function in its maturation are found in a conserved gene cluster (By similarity). {ECO:0000250}.\n",
      "['Complete proteome']\n",
      "MRHHIRPSISALDAILCPDRRIAVETCWRKAIQMDYETDTDTELVTETLVEEVSIDGMCGVY\n",
      "62\n",
      "11\n",
      "\n",
      "FUNCTION: PhoP-regulated transcription is redox-sensitive, being activated when the periplasm becomes more reducing. MgrB acts between DsbA/DsbB and PhoP/PhoQ in this pathway. Represses PhoP/PhoQ signaling, possibly by binding to the periplasmic domain of PhoQ, altering its activity and that of downstream effector PhoP. {ECO:0000255|HAMAP-Rule:MF_01596}.\n",
      "SUBUNIT: May form homooligomers. Probably interacts with the periplasmic domain of PhoQ. {ECO:0000255|HAMAP-Rule:MF_01596}.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane {ECO:0000255|HAMAP- Rule:MF_01596}; Single-pass membrane protein {ECO:0000255|HAMAP- Rule:MF_01596}.\n",
      "SIMILARITY: Belongs to the MgrB family. {ECO:0000255|HAMAP- Rule:MF_01596}.\n",
      "SEQUENCE CAUTION: Sequence=ABV12290.1; Type=Erroneous initiation; Note=Translation N-terminally shortened.; Evidence={ECO:0000305};\n",
      "['Cell inner membrane', 'Cell membrane', 'Complete proteome', 'Membrane', 'Reference proteome', 'Transmembrane', 'Transmembrane helix']\n",
      "MKKFRWVILIIVALVCLLLWAQVFNIMCDQDVQFFNGICAINKFIPW\n",
      "47\n",
      "7\n",
      "\n",
      "FUNCTION: May play a role in essential metal ion homeostasis (especially zinc homeostasis) and resistance to certain non- essential metal ions. Metallothioneins have a high content of cysteine residues that bind various heavy metals.\n",
      "INDUCTION: Upon exposure to a range of metals, particularly cadmium, zinc, copper and mercury.\n",
      "SIMILARITY: Belongs to the metallothionein superfamily. Type 14 family. {ECO:0000305}.\n",
      "['3D-structure', 'Cadmium', 'Complete proteome', 'Copper', 'Direct protein sequencing', 'Mercury', 'Metal-binding', 'Metal-thiolate cluster', 'Reference proteome', 'Zinc']\n",
      "MTSTTLVKCACEPCLCNVDPSKAIDRNGLYYCSEACADGHTGGSKGCGHTGCNCHG\n",
      "56\n",
      "4\n",
      "\n",
      "FUNCTION: This protein is involved in the repair of mismatches in DNA. It is possible that it carries out the mismatch recognition step. This protein has a weak ATPase activity (By similarity). {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the DNA mismatch repair MutS family. {ECO:0000305}.\n",
      "['ATP-binding', 'DNA damage', 'DNA repair', 'DNA-binding', 'Nucleotide-binding']\n",
      "TLLNEEVSPAVEALESLDPDSLSPRQALEWIYRLKNMV\n",
      "38\n",
      "2\n",
      "\n",
      "FUNCTION: Metallothioneins are small proteins that have a high content of cysteine residues wich allow them to bind heavy metal ions through clusters of thiolate bonds. MymT binds up to seven ions of Cu(+), with a preference for four to six Cu(+) ions, in a solvent-shielded core. MymT protects M.tuberculosis from copper toxicity (By similarity). {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the metallothionein superfamily. {ECO:0000305}.\n",
      "['Complete proteome', 'Copper', 'Metal-binding', 'Metal-thiolate cluster']\n",
      "MRVIRMTNYEAGTLLTCSHEGCGCRVRIEVPCHCAGAGDAYRCTCGDELAPVK\n",
      "53\n",
      "7\n",
      "\n",
      "FUNCTION: This enzyme has three activities: DNA binding, nicotinamide mononucleotide (NMN) adenylyltransferase and ribosylnicotinamide (RN) kinase. The DNA-binding domain binds to the nadB operator sequence in an NAD- and ATP-dependent manner. As NAD levels increase within the cell, the affinity of NadR for the nadB operator regions of nadA, nadB, and pncB increases, repressing the transcription of these genes. The RN kinase activity catalyzes the phosphorylation of RN to form nicotinamide ribonucleotide. The NMN adenylyltransferase activity catalyzes the transfer of the AMP moiety of ATP to nicotinamide ribonucleotide to form NAD(+). The NMN adenylyltransferase domain also functions as the NAD and ATP sensor (By similarity). {ECO:0000250}.\n",
      "CATALYTIC ACTIVITY: ATP + nicotinamide ribonucleotide = diphosphate + NAD(+).\n",
      "CATALYTIC ACTIVITY: ATP + 1-(beta-D-ribofuranosyl)-nicotinamide = ADP + beta-nicotinamide D-ribonucleotide.\n",
      "PATHWAY: Cofactor biosynthesis; NAD(+) biosynthesis [regulation].\n",
      "PATHWAY: Cofactor biosynthesis; NAD(+) biosynthesis; NAD(+) from nicotinamide D-ribonucleotide: step 1/1.\n",
      "SUBCELLULAR LOCATION: Cell membrane; Peripheral membrane protein. Cytoplasm {ECO:0000250}.\n",
      "['ATP-binding', 'Cell membrane', 'Cytoplasm', 'DNA-binding', 'Kinase', 'Membrane', 'Multifunctional enzyme', 'NAD', 'Nucleotide-binding', 'Pyridine nucleotide biosynthesis', 'Repressor', 'Transcription', 'Transcription regulation', 'Transferase']\n",
      "MSSFDYLKSAIKQKGCTLDEVAEPSGMTKGYLSQLLNRKIKARARRS\n",
      "47\n",
      "8\n",
      "\n",
      "FUNCTION: May be involved in mediating interactions between NapC and a quinol oxidase.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane; Single-pass membrane protein.\n",
      "['Cell inner membrane', 'Cell membrane', 'Membrane', 'Transmembrane', 'Transmembrane helix']\n",
      "MIDSAKETDRPKHRKRDEVIAFLILAVVIWPILSVAIVGGYGFLVWMSQIIFGPPGPMH\n",
      "59\n",
      "6\n",
      "\n",
      "FUNCTION: Involved in phycobilisome (PBS) degradation during nutrient deprivation. May mark the PBS for degradation by covalent association with PBS components or may disrupt the PBS via ionic interactions.\n",
      "INDUCTION: Present at very low levels in nutrient-replete cells, and at 50-fold higher levels in nitrogen- or sulfur-deprived cells and to a variable extent (<10-fold) during phosphorus deprivation.\n",
      "SIMILARITY: To chloroplast ycf18. {ECO:0000305}.\n",
      "['3D-structure', 'Complete proteome', 'Reference proteome']\n",
      "MLPPLPDFSLSVEQQFDLQKYRQQVRDISREDLEDLFIEVVRQKMAHENIFKGMIRQGS\n",
      "59\n",
      "7\n",
      "\n",
      "FUNCTION: The key enzymatic reactions in nitrogen fixation are catalyzed by the nitrogenase complex, which has 2 components: the iron protein and the molybdenum-iron protein. {ECO:0000250}.\n",
      "CATALYTIC ACTIVITY: 8 reduced ferredoxin + 8 H(+) + N(2) + 16 ATP + 16 H(2)O = 8 oxidized ferredoxin + H(2) + 2 NH(3) + 16 ADP + 16 phosphate.\n",
      "COFACTOR: Name=[4Fe-4S] cluster; Xref=ChEBI:CHEBI:49883;   Evidence={ECO:0000250}; Note=Binds 1 [4Fe-4S] cluster per dimer. {ECO:0000250};\n",
      "SUBUNIT: Homodimer. {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the NifH/BchL/ChlL family. {ECO:0000305}.\n",
      "['4Fe-4S', 'ATP-binding', 'Iron', 'Iron-sulfur', 'Metal-binding', 'Nitrogen fixation', 'Nucleotide-binding', 'Oxidoreductase']\n",
      "MAALRQIAFYGKGGIGKSTTSQNTLAALVDHHVPRIPMIIRIGGYAQ\n",
      "47\n",
      "9\n",
      "\n",
      "FUNCTION: The 2-oxoglutarate dehydrogenase complex catalyzes the overall conversion of 2-oxoglutarate to succinyl-CoA and CO(2). It contains multiple copies of 3 enzymatic components: 2-oxoglutarate dehydrogenase (E1), dihydrolipoamide succinyltransferase (E2) and lipoamide dehydrogenase (E3) (By similarity). {ECO:0000250|UniProtKB:P52993}.\n",
      "CATALYTIC ACTIVITY: Succinyl-CoA + enzyme N(6)- (dihydrolipoyl)lysine = CoA + enzyme N(6)-(S- succinyldihydrolipoyl)lysine.\n",
      "COFACTOR: Name=(R)-lipoate; Xref=ChEBI:CHEBI:83088; Evidence={ECO:0000250}; Note=Binds 1 lipoyl cofactor covalently. {ECO:0000250};\n",
      "PATHWAY: Amino-acid degradation; L-lysine degradation via saccharopine pathway; glutaryl-CoA from L-lysine: step 6/6.\n",
      "SUBUNIT: Forms a 24-polypeptide structural core with octahedral symmetry. {ECO:0000250|UniProtKB:P52993}.\n",
      "INDUCTION: By 2,4-dichlorophenol. {ECO:0000269|PubMed:15073309}.\n",
      "SIMILARITY: Belongs to the 2-oxoacid dehydrogenase family. {ECO:0000305}.\n",
      "['Acyltransferase', 'Direct protein sequencing', 'Lipoyl', 'Transferase', 'Tricarboxylic acid cycle']\n",
      "AIVEVKVPXLXE\n",
      "12\n",
      "2\n",
      "\n",
      "FUNCTION: The pyruvate dehydrogenase complex catalyzes the overall conversion of pyruvate to acetyl-CoA and CO(2). It contains multiple copies of three enzymatic components: pyruvate dehydrogenase (E1), dihydrolipoamide acetyltransferase (E2) and lipoamide dehydrogenase (E3).\n",
      "CATALYTIC ACTIVITY: Pyruvate + [dihydrolipoyllysine-residue acetyltransferase] lipoyllysine = [dihydrolipoyllysine-residue acetyltransferase] S-acetyldihydrolipoyllysine + CO(2).\n",
      "COFACTOR: Name=thiamine diphosphate; Xref=ChEBI:CHEBI:58937;\n",
      "SUBUNIT: Homodimer.\n",
      "['Glycolysis', 'Oxidoreductase', 'Pyruvate', 'Thiamine pyrophosphate']\n",
      "EVDRYWVVLAALEALADRGDIEAKVVAEAIAKFGIDPDKRNPLDC\n",
      "45\n",
      "3\n",
      "\n",
      "FUNCTION: Acts as an orphan toxin which is important for maintaining cell fitness during stress related to the stringent response. Increases the formation of persister cells. Has no known antitoxin. {ECO:0000250|UniProtKB:P64453}.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane {ECO:0000250|UniProtKB:P64453}; Multi-pass membrane protein {ECO:0000305}.\n",
      "SIMILARITY: Belongs to the GhoT/OrtT toxin family. {ECO:0000305}.\n",
      "SEQUENCE CAUTION: Sequence=AAN80332.1; Type=Erroneous initiation; Note=Translation N-terminally shortened.; Evidence={ECO:0000305};\n",
      "['Cell inner membrane', 'Cell membrane', 'Complete proteome', 'Membrane', 'Stress response', 'Toxin', 'Transmembrane', 'Transmembrane helix']\n",
      "MSLYQHMLVFYAVMAAIAFLITWFLSHDKKRIRFLSAFLVGATWPMSFPVALLFSLF\n",
      "57\n",
      "13\n",
      "\n",
      "FUNCTION: Component of the cytochrome b6-f complex, which mediates electron transfer between photosystem II (PSII) and photosystem I (PSI), cyclic electron flow around PSI, and state transitions. PetG is required for either the stability or assembly of the cytochrome b6-f complex. {ECO:0000255|HAMAP-Rule:MF_00432}.\n",
      "SUBUNIT: The 4 large subunits of the cytochrome b6-f complex are cytochrome b6, subunit IV (17 kDa polypeptide, PetD), cytochrome f and the Rieske protein, while the 4 small subunits are PetG, PetL, PetM and PetN. The complex functions as a dimer. {ECO:0000255|HAMAP-Rule:MF_00432}.\n",
      "SUBCELLULAR LOCATION: Cellular thylakoid membrane {ECO:0000255|HAMAP-Rule:MF_00432}; Single-pass membrane protein {ECO:0000255|HAMAP-Rule:MF_00432}.\n",
      "SIMILARITY: Belongs to the PetG family. {ECO:0000255|HAMAP- Rule:MF_00432}.\n",
      "['Complete proteome', 'Electron transport', 'Membrane', 'Photosynthesis', 'Reference proteome', 'Thylakoid', 'Transmembrane', 'Transmembrane helix', 'Transport']\n",
      "MVEPLLAGIVLGLVPVTLAGLFVAAWQQYKRGEEVG\n",
      "36\n",
      "2\n",
      "\n",
      "FUNCTION: Light-harvesting photosynthetic bile pigment-protein from the phycobiliprotein complex. {ECO:0000305}.\n",
      "SUBUNIT: Heterodimer of an alpha and a beta chain. {ECO:0000269|PubMed:16920511, ECO:0000305}.\n",
      "SUBCELLULAR LOCATION: Cellular thylakoid membrane {ECO:0000269|Ref.1}; Peripheral membrane protein {ECO:0000269|Ref.1}; Cytoplasmic side {ECO:0000269|Ref.1}. Note=Part of the phycobilisome rod.\n",
      "PTM: Contains one covalently linked bilin chromophore. {ECO:0000305}.\n",
      "MISCELLANEOUS: In cell culture induces a urokinase-type plasminogen activator; may therefore be useful in dissolving blood clots. {ECO:0000269|PubMed:16920511}.\n",
      "SIMILARITY: Belongs to the phycobiliprotein family. {ECO:0000255}.\n",
      "['Bile pigment', 'Chromophore', 'Direct protein sequencing', 'Electron transport', 'Membrane', 'Photosynthesis', 'Phycobilisome', 'Thylakoid', 'Transport']\n",
      "MKTPLTEAVSIADSQGRFLSSTQIQVLFGRFRQAKAGLXAAKAL\n",
      "44\n",
      "3\n",
      "\n",
      "SUBUNIT: Heterodimer of a large and a small chain.\n",
      "SUBCELLULAR LOCATION: Secreted.\n",
      "SIMILARITY: Belongs to the peptidase S8 family. {ECO:0000305}.\n",
      "['Direct protein sequencing', 'Hydrolase', 'Protease', 'Secreted', 'Serine protease']\n",
      "LTPNDPLYSQQWGLSGTYGIRANTAWDNGYQGQGKIIAVVDTGITDHPDLLANRTSPLGYDFI\n",
      "63\n",
      "6\n",
      "\n",
      "SIMILARITY: Belongs to the site-specific recombinase resolvase family. {ECO:0000305}.\n",
      "CAUTION: Could be the product of a pseudogene. {ECO:0000305}.\n",
      "['Complete proteome', 'DNA integration', 'DNA invertase', 'DNA recombination', 'DNA-binding', 'Reference proteome']\n",
      "MWHLVVLLEELCERGINFRALAQSIFAQQWGDECCKSKTICDLKVIV\n",
      "47\n",
      "10\n",
      "\n",
      "FUNCTION: Catalyzes the phosphorylation of pyruvate to phosphoenolpyruvate. {ECO:0000250}.\n",
      "CATALYTIC ACTIVITY: ATP + pyruvate + H(2)O = AMP + phosphoenolpyruvate + phosphate.\n",
      "COFACTOR: Name=Mg(2+); Xref=ChEBI:CHEBI:18420; Evidence={ECO:0000250};\n",
      "PATHWAY: Carbohydrate biosynthesis; gluconeogenesis.\n",
      "DOMAIN: The N-terminal domain contains the ATP/Pi binding site, the central domain the pyrophosphate/phosphate carrier histidine, and the C-terminal domain the pyruvate binding site. {ECO:0000250}.\n",
      "MISCELLANEOUS: The reaction takes place in three steps, mediated by a phosphocarrier histidine residue located on the surface of the central domain. The two first partial reactions are catalyzed at an active site located on the N-terminal domain, and the third partial reaction is catalyzed at an active site located on the C- terminal domain. For catalytic turnover, the central domain swivels from the concave surface of the N-terminal domain to that of the C-terminal domain (By similarity). {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the PEP-utilizing enzyme family. {ECO:0000305}.\n",
      "['ATP-binding', 'Kinase', 'Magnesium', 'Metal-binding', 'Nucleotide-binding', 'Phosphoprotein', 'Transferase']\n",
      "MSSKGEQPLVLWYNQLGMHDVDRVGGKNPSLGEMITNLSSLGVSVPNGFATTSYAFNLFLD\n",
      "61\n",
      "8\n",
      "\n",
      "FUNCTION: Ring cyclization and eight-electron oxidation of 3a-(2- amino-2-carboxyethyl)-4,5-dioxo-4,5,6,7,8,9-hexahydroquinoline- 7,9-dicarboxylic-acid to PQQ. {ECO:0000250}.\n",
      "CATALYTIC ACTIVITY: 6-(2-amino-2-carboxyethyl)-7,8-dioxo- 1,2,3,4,7,8-hexahydroquinoline-2,4-dicarboxylate + 3 O(2) = 4,5- dioxo-4,5-dihydro-1H-pyrrolo[2,3-f]quinoline-2,7,9-tricarboxylate + 2 H(2)O(2) + 2 H(2)O.\n",
      "PATHWAY: Cofactor biosynthesis; pyrroloquinoline quinone biosynthesis.\n",
      "SIMILARITY: Belongs to the PqqC family. {ECO:0000305}.\n",
      "['Oxidoreductase', 'PQQ biosynthesis']\n",
      "MNAPLTDNIPWTREEFEQKLRDKGRGYHIYHPFHVMMYEGKLTREQLQCWVA\n",
      "52\n",
      "4\n",
      "\n",
      "FUNCTION: May help in the organization of the PsaL subunit. {ECO:0000255|HAMAP-Rule:MF_00431}.\n",
      "SUBCELLULAR LOCATION: Cellular thylakoid membrane {ECO:0000255|HAMAP-Rule:MF_00431}; Single-pass membrane protein {ECO:0000255|HAMAP-Rule:MF_00431}.\n",
      "SIMILARITY: Belongs to the PsaI family. {ECO:0000255|HAMAP- Rule:MF_00431}.\n",
      "['Complete proteome', 'Membrane', 'Photosynthesis', 'Photosystem I', 'Reference proteome', 'Thylakoid', 'Transmembrane', 'Transmembrane helix']\n",
      "MTGAYAASYLPWILIPIIGWLMPAVVMGLLFLYIESDA\n",
      "38\n",
      "3\n",
      "\n",
      "FUNCTION: May help in the organization of the PsaL subunit.\n",
      "SUBCELLULAR LOCATION: Cellular thylakoid membrane {ECO:0000250}; Single-pass membrane protein {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the PsaI family. {ECO:0000305}.\n",
      "SEQUENCE CAUTION: Sequence=CAB11178.1; Type=Erroneous initiation; Evidence={ECO:0000305};\n",
      "['Complete proteome', 'Membrane', 'Photosynthesis', 'Photosystem I', 'Reference proteome', 'Thylakoid', 'Transmembrane', 'Transmembrane helix']\n",
      "MSSEFAATWLPAVFVPLIGLVTPAVFIVLIGRYITATD\n",
      "38\n",
      "4\n",
      "\n",
      "FUNCTION: May help in the organization of the PsaE and PsaF subunits. {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cellular thylakoid membrane {ECO:0000250}; Single-pass membrane protein {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the PsaJ family. {ECO:0000305}.\n",
      "['Complete proteome', 'Membrane', 'Photosynthesis', 'Photosystem I', 'Thylakoid', 'Transmembrane', 'Transmembrane helix']\n",
      "MFKIFRTKWFRSAPVLATLWLSSTAVILIGVNSYFPDYLFMPMS\n",
      "44\n",
      "4\n",
      "\n",
      "FUNCTION: May help in the organization of the PsaE and PsaF subunits. {ECO:0000255|HAMAP-Rule:MF_00522}.\n",
      "SUBCELLULAR LOCATION: Cellular thylakoid membrane {ECO:0000255|HAMAP-Rule:MF_00522}; Single-pass membrane protein {ECO:0000255|HAMAP-Rule:MF_00522}.\n",
      "SIMILARITY: Belongs to the PsaJ family. {ECO:0000255|HAMAP- Rule:MF_00522}.\n",
      "['Complete proteome', 'Membrane', 'Photosynthesis', 'Photosystem I', 'Thylakoid', 'Transmembrane', 'Transmembrane helix']\n",
      "MFKILNTKFVRSAPVVAAIWLSLTAGIIIEFNRFFPDLLFHPMS\n",
      "44\n",
      "5\n",
      "\n",
      "FUNCTION: May help in the organization of the PsaE and PsaF subunits. {ECO:0000250}.\n",
      "SUBCELLULAR LOCATION: Cellular thylakoid membrane {ECO:0000250}; Single-pass membrane protein {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the PsaJ family. {ECO:0000305}.\n",
      "['Complete proteome', 'Membrane', 'Photosynthesis', 'Photosystem I', 'Reference proteome', 'Thylakoid', 'Transmembrane', 'Transmembrane helix']\n",
      "MFKIFSTKWFRSAPVVATIWIVITAGILVEWNRFVPDLLFHPGL\n",
      "44\n",
      "6\n",
      "\n",
      "SUBCELLULAR LOCATION: Cellular thylakoid membrane {ECO:0000250}; Single-pass membrane protein {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the PsaX family. {ECO:0000305}.\n",
      "['Complete proteome', 'Direct protein sequencing', 'Membrane', 'Photosynthesis', 'Photosystem I', 'Thylakoid', 'Transmembrane', 'Transmembrane helix']\n",
      "MAKAKTPAVANTGAKPPYTFRTAWALLLLGVNFLVAAYYFHIIQ\n",
      "44\n",
      "7\n",
      "\n",
      "FUNCTION: This b-type cytochrome is tightly associated with the reaction center of photosystem II (PSII). PSII is a light-driven water:plastoquinone oxidoreductase that uses light energy to abstract electrons from H(2)O, generating O(2) and a proton gradient subsequently used for ATP formation. It consists of a core antenna complex that captures photons, and an electron transfer chain that converts photonic excitation into a charge separation. {ECO:0000255|HAMAP-Rule:MF_00643}.\n",
      "COFACTOR: Name=heme b; Xref=ChEBI:CHEBI:60344;   Evidence={ECO:0000255|HAMAP-Rule:MF_00643}; Note=With its partner (PsbE) binds heme. PSII binds additional chlorophylls, carotenoids and specific lipids. {ECO:0000255|HAMAP- Rule:MF_00643};\n",
      "SUBUNIT: Heterodimer of an alpha subunit and a beta subunit. Cyanobacterial PSII is composed of 1 copy each of membrane proteins PsbA, PsbB, PsbC, PsbD, PsbE, PsbF, PsbH, PsbI, PsbJ, PsbK, PsbL, PsbM, PsbT, PsbX, PsbY, PsbZ, Ycf12, at least 3 peripheral proteins PsbO, PsbU, PsbV and a large number of cofactors. It forms dimeric complexes. {ECO:0000255|HAMAP- Rule:MF_00643}.\n",
      "SUBCELLULAR LOCATION: Cellular thylakoid membrane {ECO:0000255|HAMAP-Rule:MF_00643}; Single-pass membrane protein {ECO:0000255|HAMAP-Rule:MF_00643}.\n",
      "SIMILARITY: Belongs to the PsbE/PsbF family. {ECO:0000255|HAMAP- Rule:MF_00643}.\n",
      "['Complete proteome', 'Electron transport', 'Heme', 'Iron', 'Membrane', 'Metal-binding', 'Photosynthesis', 'Photosystem II', 'Reference proteome', 'Thylakoid', 'Transmembrane', 'Transmembrane helix', 'Transport']\n",
      "MVNNTSNQPTSYPIFTVRWLAVHTLAVPTVFFVGAIAAMQFIQR\n",
      "44\n",
      "8\n",
      "\n",
      "FUNCTION: One of the components of the core complex of photosystem II (PSII), required for its stability and/or assembly. PSII is a light-driven water:plastoquinone oxidoreductase that uses light energy to abstract electrons from H(2)O, generating O(2) and a proton gradient subsequently used for ATP formation. It consists of a core antenna complex that captures photons, and an electron transfer chain that converts photonic excitation into a charge separation. {ECO:0000255|HAMAP-Rule:MF_00752}.\n",
      "SUBUNIT: Cyanobacterial PSII is composed of 1 copy each of membrane proteins PsbA, PsbB, PsbC, PsbD, PsbE, PsbF, PsbH, PsbI, PsbJ, PsbK, PsbL, PsbM, PsbT, PsbX, PsbY, PsbZ, Ycf12, at least 3 peripheral proteins PsbO, PsbU, PsbV and a large number of cofactors. It forms dimeric complexes. {ECO:0000255|HAMAP- Rule:MF_00752}.\n",
      "SUBCELLULAR LOCATION: Cellular thylakoid membrane {ECO:0000255|HAMAP-Rule:MF_00752}; Single-pass membrane protein {ECO:0000255|HAMAP-Rule:MF_00752}.\n",
      "SIMILARITY: Belongs to the PsbH family. {ECO:0000255|HAMAP- Rule:MF_00752}.\n",
      "['Complete proteome', 'Membrane', 'Photosynthesis', 'Photosystem II', 'Thylakoid', 'Transmembrane', 'Transmembrane helix']\n",
      "MAQRTRLGDILRPLNSEYGKVAPGWGTTPVMGVFMALFLVFLLIILQLYNKSILIQDVRVGW\n",
      "62\n",
      "12\n",
      "\n",
      "FUNCTION: One of the components of the core complex of photosystem II (PSII), required for its stability and/or assembly. PSII is a light-driven water:plastoquinone oxidoreductase that uses light energy to abstract electrons from H(2)O, generating O(2) and a proton gradient subsequently used for ATP formation. It consists of a core antenna complex that captures photons, and an electron transfer chain that converts photonic excitation into a charge separation. {ECO:0000255|HAMAP-Rule:MF_01316}.\n",
      "SUBUNIT: Cyanobacterial PSII is composed of 1 copy each of membrane proteins PsbA, PsbB, PsbC, PsbD, PsbE, PsbF, PsbH, PsbI, PsbJ, PsbK, PsbL, PsbM, PsbT, PsbX, PsbY, PsbZ, Ycf12, at least 3 peripheral proteins PsbO, PsbU, PsbV and a large number of cofactors. It forms dimeric complexes. {ECO:0000255|HAMAP- Rule:MF_01316}.\n",
      "SUBCELLULAR LOCATION: Cellular thylakoid membrane {ECO:0000255|HAMAP-Rule:MF_01316}; Single-pass membrane protein {ECO:0000255|HAMAP-Rule:MF_01316}.\n",
      "SIMILARITY: Belongs to the PsbI family. {ECO:0000255|HAMAP- Rule:MF_01316}.\n",
      "['Complete proteome', 'Membrane', 'Photosynthesis', 'Photosystem II', 'Reaction center', 'Thylakoid', 'Transmembrane', 'Transmembrane helix']\n",
      "MLALKLSVYSVVFFFIAVFVFGFLASDPARTPARKDLED\n",
      "39\n",
      "3\n",
      "\n",
      "FUNCTION: One of the components of the core complex of photosystem II (PSII). PSII is a light-driven water:plastoquinone oxidoreductase that uses light energy to abstract electrons from H(2)O, generating O(2) and a proton gradient subsequently used for ATP formation. It consists of a core antenna complex that captures photons, and an electron transfer chain that converts photonic excitation into a charge separation. {ECO:0000255|HAMAP- Rule:MF_01305}.\n",
      "SUBUNIT: Cyanobacterial PSII is composed of 1 copy each of membrane proteins PsbA, PsbB, PsbC, PsbD, PsbE, PsbF, PsbH, PsbI, PsbJ, PsbK, PsbL, PsbM, PsbT, PsbX, PsbY, PsbZ, Ycf12, at least 3 peripheral proteins PsbO, PsbU, PsbV and a large number of cofactors. It forms dimeric complexes. {ECO:0000255|HAMAP- Rule:MF_01305}.\n",
      "SUBCELLULAR LOCATION: Cellular thylakoid membrane {ECO:0000255|HAMAP-Rule:MF_01305}; Single-pass membrane protein {ECO:0000255|HAMAP-Rule:MF_01305}.\n",
      "SIMILARITY: Belongs to the PsbJ family. {ECO:0000255|HAMAP- Rule:MF_01305}.\n",
      "['Complete proteome', 'Membrane', 'Photosynthesis', 'Photosystem II', 'Reaction center', 'Reference proteome', 'Thylakoid', 'Transmembrane', 'Transmembrane helix']\n",
      "MSKLRGPDGRIPDRLPDGRPAVSWERRWTEGTLPLWLVATAGGIAVIFVLGIFFYGSYTGVGSA\n",
      "64\n",
      "10\n",
      "\n",
      "FUNCTION: One of the components of the core complex of photosystem II (PSII). PSII is a light-driven water:plastoquinone oxidoreductase that uses light energy to abstract electrons from H(2)O, generating O(2) and a proton gradient subsequently used for ATP formation. It consists of a core antenna complex that captures photons, and an electron transfer chain that converts photonic excitation into a charge separation. This subunit is found at the monomer-monomer interface. {ECO:0000255|HAMAP-Rule:MF_00438}.\n",
      "SUBUNIT: Cyanobacterial PSII is composed of 1 copy each of membrane proteins PsbA, PsbB, PsbC, PsbD, PsbE, PsbF, PsbH, PsbI, PsbJ, PsbK, PsbL, PsbM, PsbT, PsbX, PsbY, PsbZ, Ycf12, at least 3 peripheral proteins PsbO, PsbU, PsbV and a large number of cofactors. It forms dimeric complexes. {ECO:0000255|HAMAP- Rule:MF_00438}.\n",
      "SUBCELLULAR LOCATION: Cellular thylakoid membrane {ECO:0000255|HAMAP-Rule:MF_00438}; Single-pass membrane protein {ECO:0000255|HAMAP-Rule:MF_00438}.\n",
      "SIMILARITY: Belongs to the PsbM family. {ECO:0000255|HAMAP- Rule:MF_00438}.\n",
      "['Complete proteome', 'Membrane', 'Photosynthesis', 'Photosystem II', 'Reaction center', 'Thylakoid', 'Transmembrane', 'Transmembrane helix']\n",
      "METTSFGFAASLLFVGVPTIFLIGLFVSTSDGEKSSFYSDTSKGRLSPEPKK\n",
      "52\n",
      "5\n",
      "\n",
      "FUNCTION: May play a role in photosystem I and II biogenesis. {ECO:0000255|HAMAP-Rule:MF_00293}.\n",
      "SUBCELLULAR LOCATION: Cellular thylakoid membrane {ECO:0000255|HAMAP-Rule:MF_00293}; Single-pass membrane protein {ECO:0000255|HAMAP-Rule:MF_00293}.\n",
      "SIMILARITY: Belongs to the PsbN family. {ECO:0000255|HAMAP- Rule:MF_00293}.\n",
      "CAUTION: Originally thought to be a component of PSII; based on experiments in Synechocystis, N.tabacum and barley, and its absence from PSII in T.elongatus and T.vulcanus, this is probably not true. {ECO:0000255|HAMAP-Rule:MF_00293}.\n",
      "['Complete proteome', 'Membrane', 'Reference proteome', 'Thylakoid', 'Transmembrane', 'Transmembrane helix']\n",
      "METATILGILIAAAVVGITVLALDTAFGPPAAELSDPFEDHED\n",
      "43\n",
      "7\n",
      "\n",
      "FUNCTION: May play a role in photosystem I and II biogenesis. {ECO:0000255|HAMAP-Rule:MF_00293}.\n",
      "SUBCELLULAR LOCATION: Cellular thylakoid membrane {ECO:0000255|HAMAP-Rule:MF_00293}; Single-pass membrane protein {ECO:0000255|HAMAP-Rule:MF_00293}.\n",
      "SIMILARITY: Belongs to the PsbN family. {ECO:0000255|HAMAP- Rule:MF_00293}.\n",
      "CAUTION: Originally thought to be a component of PSII; based on experiments in Synechocystis, N.tabacum and barley, and its absence from PSII in T.elongatus and T.vulcanus, this is probably not true. {ECO:0000255|HAMAP-Rule:MF_00293}.\n",
      "['Complete proteome', 'Membrane', 'Reference proteome', 'Thylakoid', 'Transmembrane', 'Transmembrane helix']\n",
      "METATILGISIAAALVGITVLALYTAFGPPAAELSDPFEDHED\n",
      "43\n",
      "8\n",
      "\n",
      "FUNCTION: May play a role in photosystem I and II biogenesis. {ECO:0000255|HAMAP-Rule:MF_00293}.\n",
      "SUBCELLULAR LOCATION: Cellular thylakoid membrane {ECO:0000255|HAMAP-Rule:MF_00293}; Single-pass membrane protein {ECO:0000255|HAMAP-Rule:MF_00293}.\n",
      "SIMILARITY: Belongs to the PsbN family. {ECO:0000255|HAMAP- Rule:MF_00293}.\n",
      "CAUTION: Originally thought to be a component of PSII; based on experiments in Synechocystis, N.tabacum and barley, and its absence from PSII in T.elongatus and T.vulcanus, this is probably not true. {ECO:0000255|HAMAP-Rule:MF_00293}.\n",
      "['Complete proteome', 'Membrane', 'Thylakoid', 'Transmembrane', 'Transmembrane helix']\n",
      "MEPAIVLIISVGAALVAVTGYGIYTAFGPPSRELSDPFEDHED\n",
      "43\n",
      "9\n",
      "\n",
      "FUNCTION: Involved in the binding and/or turnover of quinones at the Q(B) site of Photosystem II. {ECO:0000255|HAMAP- Rule:MF_01388}.\n",
      "SUBUNIT: PSII consists of a core antenna complex that captures photons, and an electron transfer chain that converts photonic excitation into a charge separation. PSII forms dimeric complexes. {ECO:0000255|HAMAP-Rule:MF_01388}.\n",
      "SUBCELLULAR LOCATION: Cellular thylakoid membrane {ECO:0000255|HAMAP-Rule:MF_01388}; Single-pass membrane protein {ECO:0000255|HAMAP-Rule:MF_01388}.\n",
      "SIMILARITY: Belongs to the PsbX family. Type 2 subfamily. {ECO:0000255|HAMAP-Rule:MF_01388}.\n",
      "['Complete proteome', 'Membrane', 'Photosynthesis', 'Photosystem II', 'Thylakoid', 'Transmembrane', 'Transmembrane helix']\n",
      "MLQISNLLLAADFSAEVANNSAVGMIGSFIAAALLIVIPATAFLIFVSQKDSLDRTSTGRR\n",
      "61\n",
      "9\n",
      "\n",
      "FUNCTION: Manganese-binding polypeptide with L-arginine metabolizing enzyme activity. Component of the core of photosystem II. {ECO:0000255|HAMAP-Rule:MF_00717}.\n",
      "SUBCELLULAR LOCATION: Cellular thylakoid membrane {ECO:0000255|HAMAP-Rule:MF_00717}; Single-pass membrane protein {ECO:0000255|HAMAP-Rule:MF_00717}.\n",
      "SIMILARITY: Belongs to the PsbY family. {ECO:0000255|HAMAP- Rule:MF_00717}.\n",
      "['Complete proteome', 'Membrane', 'Photosynthesis', 'Photosystem II', 'Reference proteome', 'Thylakoid', 'Transmembrane', 'Transmembrane helix']\n",
      "MLGIDARLFLVVAPILAAVSWAAFNIGRAAVGQLQLLIKRSRA\n",
      "43\n",
      "10\n",
      "\n",
      "FUNCTION: Controls the interaction of photosystem II (PSII) cores with the light-harvesting antenna. {ECO:0000255|HAMAP- Rule:MF_00644}.\n",
      "SUBCELLULAR LOCATION: Cellular thylakoid membrane {ECO:0000255|HAMAP-Rule:MF_00644}; Multi-pass membrane protein {ECO:0000255|HAMAP-Rule:MF_00644}. Note=Associated with the photosystem II complex. {ECO:0000255|HAMAP-Rule:MF_00644}.\n",
      "SIMILARITY: Belongs to the PsbZ family. {ECO:0000255|HAMAP- Rule:MF_00644}.\n",
      "['Complete proteome', 'Membrane', 'Photosynthesis', 'Photosystem II', 'Reaction center', 'Thylakoid', 'Transmembrane', 'Transmembrane helix']\n",
      "MMLIFQIALLVLVLYSLLLVVAVPVLFSSASDWSRAKNVILVGSLLWVLMVIGVGVLSFFK\n",
      "61\n",
      "10\n",
      "\n",
      "CATALYTIC ACTIVITY: Phosphatidyl-L-serine = phosphatidylethanolamine + CO(2).\n",
      "COFACTOR: Name=pyruvate; Xref=ChEBI:CHEBI:15361; Evidence={ECO:0000250}; Note=Binds 1 pyruvoyl group covalently per subunit. {ECO:0000250};\n",
      "PATHWAY: Phospholipid metabolism; phosphatidylethanolamine biosynthesis; phosphatidylethanolamine from CDP-diacylglycerol: step 2/2.\n",
      "SIMILARITY: Belongs to the phosphatidylserine decarboxylase family. Type 1 subfamily. {ECO:0000305}.\n",
      "['Decarboxylase', 'Lipid biosynthesis', 'Lipid metabolism', 'Lyase', 'Phospholipid biosynthesis', 'Phospholipid metabolism', 'Pyruvate']\n",
      "MKDRLFLLGQHLLPHHLLSRAAGRLAECRVPWVKNSLIKAFARHFQVD\n",
      "48\n",
      "8\n",
      "\n",
      "CAUTION: B.parapertussis and B.bronchiseptica seem not to produce the pertussis toxin (S1, S2, S4, S5 and S3) and ptl proteins (PtlA, PtlB, PtlC, PtlD, PtlE, PtlF, PtlG, PtlH and PtlI) in vivo due to changes in the promoter region of the ptx-ptl operon. However, it is possible that their promoter is active under certain, as-yet-undefined conditions and that B.parapertussis and B.bronchiseptica are therefore capable of producing these proteins. {ECO:0000305}.\n",
      "['Complete proteome', 'Signal']\n",
      "MIHAHSNARLLRWAILAIAPATLGACAPNGPPGLPYPDGKPLIPINTAAPEQGSSCQTRAP\n",
      "61\n",
      "11\n",
      "\n",
      "FUNCTION: Component of the type IV secretion system ptl required for secretion of assembled pertussis toxin (PTX) through the outer membrane. {ECO:0000269|PubMed:10518754, ECO:0000269|PubMed:15322034, ECO:0000269|PubMed:8464913}.\n",
      "SUBUNIT: Forms a complex with PtlF.\n",
      "INDUCTION: Cotranscribed with ptxABCDE. Activated by the two- component regulatory system BvgS/BvgA.\n",
      "['Complete proteome', 'Reference proteome', 'Signal', 'Transport']\n",
      "MIHAHSNARLLRWAILAIAPVTLGACAPNGPPGLPYPDGKPLIPINTAAPEQGSSCQTRAP\n",
      "61\n",
      "12\n",
      "\n",
      "FUNCTION: Protein modifier that is covalently attached to lysine residues of substrate proteins, thereby targeting them for proteasomal degradation. The tagging system is termed pupylation. {ECO:0000255|HAMAP-Rule:MF_02106}.\n",
      "PATHWAY: Protein degradation; proteasomal Pup-dependent pathway. {ECO:0000255|HAMAP-Rule:MF_02106}.\n",
      "SUBUNIT: Strongly interacts with the proteasome-associated ATPase ARC through a hydrophobic interface; the interacting region of Pup lies in its C-terminal half. There is one Pup binding site per ARC hexamer ring. {ECO:0000255|HAMAP-Rule:MF_02106}.\n",
      "DOMAIN: The N-terminal unstructured half of Pup provides a signal required to initiate unfolding and degradation by the proteasome but is not needed for pupylation, while the C-terminal helical half of Pup interacts with ARC to target proteins to the proteasome. {ECO:0000255|HAMAP-Rule:MF_02106}.\n",
      "PTM: Is modified by deamidation of its C-terminal glutamine to glutamate by the deamidase Dop, a prerequisite to the subsequent pupylation process. {ECO:0000255|HAMAP-Rule:MF_02106}.\n",
      "SIMILARITY: Belongs to the prokaryotic ubiquitin-like protein family. {ECO:0000255|HAMAP-Rule:MF_02106}.\n",
      "['Coiled coil', 'Isopeptide bond', 'Ubl conjugation pathway']\n",
      "MAQEQTKRTGGGDEDDTPGGDGAAGQERREKLAEDTDDLLDEIDDVLEENAEDFVRAYVQKGGQ\n",
      "64\n",
      "11\n",
      "\n",
      "FUNCTION: Protein modifier that is covalently attached to lysine residues of substrate proteins, thereby targeting them for proteasomal degradation. The tagging system is termed pupylation (By similarity). {ECO:0000250}.\n",
      "PATHWAY: Protein degradation; proteasomal Pup-dependent pathway.\n",
      "SUBUNIT: Strongly interacts with the proteasome-associated ATPase ARC through a hydrophobic interface; the interacting region of Pup lies in its C-terminal half. There is one Pup binding site per ARC hexamer ring (By similarity). {ECO:0000250}.\n",
      "DOMAIN: The N-terminal unstructured half of Pup provides a signal required to initiate unfolding and degradation by the proteasome but is not needed for pupylation, while the C-terminal helical half of Pup interacts with ARC to target proteins to the proteasome. {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the prokaryotic ubiquitin-like protein family. {ECO:0000305}.\n",
      "['Complete proteome', 'Isopeptide bond', 'Reference proteome', 'Ubl conjugation pathway']\n",
      "MPQKFEQMQSAEQKHDEDETIAQAGTQIDDTVDALDAVLDDIESVLESNAEEYVGSFVQKGGE\n",
      "63\n",
      "7\n",
      "\n",
      "FUNCTION: Protein modifier that is covalently attached to lysine residues of substrate proteins, thereby targeting them for proteasomal degradation. The tagging system is termed pupylation. {ECO:0000255|HAMAP-Rule:MF_02106}.\n",
      "PATHWAY: Protein degradation; proteasomal Pup-dependent pathway. {ECO:0000255|HAMAP-Rule:MF_02106}.\n",
      "SUBUNIT: Strongly interacts with the proteasome-associated ATPase ARC through a hydrophobic interface; the interacting region of Pup lies in its C-terminal half. There is one Pup binding site per ARC hexamer ring. {ECO:0000255|HAMAP-Rule:MF_02106}.\n",
      "DOMAIN: The N-terminal unstructured half of Pup provides a signal required to initiate unfolding and degradation by the proteasome but is not needed for pupylation, while the C-terminal helical half of Pup interacts with ARC to target proteins to the proteasome. {ECO:0000255|HAMAP-Rule:MF_02106}.\n",
      "SIMILARITY: Belongs to the prokaryotic ubiquitin-like protein family. {ECO:0000255|HAMAP-Rule:MF_02106}.\n",
      "['Complete proteome', 'Isopeptide bond', 'Reference proteome', 'Ubl conjugation pathway']\n",
      "MPQEFEQIRSADQPLDSEESAPVAGARTDDTVDALDAVLDDIESVLETNAEEYVGSFVQKGGE\n",
      "63\n",
      "8\n",
      "\n",
      "FUNCTION: Facilitates transcription termination by a mechanism that involves Rho binding to the nascent RNA, activation of Rho's RNA-dependent ATPase activity, and release of the mRNA from the DNA template. {ECO:0000250}.\n",
      "SUBUNIT: Homohexamer. The homohexamer assembles into an open ring structure (By similarity). {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the Rho family. {ECO:0000305}.\n",
      "['ATP-binding', 'Helicase', 'Hydrolase', 'Nucleotide-binding', 'RNA-binding', 'Transcription', 'Transcription regulation', 'Transcription termination']\n",
      "MNLTDLKRKTAADLAVICQDMGLEGTARQKKQEIIFNILNARAQCDAIYGEGVLEILQ\n",
      "58\n",
      "7\n",
      "\n",
      "FUNCTION: Binds to the 23S rRNA. {ECO:0000250}.\n",
      "SUBUNIT: Part of the 50S ribosomal subunit. {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the ribosomal protein L15P family. {ECO:0000305}.\n",
      "['Ribonucleoprotein', 'Ribosomal protein', 'RNA-binding', 'rRNA-binding']\n",
      "AETLVAAKIVKDVKSGVKVLANGELTAKXLTVKVAKVSAAAKAAIEAAGGSVEEA\n",
      "55\n",
      "4\n",
      "\n",
      "FUNCTION: One of two assembly initiator proteins, it binds directly to the 5'-end of the 23S rRNA, where it nucleates assembly of the 50S subunit. {ECO:0000250}.\n",
      "FUNCTION: One of the proteins that surrounds the polypeptide exit tunnel on the outside of the subunit. {ECO:0000250}.\n",
      "SUBUNIT: Part of the 50S ribosomal subunit.\n",
      "SIMILARITY: Belongs to the ribosomal protein L24P family. {ECO:0000305}.\n",
      "['Ribonucleoprotein', 'Ribosomal protein', 'RNA-binding', 'rRNA-binding']\n",
      "MDKVKFKKGDLVKVIAGKHKGTEGPIIRVLREKSRVVIEGITNIKHVKPSQDNTEAGI\n",
      "58\n",
      "8\n",
      "\n",
      "SUBUNIT: Part of the 50S ribosomal subunit. {ECO:0000255|HAMAP- Rule:MF_01371}.\n",
      "SIMILARITY: Belongs to the ribosomal protein L30P family. {ECO:0000255|HAMAP-Rule:MF_01371}.\n",
      "['Complete proteome', 'Reference proteome', 'Ribonucleoprotein', 'Ribosomal protein']\n",
      "MKITLKRSVIGRPQHQVDTVKALGLKKIGDSREVQDSPAIRGMVNSVKHLLEVEA\n",
      "55\n",
      "5\n",
      "\n",
      "SIMILARITY: Belongs to the ribosomal protein L32P family. {ECO:0000255|HAMAP-Rule:MF_00340}.\n",
      "SEQUENCE CAUTION: Sequence=AAP10844.1; Type=Erroneous initiation; Evidence={ECO:0000305};\n",
      "['Complete proteome', 'Reference proteome', 'Ribonucleoprotein', 'Ribosomal protein']\n",
      "MYLFRRTSKAVKRKRRTHFKLSVPGMVECPSCGEAKLAHRVCKACGTYKGKEVISK\n",
      "56\n",
      "5\n",
      "\n",
      "SIMILARITY: Belongs to the ribosomal protein L33P family. {ECO:0000255|HAMAP-Rule:MF_00294}.\n",
      "SEQUENCE CAUTION: Sequence=ABK67173.1; Type=Erroneous initiation; Evidence={ECO:0000305};\n",
      "['Complete proteome', 'Ribonucleoprotein', 'Ribosomal protein']\n",
      "MASSTDVRPKITLACEVCKHRNYITKKNRRNDPDRLELKKYCPNCGKHQAHRETR\n",
      "55\n",
      "6\n",
      "\n",
      "FUNCTION: Binds the 23S rRNA and the E site tRNA.\n",
      "SUBUNIT: Part of the 50S ribosomal subunit. Contacts protein L35. {ECO:0000269|PubMed:11677599, ECO:0000269|PubMed:12535524, ECO:0000269|PubMed:12623020, ECO:0000269|PubMed:12665853, ECO:0000269|PubMed:15059283, ECO:0000269|PubMed:15554968}.\n",
      "SIMILARITY: Belongs to the ribosomal protein L33P family. {ECO:0000305}.\n",
      "SEQUENCE CAUTION: Sequence=AAF11599.1; Type=Erroneous initiation; Evidence={ECO:0000305};\n",
      "['3D-structure', 'Complete proteome', 'Direct protein sequencing', 'Reference proteome', 'Ribonucleoprotein', 'Ribosomal protein', 'RNA-binding', 'rRNA-binding', 'tRNA-binding']\n",
      "MAKDGPRIIVKMESSAGTGFYYTTTKNRRNTQAKLELKKYDPVAKKHVVFREKKV\n",
      "55\n",
      "7\n",
      "\n",
      "SUBUNIT: Part of the 50S ribosomal subunit (PubMed:786732, PubMed:10094780, PubMed:12809609, PubMed:16272117, PubMed:25310980, PubMed:24844575, PubMed:27934701, PubMed:27906160, PubMed:27906161). Cross-links to the P and E site tRNAs (PubMed:8524654). {ECO:0000269|PubMed:10094780, ECO:0000269|PubMed:12809609, ECO:0000269|PubMed:16272117, ECO:0000269|PubMed:24844575, ECO:0000269|PubMed:25310980, ECO:0000269|PubMed:27906160, ECO:0000269|PubMed:27906161, ECO:0000269|PubMed:27934701, ECO:0000269|PubMed:786732, ECO:0000269|PubMed:8524654}.\n",
      "MASS SPECTROMETRY: Mass=6254.1; Method=MALDI; Range=2-55; Evidence={ECO:0000269|PubMed:10094780};\n",
      "MISCELLANEOUS: Surface exposed on the 50S subunit.\n",
      "SIMILARITY: Belongs to the ribosomal protein L33P family. {ECO:0000305}.\n",
      "['3D-structure', 'Complete proteome', 'Direct protein sequencing', 'Methylation', 'Reference proteome', 'Ribonucleoprotein', 'Ribosomal protein', 'RNA-binding', 'tRNA-binding']\n",
      "MAKGIREKIKLVSSAGTGHFYTTTKNKRTKPEKLELKKFDPVVRQHVIYKEAKIK\n",
      "55\n",
      "8\n",
      "\n",
      "SIMILARITY: Belongs to the ribosomal protein L33P family. {ECO:0000255|HAMAP-Rule:MF_00294}.\n",
      "SEQUENCE CAUTION: Sequence=AAZ68271.1; Type=Erroneous initiation; Evidence={ECO:0000305};\n",
      "['Complete proteome', 'Ribonucleoprotein', 'Ribosomal protein']\n",
      "MAKKGSTLLVKLASSAKTGFFYVKKRNPKKLINKLSFRKYDPVVRKHVLFSEEKLK\n",
      "56\n",
      "6\n",
      "\n",
      "PTM: The protein is methylated on either Ala-2 or Lys-3.\n",
      "MASS SPECTROMETRY: Mass=6248.5; Method=Electrospray; Range=2-55; Evidence={ECO:0000269|PubMed:15473684};\n",
      "SIMILARITY: Belongs to the ribosomal protein L33P family. {ECO:0000255|HAMAP-Rule:MF_00294}.\n",
      "['Complete proteome', 'Direct protein sequencing', 'Methylation', 'Reference proteome', 'Ribonucleoprotein', 'Ribosomal protein']\n",
      "MAKAVTIKIKLVSTADTGFYYVTKKNSRTMTDKMVKKKYDPVARKHVEFKEAKIK\n",
      "55\n",
      "9\n",
      "\n",
      "SIMILARITY: Belongs to the ribosomal protein L34P family. {ECO:0000255|HAMAP-Rule:MF_00391}.\n",
      "SEQUENCE CAUTION: Sequence=CAE77669.1; Type=Erroneous initiation; Evidence={ECO:0000305};\n",
      "['Complete proteome', 'Reference proteome', 'Ribonucleoprotein', 'Ribosomal protein']\n",
      "MKRTWQPSKLKHAGVHGFRARMATENGRKVIKARRAKGRVRLSA\n",
      "44\n",
      "9\n",
      "\n",
      "FUNCTION: During stationary phase, converts 70S ribosomes to an inactive dimeric form (100S ribosomes). May form immature 90S particles, which are converted to mature 100S ribosomes by the hibernation promoting factor Hpf. {ECO:0000255|HAMAP- Rule:MF_00919}.\n",
      "SUBUNIT: Associates exclusively with 100S ribosomes. {ECO:0000255|HAMAP-Rule:MF_00919}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000255|HAMAP-Rule:MF_00919}.\n",
      "INDUCTION: Induced during stationary growth phase. {ECO:0000255|HAMAP-Rule:MF_00919}.\n",
      "SIMILARITY: Belongs to the ribosome modulation factor family. {ECO:0000255|HAMAP-Rule:MF_00919}.\n",
      "['Complete proteome', 'Cytoplasm', 'Translation regulation']\n",
      "MKRQKRDRLERAHQRGYQAGIAGRSKEMCPYQTLNQRSQWLGGWREAMADRVVMA\n",
      "55\n",
      "10\n",
      "\n",
      "FUNCTION: During stationary phase, converts 70S ribosomes to an immature dimeric form (90S ribosomes) which are converted to inactive 100S ribosomes (a process called ribosomal hibernation) by the hibernation promoting factor HPF (PubMed:7677746, PubMed:18174192). Inactivates ribosomes by covering the peptidyl transferase (PTase) center of the 23S rRNA and the entrance of peptide exit tunnel (PubMed:12473202, PubMed:15066119). However crystallization with T.thermophilus 70S ribosomes shows it binds near the 3'-end of the 16S rRNA near the anti-Shine-Dalgarno sequence, where it would sterically hinder translation inititation (PubMed:22605777). In this crystal binding of RMF induces movement of the 30S head domain away from the rest of the ribosome, presumably so they would more easily form dimers (PubMed:22605777). Also involved in protection against heat stress, but this role is not dependent on the maintenance of ribosome dimers (PubMed:15278243). {ECO:0000255|HAMAP- Rule:MF_00919, ECO:0000269|PubMed:12473202, ECO:0000269|PubMed:15066119, ECO:0000269|PubMed:15278243, ECO:0000269|PubMed:18174192, ECO:0000269|PubMed:19170772, ECO:0000269|PubMed:7677746, ECO:0000269|PubMed:8440252, ECO:0000269|PubMed:9278503}.\n",
      "SUBUNIT: Associates exclusively with 100S ribosomes (PubMed:2181444, PubMed:12473202, PubMed:15066119). Contacts 16S rRNA, might contact ribosomal protein S18 (PubMed:22605777). {ECO:0000269|PubMed:12473202, ECO:0000269|PubMed:15066119, ECO:0000269|PubMed:2181444, ECO:0000269|PubMed:22605777}.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000305}.\n",
      "INDUCTION: Induced during stationary growth phase (PubMed:8440252). Requires ppGpp for translation (PubMed:11532026). {ECO:0000255|HAMAP-Rule:MF_00919, ECO:0000269|PubMed:11532026, ECO:0000269|PubMed:8440252}.\n",
      "DISRUPTION PHENOTYPE: Non-essential gene, 100S ribosome dimers are not formed, decreased cell viability during stationary phase (PubMed:8440252). A quadruple raiA-hpf-rmf-sra knockout strain is significantly outcompeted by wild-type after 4 days growth (PubMed:17277072). Very high sensitivity to aminoglycoside antiobiotic gentamicin in stationary phase cultures (PubMed:26324267). {ECO:0000269|PubMed:17277072, ECO:0000269|PubMed:26324267, ECO:0000269|PubMed:8440252}.\n",
      "MISCELLANEOUS: When cells are transferred to rich nutritious culture medium, RMF is quickly released from 100S ribosomes, which dissociate into 70S ribosomes (PubMed:12473202). It indicates that this interconversion is a major system regulating translation activity during the transition of growth phases. {ECO:0000305|PubMed:12473202}.\n",
      "SIMILARITY: Belongs to the ribosome modulation factor family. {ECO:0000255|HAMAP-Rule:MF_00919}.\n",
      "SEQUENCE CAUTION: Sequence=J03186; Type=Frameshift; Positions=4, 11, 30; Evidence={ECO:0000305};\n",
      "['3D-structure', 'Complete proteome', 'Cytoplasm', 'Direct protein sequencing', 'Reference proteome', 'RNA-binding', 'rRNA-binding', 'Stress response', 'Translation regulation']\n",
      "MKRQKRDRLERAHQRGYQAGIAGRSKEMCPYQTLNQRSQWLGGWREAMADRVVMA\n",
      "55\n",
      "11\n",
      "\n",
      "FUNCTION: Protein S19 forms a complex with S13 that binds strongly to the 16S ribosomal RNA. {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the ribosomal protein S19P family. {ECO:0000305}.\n",
      "['Ribonucleoprotein', 'Ribosomal protein', 'RNA-binding', 'rRNA-binding']\n",
      "GHAKKDKKIQKK\n",
      "12\n",
      "3\n",
      "\n",
      "FUNCTION: Involved in the hydrocarbon hydroxylating system, which transfers electrons from NADH to rubredoxin reductase and then through rubredoxin to alkane 1 monooxygenase. {ECO:0000269|PubMed:14574114}.\n",
      "COFACTOR: Name=Fe(3+); Xref=ChEBI:CHEBI:29034; Evidence={ECO:0000250}; Note=Binds 1 Fe(3+) ion per subunit. {ECO:0000250};\n",
      "PATHWAY: Hydrocarbon metabolism; alkane degradation.\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000250}.\n",
      "INDUCTION: Constitutively expressed. {ECO:0000269|PubMed:12730186}.\n",
      "SIMILARITY: Belongs to the rubredoxin family. {ECO:0000305}.\n",
      "['Complete proteome', 'Cytoplasm', 'Electron transport', 'Iron', 'Metal-binding', 'Reference proteome', 'Transport']\n",
      "MKKWQCVVCGLIYDEAKGWPEEGIEAGTRWEDVPEDWLCPDCGVGKLDFEMIEIG\n",
      "55\n",
      "12\n",
      "\n",
      "FUNCTION: Serves as an electron acceptor for pyruvate ferredoxin oxidoreductase (PFOR). {ECO:0000250}.\n",
      "COFACTOR: Name=Fe(3+); Xref=ChEBI:CHEBI:29034; Evidence={ECO:0000250}; Note=Binds 1 Fe(3+) ion per subunit. {ECO:0000250};\n",
      "SUBUNIT: Monomer. {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the rubredoxin family. {ECO:0000305}.\n",
      "['Complete proteome', 'Electron transport', 'Iron', 'Metal-binding', 'Reference proteome', 'Transport']\n",
      "MEQWKCNICGYIYNPETGDPEGDIPAGTSFESLPDSWMCPVCGAGKEEFTKI\n",
      "52\n",
      "6\n",
      "\n",
      "FUNCTION: Essential subunit of the Sec protein translocation channel SecYEG. Clamps together the 2 halves of SecY. May contact the channel plug during translocation. {ECO:0000255|HAMAP- Rule:MF_00422}.\n",
      "SUBUNIT: Component of the Sec protein translocase complex. Heterotrimer consisting of SecY, SecE and SecG subunits. The heterotrimers can form oligomers, although 1 heterotrimer is thought to be able to translocate proteins. Interacts with the ribosome. Interacts with SecDF, and other proteins may be involved. Interacts with SecA. {ECO:0000255|HAMAP-Rule:MF_00422}.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane {ECO:0000255|HAMAP- Rule:MF_00422}; Single-pass membrane protein {ECO:0000255|HAMAP- Rule:MF_00422}.\n",
      "SIMILARITY: Belongs to the SecE/SEC61-gamma family. {ECO:0000255|HAMAP-Rule:MF_00422}.\n",
      "['Cell inner membrane', 'Cell membrane', 'Complete proteome', 'Membrane', 'Protein transport', 'Reference proteome', 'Translocation', 'Transmembrane', 'Transmembrane helix', 'Transport']\n",
      "MFRFIKDSILELKKVTWPKYNEVVGNGKQVFWLVLFVSIFLGIVDYLMFLVVTYVF\n",
      "56\n",
      "7\n",
      "\n",
      "FUNCTION: Antitoxin component of a toxin-antitoxin (TA) module. Antitoxin that binds cognate toxin SpoIISA and neutralizes its toxic activity; unlike most antitoxins it does not seem to be highly labile upon expression in E.coli. {ECO:0000269|PubMed:11371520}.\n",
      "SUBUNIT: The isolated protein is unfolded; X-ray data suggests the inactive complex forms a heterotetramer of SpoIISA(2)-SpoIISB(2), which inactivates the toxic activity of SpoIISA.\n",
      "INTERACTION: O34853:spoIISA; NbExp=3; IntAct=EBI-4406141, EBI-4406150;\n",
      "INDUCTION: Expressed during exponential growth and sporulation. A member of the spoIISA-spoIISB operon, it also has its own promoter. {ECO:0000269|PubMed:11371520}.\n",
      "DISRUPTION PHENOTYPE: Decreases sporulation efficiency by 4 orders of magnitude. A double spoIISA-spoIISB disruption sporulates normally, suggesting its only role is to neutralize SpoIISA. {ECO:0000269|PubMed:11371520}.\n",
      "SIMILARITY: Belongs to the SpoIISB antitoxin family. {ECO:0000305}.\n",
      "['3D-structure', 'Complete proteome', 'Reference proteome', 'Sporulation']\n",
      "MERAFQNRCEPRAAKPFKILKKRSTTSVASYQVSPHTARIFKENERLIDEYKRKKA\n",
      "56\n",
      "8\n",
      "\n",
      "FUNCTION: May play some important role in either sporulation or the dormant spore.\n",
      "SIMILARITY: Belongs to the alpha/beta-type SASP family. {ECO:0000305}.\n",
      "['Complete proteome', 'DNA-binding', 'Reference proteome', 'Sporulation']\n",
      "MSRRRGIMSSKLKEELAKELGFYDTVQAEGWGAIRAKDAGNMVKRAVELAQQQLKQ\n",
      "56\n",
      "9\n",
      "\n",
      "DEVELOPMENTAL STAGE: Expressed only in the mother cell compartment of sporulating cells. {ECO:0000269|PubMed:9852018}.\n",
      "INDUCTION: Expression is GerE and sigma K-dependent.\n",
      "['Complete proteome', 'Direct protein sequencing', 'Reference proteome', 'Sporulation']\n",
      "MSENRHENEENRRDAAVAKVQNSGNAKVVVSVNTDQDQAQAQSQDGED\n",
      "48\n",
      "9\n",
      "\n",
      "SUBCELLULAR LOCATION: Spore core {ECO:0000255|HAMAP- Rule:MF_01505}.\n",
      "INDUCTION: Expressed only in the forespore compartment of sporulating cells. {ECO:0000255|HAMAP-Rule:MF_01505}.\n",
      "SIMILARITY: Belongs to the SspN family. {ECO:0000255|HAMAP- Rule:MF_01505}.\n",
      "['Sporulation']\n",
      "MGNPKKNSKDFAPNHIGTQSKKAGGNKGKQMQDQTGKQPIVDNG\n",
      "44\n",
      "10\n",
      "\n",
      "SUBCELLULAR LOCATION: Spore core {ECO:0000305}.\n",
      "INDUCTION: Expressed only in the forespore compartment of sporulating cells. Expression is sigma F and sigma G-dependent. {ECO:0000269|PubMed:10333516}.\n",
      "SIMILARITY: Belongs to the SspN family. {ECO:0000305}.\n",
      "['Complete proteome', 'Reference proteome', 'Sporulation']\n",
      "MGNNKKNGQPQYAPSHLGTKPVKYKANKGEKMHDTSGQRPIIMQTKGE\n",
      "48\n",
      "10\n",
      "\n",
      "SUBCELLULAR LOCATION: Spore core {ECO:0000255|HAMAP- Rule:MF_00665}.\n",
      "INDUCTION: Expressed only in the forespore compartment of sporulating cells. {ECO:0000255|HAMAP-Rule:MF_00665}.\n",
      "SIMILARITY: Belongs to the SspO family. {ECO:0000255|HAMAP- Rule:MF_00665}.\n",
      "['Complete proteome', 'Reference proteome', 'Sporulation']\n",
      "MAKRKANHVINGMNNAKRQGNGAGYIENDQHILTEAERQNNKKRKTNQ\n",
      "48\n",
      "11\n",
      "\n",
      "SUBCELLULAR LOCATION: Spore core {ECO:0000255|HAMAP- Rule:MF_00666}.\n",
      "INDUCTION: Expressed only in the forespore compartment of sporulating cells. {ECO:0000255|HAMAP-Rule:MF_00666}.\n",
      "SIMILARITY: Belongs to the SspP family. {ECO:0000255|HAMAP- Rule:MF_00666}.\n",
      "SEQUENCE CAUTION: Sequence=AAS42544.1; Type=Erroneous initiation; Evidence={ECO:0000305};\n",
      "['Complete proteome', 'Sporulation']\n",
      "MSQTMSKNNREAKEKKGQPEPLSGSHKVKNRNHSRQKHHAHHDM\n",
      "44\n",
      "11\n",
      "\n",
      "FUNCTION: Part of the twin-arginine translocation (Tat) system that transports large folded proteins containing a characteristic twin-arginine motif in their signal peptide across membranes. TatA could form the protein-conducting channel of the Tat system. {ECO:0000255|HAMAP-Rule:MF_00236}.\n",
      "SUBUNIT: The Tat system comprises two distinct complexes: a TatABC complex, containing multiple copies of TatA, TatB and TatC subunits, and a separate TatA complex, containing only TatA subunits. Substrates initially bind to the TatABC complex, which probably triggers association of the separate TatA complex to form the active translocon. {ECO:0000255|HAMAP-Rule:MF_00236}.\n",
      "SUBCELLULAR LOCATION: Cell inner membrane {ECO:0000255|HAMAP- Rule:MF_00236}; Single-pass membrane protein {ECO:0000255|HAMAP- Rule:MF_00236}.\n",
      "SIMILARITY: Belongs to the TatA/E family. {ECO:0000255|HAMAP- Rule:MF_00236}.\n",
      "['Cell inner membrane', 'Cell membrane', 'Membrane', 'Protein transport', 'Translocation', 'Transmembrane', 'Transmembrane helix', 'Transport']\n",
      "MALGPWQIFLILVIILVLFGAGKLPDVMSDLGKGIRNLKQELKDNKLASTEDESNL\n",
      "56\n",
      "10\n",
      "\n",
      "CATALYTIC ACTIVITY: D-xylopyranose = D-xylulose.\n",
      "COFACTOR: Name=Mg(2+); Xref=ChEBI:CHEBI:18420; Evidence={ECO:0000305}; Note=Binds 2 magnesium ions per subunit. {ECO:0000305};\n",
      "SUBUNIT: Homotetramer.\n",
      "SUBCELLULAR LOCATION: Cytoplasm.\n",
      "SIMILARITY: Belongs to the xylose isomerase family. {ECO:0000305}.\n",
      "['Carbohydrate metabolism', 'Cytoplasm', 'Direct protein sequencing', 'Isomerase', 'Magnesium', 'Metal-binding', 'Pentose shunt', 'Xylose metabolism']\n",
      "GVTFHDDDLIPF\n",
      "12\n",
      "4\n",
      "\n",
      "SUBCELLULAR LOCATION: Cytoplasm {ECO:0000255|HAMAP-Rule:MF_01103}.\n",
      "SIMILARITY: Belongs to the UPF0291 family. {ECO:0000255|HAMAP- Rule:MF_01103}.\n",
      "['Complete proteome', 'Cytoplasm', 'Reference proteome']\n",
      "MVSKEKINRINELARIAKERVLTKLEEEEQQILRKEYIEAFRKSFRKQLDNIELVD\n",
      "56\n",
      "11\n",
      "\n",
      "SUBCELLULAR LOCATION: Cell membrane {ECO:0000255|HAMAP- Rule:MF_01361}; Multi-pass membrane protein {ECO:0000255|HAMAP- Rule:MF_01361}.\n",
      "SIMILARITY: Belongs to the UPF0391 family. {ECO:0000255|HAMAP- Rule:MF_01361}.\n",
      "['Cell membrane', 'Complete proteome', 'Membrane', 'Reference proteome', 'Transmembrane', 'Transmembrane helix']\n",
      "MLTWALIFLVVAVVAALFGFGGIAGAAASIAQILFFVFLVLLVISLIMHFTRRSRL\n",
      "56\n",
      "12\n",
      "\n",
      "Breaking - got the sequences\n",
      "7: 1 9: 2 10: 1 12: 4 13: 1 14: 1 15: 1 18: 2 19: 5 20: 3 21: 5 22: 3 23: 1 24: 4 25: 2 26: 2 28: 1 29: 1 30: 3 31: 2 34: 1 35: 2 36: 2 37: 2 38: 4 39: 3 40: 3 41: 4 42: 3 43: 10 44: 11 45: 3 46: 6 47: 10 48: 11 49: 5 50: 6 51: 11 52: 6 53: 7 54: 2 55: 12 56: 12 57: 13 58: 8 59: 7 60: 5 61: 12 62: 12 63: 8 64: 11 65: 7 66: 5 67: 11 68: 8 69: 9 70: 3 71: 10 72: 6 73: 1 74: 7 75: 5 76: 8 77: 3 78: 2 79: 1 80: 2 81: 3 83: 1 84: 4 85: 2 86: 1 87: 1 88: 1 89: 2 90: 3 91: 4 92: 1 95: 1 96: 2 97: 2 99: 1 101: 1 102: 3 103: 2 105: 1 110: 1 120: 2 128: 1 131: 1 207: 1 225: 1 255: 1\n"
     ]
    }
   ],
   "source": [
    "\"\"\"\n",
    "Getting negative sequnces of the same length as the postitive training examples\n",
    "for bacteriocin detection i.e. BAGEL sequences\n",
    "\"\"\"\n",
    "\n",
    "from Bio import SeqIO\n",
    "from collections import defaultdict\n",
    "flag = 0\n",
    "count = 0\n",
    "\n",
    "keys_track = defaultdict(int)\n",
    "keys_done = []\n",
    "keys_description_done = []\n",
    "out_handle = open('negative_bacteriocin_set', 'w')\n",
    "for record in SeqIO.parse(\"uniprot_sprot_bacteria.dat\", \"swiss\"):\n",
    "    if 'keywords' in record.annotations and 'comment' in record.annotations: \n",
    "        if 'Bacteriocin' not in record.annotations['keywords'] and \\\n",
    "            'Plasmid' not in record.annotations['keywords'] and \\\n",
    "            'Antimicrobial' not in record.annotations['keywords'] and \\\n",
    "            'Antibiotic' not in record.annotations['keywords'] and \\\n",
    "            len(record.annotations['comment']) >= 125 and \\\n",
    "            len(record.seq) in merged_class and \\\n",
    "            len(record.seq) not in keys_done and \\\n",
    "            str(record.description) not in keys_description_done:\n",
    "                if keys_track[len(record.seq)] + 1 < merged_class[len(record.seq)]:\n",
    "                    keys_track[len(record.seq)] = keys_track[len(record.seq)] + 1\n",
    "                    flag = 1\n",
    "                elif keys_track[len(record.seq)] + 1 == merged_class[len(record.seq)]:\n",
    "                    keys_track[len(record.seq)] = keys_track[len(record.seq)] + 1\n",
    "                    keys_done.append(len(record.seq))\n",
    "                    flag = 1\n",
    "        if flag == 1:\n",
    "            print record.annotations['comment']\n",
    "            print record.annotations['keywords']\n",
    "            print record.seq\n",
    "            print len(record.seq)\n",
    "            print keys_track[len(record.seq)]\n",
    "            print\n",
    "            # NO 2 sequences should have the same description\n",
    "            keys_description_done.append(str(record.description))\n",
    "            \n",
    "            out_handle.write('>%s|%s|%s|%s\\n%s\\n' % (record.name, record.id, record.description, \n",
    "                                                     str(len(record.seq)),\n",
    "                                                     record.seq))\n",
    "        flag = 0\n",
    "    if keys_track == merged_class:\n",
    "        print \"Breaking - got the sequences\"\n",
    "        break\n",
    "        #break\n",
    "\n",
    "out_handle.close()\n",
    "keylist = keys_track.keys()\n",
    "keylist.sort()\n",
    "for key in keylist:\n",
    "    print \"%s: %s\" % (key, keys_track[key]),"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FUNCTION: Has bactericidal activity against E.coli ATCC 29522 and S.aureus. Inhibits growth of S.marcescens and B.cereus ATCC 14579. A synthetic peptide has antibacterial activity against E.coli ATCC 29522 (MIC=8 ug/ml), S.aureus (MIC=64 ug/ml), S.marcescens (MIC=32 ug/ml), B.cereus ATCC 14579 (MIC=64 ug/ml), B.subtilis (MIC=32 ug/ml), L.plantarum ATCC 8014 (MIC=32 ug/ml), B.flexus (MIC=32 ug/ml), S.enteritidis ATCC 13076 (MIC=4 ug/ml), Enterobacter spp (MIC=32 ug/ml), B.anthracis (MIC=128 ug/ml), B.licheniformis (MIC>128 ug/ml) and L.lactis ATCC 11454 (MIC>128 ug/ml). {ECO:0000269|PubMed:23508455}.\n",
      "BIOPHYSICOCHEMICAL PROPERTIES: pH dependence:   Active between pH 2 and 12. {ECO:0000269|PubMed:23508455}; Temperature dependence:   Active between 37 and 90 degrees Celsius. Activity is reduced at   100 degrees Celsius and absent at 110 degrees Celsius. Activity   is unaffected by storage at -20 degrees Celsius.   {ECO:0000269|PubMed:23508455};\n",
      "MASS SPECTROMETRY: Mass=1316.734; Method=Electrospray; Range=1-12; Evidence={ECO:0000269|PubMed:23508455};\n",
      "MISCELLANEOUS: A synthetic peptide has no antibacterial activity against S.agalactiae ATCC 12386, S.epidermidis, P.aeruginosa and methicillin-resistant S.aureus (MRSA). {ECO:0000269|PubMed:23508455}.\n",
      "['Antibiotic', 'Antimicrobial', 'Direct protein sequencing']\n",
      "YSKSLPLSVLNP\n",
      "\n",
      "FUNCTION: Has antibacterial activity against Gram-positive bacteria L.plantarum strain ATCC 8014 and B.cereus strain ATCC 14579 as well as against Gram-negative bacteria E.coli strain ATCC 25922 and S.enteritidis strain ATCC 13076. {ECO:0000269|Ref.1}.\n",
      "['Antibiotic', 'Antimicrobial', 'Direct protein sequencing']\n",
      "FCKSLPLPLSVK\n",
      "\n",
      "FUNCTION: Degrades heparin and heparan sulfate. {ECO:0000250|UniProtKB:Q05819}.\n",
      "CATALYTIC ACTIVITY: Eliminative cleavage of polysaccharides containing (1->4)-linked D-glucuronate or L-iduronate residues and (1->4)-alpha-linked 2-sulfoamino-2-deoxy-6-sulfo-D-glucose residues to give oligosaccharides with terminal 4-deoxy-alpha-D- gluc-4-enuronosyl groups at their non-reducing ends.\n",
      "ENZYME REGULATION: Inhibited by Cu(2+) ion, lead and some agents that modify histidine and cysteine residues. Activated by KCl and by reducing agents, such as dithiothreitol and 2-mercaptoethanol.\n",
      "BIOPHYSICOCHEMICAL PROPERTIES: pH dependence:   Optimum pH is 7.2.; Temperature dependence:   Optimum temperature is 45 degrees Celsius.;\n",
      "PTM: The N-terminus is blocked. {ECO:0000269|PubMed:10920269}.\n",
      "['Direct protein sequencing', 'Heparin-binding', 'Lyase']\n",
      "MADEALQHTFFA\n",
      "\n",
      "FUNCTION: The 2-oxoglutarate dehydrogenase complex catalyzes the overall conversion of 2-oxoglutarate to succinyl-CoA and CO(2). It contains multiple copies of 3 enzymatic components: 2-oxoglutarate dehydrogenase (E1), dihydrolipoamide succinyltransferase (E2) and lipoamide dehydrogenase (E3) (By similarity). {ECO:0000250|UniProtKB:P52993}.\n",
      "CATALYTIC ACTIVITY: Succinyl-CoA + enzyme N(6)- (dihydrolipoyl)lysine = CoA + enzyme N(6)-(S- succinyldihydrolipoyl)lysine.\n",
      "COFACTOR: Name=(R)-lipoate; Xref=ChEBI:CHEBI:83088; Evidence={ECO:0000250}; Note=Binds 1 lipoyl cofactor covalently. {ECO:0000250};\n",
      "PATHWAY: Amino-acid degradation; L-lysine degradation via saccharopine pathway; glutaryl-CoA from L-lysine: step 6/6.\n",
      "SUBUNIT: Forms a 24-polypeptide structural core with octahedral symmetry. {ECO:0000250|UniProtKB:P52993}.\n",
      "INDUCTION: By 2,4-dichlorophenol. {ECO:0000269|PubMed:15073309}.\n",
      "SIMILARITY: Belongs to the 2-oxoacid dehydrogenase family. {ECO:0000305}.\n",
      "['Acyltransferase', 'Direct protein sequencing', 'Lipoyl', 'Transferase', 'Tricarboxylic acid cycle']\n",
      "AIVEVKVPXLXE\n",
      "\n",
      "FUNCTION: Peptidase. Has bacteriolytic activity. {ECO:0000269|Ref.1}.\n",
      "BIOPHYSICOCHEMICAL PROPERTIES: pH dependence:   Optimum pH is 7.5. {ECO:0000269|Ref.1}; Temperature dependence:   Optimum temperature is 80 degrees Celsius. {ECO:0000269|Ref.1};\n",
      "SUBUNIT: Monomer. {ECO:0000269|Ref.1}.\n",
      "SUBCELLULAR LOCATION: Secreted {ECO:0000269|Ref.1}.\n",
      "['Antibiotic', 'Antimicrobial', 'Bacteriolytic enzyme', 'Direct protein sequencing', 'Hydrolase', 'Protease', 'Secreted']\n",
      "ATVQGGIXYRMP\n",
      "\n",
      "FUNCTION: Protein S19 forms a complex with S13 that binds strongly to the 16S ribosomal RNA. {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the ribosomal protein S19P family. {ECO:0000305}.\n",
      "['Ribonucleoprotein', 'Ribosomal protein', 'RNA-binding', 'rRNA-binding']\n",
      "GHAKKDKKIQKK\n",
      "\n",
      "FUNCTION: Protein S19 forms a complex with S13 that binds strongly to the 16S ribosomal RNA. {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the ribosomal protein S19P family. {ECO:0000305}.\n",
      "['Ribonucleoprotein', 'Ribosomal protein', 'RNA-binding', 'rRNA-binding']\n",
      "GHAKGDKKVSKK\n",
      "\n",
      "FUNCTION: Protein S19 forms a complex with S13 that binds strongly to the 16S ribosomal RNA. {ECO:0000250}.\n",
      "SIMILARITY: Belongs to the ribosomal protein S19P family. {ECO:0000305}.\n",
      "['Ribonucleoprotein', 'Ribosomal protein', 'RNA-binding', 'rRNA-binding']\n",
      "GHNKKDKKMQKK\n",
      "\n",
      "INDUCTION: By superoxide.\n",
      "['Direct protein sequencing']\n",
      "GNALGKEGLKAI\n",
      "\n",
      "MISCELLANEOUS: On the 2D-gel the determined pI of this unknown protein is: 5.4, its MW is: 29.5 kDa.\n",
      "['Direct protein sequencing']\n",
      "MIYSTEVVNMNM\n",
      "\n",
      "CATALYTIC ACTIVITY: D-xylopyranose = D-xylulose.\n",
      "COFACTOR: Name=Mg(2+); Xref=ChEBI:CHEBI:18420; Evidence={ECO:0000305}; Note=Binds 2 magnesium ions per subunit. {ECO:0000305};\n",
      "SUBUNIT: Homotetramer.\n",
      "SUBCELLULAR LOCATION: Cytoplasm.\n",
      "SIMILARITY: Belongs to the xylose isomerase family. {ECO:0000305}.\n",
      "['Carbohydrate metabolism', 'Cytoplasm', 'Direct protein sequencing', 'Isomerase', 'Magnesium', 'Metal-binding', 'Pentose shunt', 'Xylose metabolism']\n",
      "GVTFHDDDLIPF\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for record in SeqIO.parse(\"uniprot_sprot_bacteria.dat\", \"swiss\"):\n",
    "    if 'keywords' in record.annotations and 'comment' in record.annotations: \n",
    "#         for kw in record.annotations['keywords']:\n",
    "#             if 'bacteriocin' in kw:\n",
    "#                 #print 'yes'\n",
    "#                 flag = 1\n",
    "#             elif 'Bacteriocin' in kw and 'plasmid' not in kw:\n",
    "#                 flag = 1\n",
    "#             #else:\n",
    "            #    print \"no\"\n",
    "#         if 'Bacteriocin' in record.annotations['keywords'] and 'Plasmid' not in record.annotations['keywords']:\n",
    "#             count += 1\n",
    "#             flag = 1\n",
    "        if 'Bacteriocin' not in record.annotations['keywords'] and \\\n",
    "            'Plasmid' not in record.annotations['keywords'] and \\\n",
    "            len(record.seq) == 12:\n",
    "                print record.annotations['comment']\n",
    "                print record.annotations['keywords']\n",
    "                print record.seq\n",
    "                print"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RecName: Full=12-alpha-hydroxysteroid dehydrogenase; EC=1.1.1.176; Flags: Fragment;\n"
     ]
    }
   ],
   "source": [
    "for record in SeqIO.parse(\"uniprot_sprot_bacteria.dat\", \"swiss\"):\n",
    "    print record.q\n",
    "    break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "\"\"\"\n",
    "Writing down all trembl bacteria sequences in a fasta file\n",
    "\"\"\"\n",
    "out_handle = open('trembl_bacteria_seqs.fa', 'w')\n",
    "for record in SeqIO.parse(\"uniprot_trembl_bacteria.dat\", \"swiss\"):\n",
    "    out_handle.write('>%s|%s|%s|%s\\n%s\\n' % (record.name, record.id, record.description, str(len(record.seq)),\n",
    "                                             record.seq))\n",
    "\n",
    "out_handle.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<function close>"
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\"\"\"\n",
    "Writing down all swissprot bacteria sequences in a fasta file\n",
    "\"\"\"\n",
    "out_handle = open('swissprot_bacteria_seqs.fa', 'w')\n",
    "for record in SeqIO.parse(\"uniprot_sprot_bacteria.dat\", \"swiss\"):\n",
    "    out_handle.write('>%s|%s|%s|%s\\n%s\\n' % (record.name, record.id, record.description, str(len(record.seq)),\n",
    "                                             record.seq))\n",
    "\n",
    "out_handle.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31\n",
      "255\n"
     ]
    }
   ],
   "source": [
    "min_ = 10000000\n",
    "max_ = -1 \n",
    "for record in SeqIO.parse('less_than_30_pos_neg_bacteriocin.fa', 'fasta'):\n",
    "    length = len(record.seq)\n",
    "    #print length\n",
    "    if int(length) < min_:\n",
    "        min_ = int(length)\n",
    "    if int(length) > max_:\n",
    "        max_ = int(length)\n",
    "    #break\n",
    "print min_\n",
    "print max_"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## Making 4 more negative datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30\n",
      "1\n",
      "\n",
      "79\n",
      "1\n",
      "\n",
      "73\n",
      "1\n",
      "\n",
      "128\n",
      "1\n",
      "\n",
      "120\n",
      "1\n",
      "\n",
      "131\n",
      "1\n",
      "\n",
      "120\n",
      "2\n",
      "\n",
      "110\n",
      "1\n",
      "\n",
      "88\n",
      "1\n",
      "\n",
      "86\n",
      "1\n",
      "\n",
      "83\n",
      "1\n",
      "\n",
      "95\n",
      "1\n",
      "\n",
      "78\n",
      "1\n",
      "\n",
      "76\n",
      "1\n",
      "\n",
      "85\n",
      "1\n",
      "\n",
      "80\n",
      "1\n",
      "\n",
      "78\n",
      "2\n",
      "\n",
      "80\n",
      "2\n",
      "\n",
      "85\n",
      "2\n",
      "\n",
      "92\n",
      "1\n",
      "\n",
      "74\n",
      "1\n",
      "\n",
      "255\n",
      "1\n",
      "\n",
      "105\n",
      "1\n",
      "\n",
      "91\n",
      "1\n",
      "\n",
      "90\n",
      "1\n",
      "\n",
      "91\n",
      "2\n",
      "\n",
      "97\n",
      "1\n",
      "\n",
      "99\n",
      "1\n",
      "\n",
      "90\n",
      "2\n",
      "\n",
      "91\n",
      "3\n",
      "\n",
      "89\n",
      "1\n",
      "\n",
      "90\n",
      "3\n",
      "\n",
      "91\n",
      "4\n",
      "\n",
      "68\n",
      "1\n",
      "\n",
      "89\n",
      "2\n",
      "\n",
      "96\n",
      "1\n",
      "\n",
      "97\n",
      "2\n",
      "\n",
      "30\n",
      "2\n",
      "\n",
      "40\n",
      "1\n",
      "\n",
      "30\n",
      "3\n",
      "\n",
      "207\n",
      "1\n",
      "\n",
      "87\n",
      "1\n",
      "\n",
      "50\n",
      "1\n",
      "\n",
      "225\n",
      "1\n",
      "\n",
      "84\n",
      "1\n",
      "\n",
      "72\n",
      "1\n",
      "\n",
      "76\n",
      "2\n",
      "\n",
      "77\n",
      "1\n",
      "\n",
      "81\n",
      "1\n",
      "\n",
      "71\n",
      "1\n",
      "\n",
      "84\n",
      "2\n",
      "\n",
      "96\n",
      "2\n",
      "\n",
      "74\n",
      "2\n",
      "\n",
      "81\n",
      "2\n",
      "\n",
      "101\n",
      "1\n",
      "\n",
      "75\n",
      "1\n",
      "\n",
      "75\n",
      "2\n",
      "\n",
      "67\n",
      "1\n",
      "\n",
      "81\n",
      "3\n",
      "\n",
      "102\n",
      "1\n",
      "\n",
      "41\n",
      "1\n",
      "\n",
      "103\n",
      "1\n",
      "\n",
      "84\n",
      "3\n",
      "\n",
      "31\n",
      "1\n",
      "\n",
      "102\n",
      "2\n",
      "\n",
      "102\n",
      "3\n",
      "\n",
      "103\n",
      "2\n",
      "\n",
      "84\n",
      "4\n",
      "\n",
      "47\n",
      "1\n",
      "\n",
      "66\n",
      "1\n",
      "\n",
      "68\n",
      "2\n",
      "\n",
      "34\n",
      "1\n",
      "\n",
      "77\n",
      "2\n",
      "\n",
      "41\n",
      "2\n",
      "\n",
      "71\n",
      "2\n",
      "\n",
      "77\n",
      "3\n",
      "\n",
      "35\n",
      "1\n",
      "\n",
      "42\n",
      "1\n",
      "\n",
      "70\n",
      "1\n",
      "\n",
      "31\n",
      "2\n",
      "\n",
      "35\n",
      "2\n",
      "\n",
      "75\n",
      "3\n",
      "\n",
      "75\n",
      "4\n",
      "\n",
      "74\n",
      "3\n",
      "\n",
      "75\n",
      "5\n",
      "\n",
      "76\n",
      "3\n",
      "\n",
      "74\n",
      "4\n",
      "\n",
      "74\n",
      "5\n",
      "\n",
      "65\n",
      "1\n",
      "\n",
      "74\n",
      "6\n",
      "\n",
      "74\n",
      "7\n",
      "\n",
      "59\n",
      "1\n",
      "\n",
      "65\n",
      "2\n",
      "\n",
      "56\n",
      "1\n",
      "\n",
      "58\n",
      "1\n",
      "\n",
      "60\n",
      "1\n",
      "\n",
      "70\n",
      "2\n",
      "\n",
      "71\n",
      "3\n",
      "\n",
      "68\n",
      "3\n",
      "\n",
      "71\n",
      "4\n",
      "\n",
      "72\n",
      "2\n",
      "\n",
      "47\n",
      "2\n",
      "\n",
      "63\n",
      "1\n",
      "\n",
      "71\n",
      "5\n",
      "\n",
      "57\n",
      "1\n",
      "\n",
      "49\n",
      "1\n",
      "\n",
      "50\n",
      "2\n",
      "\n",
      "60\n",
      "2\n",
      "\n",
      "72\n",
      "3\n",
      "\n",
      "40\n",
      "2\n",
      "\n",
      "58\n",
      "2\n",
      "\n",
      "67\n",
      "2\n",
      "\n",
      "63\n",
      "2\n",
      "\n",
      "59\n",
      "2\n",
      "\n",
      "69\n",
      "1\n",
      "\n",
      "62\n",
      "1\n",
      "\n",
      "69\n",
      "2\n",
      "\n",
      "40\n",
      "3\n",
      "\n",
      "42\n",
      "2\n",
      "\n",
      "48\n",
      "1\n",
      "\n",
      "51\n",
      "1\n",
      "\n",
      "49\n",
      "2\n",
      "\n",
      "69\n",
      "3\n",
      "\n",
      "50\n",
      "3\n",
      "\n",
      "43\n",
      "1\n",
      "\n",
      "69\n",
      "4\n",
      "\n",
      "69\n",
      "5\n",
      "\n",
      "69\n",
      "6\n",
      "\n",
      "76\n",
      "4\n",
      "\n",
      "53\n",
      "1\n",
      "\n",
      "38\n",
      "1\n",
      "\n",
      "71\n",
      "6\n",
      "\n",
      "70\n",
      "3\n",
      "\n",
      "72\n",
      "4\n",
      "\n",
      "45\n",
      "1\n",
      "\n",
      "72\n",
      "5\n",
      "\n",
      "72\n",
      "6\n",
      "\n",
      "37\n",
      "1\n",
      "\n",
      "37\n",
      "2\n",
      "\n",
      "76\n",
      "5\n",
      "\n",
      "76\n",
      "6\n",
      "\n",
      "44\n",
      "1\n",
      "\n",
      "49\n",
      "3\n",
      "\n",
      "42\n",
      "3\n",
      "\n",
      "39\n",
      "1\n",
      "\n",
      "44\n",
      "2\n",
      "\n",
      "51\n",
      "2\n",
      "\n",
      "67\n",
      "3\n",
      "\n",
      "38\n",
      "2\n",
      "\n",
      "38\n",
      "3\n",
      "\n",
      "50\n",
      "4\n",
      "\n",
      "36\n",
      "1\n",
      "\n",
      "41\n",
      "3\n",
      "\n",
      "38\n",
      "4\n",
      "\n",
      "43\n",
      "2\n",
      "\n",
      "41\n",
      "4\n",
      "\n",
      "61\n",
      "1\n",
      "\n",
      "76\n",
      "7\n",
      "\n",
      "63\n",
      "3\n",
      "\n",
      "68\n",
      "4\n",
      "\n",
      "64\n",
      "1\n",
      "\n",
      "61\n",
      "2\n",
      "\n",
      "69\n",
      "7\n",
      "\n",
      "66\n",
      "2\n",
      "\n",
      "55\n",
      "1\n",
      "\n",
      "49\n",
      "4\n",
      "\n",
      "58\n",
      "3\n",
      "\n",
      "64\n",
      "2\n",
      "\n",
      "65\n",
      "3\n",
      "\n",
      "67\n",
      "4\n",
      "\n",
      "63\n",
      "4\n",
      "\n",
      "69\n",
      "8\n",
      "\n",
      "68\n",
      "5\n",
      "\n",
      "64\n",
      "3\n",
      "\n",
      "60\n",
      "3\n",
      "\n",
      "60\n",
      "4\n",
      "\n",
      "65\n",
      "4\n",
      "\n",
      "58\n",
      "4\n",
      "\n",
      "61\n",
      "3\n",
      "\n",
      "66\n",
      "3\n",
      "\n",
      "61\n",
      "4\n",
      "\n",
      "63\n",
      "5\n",
      "\n",
      "64\n",
      "4\n",
      "\n",
      "59\n",
      "3\n",
      "\n",
      "65\n",
      "5\n",
      "\n",
      "60\n",
      "5\n",
      "\n",
      "67\n",
      "5\n",
      "\n",
      "59\n",
      "4\n",
      "\n",
      "61\n",
      "5\n",
      "\n",
      "59\n",
      "5\n",
      "\n",
      "59\n",
      "6\n",
      "\n",
      "63\n",
      "6\n",
      "\n",
      "61\n",
      "6\n",
      "\n",
      "61\n",
      "7\n",
      "\n",
      "69\n",
      "9\n",
      "\n",
      "65\n",
      "6\n",
      "\n",
      "59\n",
      "7\n",
      "\n",
      "67\n",
      "6\n",
      "\n",
      "63\n",
      "7\n",
      "\n",
      "61\n",
      "8\n",
      "\n",
      "68\n",
      "6\n",
      "\n",
      "76\n",
      "8\n",
      "\n",
      "67\n",
      "7\n",
      "\n",
      "68\n",
      "7\n",
      "\n",
      "71\n",
      "7\n",
      "\n",
      "71\n",
      "8\n",
      "\n",
      "71\n",
      "9\n",
      "\n",
      "57\n",
      "2\n",
      "\n",
      "49\n",
      "5\n",
      "\n",
      "55\n",
      "2\n",
      "\n",
      "55\n",
      "3\n",
      "\n",
      "53\n",
      "2\n",
      "\n",
      "51\n",
      "3\n",
      "\n",
      "66\n",
      "4\n",
      "\n",
      "66\n",
      "5\n",
      "\n",
      "67\n",
      "8\n",
      "\n",
      "68\n",
      "8\n",
      "\n",
      "71\n",
      "10\n",
      "\n",
      "67\n",
      "9\n",
      "\n",
      "65\n",
      "7\n",
      "\n",
      "67\n",
      "10\n",
      "\n",
      "67\n",
      "11\n",
      "\n",
      "62\n",
      "2\n",
      "\n",
      "61\n",
      "9\n",
      "\n",
      "61\n",
      "10\n",
      "\n",
      "54\n",
      "1\n",
      "\n",
      "64\n",
      "5\n",
      "\n",
      "43\n",
      "3\n",
      "\n",
      "53\n",
      "3\n",
      "\n",
      "47\n",
      "3\n",
      "\n",
      "48\n",
      "2\n",
      "\n",
      "56\n",
      "2\n",
      "\n",
      "39\n",
      "2\n",
      "\n",
      "61\n",
      "11\n",
      "\n",
      "51\n",
      "4\n",
      "\n",
      "62\n",
      "3\n",
      "\n",
      "64\n",
      "6\n",
      "\n",
      "61\n",
      "12\n",
      "\n",
      "58\n",
      "5\n",
      "\n",
      "58\n",
      "6\n",
      "\n",
      "55\n",
      "4\n",
      "\n",
      "55\n",
      "5\n",
      "\n",
      "57\n",
      "3\n",
      "\n",
      "56\n",
      "3\n",
      "\n",
      "62\n",
      "4\n",
      "\n",
      "63\n",
      "8\n",
      "\n",
      "57\n",
      "4\n",
      "\n",
      "57\n",
      "5\n",
      "\n",
      "53\n",
      "4\n",
      "\n",
      "53\n",
      "5\n",
      "\n",
      "47\n",
      "4\n",
      "\n",
      "62\n",
      "5\n",
      "\n",
      "62\n",
      "6\n",
      "\n",
      "Second Breaking - got the sequences\n",
      "30: 13 31: 6 34: 7 35: 12 36: 1 37: 4 38: 6 39: 2 40: 7 41: 4 42: 5 43: 3 44: 2 45: 1 47: 4 48: 2 49: 5 50: 4 51: 4 53: 5 54: 1 55: 5 56: 3 57: 5 58: 6 59: 12 60: 11 61: 13 62: 6 63: 8 64: 6 65: 15 66: 11 67: 14 68: 14 69: 20 70: 10 71: 10 72: 6 73: 1 74: 7 75: 5 76: 8 77: 3 78: 2 79: 1 80: 2 81: 3 83: 1 84: 4 85: 2 86: 1 87: 1 88: 1 89: 2 90: 3 91: 4 92: 1 95: 1 96: 2 97: 2 99: 1 101: 1 102: 3 103: 2 105: 1 110: 1 120: 2 128: 1 131: 1 207: 1 225: 1 255: 1\n"
     ]
    }
   ],
   "source": [
    "\"\"\"\n",
    "Creating a second negative bacteriocin dataset with almost same distribution\n",
    "\"\"\"\n",
    "flag = 0\n",
    "count = 0\n",
    "\n",
    "keys_track = defaultdict(int)\n",
    "keys_done = []\n",
    "keys_description_done = []\n",
    "new_in_neg_dataset_ids = []\n",
    "out_handle = open('extra_second_negative_bacteriocin_proper_distribution', 'w')\n",
    "for record in SeqIO.parse(\"redundant_negative_bacteriocin_set_cd_hit_0.5\", \"fasta\"):\n",
    "    id_ = record.description.split(\"|\")[0]\n",
    "    if  id_ not in already_in_neg_dataset_ids and \\\n",
    "        len(record.seq) >= 30 and \\\n",
    "        len(record.seq) in pos_bacteriocin_distribution and \\\n",
    "        len(record.seq) not in keys_done:\n",
    "            if keys_track[len(record.seq)] + 1 < pos_bacteriocin_distribution[len(record.seq)]:\n",
    "                keys_track[len(record.seq)] = keys_track[len(record.seq)] + 1\n",
    "                flag = 1\n",
    "            elif keys_track[len(record.seq)] + 1 == pos_bacteriocin_distribution[len(record.seq)]:\n",
    "                keys_track[len(record.seq)] = keys_track[len(record.seq)] + 1\n",
    "                keys_done.append(len(record.seq))\n",
    "                flag = 1\n",
    "    if flag == 1:\n",
    "        print len(record.seq)\n",
    "        print keys_track[len(record.seq)]\n",
    "        print\n",
    "        new_in_neg_dataset_ids.append(id_)\n",
    "        out_handle.write('>%s\\n%s\\n' % (record.description, record.seq))\n",
    "        \n",
    "    flag = 0\n",
    "    if sum(keys_track.values()) == 346:\n",
    "        print \"Breaking - got the sequences\"\n",
    "        break\n",
    "        #break\n",
    "\n",
    "for record in SeqIO.parse(\"redundant_negative_bacteriocin_set_cd_hit_0.5\", \"fasta\"):\n",
    "    id_ = record.description.split(\"|\")[0]\n",
    "    \n",
    "    if  id_ not in already_in_neg_dataset_ids and \\\n",
    "        id_ not in new_in_neg_dataset_ids and \\\n",
    "        len(record.seq) >= 30 and \\\n",
    "        len(record.seq) <= 70:\n",
    "            #print \"hello\"\n",
    "            keys_track[len(record.seq)] = keys_track[len(record.seq)] + 1\n",
    "            out_handle.write('>%s\\n%s\\n' % (record.description, record.seq))\n",
    "        \n",
    "    if sum(keys_track.values()) == 346:\n",
    "        print \"Second Breaking - got the sequences\"\n",
    "        break\n",
    "\n",
    "out_handle.close()\n",
    "keylist = keys_track.keys()\n",
    "keylist.sort()\n",
    "for key in keylist:\n",
    "    print \"%s: %s\" % (key, keys_track[key]),"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "43 : 3 -> 10\n",
      "30 : 13 -> 3\n",
      "31 : 6 -> 2\n",
      "34 : 7 -> 1\n",
      "35 : 12 -> 2\n",
      "36 : 1 -> 2\n",
      "37 : 4 -> 2\n",
      "38 : 6 -> 4\n",
      "39 : 2 -> 3\n",
      "40 : 7 -> 3\n",
      "42 : 5 -> 3\n",
      "44 : 2 -> 11\n",
      "45 : 1 -> 3\n",
      "47 : 4 -> 10\n",
      "48 : 2 -> 11\n",
      "50 : 4 -> 6\n",
      "51 : 4 -> 11\n",
      "53 : 5 -> 7\n",
      "54 : 1 -> 2\n",
      "55 : 5 -> 12\n",
      "56 : 3 -> 12\n",
      "57 : 5 -> 13\n",
      "58 : 6 -> 8\n",
      "59 : 12 -> 7\n",
      "60 : 11 -> 5\n",
      "61 : 13 -> 12\n",
      "62 : 6 -> 12\n",
      "64 : 6 -> 11\n",
      "65 : 15 -> 7\n",
      "66 : 11 -> 5\n",
      "67 : 14 -> 11\n",
      "68 : 14 -> 8\n",
      "69 : 20 -> 9\n",
      "70 : 10 -> 3\n"
     ]
    }
   ],
   "source": [
    "diffkeys = [k for k in keys_track if keys_track[k] != pos_bacteriocin_distribution[k]]\n",
    "for k in diffkeys:\n",
    "    print k, ':', keys_track[k], '->', pos_bacteriocin_distribution[k]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "346\n"
     ]
    }
   ],
   "source": [
    "new_second_neg_dataset_ids = []\n",
    "for record in SeqIO.parse(\"extra_second_negative_bacteriocin_proper_distribution\", \"fasta\"):\n",
    "    id_ = record.description.split(\"|\")[0]\n",
    "    new_second_neg_dataset_ids.append(id_)\n",
    "    \n",
    "print len(set(new_second_neg_dataset_ids))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "128\n",
      "1\n",
      "\n",
      "131\n",
      "1\n",
      "\n",
      "120\n",
      "1\n",
      "\n",
      "79\n",
      "1\n",
      "\n",
      "80\n",
      "1\n",
      "\n",
      "80\n",
      "2\n",
      "\n",
      "86\n",
      "1\n",
      "\n",
      "95\n",
      "1\n",
      "\n",
      "83\n",
      "1\n",
      "\n",
      "78\n",
      "1\n",
      "\n",
      "92\n",
      "1\n",
      "\n",
      "91\n",
      "1\n",
      "\n",
      "110\n",
      "1\n",
      "\n",
      "90\n",
      "1\n",
      "\n",
      "90\n",
      "2\n",
      "\n",
      "91\n",
      "2\n",
      "\n",
      "90\n",
      "3\n",
      "\n",
      "99\n",
      "1\n",
      "\n",
      "91\n",
      "3\n",
      "\n",
      "89\n",
      "1\n",
      "\n",
      "89\n",
      "2\n",
      "\n",
      "91\n",
      "4\n",
      "\n",
      "88\n",
      "1\n",
      "\n",
      "255\n",
      "1\n",
      "\n",
      "73\n",
      "1\n",
      "\n",
      "207\n",
      "1\n",
      "\n",
      "120\n",
      "2\n",
      "\n",
      "225\n",
      "1\n",
      "\n",
      "85\n",
      "1\n",
      "\n",
      "85\n",
      "2\n",
      "\n",
      "105\n",
      "1\n",
      "\n",
      "78\n",
      "2\n",
      "\n",
      "96\n",
      "1\n",
      "\n",
      "96\n",
      "2\n",
      "\n",
      "87\n",
      "1\n",
      "\n",
      "97\n",
      "1\n",
      "\n",
      "101\n",
      "1\n",
      "\n",
      "97\n",
      "2\n",
      "\n",
      "103\n",
      "1\n",
      "\n",
      "102\n",
      "1\n",
      "\n",
      "102\n",
      "2\n",
      "\n",
      "103\n",
      "2\n",
      "\n",
      "102\n",
      "3\n",
      "\n",
      "84\n",
      "1\n",
      "\n",
      "81\n",
      "1\n",
      "\n",
      "81\n",
      "2\n",
      "\n",
      "84\n",
      "2\n",
      "\n",
      "77\n",
      "1\n",
      "\n",
      "77\n",
      "2\n",
      "\n",
      "81\n",
      "3\n",
      "\n",
      "84\n",
      "3\n",
      "\n",
      "77\n",
      "3\n",
      "\n",
      "84\n",
      "4\n",
      "\n",
      "75\n",
      "1\n",
      "\n",
      "75\n",
      "2\n",
      "\n",
      "75\n",
      "3\n",
      "\n",
      "75\n",
      "4\n",
      "\n",
      "75\n",
      "5\n",
      "\n",
      "74\n",
      "1\n",
      "\n",
      "74\n",
      "2\n",
      "\n",
      "74\n",
      "3\n",
      "\n",
      "74\n",
      "4\n",
      "\n",
      "74\n",
      "5\n",
      "\n",
      "74\n",
      "6\n",
      "\n",
      "72\n",
      "1\n",
      "\n",
      "72\n",
      "2\n",
      "\n",
      "72\n",
      "3\n",
      "\n",
      "74\n",
      "7\n",
      "\n",
      "72\n",
      "4\n",
      "\n",
      "71\n",
      "1\n",
      "\n",
      "76\n",
      "1\n",
      "\n",
      "72\n",
      "5\n",
      "\n",
      "76\n",
      "2\n",
      "\n",
      "76\n",
      "3\n",
      "\n",
      "76\n",
      "4\n",
      "\n",
      "72\n",
      "6\n",
      "\n",
      "76\n",
      "5\n",
      "\n",
      "71\n",
      "2\n",
      "\n",
      "71\n",
      "3\n",
      "\n",
      "76\n",
      "6\n",
      "\n",
      "76\n",
      "7\n",
      "\n",
      "76\n",
      "8\n",
      "\n",
      "71\n",
      "4\n",
      "\n",
      "71\n",
      "5\n",
      "\n",
      "71\n",
      "6\n",
      "\n",
      "71\n",
      "7\n",
      "\n",
      "65\n",
      "1\n",
      "\n",
      "61\n",
      "1\n",
      "\n",
      "59\n",
      "1\n",
      "\n",
      "66\n",
      "1\n",
      "\n",
      "69\n",
      "1\n",
      "\n",
      "68\n",
      "1\n",
      "\n",
      "40\n",
      "1\n",
      "\n",
      "67\n",
      "1\n",
      "\n",
      "70\n",
      "1\n",
      "\n",
      "59\n",
      "2\n",
      "\n",
      "68\n",
      "2\n",
      "\n",
      "68\n",
      "3\n",
      "\n",
      "65\n",
      "2\n",
      "\n",
      "59\n",
      "3\n",
      "\n",
      "70\n",
      "2\n",
      "\n",
      "69\n",
      "2\n",
      "\n",
      "70\n",
      "3\n",
      "\n",
      "61\n",
      "2\n",
      "\n",
      "37\n",
      "1\n",
      "\n",
      "35\n",
      "1\n",
      "\n",
      "35\n",
      "2\n",
      "\n",
      "63\n",
      "1\n",
      "\n",
      "71\n",
      "8\n",
      "\n",
      "67\n",
      "2\n",
      "\n",
      "31\n",
      "1\n",
      "\n",
      "34\n",
      "1\n",
      "\n",
      "68\n",
      "4\n",
      "\n",
      "69\n",
      "3\n",
      "\n",
      "68\n",
      "5\n",
      "\n",
      "69\n",
      "4\n",
      "\n",
      "Second Breaking - got the sequences\n",
      "31: 1 34: 1 35: 4 37: 1 40: 1 59: 3 61: 2 63: 1 65: 2 66: 1 67: 2 68: 5 69: 4 70: 6 71: 8 72: 18 73: 23 74: 23 75: 26 76: 16 77: 29 78: 43 79: 52 80: 30 81: 3 83: 1 84: 4 85: 2 86: 1 87: 1 88: 1 89: 2 90: 3 91: 4 92: 1 95: 1 96: 2 97: 2 99: 1 101: 1 102: 3 103: 2 105: 1 110: 1 120: 2 128: 1 131: 1 207: 1 225: 1 255: 1\n"
     ]
    }
   ],
   "source": [
    "\"\"\"\n",
    "Creating a third negative bacteriocin dataset with almost same distribution\n",
    "\"\"\"\n",
    "flag = 0\n",
    "count = 0\n",
    "\n",
    "keys_track = defaultdict(int)\n",
    "keys_done = []\n",
    "keys_description_done = []\n",
    "new_in_neg_dataset_ids = []\n",
    "out_handle = open('extra_third_negative_bacteriocin_proper_distribution', 'w')\n",
    "for record in SeqIO.parse(\"redundant_negative_bacteriocin_set_cd_hit_0.5\", \"fasta\"):\n",
    "    id_ = record.description.split(\"|\")[0]\n",
    "    if  id_ not in already_in_neg_dataset_ids and \\\n",
    "        id_ not in new_second_neg_dataset_ids and \\\n",
    "        len(record.seq) >= 30 and \\\n",
    "        len(record.seq) in pos_bacteriocin_distribution and \\\n",
    "        len(record.seq) not in keys_done:\n",
    "            if keys_track[len(record.seq)] + 1 < pos_bacteriocin_distribution[len(record.seq)]:\n",
    "                keys_track[len(record.seq)] = keys_track[len(record.seq)] + 1\n",
    "                flag = 1\n",
    "            elif keys_track[len(record.seq)] + 1 == pos_bacteriocin_distribution[len(record.seq)]:\n",
    "                keys_track[len(record.seq)] = keys_track[len(record.seq)] + 1\n",
    "                keys_done.append(len(record.seq))\n",
    "                flag = 1\n",
    "    if flag == 1:\n",
    "        print len(record.seq)\n",
    "        print keys_track[len(record.seq)]\n",
    "        print\n",
    "        new_in_neg_dataset_ids.append(id_)\n",
    "        out_handle.write('>%s\\n%s\\n' % (record.description, record.seq))\n",
    "        \n",
    "    flag = 0\n",
    "    if sum(keys_track.values()) == 346:\n",
    "        print \"Breaking - got the sequences\"\n",
    "        break\n",
    "        #break\n",
    "\n",
    "for record in SeqIO.parse(\"redundant_negative_bacteriocin_set_cd_hit_0.5\", \"fasta\"):\n",
    "    id_ = record.description.split(\"|\")[0]\n",
    "    \n",
    "    if  id_ not in already_in_neg_dataset_ids and \\\n",
    "        id_ not in new_second_neg_dataset_ids and \\\n",
    "        id_ not in new_in_neg_dataset_ids and \\\n",
    "        len(record.seq) >= 30 and \\\n",
    "        len(record.seq) <= 80:\n",
    "            #print \"hello\"\n",
    "            keys_track[len(record.seq)] = keys_track[len(record.seq)] + 1\n",
    "            out_handle.write('>%s\\n%s\\n' % (record.description, record.seq))\n",
    "        \n",
    "    if sum(keys_track.values()) == 346:\n",
    "        print \"Second Breaking - got the sequences\"\n",
    "        break\n",
    "\n",
    "out_handle.close()\n",
    "keylist = keys_track.keys()\n",
    "keylist.sort()\n",
    "for key in keylist:\n",
    "    print \"%s: %s\" % (key, keys_track[key]),"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "346"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sum(keys_track.values())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31 : 1 -> 2\n",
      "35 : 4 -> 2\n",
      "37 : 1 -> 2\n",
      "40 : 1 -> 3\n",
      "71 : 8 -> 10\n",
      "59 : 3 -> 7\n",
      "61 : 2 -> 12\n",
      "63 : 1 -> 8\n",
      "65 : 2 -> 7\n",
      "66 : 1 -> 5\n",
      "67 : 2 -> 11\n",
      "68 : 5 -> 8\n",
      "69 : 4 -> 9\n",
      "70 : 6 -> 3\n",
      "72 : 18 -> 6\n",
      "73 : 23 -> 1\n",
      "74 : 23 -> 7\n",
      "75 : 26 -> 5\n",
      "76 : 16 -> 8\n",
      "77 : 29 -> 3\n",
      "78 : 43 -> 2\n",
      "80 : 30 -> 2\n",
      "79 : 52 -> 1\n"
     ]
    }
   ],
   "source": [
    "diffkeys = [k for k in keys_track if keys_track[k] != pos_bacteriocin_distribution[k]]\n",
    "for k in diffkeys:\n",
    "    print k, ':', keys_track[k], '->', pos_bacteriocin_distribution[k]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done\n",
      "692\n"
     ]
    }
   ],
   "source": [
    "\"\"\"\n",
    "second negative set\n",
    "\"\"\"\n",
    "out_handle = open('extra_second_negative_bacteriocin_set_less_than_30', 'w')\n",
    "count = 0\n",
    "for record in SeqIO.parse(\"redundant_negative_bacteriocin_set_cd_hit_0.5\", \"fasta\"):\n",
    "    id_ = record.description.split(\"|\")[0]\n",
    "    if count == 346:\n",
    "        break\n",
    "    if len(record.seq) >= 30 and len(record.seq) <= 255 and id_ not in already_in_neg_dataset_ids:\n",
    "        out_handle.write('>%s\\n%s\\n' % (record.description, record.seq))\n",
    "        count = count + 1\n",
    "        already_in_neg_dataset_ids.append(id_)\n",
    "\n",
    "print \"Done\"\n",
    "print len(already_in_neg_dataset_ids)\n",
    "out_handle.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done\n",
      "1038\n"
     ]
    }
   ],
   "source": [
    "\"\"\"\n",
    "third negative set\n",
    "\"\"\"\n",
    "out_handle = open('extra_third_negative_bacteriocin_set_less_than_30', 'w')\n",
    "count = 0\n",
    "for record in SeqIO.parse(\"redundant_negative_bacteriocin_set_cd_hit_0.5\", \"fasta\"):\n",
    "    id_ = record.description.split(\"|\")[0]\n",
    "    if count == 346:\n",
    "        break\n",
    "    if len(record.seq) >= 30 and len(record.seq) <= 255 and id_ not in already_in_neg_dataset_ids:\n",
    "        out_handle.write('>%s\\n%s\\n' % (record.description, record.seq))\n",
    "        count = count + 1\n",
    "        already_in_neg_dataset_ids.append(id_)\n",
    "\n",
    "print \"Done\"\n",
    "print len(already_in_neg_dataset_ids)\n",
    "out_handle.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAnYAAAG3CAYAAAA5CBh1AAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAIABJREFUeJzt3X28XVV95/HPVxAxxWgMQYslBKQVFVu1sS1tHR+wo6CW\nqiioHWVkjNRxfKxWESpoW5FRGQq1QG3FOuNjdIoREQULYikqqEWKgKNGoII8GEAMGoHf/LF39HC4\nyb0X7r3nnHU/79drv+7da62zz+9k5cA3+zFVhSRJkibfvUZdgCRJkuaGwU6SJKkRBjtJkqRGGOwk\nSZIaYbCTJElqhMFOkiSpEQY7SZKkRhjsJEmSGmGwkyRJasS2oy5gVHbcccdatWrVqMuQJEma1oUX\nXnh9Va2YbtyiDXarVq3iggsuGHUZkiRJ00ryvZmM81CsJElSIwx2kiRJjTDYSZIkNcJgJ0mS1AiD\nnSRJUiMMdpIkSY0w2EmSJDXCYCdJktQIg50kSVIjDHaSJEmNMNhJkiQ1wmAnSZLUCIOdJElSIwx2\nkiRJjTDYSZIkNcJgJ0mS1IhtR12AtNBWvfG0eX+P9Uc/fd7fQ5KkYe6xkyRJaoTBTpIkqREGO0mS\npEYY7CRJkhphsJMkSWqEwU6SJKkRCx7skuyR5KQkFyW5PcnZQ/1PTFJbWM4YGHfwFsYcutCfSZIk\naRyM4j52jwT2A84H7j1F/1eBvYfaVgIfAU6fYvyTgVsH1r8zBzVKkiRNnFEEu3VVdSpAkrXAjoOd\nVXUzXej7uSSPB+4APjrF9r5SVbfMU62SJEkTY8EPxVbVHXfjZc8Hzqmq7891PZIkSa0Y+4snkvwa\n8BjgQ1sY8u0ktyW5LMnLFrA0SZKksTIJz4o9CPgZ8PGh9quBI4AvA9v0405MsqSqjp1qQ0nWAGsA\nVq5cOW8FS5IkjcKkBLvPVtUPBxur6gzgjIGm05NsD7w5yXFTHfKtqpOBkwFWr15d81izJEnSghvr\nQ7FJfgN4OFs+DDtsLbAc2HXeipIkSRpTYx3s6PbW3QqcOsPx7oWTJEmL1iQEu3WzuJ3JAcANwPfm\nryRJkqTxtODn2CVZQneDYoCHAEuTHNCvf7qqNvbjfgdYBbxmC9tZS3e/u4vpPseB/fLKu3lLFUmS\npIk2iosndgI+NtS2eX03YH3/+0HATUz9tAmAy4GXArsAAS4BXlRVH5jLYiVJkibFgge7qlpPF8Sm\nG/dq4NVb6T8MOGzuKpMkSZps436OnSRJkmbIYCdJktQIg50kSVIjDHaSJEmNMNhJkiQ1wmAnSZLU\nCIOdJElSIwx2kiRJjTDYSZIkNcJgJ0mS1AiDnSRJUiMMdpIkSY0w2EmSJDXCYCdJktQIg50kSVIj\nDHaSJEmNMNhJkiQ1wmAnSZLUCIOdJElSIwx2kiRJjTDYSZIkNcJgJ0mS1AiDnSRJUiMMdpIkSY0w\n2EmSJDXCYCdJktQIg50kSVIjDHaSJEmNMNhJkiQ1wmAnSZLUCIOdJElSIwx2kiRJjTDYSZIkNcJg\nJ0mS1AiDnSRJUiMMdpIkSY0w2EmSJDViwYNdkj2SnJTkoiS3Jzl7ijHrk9TQcs0U4x6R5KwkG5N8\nP8lbk2yzIB9EkiRpzGw7gvd8JLAfcD5w762M+yBw/MD6psHOJMuAM4FLgP2BhwLvogurh89hvZIk\nSRNhFMFuXVWdCpBkLbDjFsZdXVXnb2U7hwL3BZ5dVTcDn0uyFDgyyTF9myRJ0qKx4Idiq+qOOdrU\nvsAZQwHuw3Rh7wlz9B6SJEkTY5wvnjgkyaYkNyVZm2TXof49gUsHG6rqCmBj3ydJkrSojOJQ7Eyc\nSncO3lXAw4G3AOcmeVRV3dSPWQbcOMVrN/R9d5FkDbAGYOXKlXNdsyRJ0kiN5R67qnpVVX2oqs6t\nqpOBpwI7Awffw+2eXFWrq2r1ihUr5qJUSZKksTGWwW5YVV0MXAY8dqB5A3D/KYYv6/skSZIWlYkI\ndr0aWr+UoXPpkuwCLGHo3DtJkqTFYCKCXZK96ELchQPNpwNPTXK/gbYDgVuBcxawPEmSpLGw4BdP\nJFlCd4NigIcAS5Mc0K9/GngS8AJgHXAN3cUThwNXAKcMbOpE4JXAJ5K8A9gdOBJ4t/ewkyRJi9Eo\nrordCfjYUNvm9d2AK4EH0z114gHADcBngMMGA1tVbUiyD3ACXQi8ETiWLtxJkiQtOgse7KpqPZBp\nhu0zw21dAjz5ntYkSZLUgok4x06SJEnTM9hJkiQ1wmAnSZLUCIOdJElSIwx2kiRJjTDYSZIkNcJg\nJ0mS1AiDnSRJUiMMdpIkSY0w2EmSJDXCYCdJktQIg50kSVIjDHaSJEmNMNhJkiQ1wmAnSZLUCIOd\nJElSIwx2kiRJjTDYSZIkNcJgJ0mS1AiDnSRJUiMMdpIkSY0w2EmSJDXCYCdJktQIg50kSVIjDHaS\nJEmNMNhJkiQ1wmAnSZLUCIOdJElSIwx2kiRJjTDYSZIkNcJgJ0mS1AiDnSRJUiMMdpIkSY0w2EmS\nJDXCYCdJktQIg50kSVIjDHaSJEmNWPBgl2SPJCcluSjJ7UnOHurfOcm7klyc5MdJrkzy/iQ7D407\nOElNsRy6oB9IkiRpTGw7gvd8JLAfcD5w7yn6HwvsD7wX+BLwIOBI4Lwke1XVLUPjnwzcOrD+nbku\nWJIkaRKMItitq6pTAZKsBXYc6v8isGdV3ba5IclXgcuA5wDvHxr/lSnCniRJ0qKz4MGuqu6Ypv/G\nKdouT7IR2HmKl0iSJIkJuXgiya8DS4DLp+j+dpLbklyW5GULXJokSdLYGMWh2FlJci/gOOBbwCcH\nuq4GjgC+DGwDHAScmGRJVR274IVKkiSN2NgHO+DtwN7AE6rqZ5sbq+oM4IyBcacn2R54c5Ljpjrk\nm2QNsAZg5cqV81u1JEnSAhvrQ7FJXg68HnhxVX1pBi9ZCywHdp2qs6pOrqrVVbV6xYoVc1ipJEnS\n6I1tsEvyHOB44A1V9ZEZvqzmsSRJkqSxNpbBLskTgf8DHF9V75zFSw8AbgC+Nx91SZIkjbMFP8cu\nyRK6GxQDPARYmuSAfv3TdIdR/wm4FPhIkt8ZePl1VfXtfjtr6W5yfDHd5ziwX1453S1VJEmSWjSK\niyd2Aj421LZ5fTfgt4H7A78BnDc07v3Awf3vlwMvBXYBAlwCvKiqPjD3JUuSJI2/UdygeD1dENuS\nU/pluu0cBhw2J0VJkiQ1YCzPsZMkSdLsGewkSZIaYbCTJElqhMFOkiSpEQY7SZKkRhjsJEmSGmGw\nkyRJasQoblAsTWnVG08bdQmSJE0099hJkiQ1wmAnSZLUCIOdJElSIwx2kiRJjTDYSZIkNcJgJ0mS\n1AiDnSRJUiMMdpIkSY0w2EmSJDXCYCdJktQIg50kSVIjDHaSJEmNMNhJkiQ1wmAnSZLUCIOdJElS\nIwx2kiRJjTDYSZIkNcJgJ0mS1AiDnSRJUiMMdpIkSY0w2EmSJDXCYCdJktQIg50kSVIjDHaSJEmN\nmHGwS/KiJMu30PfAJC+au7IkSZI0W7PZY/c+4KFb6Nut75ckSdKIzCbYZSt9y4Gb72EtkiRJuge2\n3Vpnkv2B/Qeajkhy3dCw7YHHA1+Z49okSZI0C1sNdsBOwKMG1h8KPHhozCbgs8BfzGFdkiRJmqWt\nHoqtqr+rqsdV1eOAc4Bnb14fWH6vqg6pqu/O5A2T7JHkpCQXJbk9ydlTjEmSw5JcmeTWJF9I8ugp\nxj0iyVlJNib5fpK3Jtlmhp9dkiSpKTM+x66qnlRVl87Bez4S2A+4DLh8C2PeCBwBvAN4JnALcGaS\nn+8tTLIMOBMousPFbwVeBxw1BzVKkiRNnOkOxd5Jkp2BZwC/Qndu3aCqqj+bwWbWVdWp/fbWAjsO\nvcf2dMHu7VV1Qt/2r8B64BXA4f3QQ4H70u1FvBn4XJKlwJFJjunbJEmSFo0ZB7skzwI+BGwDXEt3\nbt2gAqYNdlV1xzRDfhdYCnx04DU/TrIO2JdfBLt9gTOGAtyH6fbyPQFYN10tkiRJLZnN7U7+iu4i\niQdV1UOqarehZfc5qmlP4HbgW0Pt3+z7Bsfd6dBwVV0BbBwaJ0mStCjMJtjtAvx1Vf1wvorpLQNu\nqarbh9o3AEuSbDcw7sYpXr+h75MkSVpUZhPszgMeNl+FLIQka5JckOSC664bvh2fJEnSZJtNsHst\nsCbJi5PsnGTJ8DJHNW0AdpjitiXLgI1VtWlg3P2neP2yvu8uqurkqlpdVatXrFgxR+VKkiSNh9lc\nFXtR//N9dBdKTGUu7iF3ab+dPehuibLZ8Dl1lzJ0Ll2SXYAlQ+MkSZIWhdkEu5ew5UA3l86je+7s\nc+mfZtHvDXwmcPLAuNOB1ye5X1X9qG87ELiV7mbKkiRJi8qMg11VnTIXb9iHtP361YcAS5Mc0K9/\nuqo2Jjma7rm0G+j2vr2W7rDx8QObOhF4JfCJJO8AdgeOBN7tPewkSdJiNKsbFM+RnYCPDbVtXt+N\n7kbER9MFuTcBy4ELgD+oqh9sfkFVbUiyD3AC3T3rbgSOpQt3kiRJi85sblB8HdMciq2qnabbTlWt\nBzLNmAL+sl+2Nu4S4MnTvackSdJiMJs9dn/DXYPdMmAfuidF/MNcFSVJkqTZm805dkdO1Z4kdI//\n+tkc1SRJkqS7YTb3sZtSf9j0vcAr7nk5kiRJurvucbDr7Q5sN+0oSZIkzZvZXDzx8imatwMeDryQ\nu17pKkmSpAU0m4snTpii7afAVcB7gKPmpCJJkiTdLbO5eGKuDttKkiRpHhjWJEmSGjGrYJdk9yR/\nm+QbSf6j//meJLvPV4GSJEmamdlcPPGbwD8DPwE+BfwAeBDwHOCFSZ5UVV+dlyolSZI0rdlcPPFO\n4GvAvlW1cXNjkiXAp/t+H+8lSZI0IrM5FPtbwDGDoQ6gX38n8NtzWZgkSZJmZzbB7lZg+Rb6Hkh3\niFaSJEkjMptgdxpwdJLfH2zs198OrJvLwiRJkjQ7sznH7rXAqcA5Sa4FrgV2oruA4jzgdXNfniRJ\nkmZqNjcovgH4/SRPAx4H/DJwNfClqvrsPNUnSZKkGdrqodgkv5zk40meurmtqj5TVW+rqpdX1du6\nYfl4kp3mvVpJkiRt0XTn2P0psDuwtT1ynwV2w0OxkiRJIzVdsHsGcGJV1ZYG9H0nAfvPZWGSJEma\nnemC3a7AJTPYzjeBVfe4GkmSJN1t0wW7W4GlM9jODv1YSZIkjch0we6rwB/OYDv792MlSZI0ItMF\nu/cAhyR58ZYGJHkR8F+BE+ayMEmSJM3OVu9jV1UfT3Ic8L4krwA+A1wBFLASeCqwGji2qv7vfBcr\nSZKkLZv2BsVV9bokZwOvprv9yX36rp8C/wLsX1WfmrcKJUmSNCMzevJEVa0D1iXZFljeN99QVbfN\nW2WSJEmaldk8K5Y+yP1gnmqRJEnSPTDdxROSJEmaEAY7SZKkRhjsJEmSGmGwkyRJaoTBTpIkqREG\nO0mSpEYY7CRJkhphsJMkSWqEwU6SJKkRBjtJkqRGjGWwS3J2ktrCsnc/Zv0UfdeMunZJkqRRmdWz\nYhfQy4GlQ21vBR4DfGWg7YPA8QPrm+a5LkmSpLE1lsGuqi4ZXE+yHbAa+EhV3TbQdXVVnb+gxUmS\nJI2psTwUO4WnAcuAD426EEmSpHE1KcHuIOAq4Nyh9kOSbEpyU5K1SXYdQW2SJEljYSwPxQ5KsgT4\nQ+CkqqqBrlOB8+kC38OBtwDnJnlUVd208JVKkiSN1tgHO+CZwC8xdBi2ql41sHpukvOArwMHA8dN\ntaEka4A1ACtXrpyPWiVJkkZmEg7FHgT8v6q6YGuDqupi4DLgsVsZc3JVra6q1StWrJjjMiVJkkZr\nrINdkvsD+zLziyZq+iGSJEltGutgBzwLuA8zCHZJ9gL2BC6c76IkSZLG0bifY3cQ8G9V9c3BxiRP\nB14ArAOuobt44nDgCuCUBa5RkiRpLIxtsEuyI7APcMQU3VcCD6Z76sQDgBuAzwCHVdXNC1akJEnS\nGBnbYFdV1wP33kLfRXShT5IkSb1xP8dOkiRJM2SwkyRJaoTBTpIkqREGO0mSpEYY7CRJkhphsJMk\nSWqEwU6SJKkRBjtJkqRGGOwkSZIaYbCTJElqhMFOkiSpEQY7SZKkRhjsJEmSGmGwkyRJaoTBTpIk\nqREGO0mSpEYY7CRJkhphsJMkSWqEwU6SJKkRBjtJkqRGGOwkSZIaYbCTJElqhMFOkiSpEQY7SZKk\nRhjsJEmSGmGwkyRJaoTBTpIkqREGO0mSpEYY7CRJkhphsJMkSWqEwU6SJKkRBjtJkqRGGOwkSZIa\nYbCTJElqhMFOkiSpEQY7SZKkRhjsJEmSGjGWwS7JwUlqiuXQgTFJcliSK5PcmuQLSR49yrolSZJG\nadtRFzCNJwO3Dqx/Z+D3NwJHAK8HLgVeC5yZZK+qumbhSpQkSRoP4x7svlJVtww3JtmeLti9vapO\n6Nv+FVgPvAI4fCGLlCRJGgdjeSh2Bn4XWAp8dHNDVf0YWAfsO6qiJEmSRmncg923k9yW5LIkLxto\n3xO4HfjW0Phv9n2SJEmLzrgeir2a7vy5LwPbAAcBJyZZUlXHAsuAW6rq9qHXbQCWJNmuqjYtaMWS\nJEkjNpbBrqrOAM4YaDq9P6/uzUmOu7vbTbIGWAOwcuXKe1akJEnSmBn3Q7GD1gLLgV3p9sztkGSb\noTHLgI1b2ltXVSdX1eqqWr1ixYr5rVaSJGmBTVKwq4HfL6U7RLvH0Jg9+z5JkqRFZ5KC3QHADcD3\ngPOAm4Hnbu5MsgR4JnD6SKqTJEkasbE8xy7JWuB84GK6Gg/sl1dW1R3AT5IcDRyRZAO/uEHxvYDj\nR1O1JEnSaI1lsAMuB14K7AIEuAR4UVV9YGDM0XRB7k10595dAPxBVf1ggWuVJEkaC2MZ7KrqMOCw\nacYU8Jf9IkmStOhN0jl2kiRJ2gqDnSRJUiMMdpIkSY0w2EmSJDXCYCdJktQIg50kSVIjDHaSJEmN\nMNhJkiQ1wmAnSZLUCIOdJElSIwx2kiRJjTDYSZIkNcJgJ0mS1AiDnSRJUiMMdpIkSY0w2EmSJDXC\nYCdJktQIg50kSVIjDHaSJEmNMNhJkiQ1wmAnSZLUCIOdJElSIwx2kiRJjTDYSZIkNcJgJ0mS1AiD\nnSRJUiMMdpIkSY0w2EmSJDXCYCdJktQIg50kSVIjDHaSJEmNMNhJkiQ1wmAnSZLUCIOdJElSIwx2\nkiRJjTDYSZIkNcJgJ0mS1IixDHZJnpfktCRXJ7klyYVJnj80Zn2SGlquGVXNkiRJo7btqAvYgtcA\n3wVeBVwP7Ad8MMmOVXX8wLgPAoPrmxauREmSpPEyrsHumVV1/cD655PsDLyWOwe5q6vq/IUtTZIk\naTyN5aHYoVC32deAnRe6FkmSpEkxlsFuC/YGLh9qOyTJpiQ3JVmbZNdRFCZJkjQOxvVQ7J0k2Qf4\nI+AlA82nAucDVwEPB94CnJvkUVV108JXKUmSNFpjH+ySrKK7SOLUqjplc3tVvWpg2LlJzgO+DhwM\nHLeFba0B1gCsXLlyXuqVJEkalbE+FJvkgcDpwPeAF25tbFVdDFwGPHYrY06uqtVVtXrFihVzWqsk\nSdKojW2wS7IE+BSwHfCMqto4g5fV/FYlSZI0vsYy2CXZFvgY8KvA06rq2hm8Zi9gT+DCeS5PkiRp\nLI3rOXbvobsp8auA5UmWD/R9DXgK8AJgHXAN3cUThwNXAKcsaKWSJEljYlyD3X/uf051EcRuwJXA\ng+luVvwA4AbgM8BhVXXzglQoSZI0ZsYy2FXVqhkM22e+65AkSZokY3mOnSRJkmbPYCdJktQIg50k\nSVIjDHaSJEmNMNhJkiQ1wmAnSZLUCIOdJElSIwx2kiRJjTDYSZIkNWIsnzwhTbpVbzxt3t9j/dFP\nn/f3kCRNFvfYSZIkNcJgJ0mS1AiDnSRJUiMMdpIkSY0w2EmSJDXCYCdJktQIg50kSVIjDHaSJEmN\nMNhJkiQ1wmAnSZLUCIOdJElSIwx2kiRJjTDYSZIkNcJgJ0mS1AiDnSRJUiMMdpIkSY0w2EmSJDVi\n21EX0LJVbzxt3t9j/dFPn/f30HhaiL9f4N8xSZok7rGTJElqhMFOkiSpEQY7SZKkRniOnWZkoc7n\nkiRJd5977CRJkhphsJMkSWqEwU6SJKkRBjtJkqRGGOwkSZIaMdFXxSZ5BHA8sDdwI/Be4Kiqun2k\nhUkN8QkqkjQ5JjbYJVkGnAlcAuwPPBR4F91eyMNHWJokSdJITGywAw4F7gs8u6puBj6XZClwZJJj\n+jZJE8C9gpI0Nyb5HLt9gTOGAtyH6cLeE0ZTkiRJ0uhM8h67PYHPDzZU1RVJNvZ960ZSlSRpUWpl\nz/NCPWmolc8ybkcDJnmP3TK6CyaGbej7JEmSFpVU1ahruFuS/Ax4fVX9r6H2q4B/rKrDpnjNGmBN\nv/ow4LJ5L3Ry7AhcP+oidI84h21wHiefczj5xnEOd62qFdMNmuRDsRuA+0/Rvqzvu4uqOhk4eT6L\nmlRJLqiq1aOuQ3efc9gG53HyOYeTb5LncJIPxV5Kdy7dzyXZBVjS90mSJC0qkxzsTgeemuR+A20H\nArcC54ymJEmSpNGZ5GB3IvBT4BNJntKfP3ck8G7vYXe3eIh68jmHbXAeJ59zOPkmdg4n9uIJ+Pkj\nxU7gzo8UO9JHikmSpMVoooOdJEmSfmGSD8VqFpIcnKSmWA4dGJMkhyW5MsmtSb6Q5NGjrHsxS7JH\nkpOSXJTk9iRnTzFmRnOW5BFJzkqyMcn3k7w1yTYL8kEWsRnO4fopvpfXTDHOORyBJM9LclqSq5Pc\nkuTCJM8fGuP3cIzNcA6b+R5O8u1OdPc8me4Ck82+M/D7G4EjgNfTXVn8WuDMJHtV1V3+gmvePRLY\nDzgfuPcWxkw7Z0mWAWcClwD7Aw8F3kX3D7vD5/MDaEZzCPBB4PiB9U2Dnc7hSL0G+C7wKrr7mu0H\nfDDJjlW1ec78Ho63mcwhtPI9rCqXRbAABwMF7LCF/u2Bm4A/H2j7JeA64C9GXf9iXIB7Dfy+Fjj7\n7swZ8Ca6ezsuHWh7A7BxsM1l4eewb18PvHOa7TiHo5vDHado+yDw3f53v4djvkw3h/16M99DD8Vq\ns98FlgIf3dxQVT+me+buvqMqajGrqjumGTLTOdsXOKPufLX4h4H7Ak+Ym2o1lRnM4Uw5hyNSVVM9\nfeBrwM79734Px9wM5nCmJmIODXaLz7eT3JbksiQvG2jfE7gd+NbQ+G8ydCNojY2ZztmeDN20u6qu\noPtXpnM7Hg5JsinJTUnWJtl1qN85HC97A5f3v/s9nEyDc7hZE99Dz7FbPK6mOwfky8A2wEHAiUmW\nVNWxdI9iu6XuequYDcCSJNtV1SY0TmY6Z8vobgc0bEPfp9E6le4cvKuAhwNvAc5N8qiquqkf4xyO\niST7AH8EvKRv8ns4YaaYQ2joe2iwWySq6gzgjIGm05NsD7w5yXEjKkta9KrqVQOr5yY5D/g63Xmx\nfjfHSJJVdOdmnVpVp4y0GN0tW5rDlr6HHopd3NYCy4Fd6f7FscMUl20vAza6t24szXTONgD3n+L1\ny/o+jZGquhi4DHjsQLNzOGJJHkj3KMvvAS8c6PJ7OCG2Mod3McnfQ4Pd4jZ4d+pL6Q7R7jE05i7n\nFGhszHTOLmXo/I8kuwBLcG7H1fCd453DEUqyBPgUsB3wjKraONDt93ACTDOHWzKR30OD3eJ2AHAD\n3b9ezgNuBp67ubP/IjyT7l84Gj8znbPTgacmud9A24F09zM8ZwHq1Cwk2Yvufx4XDjQ7hyOSZFvg\nY8CvAk+rqmuHhvg9HHMzmMOpXjOx30PPsVskkqylOzH0Yrp5P7BfXtnfkuEnSY4GjkiygV/cZPNe\n3PmGjVog/f8c9utXHwIsTXJAv/7pqto4wzk7EXgl8Ikk7wB2B44E3j102b7m2HRzCDwJeAHdrTGu\noTtp+3DgCuCUgU05h6PzHro5fBWwPMnygb6vVdVM/9vpHI7OVucQeAotfQ9HfSM9l4VZgL+iO19g\nI92/Li4E/svQmABvprsq6FbgXOAxo659sS7AKrpDAVMtq2YzZ8AjgM/3Y64G3gZsM+rP2Poy3RwC\nvw6cRXcz25/R/U/lFGBn53A8Frob1/o9nOBlujls7XuYvlBJkiRNOM+xkyRJaoTBTpIkqREGO0mS\npEYY7CRJkhphsJMkSWqEwU6SJKkRBjtJ8yLJwUkuTPKjJBuSfC3Ju0dd17hLckqSC0ZcwxuSPHGK\n9kryihGUJGmGDHaS5lySNwHvBc4Ang28CDgV+MNR1qUZewPwxFEXIWn2fKSYpPnwCuCkqjpsoG1d\nkqNGVZAkLQbusZM0Hx5A91ieO6mhR90k2T7JMUmuTPLTJP+WZL+hMfdJckKSG5P8MMmxSV6TpAbG\nHNwfJtxh6LXrk7xzqG3/JBck+UmSa/r3v/dA/5FJrk/ymCTnJ9nYH0Z+/PDnSfLSJN/ot/WDJGuT\n3H+g//FJzum3cUOSvxt6gPiMJFmZ5MP959+Y5IwkDxvoX9V//uclOSnJTUmuSnJUknsNbeu5Sb6V\n5NYk/9x/zkpy8OY/M2A58Ja+vYYOy26T5K+SXJfk2iR/k+Q+s/1MkuaHwU7SfPgq8D+SvHjogdvD\n1gIH0z3L+JnAV4BPJnn0wJijgf9G90zGFwK7Aq+7O0UleR7wCeDLdIeFjwLWAG8fGroEeD9wEvAc\n4Kd0D/5eMrCtw/v+c4A/Av4EuAnYoe//PeBMuoB7APBqugeRv2+WNT8Q+CLwMOBQ4HnALwFnJrnv\n0PBjgFv69/vfwJ/3v2/e1mrgw3Tz8yzgk8BHhrbxrP5z/D2wd798daD/dcDOwB8D/xN4Gd3D1SWN\ng1E/rNbFxaW9he6h2t+he8j2HcC/A28Flg6M2afvf8LQa78AfKz/fTndw7b/bKD/XsCl9DsA+7aD\n+23tMLSt9cA7+98DfA9439CYl/TvsbxfP7Lf1pMHxjy6b3tav/4AYCPw7q38GZwL/PNQ25P77ey1\nldedAlzp2fDpAAAELklEQVQwsP424AbggQNty+jC13/v11f12/3HoW19HfjwwPrHgIuhe0543/aG\n/rUHD7RdDxw5RW0FfGGo7Z+A80f9d87FxaVb3GMnac5V1UXAw+n2ir2HLlQdAVwwcLj0KXR7s/4l\nybabF+AsYHU/5lHA9nQXXmze9h2D67Pwa8BK4KND7/f5/j32Ghi7CTh7YP2S/uev9D/3Bu7LFva+\n9Xv29p7ivb4I/Az4zVnU/RTgc8DNA9v5EXAhv/hz2uyzQ+uXDNQM8DhgXVUNHhL/5Cxqmcl7SBoh\nL56QNC+q6qfAun4hySF0V8oeAhwH7Ag8mC7oDLu9//ng/ue1Q/3D6zOxY//z01vo32Xg9x/1ARKA\nqtqUBLoACN2eRICrt7CtZcA2dKH2PdO813R2BH4HOHCKvrOG1m8cWt/EL2qG7s/zuqExw+vTme49\nJI2QwU7Sgqiqv09yDLBn3/RD4D/ozk/bks0XYOzUj2dgfdBP+p/bDbUvG/h98+vXAF+b4r2+u5U6\nht3Q//xlusOWw26kO2x5JFMHye/P4r1+SLdX7W1T9P1oFtuB7s9zxVDb8LqkCWawkzTnkuxUVdcO\nta0A7g/8oG86i+5E/Fuq6tItbOobdKFtf7rz6uiv8tx/aNxV/c+HA//Sj/ttYOnAmMvoguSqqvq7\nu/GxBv0r3Xl5Lwb+dLizqn6c5HzgYVX11nv4XmfRXTDx71V16z3c1leAZyY5bOBw7FT3FnQvnDSh\nDHaS5sM3kpxKdz7WtXRXsv4p3QUH7+/HfI7uBsafS/IOugssltJdqLB9Vb2pqm5IcjJwVJLb+jEv\npb/ydMCX6ULbXyc5Angg3UUBN28eUFV3JHkd8IEkS4HT6QLM7nR7DQ+oqo0z+XBVdWOStwF/mWQ7\nur1y9wGeDhxVVf/Rv/9ZSe6gu/r3R3Tn+D0deHNVXT6T9wLeTXcF6ueTHN9/zgcBTwC+WFUfmuF2\nAN4BfAn4cJL30QXhl/Z9dwyMuxR4epLP0F1le1lVzXbvoKQR8OIJSfPhrXRXav41Xbh7G10o+62q\n+i78/J52zwb+ge5WIGfQ3T5kb7qLDDZ7Qz/mz4EP0R3GvNOjyapqE91tOjaHqNfR3X5kw9C4j9Dt\n7Xs03RWinwBeTnc7j02z+YBV9fb+PZ5CdzHHSXRXy/6o7/8i8J/oDnV+gO5cwzcAV/KLvZYzeZ/r\n6c6xuxQ4lu7P8xi6vZ8XzbLmC4Dn01288U90t3L5k7775oGhrwd+DJxGt5dvNhd7SBqh3PniKEka\nf/3zSo+vqoy6lkmX5I/pgufum0O3pMnloVhJWkSS/C3dYfANwGOBw4HTDHVSGwx2krS4LKe7Bcty\nuqt7P0J3iFhSAzwUK0mS1AgvnpAkSWqEwU6SJKkRBjtJkqRGGOwkSZIaYbCTJElqhMFOkiSpEf8f\n+7W1VoGX0XsAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x7f3c7b794250>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.tick_params(\n",
    "    axis='y')\n",
    "plt.tick_params(\n",
    "    axis='x')\n",
    "plt.hist(sizes, bins=20)\n",
    "plt.xlabel(\"Sequence length\", fontsize = 15)\n",
    "plt.ylabel(\"Count\", fontsize = 15)\n",
    "plt.tick_params(axis='both', which='major', labelsize=15)\n",
    "#plt.tick_params(axis='both', which='major', labelsize=15)\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "346 30 255\n"
     ]
    }
   ],
   "source": [
    "sizes = [len(rec.seq) for rec in SeqIO.parse(\"extra_second_negative_bacteriocin_proper_distribution\", \"fasta\")]\n",
    "print (len(sizes), min(sizes), max(sizes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAmgAAAHICAYAAAD6GxY6AAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAIABJREFUeJzt3XmYZVV97vHvKw0yqiidDgLSKjigUdRWDE4kEAcGcSCI\nQW+TqyFGY/Bq1I6JEZOYtInXxDGKI4lIgoqK4jUigsaoaEMQGUJQaWRooBEQnEDgd//Yq+xDpaqr\ngK46q7q/n+c5T+2zx98+u07X22vtc1aqCkmSJPXjbuMuQJIkSbdnQJMkSeqMAU2SJKkzBjRJkqTO\nGNAkSZI6Y0CTJEnqjAFN0pxK8uMkD9jA+9wnyWUbcp8bSpLXJXn/XdzH4Um+sKFqkrTwGNCkeZDk\nI0muTHJDkv9O8uJp1vvzJJVkv0nzH53kKy3sXJXkqGm2X9q2/9wUxz96g53QNJKcPvncqmrbqvr+\nXB97Q0jy4SR/dVf2UVV/XVVTXt87sI/jquqpd2UfkhY2A5o0P1YCD6iqewDPBP4qyWNGV0jyQOC3\ngTWT5u8AfB54L3AfYDdgptaVvZLsvYFq1ywlWTTuGiRtHAxo0jyoqnOr6qcTT9vjgZNWexfwWuDm\nSfNfCfxba1W5qapurKoLZjjk3wJvmm5hkgOTnJ3k+iRfS/KIkWWPTvKfSW5M8rEk/zrRqpRk+ySf\nTbI2yXVteue27E3Ak4B3tpa+d7b5lWS3JHu1VsTNRo717CTntOm7JVmR5HtJfpjkhCT3Xt9Jtu7E\na5KsTnL4yPwD2jnckOTSya2HSZ7Yzvv6tvyIJEcChwOvafV/pq173ySfaOd8cZI/GtnP0Uk+3loo\nbwCOaPM+MrLOM5Oc1451epKHjizbJcmJbd8/HHnNjkjy1ZH1KslLklzU9vOuJJnmNXlcklXt3K9K\n8taRZY8fOe9vJ9lnZNn9k3y5XfdTkrxz4jwyRZdye833m+najbTqLk/yg3a9/nRkP5u16/i9duwz\nk+zSlj2k1XJtkguTHDqy3f5Jzm/bXJ7kj6f8JZEWqqry4cPHPDyAdwM/ZQhnZwHbjiz7beDTbXo1\nsN/Isi8BbwO+BlwNfAa43zTHWNr2vx1w+cR+gI8AR7fpR7X97AVsBixvx7w7sAVwCXAUsDnwHIbA\n+Fdt2/sAzwW2bsf4GPCpkeOfDrx4Uk0F7Namvwf81siyjwEr2vRRwDeAnVst7wWOn+Y89wFuAd7a\n1n0K8BPgwSPLf43hP6GPAK4CntWW7QrcCDy/neN9gD3bsg9PnGt7fjfgTODP22vzAOD7wNPa8qOB\nXwDPautu1eZ9pC1/UKvrt9qxXgN8t+1rM+DbwN8D2wBbAk9s2x0BfHXSa/hZ4F7A/YC1wNOneW2+\nDrywTW8LPL5N7wT8ENi/1fpb7fnike0mXs8nt9foIyOv52WTjrOadb9f01471v1Ovq+9Po8EbgIe\n2pa/GvgO8GAgbfl92mtyKfC7wCKG39trgD3admuAJ7Xp7YFHj/s97sPHhnzYgibNk6p6KUOoeRJw\nIsMfKZJsB/w1wx+5qezMEKKOYvjjfDFw/AyH+xlDC9pU91MdCby3qs6oqlur6thWy+PbYxHw9qr6\nRVWdCHxz5Bx+WFWfqKqfVtWN7RhPmfHk1zmeIRhNnPf+I+fyEuBPq+qyqrqJIegckvV3G76+hlbF\nLwMnA4e2Ok+vqu9U1W1VdU47xkSdvwN8saqOb+f4w6o6e5r9P5YhwPxFVd1cw7107wMOG1nn61X1\nqXasn03a/nnAyVV1SlX9AngLQ0jZG3gccF/g1VX1k6r6eVV9lemtrKrrq+oHwGnAntOs9wtgtyQ7\nVNWPq+obbf4LgM9V1edaracAq4D9k9yvnevE6/kVhv8IzNZsrt0bq+pnVfVthmD6yDb/xcCfVdWF\nNfh2Vf0QOBBYXVUfqqpbquo/gU8w/Gdm4jz3SHKPqrquqs66A/VK3TOgSfOoBaKvMoSuP2izjwb+\nuapWT7PZz4BPVtW3qurnwBuBvZPcc4bDvR9YkuSgSfN3BV7VurmuT3I9sAtDWLgvcHlV1cj6l05M\nJNk6yXuTXNK69L4C3Gu023IGHwWek+TuDK1zZ1XVJSN1fXKkpguAW4El0+zruqr6ycjzS1r9ZOhO\nPa11Hf6IIUDs0NbbhaElbzZ2Be476bV63aSaLp16U2j1TJwfVXVbW3+nVsclVXXLLGu5cmT6pwyt\nY1N5EUPL3X8l+VaSA0fO5bcnncsTgR1bnVO9nrM1m2s3Xf3TXY9dGe6lHK33cOBX2/LnMgT8S1rX\n7K/fgXql7hnQpPFYxLp70PYF/ijD/VlXMvzBOiHJa9vycxi6iCaMTk+rqm5mCHN/ydB1NOFS4E1V\nda+Rx9ZVdTxDt9FOk+5v2mVk+lUMXVF71fCBhye3+RPrr7e2qjqf4Q//Mxhasj46qa5nTKpry6q6\nfJrdbZ9km5Hn9wOuaNMfBU4CdqmqewLvGanxUv7n/X+/LHHS80uBiyfVtF1V7b+ebUZdwRA0AGiv\n6y4M3c+XAveboYXwDquqi6rq+cCvAG8GPt5ep0sZ/iMwei7bVNVKhus+1es54ScM3doT57EZsHhk\n+R29dqOmux6XAl+etM9tq+oP2nl+q6oObuf5KeCEWRxLWjAMaNIcS/IrSQ5Lsm27IfppDN18p7ZV\n9gUeztBltSfDH/XfZ/jQAMCHgGcn2TPJ5sDrGe5P+tEsDv/PDPc2PX1k3vuAl7RWpiTZJsNN9dsx\n3Id0K/CHSRYlOZihK27Cdgwtete3m8DfMOl4VzHcp7U+H2Xorn0ywz1oE94DvCnJrgBJFrfjr88b\nk2yR5EkMXWIT+9sOuLaqfp7kcQxhcMJxwH5JDm3neJ8kE92Fk+v/JnBjktcm2apdv4cneewMdU04\nATggyb7t2r2KoTv5a23fa4CV7RpsmeQJs9zvtJK8IMni1lp3fZt9G8N9iAcleVo7jy0z3Py/c2vF\nXMW61/OJwGjL638DW7bfk82BP2O412zCnbl2E94P/GWS3dvv4yOS3IfhnrsHJXlhks3b47FJHtpq\nPDzJPVvX8Q3tHKWNhgFNmnvF0J15GXAdw31Ir6iqk+CX93VdOfFgCEjXVdWP2/IvMXSrncxwc/9u\n3D5wTH/gqlsZbnC/98i8VcDvAe9s9XyX4ab0iVa35zB0k13PcN/SZ2n3ywH/wHAP1TUMN4V/ftIh\n38Zw79F1Sd4+TVkT94N9qaqumbTtScAXktzY9r/Xek7vylb/FQyh6yVV9V9t2UuBv2j7+XNGWlfa\nPVz7M4Sla4GzWXc/1AcY7mu6Psmn2ut3IENwvrid9/uBmbqXJ451IcNr+I627UHAQe1+tlvb892A\nHzD8fjxvNvudwdOB85L8mOE1Pazd+3UpcDDD79JahhaqV7Pu78DvMLze1zIE738aOY8fMbym72do\n/ftJq3fCHb12o97KcH2+wBC0PgBs1e5xfCrD/X5XMFzvN7MuGL4QWN262l/C0P0pbTRy+1tNJOn2\nkpwBvKeqPjTuWjR/Mnw1yW5V9YJx1yJtimxBk3Q7SZ6S5Fdb999yhq+pmNxSJkmaQ37rtaTJHszQ\n5bQNw3d+HVJVa9a/iSRpQ7KLU5IkqTN2cUqSJHXGgCZpwcvIuJCStDEwoEm608YRjJJ8OG3w9jnY\n98ToAzdkGEz84EnLfyfDKAo/SfKprGcw97vy2rTvzjs+yRVJfpTkP5LsNbJ8nyS3ZRjUfeKxfGT5\nh5PcPGn5bEd7kNQBA5okrfMKYOc2SsKRwEeS7AiQ5GEMg4C/kGEIo58C756jOrYFvgU8huE77I4F\nTk4yOrzTFe2b9Scex07ax99OWn7rHNUqaQ4Y0CTNiSQHJjm7fenr15I8YmTZ6iR/nOSc1kL0r0m2\nHFn+miRrWgvSi5NUkt2SHMnwhaSvaa1CowN67znV/pLskOSzrY5rk/x7kin/7WsDdU98KW8Bm7Nu\nqKvDgc9U1Vfalwi/nmFc0e2mOPd/Zhgq6TOtzte0+c9Mcl6r5fQkD52mju9X1Vurak0bv/UYYAuG\nT9hK2gQY0CRtcEkeBXyQYciq+zC0PJ2UYZD0CYcyfOv9/Rm+a+2Itu3TgVcC+zF8y/4+Exu0oHIc\n61qHDpppfwwjBlzGMHbkEoZv0p/24+stzP0cOAM4nWEIJICHAd8eqeV7DCMsPGjyPqrqhQyjAxzU\n6vzbJA9iGEXhFa2WzzEEuC2mq2Wkpj0ZAtp3R2b/SpKrklyc5O9z+3E0AV7aAumZSZ470zEk9cWA\nJmkuHAm8t6rOaC1AxzKEmcePrPP2qrqiqq4FPsMwnBIMQetDVXVeVf0UOHqWx5xuf78AdgR2rapf\nVNW/13q+X6iqDmQYy3N/4AttTEsYuh0nj396Q1t3Np4HnFxVp7TxI9/CMGzW3uvbKMk9GMZUfePI\n+Kv/1c5vR+A3GbpC3zqy2duB3RkGEn898OFsgHE+Jc0fA5qkubAr8KrWlXd9kusZugrvO7LOlSPT\nP2UIQLR1Lh1ZNjq9PtPt7+8YWp6+kOT7SVbMtKMW5P4f8NQkz2yzfwzcY9Kq9wRunGV99wUuGTnG\nbQznttN0GyTZiiFsfqOq/mZk2yur6vyquq2qLgZeAzx3ZPlZbYzXW6rqcwytjs+ZZZ2SOmBAkzQX\nLgXeVFX3GnlsXVXHz2LbNcDOI893mbT8Dn27dlXdWFWvqqoHAM8EXplk31luvgh4YJs+j3WDqpPk\ngQzdjv893aEnPb+CIbhObB+Gc7t8qo1bd/CnGLpnf3+GOov1/3teQGbYh6SOGNAk3VWbJ9ly5LEI\neB/wkiR7ZbBNkgOmuqF+CicAv5vkoUm2ZuiiG3UV8IDZFtc+rLBbC0Q/Am4FbptivYckeUaSrZJs\nnuQFwJOBL7dVjgMOSvKkdr/XXwInVtV0LWiT6zwBOCDJvkk2Z7g37ibga1PUsjnwceBnwPKRbtaJ\n5b+RZNf22u4CvBn49MjyQ5Jsm+RuSZ4KvAA4aYaXSlJHDGiS7qrPMQSJicfRVbUK+D3gncB1DF2M\nR8xmZ61r8e3AaW27b7RFE5+u/ACwR+s6/dQsdrk78EWGLsqvA++uqtOmWC8M97tdDawFjgKeV1Vn\ntbrOA17CENSuZhir9KXrOe7fAH/W6vzjqrqQISi9A7gGOIjhQwQ3T7Ht3sCBwFOB60e+y+xJbfmj\nGILdT9rPc4A/Gtn+KIaWuesZunh/r6pOX0+tkjrjWJySuta+iuJc4O5Vdcu465Gk+WALmqTuJHl2\nkrsn2Z6h++4zhjNJmxIDmqQe/T5DN+L3GO4Z+4PxliNJ88suTkmSpM7YgiZJktSZReMu4K7YYYcd\naunSpeMuQ5IkaUZnnnnmNVW1eDbrLuiAtnTpUlatWjXzipIkSWOW5JKZ1xrYxSlJktQZA5okSVJn\nDGiSJEmdMaBJkiR1xoAmSZLUGQOaJElSZwxokiRJnTGgSZIkdcaAJkmS1BkDmiRJUmcMaJIkSZ0x\noEmSJHXGgCZJktQZA5okSVJnDGiSJEmdMaBJkiR1ZtG4C9DGZ+mKk+f8GKtXHjDnx5AkaVxsQZMk\nSeqMAU2SJKkzBjRJkqTOGNAkSZI6Y0CTJEnqjAFNkiSpMwY0SZKkzhjQJEmSOmNAkyRJ6owBTZIk\nqTMGNEmSpM4Y0CRJkjpjQJMkSeqMAU2SJKkzBjRJkqTOGNAkSZI6Y0CTJEnqjAFNkiSpMwY0SZKk\nzhjQJEmSOmNAkyRJ6owBTZIkqTMGNEmSpM4Y0CRJkjpjQJMkSeqMAU2SJKkzBjRJkqTOGNAkSZI6\nY0CTJEnqjAFNkiSpMwY0SZKkzhjQJEmSOmNAkyRJ6owBTZIkqTMGNEmSpM4Y0CRJkjpjQJMkSeqM\nAU2SJKkzBjRJkqTOGNAkSZI6Y0CTJEnqzJwFtCQfTHJ1knNH5t07ySlJLmo/tx9Z9idJvpvkwiRP\nm6u6JEmSejeXLWgfBp4+ad4K4NSq2h04tT0nyR7AYcDD2jbvTrLZHNYmSZLUrTkLaFX1FeDaSbMP\nBo5t08cCzxqZ/y9VdVNVXQx8F3jcXNUmSZLUs/m+B21JVa1p01cCS9r0TsClI+td1ub9D0mOTLIq\nyaq1a9fOXaWSJEljMrYPCVRVAXUntjumqpZV1bLFixfPQWWSJEnjNd8B7aokOwK0n1e3+ZcDu4ys\nt3ObJ0mStMmZ74B2ErC8TS8HPj0y/7Akd09yf2B34JvzXJskSVIXFs3VjpMcD+wD7JDkMuANwErg\nhCQvAi4BDgWoqvOSnACcD9wCvKyqbp2r2iRJkno2ZwGtqp4/zaJ9p1n/TcCb5qoeSZKkhcKRBCRJ\nkjpjQJMkSeqMAU2SJKkzBjRJkqTOGNAkSZI6Y0CTJEnqjAFNkiSpMwY0SZKkzhjQJEmSOmNAkyRJ\n6owBTZIkqTMGNEmSpM4Y0CRJkjpjQJMkSeqMAU2SJKkzBjRJkqTOGNAkSZI6Y0CTJEnqjAFNkiSp\nMwY0SZKkzhjQJEmSOmNAkyRJ6owBTZIkqTMGNEmSpM4Y0CRJkjpjQJMkSeqMAU2SJKkzBjRJkqTO\nGNAkSZI6Y0CTJEnqjAFNkiSpMwY0SZKkzhjQJEmSOmNAkyRJ6owBTZIkqTMGNEmSpM4Y0CRJkjpj\nQJMkSeqMAU2SJKkzBjRJkqTOGNAkSZI6Y0CTJEnqjAFNkiSpMwY0SZKkzhjQJEmSOmNAkyRJ6owB\nTZIkqTMGNEmSpM4Y0CRJkjpjQJMkSeqMAU2SJKkzBjRJkqTOGNAkSZI6Y0CTJEnqzFgCWpL/k+S8\nJOcmOT7JlknuneSUJBe1n9uPozZJkqRxm/eAlmQn4I+AZVX1cGAz4DBgBXBqVe0OnNqeS5IkbXLG\n1cW5CNgqySJga+AK4GDg2Lb8WOBZY6pNkiRprOY9oFXV5cBbgB8Aa4AfVdUXgCVVtaatdiWwZKrt\nkxyZZFWSVWvXrp2XmiVJkubTOLo4t2doLbs/cF9gmyQvGF2nqgqoqbavqmOqallVLVu8ePGc1ytJ\nkjTfxtHFuR9wcVWtrapfACcCewNXJdkRoP28egy1SZIkjd04AtoPgMcn2TpJgH2BC4CTgOVtneXA\np8dQmyRJ0tgtmu8DVtUZST4OnAXcAvwncAywLXBCkhcBlwCHzndtkiRJPZj3gAZQVW8A3jBp9k0M\nrWmSJEmbNEcSkCRJ6owBTZIkqTMGNEmSpM4Y0CRJkjpjQJMkSeqMAU2SJKkzBjRJkqTOGNAkSZI6\nY0CTJEnqjAFNkiSpM2MZ6knjsXTFyeMuQZIkzYItaJIkSZ0xoEmSJHXGgCZJktQZA5okSVJnDGiS\nJEmdMaBJkiR1xoAmSZLUGQOaJElSZwxokiRJnTGgSZIkdcaAJkmS1BkDmiRJUmcMaJIkSZ0xoEmS\nJHXGgCZJktQZA5okSVJnDGiSJEmdMaBJkiR1xoAmSZLUGQOaJElSZwxokiRJnTGgSZIkdcaAJkmS\n1BkDmiRJUmcMaJIkSZ0xoEmSJHXGgCZJktQZA5okSVJnDGiSJEmdMaBJkiR1xoAmSZLUGQOaJElS\nZwxokiRJnTGgSZIkdcaAJkmS1JlZBbQkT5jNPEmSJN11s21Be8cs50mSJOkuWrS+hUl+HdgbWJzk\nlSOL7gFsNpeFSZIkbarWG9CALYBt23rbjcy/AThkroqSJEnalK03oFXVl4EvJ/lwVV0yTzVJkiRt\n0mZqQZtw9yTHAEtHt6mq35yLoiRJkjZlsw1oHwPeA7wfuHXuyunT0hUnz/kxVq88YM6PIUmSFobZ\nBrRbquof57QSSZIkAbP/mo3PJHlpkh2T3HviMaeVSZIkbaJm24K2vP189ci8Ah5wZw6a5F4M3aUP\nb/v538CFwL8y3Oe2Gji0qq67M/uXJElayGbVglZV95/icafCWfM24PNV9RDgkcAFwArg1KraHTi1\nPZckSdrkzKoFLcn/mmp+Vf3THT1gknsCTwaOaPu4Gbg5ycHAPm21Y4HTgdfe0f1LkiQtdLPt4nzs\nyPSWwL7AWcAdDmjA/YG1wIeSPBI4EzgKWFJVa9o6VwJLpto4yZHAkQD3u9/97sThJUmS+jargFZV\nLx993u4h+5e7cMxHAy+vqjOSvI1J3ZlVVUlqmlqOAY4BWLZs2ZTrSJIkLWSz/RTnZD9haAm7My4D\nLquqM9rzjzMEtquS7AjQfl59J/cvSZK0oM32HrTPMHzaEoZB0h8KnHBnDlhVVya5NMmDq+pChu7S\n89tjObCy/fz0ndm/JEnSQjfbe9DeMjJ9C3BJVV12F477cuC4JFsA3wd+l6E174QkLwIuAQ69C/uX\nJElasGZ7D9qXkyxh3YcFLrorB62qs4FlUyza967sV5IkaWMwq3vQkhwKfBP4bYaWrTOSHDKXhUmS\nJG2qZtvF+afAY6vqaoAki4EvMtzgL0mSpA1otp/ivNtEOGt+eAe2lSRJ0h0w2xa0zyf5N+D49vx5\nwOfmpiRJkqRN23oDWpLdGL7h/9VJngM8sS36OnDcXBcnSZK0KZqpBe0fgD8BqKoTgRMBkvxaW3bQ\nnFYnSZK0CZrpPrIlVfWdyTPbvKVzUpEkSdImbqaAdq/1LNtqQxYiSZKkwUwBbVWS35s8M8mLgTPn\npiRJkqRN20z3oL0C+GSSw1kXyJYBWwDPnsvCJEmSNlXrDWhVdRWwd5LfAB7eZp9cVV+a88okSZI2\nUbMdi/M04LQ5rkWSJEk4GoAkSVJ3DGiSJEmdMaBJkiR1xoAmSZLUGQOaJElSZwxokiRJnTGgSZIk\ndcaAJkmS1BkDmiRJUmcMaJIkSZ0xoEmSJHXGgCZJktQZA5okSVJnDGiSJEmdMaBJkiR1xoAmSZLU\nGQOaJElSZwxokiRJnTGgSZIkdcaAJkmS1BkDmiRJUmcMaJIkSZ0xoEmSJHXGgCZJktQZA5okSVJn\nDGiSJEmdMaBJkiR1xoAmSZLUGQOaJElSZwxokiRJnTGgSZIkdcaAJkmS1BkDmiRJUmcMaJIkSZ0x\noEmSJHXGgCZJktQZA5okSVJnDGiSJEmdMaBJkiR1xoAmSZLUGQOaJElSZwxokiRJnTGgSZIkdWZs\nAS3JZkn+M8ln2/N7JzklyUXt5/bjqk2SJGmcxtmCdhRwwcjzFcCpVbU7cGp7LkmStMkZS0BLsjNw\nAPD+kdkHA8e26WOBZ813XZIkST0YVwvaPwCvAW4bmbekqta06SuBJVNtmOTIJKuSrFq7du0clylJ\nkjT/5j2gJTkQuLqqzpxunaoqoKZZdkxVLauqZYsXL56rMiVJksZm0RiO+QTgmUn2B7YE7pHkI8BV\nSXasqjVJdgSuHkNtkiRJYzfvLWhV9SdVtXNVLQUOA75UVS8ATgKWt9WWA5+e79okSZJ60NP3oK0E\nfivJRcB+7bkkSdImZxxdnL9UVacDp7fpHwL7jrMeSZKkHvTUgiZJkiQMaJIkSd0xoEmSJHXGgCZJ\nktQZA5okSVJnDGiSJEmdMaBJkiR1xoAmSZLUGQOaJElSZwxokiRJnTGgSZIkdcaAJkmS1BkDmiRJ\nUmcMaJIkSZ0xoEmSJHXGgCZJktQZA5okSVJnDGiSJEmdMaBJkiR1ZtG4C5DujKUrTp6X46xeecC8\nHEeSpFG2oEmSJHXGgCZJktQZA5okSVJnDGiSJEmdMaBJkiR1xk9xdmK+PpUoSZL6ZwuaJElSZwxo\nkiRJnTGgSZIkdcaAJkmS1BkDmiRJUmcMaJIkSZ0xoEmSJHXGgCZJktQZA5okSVJnDGiSJEmdMaBJ\nkiR1xoAmSZLUGQOaJElSZwxokiRJnTGgSZIkdcaAJkmS1BkDmiRJUmcMaJIkSZ0xoEmSJHXGgCZJ\nktQZA5okSVJnDGiSJEmdMaBJkiR1xoAmSZLUGQOaJElSZwxokiRJnTGgSZIkdcaAJkmS1Jl5D2hJ\ndklyWpLzk5yX5Kg2/95JTklyUfu5/XzXJkmS1INxtKDdAryqqvYAHg+8LMkewArg1KraHTi1PZck\nSdrkzHtAq6o1VXVWm74RuADYCTgYOLatdizwrPmuTZIkqQdjvQctyVLgUcAZwJKqWtMWXQksmWab\nI5OsSrJq7dq181KnJEnSfBpbQEuyLfAJ4BVVdcPosqoqoKbarqqOqaplVbVs8eLF81CpJEnS/BpL\nQEuyOUM4O66qTmyzr0qyY1u+I3D1OGqTJEkat3F8ijPAB4ALquqtI4tOApa36eXAp+e7NkmSpB4s\nGsMxnwC8EPhOkrPbvNcBK4ETkrwIuAQ4dAy1SZIkjd28B7Sq+iqQaRbvO5+1SJIk9ciRBCRJkjpj\nQJMkSeqMAU2SJKkzBjRJkqTOjONTnNKCsXTFyXN+jNUrD5jzY0iSFhZb0CRJkjpjQJMkSeqMAU2S\nJKkzBjRJkqTOGNAkSZI6Y0CTJEnqjAFNkiSpMwY0SZKkzhjQJEmSOmNAkyRJ6owBTZIkqTMGNEmS\npM4Y0CRJkjpjQJMkSeqMAU2SJKkzBjRJkqTOGNAkSZI6Y0CTJEnqjAFNkiSpMwY0SZKkzhjQJEmS\nOmNAkyRJ6owBTZIkqTMGNEmSpM4Y0CRJkjpjQJMkSeqMAU2SJKkzi8ZdgLSpW7ri5Dk/xuqVB8z5\nMSRJG44taJIkSZ0xoEmSJHXGgCZJktQZA5okSVJnDGiSJEmdMaBJkiR1xoAmSZLUGQOaJElSZwxo\nkiRJnTGgSZIkdcaAJkmS1BkDmiRJUmccLF3SBuPA75K0YdiCJkmS1BkDmiRJUmcMaJIkSZ0xoEmS\nJHXGgCZJktQZA5okSVJnDGiSJEmdMaBJkiR1xoAmSZLUme5GEkjydOBtwGbA+6tq5ZhLkha8+fiG\nf2ljMF/vlfkYEWNjGtljYzqX2eqqBS3JZsC7gGcAewDPT7LHeKuSJEmaX10FNOBxwHer6vtVdTPw\nL8DBY65JkiRpXqWqxl3DLyU5BHh6Vb24PX8hsFdV/eHIOkcCR7anDwYunPdCF4YdgGvGXYTuEq/h\nwub1W/i8hgtfb9dw16paPJsVu7sHbSZVdQxwzLjr6F2SVVW1bNx16M7zGi5sXr+Fz2u48C3ka9hb\nF+flwC5Tcl23AAAGcUlEQVQjz3du8yRJkjYZvQW0bwG7J7l/ki2Aw4CTxlyTJEnSvOqqi7Oqbkny\nh8C/MXzNxger6rwxl7VQ2Q288HkNFzav38LnNVz4Fuw17OpDApIkSeqvi1OSJGmTZ0CTJEnqjAFt\nI5BkdZLvJDk7yao2795JTklyUfu5/bjr1DpJPpjk6iTnjsyb9pol+ZMk301yYZKnjadqjZrmGh6d\n5PL2Xjw7yf4jy7yGHUmyS5LTkpyf5LwkR7X5vg8XiPVcw43ifeg9aBuBJKuBZVV1zci8vwWuraqV\nSVYA21fVa8dVo24vyZOBHwP/VFUPb/OmvGZtuLPjGUbauC/wReBBVXXrmMoX017Do4EfV9VbJq3r\nNexMkh2BHavqrCTbAWcCzwKOwPfhgrCea3goG8H70Ba0jdfBwLFt+liGX1p1oqq+Alw7afZ01+xg\n4F+q6qaquhj4LsM/MBqjaa7hdLyGnamqNVV1Vpu+EbgA2AnfhwvGeq7hdBbUNTSgbRwK+GKSM9tQ\nWABLqmpNm74SWDKe0nQHTHfNdgIuHVnvMtb/j5DG6+VJzmldoBPdY17DjiVZCjwKOAPfhwvSpGsI\nG8H70IC2cXhiVe0JPAN4Wet6+aUa+rHty15AvGYL1j8CDwD2BNYA/3e85WgmSbYFPgG8oqpuGF3m\n+3BhmOIabhTvQwPaRqCqLm8/rwY+ydBke1Xrn5/op796fBVqlqa7Zg6BtkBU1VVVdWtV3Qa8j3Xd\nJ17DDiXZnOEP+3FVdWKb7ftwAZnqGm4s70MD2gKXZJt2cyRJtgGeCpzLMETW8rbacuDT46lQd8B0\n1+wk4LAkd09yf2B34JtjqE8zmPjD3jyb4b0IXsPuJAnwAeCCqnrryCLfhwvEdNdwY3kfdjXUk+6U\nJcAnh99TFgEfrarPJ/kWcEKSFwGXMHyqRZ1IcjywD7BDksuANwArmeKaVdV5SU4AzgduAV7W66eO\nNiXTXMN9kuzJ0C22Gvh98Bp26gnAC4HvJDm7zXsdvg8Xkumu4fM3hvehX7MhSZLUGbs4JUmSOmNA\nkyRJ6owBTZIkqTMGNEmSpM4Y0CRJkjpjQJM0r5L8aZLz2jAsZyfZa9w13RVJPpzkkDnY7+tGppcm\nOXd960vauBjQJM2bJL8OHAg8uqoeAezH7cfG0zqvm3kVSRsrA5qk+bQjcE1V3QRQVddU1RUASR6T\n5MtJzkzybyPD7Twmybfb4+8mWpKSHJHknRM7TvLZJPu06acm+XqSs5J8rI3VR5LVSd7Y5n8nyUPa\n/G2TfKjNOyfJc9e3n+ms5xxOT/LmJN9M8t9JntTmb53khCTnJ/lkkjOSLEuyEtiqtTAe13a/WZL3\ntdbHLyTZasNcEkk9MqBJmk9fAHZpIeXdSZ4CvxxP7x3AIVX1GOCDwJvaNh8CXl5Vj5zNAZLsAPwZ\nsF9VPRpYBbxyZJVr2vx/BP64zXs98KOq+rXWsvelWexn8nHXdw4Ai6rqccArGEYdAHgpcF1V7dFq\neAxAVa0AflZVe1bV4W3d3YF3VdXDgOuB587m9ZC0MDnUk6R5U1U/TvIY4EnAbwD/mmQFQ/h5OHBK\nG7ZsM2BNknsB96qqr7Rd/DPwjBkO83hgD+A/2r62AL4+snxiUOwzgee06f2Aw0bqvC7JgTPsZ7IH\nT3UO0xx3aZt+IvC2dsxzk5yznv1fXFUTw9mM7kPSRsiAJmletbHvTgdOT/IdhgGpzwTOq6pfH123\nBbTp3MLtewG2nNgMOKWqnj/Ndje1n7ey/n8DZ9rPVOv/j3O4E8edzk0j07cCdnFKGzG7OCXNmyQP\nTrL7yKw9GQakvhBY3D5EQJLNkzysqq4Hrk/yxLb+4SPbrgb2THK3JLsAj2vzvwE8IclubV/bJHnQ\nDKWdArxspM7t78R+pjyHGY77H7TBuJPsAfzayLJftG5TSZsgA5qk+bQtcGy7Kf4chi7Eo6vqZuAQ\n4M1Jvg2cDezdtvld4F1JzmZopZrwH8DFwPnA24GzAKpqLXAEcHw7xteBh8xQ118B2yc5tx3/N+7o\nfmY4h+m8myHUnd9qOA/4UVt2DHDOyIcEJG1CUlXjrkGSZiXJUuCzVfXwMZeyQSTZDNi8qn6e5IHA\nF4EHt7AnaRPmPWiSND5bA6e1rswALzWcSQJb0CRJkrrjPWiSJEmdMaBJkiR1xoAmSZLUGQOaJElS\nZwxokiRJnfn/ZNh/p8oL+FMAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x7f2b30dec8d0>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.hist(sizes, bins=20)\n",
    "plt.title(\"%i Negative bacteriocin sequences\\nLengths %i to %i\" \\\n",
    "            % (len(sizes),min(sizes),max(sizes)))\n",
    "plt.xlabel(\"Sequence length\")\n",
    "plt.ylabel(\"Count\")\n",
    "\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "999 6 94\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAmgAAAHICAYAAAD6GxY6AAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAIABJREFUeJzt3XmYZVV5tvH7kUaQQQHpjzDaRHBANA4tjlEjxgkIJjGI\nUYMjMdGoiVOrUTEJBqOXiUaNEhxIRBRxQjEGRNAkKtogMxJQwGZuBVQQQfD9/tir9FCp6q5u+tRZ\n1X3/rquuOmftYb37bKAe1tpn71QVkiRJ6sedJl2AJEmSbs+AJkmS1BkDmiRJUmcMaJIkSZ0xoEmS\nJHXGgCZJktQZA5qksUpyQ5LfXMf7fFySy9blPteVJK9PcsQd3MezkpywrmqStPAY0KR5kOS+Sb6S\n5MdJLkry+9OWv7C135DkS0l2GFm2VZIjk1zTfg5ZRT9LklSSL05r/+iqtltXkpyS5IWjbVW1RVV9\nf9x9rwtJPpLk7+7IPqrqrVX1wtWvucp9HFVVT7wj+5C0sBnQpDFLsgj4HPAFYBvgYOCjSe7Vlj8O\neCuwf1t+MXD0yC7+EdgMWALsBTwnyfNW0+3Dkjxy3R2F5qKda0m6wwxo0vjdB9gB+Mequq2qvgL8\nD/Cctnxf4NiqOreqbgH+FnhMknu25fsBb6+qn1XVJcAHgeevps9/AA6dbWGSfZOckeT6JF9P8oCR\nZQ9O8p0kP03yySSfmBpVSrJ1ki8kWZnkuvZ6p7bsUOC3gfe0kcD3tPZKsluShyW5KslGI339fpKz\n2us7JVmW5HtJfpTkmCTbrOog23TiD5NckuRZI+37tGP4SZIV00cPkzy6Hff1bflzkxwMPAt4Tav/\n823dHZJ8qh3zxUleNrKfQ5Ic20YofwI8t7V9dGSd30tybuvrlCT3HVm2c5JPt33/aOQze26S/x5Z\nr5K8OMmFbT/vTZJZPpO9kixvx351kneOLHv4yHGf2f7nYGrZrkm+2s77iUneM3UcmWFKuX3mT1jd\nuRsZ1T0oyQ/a+XrDyH42aufxe63v05Ls3Jbdp9VybZILkhwwst1Tk5zXtrk8yatm/IdEWqAMaNJk\nBNhzFctYzfLZlk15H3CvqT+gt9s4eRDwIeBPgbsDHwCOS7JJkjsDnwE+wjCadzQwOh17J+DDwD2A\nXYCbgPcAVNUbgP8CXtqmNV862m9VnQrcCDx+pPmPgY+1138BPA14LEOgvQ547yqO8TeAbYEdgYOA\nw5Pcuy27EfgTYCtgH+DPkjytHf89gP8A/hlYDDwQOKOqDgeOAv6h1b9fkjsBnwfObP3sDbwiyZNG\n6tgfOLb1ddRogRlGSY8GXtH6+iLw+SR3bkH1C8ClDKOjOwIfX8Xx7gs8FHgAcADwpFnWexfwrqq6\nK3BP4JhWy47A8cDfMZzbVwGfSrK4bfcx4LT2mf5t+0znai7n7tHAvRk+wzeNBNW/Ap4JPBW4K8P/\nfPwsyebAia2u/wccCLwvyR5tuw8Cf1pVWzL8+/CVNahX6p4BTRq/C4BrgFcn2TjJExn+kG3Wln8J\n+KMkD0hyF+BNQE1b/tokWybZjeEP2Gas2k0MI2gzXU91MPCBqjq1jegdCdwMPLz9LALeXVW/qKpP\nA9+a2rCqflRVn2qjeT9tfTx2DT6Loxn+GJNkS4Y/ylPTuS8G3lBVl1XVzcAhwNOz6mnDN1bVzVX1\nVYbwcUCr85SqOruqfllVZ7U+pur8Y+DLVXV0O8YfVdUZs+z/ocDiqvqbqrqlXUv3rwxhYco3quqz\nra+bpm3/DOD4qjqxqn4BvAO4C/BIhunqHYBXV9WNVfXzqvpvZndYVV1fVT8ATmYIljP5BbBbkm2r\n6oaq+mZrfzbwxar6Yqv1RGA58NQku7Rjnfo8v8YQTOdqLufuLVV1U1WdyRB4f6u1vxD466q6oAZn\nVtWPGALpJVX14aq6taq+A3wK+KOR49wjyV2r6rqqOn0N6pW6Z0CTxqz9YX4aw0jOVcArGUY1LmvL\nv8zwB+1TwCXt56dTy4GXAT8HLmS4lu3okWWrcgSwXZL9prXfA3hlm+a6Psn1wM4MYWEH4PKqqpH1\nV0y9SLJZkg8kubRN6X0N2Coj05ar8THgD5JsAvwBcHpVXTpS12dGajofuA3YbpZ9XVdVN468v7TV\nT4bp1JPb1OGPGQLEtm29nYHvzbHeewA7TPusXj+tphUzbwqtnqnjo6p+2dbfsdVxaVXdOsdarhp5\n/TNgi1nWewFwL+C7Sb6dZN+RY/mjacfyaGD7VudMn+dczeXczVb/bOfjHgzXUo7W+yyGkVOAP2QI\n+Je2qdlHrEG9UvcMaNI8qKqzquqxVXX3qnoS8JvcfmTqvVW1e1VtxxDUFgHntGXXVtWzquo3qup+\nDP/efmuGbqb3eQvwFobpqtHrlVYAh1bVViM/m1XV0cCVwI7Trm/aeeT1KxmmqR7WptAe09qn1h8N\ndjPVdB7DH/6ncPvpzam6njKtrk2r6vJZdrd1mwabsgtwRXv9MeA4YOequhvw/pEaVzBM/c1Y4rT3\nK4CLp9W0ZVU9dRXbjLqCIWgA0D7XnYHL2753Wc0I4Rqrqgur6pkM04JvA45tn9MK4N+nHcvmVXUY\nw3mf6fOcciMjo7YtkC8eWb6m527UbOdjBfDVafvcoqr+rB3nt6tq/3acn6VN5UrrCwOaNA/a9OWm\nbQTqVQyjFh9pyzZNsmcGuwCHM1xDdF1bfs8kd28XUz+FYYpyrreC+HdgU+DJI23/Cry4jTIlyeYZ\nLqrfEvgGw8jHS5MsSrI/w1TclC0Zpk+vbxeBv3laf1czhM9V+RjwcoZw98mR9vcDh7ZrxEiyuPW/\nKm9p13P9NsOU2NT+tgSuraqfJ9mLIQxOOQp4QpID2jHePcnUdOH0+r8F/DTJa5PcpZ2DPZM8dDV1\nTTkG2CfJ3kk2Zgi4NwNfb/u+EjisnYNNkzxqjvudVZJnJ1ncRuuub82/BD4K7JfkSe04Ns1w8f9O\nbRRzOb/+PB/N8OWUKf8LbNr+OdkY+Gtgk5Hla3PuphwB/G2S3ds/jw9IcneG6/PuleQ57dKAjZM8\nNMMta+6c4V5xd2sj1D9pxyitNwxo0vx4DsMf42sYLpL+3XatDgwB6mPADQx/tL8BvHFk24cAZzNM\ne/498KyqOncunVbVbQzXtG0z0rYceBHDxf3XARcBz23LbmGYenwBwx/3ZzP8oZyq9Z8YrqH6IfBN\nhuvjRr2L4dqj65K8e5aypq4H+0pV/XDatscBJyT5adv/w1ZxeFe1+q9gCF0vrqrvtmV/DvxN28+b\nGBldaddwPZUhLF0LnMGvr4f6IMN1Tdcn+Wz7/PZluN7r4nbcRwB3W0Vdv1JVFzB8hv/ctt0P2K9d\nz3Zbe78b8AOGaetnzGW/q/Fk4NwkNzB8pge2a79WMHyh4fXASoYRqlfz678Df8zweV/LELz/beQ4\nfszwmR7BMPp3I7efZl/TczfqnQzn5wSGoPVB4C41XOP4RIbr/a5gON9v49fB8DnAJW2q/cUM05/S\neiO3v9REkm4vyanA+6vqw5OuRfMnw61JdquqZ0+6FmlD5AiapNtJ8tgkv9Gm/w5iuK3D9JEySdIY\neddrSdPdm2HKaXPg+8DTq+rKyZYkSRsWpzglSZI64xSnJElSZwxoktZ7GXlupCQtBAY0SWMziWCU\n5CNpD3cf0/5fnuGh6TcmOT/D8zZnWu92D01fi362SnJkkmvazyGzrPfYDA8jH9sxS5p/fklAkuYo\nyQsZ7hG3D8PjjH6T4V5s4/CPDHfvX8Jwt/yTklw6eruTdtPYdwGnjqkGSRPiCJqkiUiyb5Iz2k1h\nv57kASPLLknyqiRnJflxkk8k2XRk+WuSXJnkiiQvbCNIuyU5mOGGpa9JckOS0Qd+P3Cm/SXZNskX\nWh3XJvmvJP/nv42t7c3AX1bVee3B3t+rqmtnWPfJDDeEfUar48zWvkOS41o/FyV50So+ov2At9fw\nYPpLGG7g+vxp67yS4Qav30XSesWAJmneJXkQ8CHgT4G7Ax8AjsvwEPUpBzDcFX9XhnuxPbdt+2Tg\nr4AnMNyF/3FTG1TV4QxPFfiH9tzG/Va3P4aQcxnDsyW3YwhWM329faf2s2eSFW2a8y0zhbmq+hLw\nVuATrY6pJxV8vPW1A/B04K1JHr+qz2pEgD1/9WZ4rNLzgb+Z4/aSFhADmqRJOBj4QFWdWlW3VdWR\nDI+TevjIOu+uqivaCNXnGR63BEPQ+nBVnVtVPwMOmWOfs+3vFwzPRr1HVf2iqv6rZr7/0E7t9xOB\n+wO/AzyTYcpztZLsDDwKeG1V/byqzmB4dNKfzLLJl4DXJtkyyW4MYWyzkeXvBt5YVTfMpX9JC4sB\nTdIk3AN4ZZtWvD7J9cDODCNLU64aef0zYIv2egeG50hOGX29KrPt7+0MzyM9Icn3kyybZfub2u9/\nqKrr27TjBxie6zkXOzA8wP2nI22XAjvOsv7LgJ8DFwKfY3iG6WUASfYDtqyqT8yxb0kLjF8SkDQJ\nK4BDq+rQtdj2Sn49mgVDsBu1RnffboHplQyBcU/gK0m+XVUnTVv1AuCWaftfVV/Tl10BbJNky5GQ\ntgvDw8dnqutaRh4AnuStwLfa272BpUmmQufdgNuS3L+q9l9FTZIWCEfQJI3bxkk2HflZBPwr8OIk\nD8tg8yT7JNlyDvs7Bnhekvsm2Qx447TlVzN8u3JO2pcVdksS4MfAbcAvp6/XplM/wfAFhC2T7MQw\nVfuFWXZ9NbBk6hq1qloBfB34+/Y5PIBhenTGW3EkuWeSuyfZKMlTWl9Tt9J4I3AvhmnaBwLHMXym\nz5vrcUvqmwFN0rh9kWF6cOrnkKpaDrwIeA/DbSou4tcX7a9SVf0Hw/VXJ7ftvtkW3dx+fxDYo02d\nfnYOu9wd+DJwA/AN4H1VdfIs6760rXdFW/djDF92mMkn2+8fJTm9vX4mw20zrgA+A7y5qr48y/YP\nAc4Gfgr8PfCsqjoXhlG/qrpq6ofhc71xpm+USlqYfBanpAUtyX2Bc4BNqurWSdcjSeuCI2iSFpwk\nv59kkyRbA28DPm84k7Q+MaBJWoj+FLgG+B7DNWN/NtlyJGndcopTkiSpM46gSZIkdWZB3wdt2223\nrSVLlky6DEmSpNU67bTTflhVi+ey7oIOaEuWLGH58uWTLkOSJGm1klw613Wd4pQkSeqMAU2SJKkz\nBjRJkqTOGNAkSZI6Y0CTJEnqjAFNkiSpMwY0SZKkzhjQJEmSOmNAkyRJ6owBTZIkqTMGNEmSpM4Y\n0CRJkjpjQJMkSeqMAU2SJKkzBjRJkqTOGNAkSZI6s2jSBUhrY8my4+eln0sO22de+pEkaZQjaJIk\nSZ0xoEmSJHXGgCZJktQZA5okSVJnDGiSJEmdMaBJkiR1xoAmSZLUGQOaJElSZwxokiRJnTGgSZIk\ndcaAJkmS1BkDmiRJUmcMaJIkSZ0xoEmSJHXGgCZJktQZA5okSVJnDGiSJEmdMaBJkiR1xoAmSZLU\nGQOaJElSZwxokiRJnTGgSZIkdcaAJkmS1JmxBbQkH0pyTZJzZlj2yiSVZNuRttcluSjJBUmeNK66\nJEmSejfOEbSPAE+e3phkZ+CJwA9G2vYADgTu17Z5X5KNxlibJElSt8YW0Krqa8C1Myz6R+A1QI20\n7Q98vKpurqqLgYuAvcZVmyRJUs/m9Rq0JPsDl1fVmdMW7QisGHl/WWubaR8HJ1meZPnKlSvHVKkk\nSdLkzFtAS7IZ8HrgTXdkP1V1eFUtraqlixcvXjfFSZIkdWTRPPZ1T2BX4MwkADsBpyfZC7gc2Hlk\n3Z1amyRJ0gZn3kbQqursqvp/VbWkqpYwTGM+uKquAo4DDkyySZJdgd2Bb81XbZIkST0Z5202jga+\nAdw7yWVJXjDbulV1LnAMcB7wJeAlVXXbuGqTJEnq2dimOKvqmatZvmTa+0OBQ8dVjyRJ0kLhkwQk\nSZI6Y0CTJEnqjAFNkiSpMwY0SZKkzhjQJEmSOmNAkyRJ6owBTZIkqTMGNEmSpM4Y0CRJkjpjQJMk\nSeqMAU2SJKkzBjRJkqTOGNAkSZI6Y0CTJEnqjAFNkiSpMwY0SZKkzhjQJEmSOmNAkyRJ6owBTZIk\nqTMGNEmSpM4smnQBWv8sWXb8pEuQJGlBcwRNkiSpMwY0SZKkzhjQJEmSOmNAkyRJ6owBTZIkqTMG\nNEmSpM4Y0CRJkjpjQJMkSeqMAU2SJKkzBjRJkqTOGNAkSZI6Y0CTJEnqjAFNkiSpMwY0SZKkzhjQ\nJEmSOmNAkyRJ6owBTZIkqTMGNEmSpM4sGteOk3wI2Be4pqr2bG1vB/YDbgG+Bzyvqq5vy14HvAC4\nDXhZVf3nuGqTNjRLlh0/L/1cctg+89KPJK3vxjmC9hHgydPaTgT2rKoHAP8LvA4gyR7AgcD92jbv\nS7LRGGuTJEnq1tgCWlV9Dbh2WtsJVXVre/tNYKf2en/g41V1c1VdDFwE7DWu2iRJkno2yWvQng/8\nR3u9I7BiZNllre3/SHJwkuVJlq9cuXLMJUqSJM2/iQS0JG8AbgWOWtNtq+rwqlpaVUsXL1687ouT\nJEmasLF9SWA2SZ7L8OWBvauqWvPlwM4jq+3U2iRJkjY48zqCluTJwGuA36uqn40sOg44MMkmSXYF\ndge+NZ+1SZIk9WKct9k4GngcsG2Sy4A3M3xrcxPgxCQA36yqF1fVuUmOAc5jmPp8SVXdNq7aJEmS\neja2gFZVz5yh+YOrWP9Q4NBx1SNJkrRQ+CQBSZKkzhjQJEmSOmNAkyRJ6owBTZIkqTMGNEmSpM4Y\n0CRJkjoz708S0OQsWXb8pEuQJElz4AiaJElSZwxokiRJnTGgSZIkdcaAJkmS1BkDmiRJUmcMaJIk\nSZ0xoEmSJHXGgCZJktQZA5okSVJnDGiSJEmdMaBJkiR1xoAmSZLUGQOaJElSZwxokiRJnTGgSZIk\ndcaAJkmS1BkDmiRJUmcMaJIkSZ1ZNOkCpJ4tWXb82Pu45LB9xt6HJGlhcQRNkiSpMwY0SZKkzhjQ\nJEmSOmNAkyRJ6owBTZIkqTMGNEmSpM54mw1pwubjVh6SpIXFETRJkqTOGNAkSZI6Y0CTJEnqjAFN\nkiSpMwY0SZKkzhjQJEmSOjO2gJbkQ0muSXLOSNs2SU5McmH7vfXIstcluSjJBUmeNK66JEmSejfO\nEbSPAE+e1rYMOKmqdgdOau9JsgdwIHC/ts37kmw0xtokSZK6NbaAVlVfA66d1rw/cGR7fSTwtJH2\nj1fVzVV1MXARsNe4apMkSerZfF+Dtl1VXdleXwVs117vCKwYWe+y1vZ/JDk4yfIky1euXDm+SiVJ\nkiZkYl8SqKoCai22O7yqllbV0sWLF4+hMkmSpMma72dxXp1k+6q6Msn2wDWt/XJg55H1dmptGwyf\nxyhJkqbM9wjaccBB7fVBwOdG2g9MskmSXYHdgW/Nc22SJEldGNsIWpKjgccB2ya5DHgzcBhwTJIX\nAJcCBwBU1blJjgHOA24FXlJVt42rNkmSpJ6NLaBV1TNnWbT3LOsfChw6rnokSZIWCp8kIEmS1BkD\nmiRJUmcMaJIkSZ0xoEmSJHXGgCZJktQZA5okSVJnDGiSJEmdMaBJkiR1xoAmSZLUGQOaJElSZwxo\nkiRJnTGgSZIkdcaAJkmS1BkDmiRJUmcMaJIkSZ0xoEmSJHXGgCZJktQZA5okSVJnDGiSJEmdMaBJ\nkiR1xoAmSZLUGQOaJElSZwxokiRJnTGgSZIkdcaAJkmS1BkDmiRJUmcMaJIkSZ0xoEmSJHXGgCZJ\nktQZA5okSVJnDGiSJEmdmVNAS/KoubRJkiTpjpvrCNo/z7FNkiRJd9CiVS1M8gjgkcDiJH81suiu\nwEbjLEySJGlDtcqABtwZ2KKtt+VI+0+Ap4+rKEmSpA3ZKgNaVX0V+GqSj1TVpfNUkyRJ0gZtdSNo\nUzZJcjiwZHSbqnr8OIqSJEnakM01oH0SeD9wBHDb+MqRJEnSXAParVX1L2OtRJIkScDcb7Px+SR/\nnmT7JNtM/axtp0n+Msm5Sc5JcnSSTds+T0xyYfu99druX5IkaSGba0A7CHg18HXgtPazfG06TLIj\n8DJgaVXtyXC7jgOBZcBJVbU7cFJ7L0mStMGZ0xRnVe06hn7vkuQXwGbAFcDrgMe15UcCpwCvXcf9\nSpIkdW9OAS3Jn8zUXlX/tqYdVtXlSd4B/AC4CTihqk5Isl1VXdlWuwrYbpZaDgYOBthll13WtHtJ\nkqTuzfVLAg8deb0psDdwOrDGAa1dW7Y/sCtwPfDJJM8eXaeqKknNtH1VHQ4cDrB06dIZ15EkSVrI\n5jrF+Rej75NsBXx8Lft8AnBxVa1s+/o0w+Okrk6yfVVdmWR74Jq13L8kSdKCNtcvCUx3I8MI2Nr4\nAfDwJJslCcNo3PnAcQxfRqD9/txa7l+SJGlBm+s1aJ8HpqYTNwLuCxyzNh1W1alJjmWYIr0V+A7D\nlOUWwDFJXgBcChywNvuXJEla6OZ6Ddo7Rl7fClxaVZetbadV9WbgzdOab2YYTZMkSdqgzWmKsz00\n/bvAlsDWwC3jLEqSJGlDNqeAluQA4FvAHzFMPZ6a5OnjLEySJGlDNdcpzjcAD62qawCSLAa+DBw7\nrsIkSZI2VHP9FuedpsJZ86M12FaSJElrYK4jaF9K8p/A0e39M4AvjqckSZKkDdsqA1qS3YDtqurV\nSf4AeHRb9A3gqHEXJ0mStCFa3QjaPzE8xJyq+jTwaYAk92/L9htrdZIkSRug1V1Htl1VnT29sbUt\nGUtFkiRJG7jVBbStVrHsLuuyEEmSJA1WF9CWJ3nR9MYkLwROG09JkiRJG7bVXYP2CuAzSZ7FrwPZ\nUuDOwO+PszBJkqQN1SoDWlVdDTwyye8Ae7bm46vqK2OvTJIkaQM1p/ugVdXJwMljrkWSJEn4NABJ\nkqTuGNAkSZI6Y0CTJEnqjAFNkiSpMwY0SZKkzhjQJEmSOmNAkyRJ6owBTZIkqTMGNEmSpM4Y0CRJ\nkjpjQJMkSeqMAU2SJKkzBjRJkqTOGNAkSZI6s2jSBUhafyxZdvzY+7jksH3G3ockTZojaJIkSZ0x\noEmSJHXGgCZJktQZA5okSVJnDGiSJEmdMaBJkiR1xoAmSZLUGQOaJElSZwxokiRJnTGgSZIkdcaA\nJkmS1JmJBLQkWyU5Nsl3k5yf5BFJtklyYpIL2++tJ1GbJEnSpE1qBO1dwJeq6j7AbwHnA8uAk6pq\nd+Ck9l6SJGmDM+8BLcndgMcAHwSoqluq6npgf+DIttqRwNPmuzZJkqQeTGIEbVdgJfDhJN9JckSS\nzYHtqurKts5VwHYzbZzk4CTLkyxfuXLlPJUsSZI0fyYR0BYBDwb+paoeBNzItOnMqiqgZtq4qg6v\nqqVVtXTx4sVjL1aSJGm+TSKgXQZcVlWntvfHMgS2q5NsD9B+XzOB2iRJkiZu3gNaVV0FrEhy79a0\nN3AecBxwUGs7CPjcfNcmSZLUg0UT6vcvgKOS3Bn4PvA8hrB4TJIXAJcCB0yoNkmSpImaSECrqjOA\npTMs2nu+a5EkSeqNTxKQJEnqjAFNkiSpMwY0SZKkzhjQJEmSOmNAkyRJ6owBTZIkqTMGNEmSpM4Y\n0CRJkjpjQJMkSeqMAU2SJKkzBjRJkqTOGNAkSZI6Y0CTJEnqjAFNkiSpMwY0SZKkzhjQJEmSOmNA\nkyRJ6owBTZIkqTMGNEmSpM4Y0CRJkjpjQJMkSeqMAU2SJKkzBjRJkqTOGNAkSZI6Y0CTJEnqjAFN\nkiSpMwY0SZKkzhjQJEmSOmNAkyRJ6owBTZIkqTMGNEmSpM4Y0CRJkjpjQJMkSeqMAU2SJKkzBjRJ\nkqTOGNAkSZI6Y0CTJEnqjAFNkiSpMwY0SZKkzkwsoCXZKMl3knyhvd8myYlJLmy/t55UbZIkSZM0\nyRG0lwPnj7xfBpxUVbsDJ7X3kiRJG5yJBLQkOwH7AEeMNO8PHNleHwk8bb7rkiRJ6sGkRtD+CXgN\n8MuRtu2q6sr2+ipgu5k2THJwkuVJlq9cuXLMZUqSJM2/eQ9oSfYFrqmq02Zbp6oKqFmWHV5VS6tq\n6eLFi8dVpiRJ0sQsmkCfjwJ+L8lTgU2Buyb5KHB1ku2r6sok2wPXTKA2SZKkiZv3EbSqel1V7VRV\nS4ADga9U1bOB44CD2moHAZ+b79okSZJ60NN90A4DfjfJhcAT2ntJkqQNziSmOH+lqk4BTmmvfwTs\nPcl6JEmSetDTCJokSZIwoEmSJHXHgCZJktQZA5okSVJnDGiSJEmdMaBJkiR1xoAmSZLUGQOaJElS\nZwxokiRJnTGgSZIkdcaAJkmS1JmJPotTktbUkmXHj72PSw7bZ+x9SNKqOIImSZLUGQOaJElSZwxo\nkiRJnTGgSZIkdcaAJkmS1BkDmiRJUmcMaJIkSZ0xoEmSJHXGgCZJktQZA5okSVJnDGiSJEmdMaBJ\nkiR1xoAmSZLUGQOaJElSZwxokiRJnTGgSZIkdcaAJkmS1BkDmiRJUmcMaJIkSZ0xoEmSJHXGgCZJ\nktQZA5okSVJnDGiSJEmdMaBJkiR1xoAmSZLUGQOaJElSZwxokiRJnZn3gJZk5yQnJzkvyblJXt7a\nt0lyYpIL2++t57s2SZKkHkxiBO1W4JVVtQfwcOAlSfYAlgEnVdXuwEntvSRJ0gZn3gNaVV1ZVae3\n1z8Fzgd2BPYHjmyrHQk8bb5rkyRJ6sFEr0FLsgR4EHAqsF1VXdkWXQVsN8s2BydZnmT5ypUr56VO\nSZKk+TSxgJZkC+BTwCuq6iejy6qqgJppu6o6vKqWVtXSxYsXz0OlkiRJ82siAS3Jxgzh7Kiq+nRr\nvjrJ9m359sA1k6hNkiRp0ibxLc4AHwTOr6p3jiw6DjiovT4I+Nx81yZJktSDRRPo81HAc4Czk5zR\n2l4PHAYck+QFwKXAAROoTZIkaeLmPaBV1X8DmWXx3vNZiyRJUo98koAkSVJnDGiSJEmdMaBJkiR1\nxoAmSZLUGQOaJElSZwxokiRJnZnEfdAWnCXLjp90CZIkaQPiCJokSVJnHEGTpPXYfMwAXHLYPmPv\nQ9rQOIL9+fhaAAAIyElEQVQmSZLUGQOaJElSZwxokiRJnTGgSZIkdcaAJkmS1BkDmiRJUmcMaJIk\nSZ0xoEmSJHXGgCZJktQZA5okSVJnDGiSJEmd8VmckjTNfDy/UpJWxRE0SZKkzhjQJEmSOmNAkyRJ\n6owBTZIkqTMGNEmSpM4Y0CRJkjpjQJMkSeqMAU2SJKkzBjRJkqTO+CQBSdKCMB9PeLjksH3G3ges\nX8cyHzbEz8sRNEmSpM44giZJukPWp2eXrk/HooXNETRJkqTOGNAkSZI6Y0CTJEnqjAFNkiSpM35J\nQJKk9ZBfeFjYHEGTJEnqTHcBLcmTk1yQ5KIkyyZdjyRJ0nzrKqAl2Qh4L/AUYA/gmUn2mGxVkiRJ\n86urgAbsBVxUVd+vqluAjwP7T7gmSZKkedXblwR2BFaMvL8MeNjoCkkOBg5ub29IcsE81aa1ty3w\nw0kXoXXG87n+8FyuPzyXd1DeNi/d3GOuK/YW0Farqg4HDp90HZq7JMuraumk69C64flcf3gu1x+e\ny/VPb1OclwM7j7zfqbVJkiRtMHoLaN8Gdk+ya5I7AwcCx024JkmSpHnV1RRnVd2a5KXAfwIbAR+q\nqnMnXJbuOKek1y+ez/WH53L94blcz6SqJl2DJEmSRvQ2xSlJkrTBM6BJkiR1xoCmdSrJzklOTnJe\nknOTvLy1b5PkxCQXtt9bT7pWzU2SjZJ8J8kX2nvP5QKUZKskxyb5bpLzkzzCc7lwJfnL9t/Yc5Ic\nnWRTz+f6xYCmde1W4JVVtQfwcOAl7XFdy4CTqmp34KT2XgvDy4HzR957LhemdwFfqqr7AL/FcE49\nlwtQkh2BlwFLq2pPhi/VHYjnc71iQNM6VVVXVtXp7fVPGf4I7MjwyK4j22pHAk+bTIVaE0l2AvYB\njhhp9lwuMEnuBjwG+CBAVd1SVdfjuVzIFgF3SbII2Ay4As/nesWAprFJsgR4EHAqsF1VXdkWXQVs\nN6GytGb+CXgN8MuRNs/lwrMrsBL4cJuuPiLJ5nguF6Squhx4B/AD4Ergx1V1Ap7P9YoBTWORZAvg\nU8Arquono8tquLeL93fpXJJ9gWuq6rTZ1vFcLhiLgAcD/1JVDwJuZNr0l+dy4WjXlu3PELx3ADZP\n8uzRdTyfC58BTetcko0ZwtlRVfXp1nx1ku3b8u2BayZVn+bsUcDvJbkE+Djw+CQfxXO5EF0GXFZV\np7b3xzIENs/lwvQE4OKqWllVvwA+DTwSz+d6xYCmdSpJGK5zOb+q3jmy6DjgoPb6IOBz812b1kxV\nva6qdqqqJQwXIH+lqp6N53LBqaqrgBVJ7t2a9gbOw3O5UP0AeHiSzdp/c/dmuN7X87ke8UkCWqeS\nPBr4L+Bsfn3d0usZrkM7BtgFuBQ4oKqunUiRWmNJHge8qqr2TXJ3PJcLTpIHMnzZ487A94HnMfxP\nuudyAUryFuAZDN+c/w7wQmALPJ/rDQOaJElSZ5zilCRJ6owBTZIkqTMGNEmSpM4Y0CRJkjpjQJMk\nSeqMAU3SHZLkDUnOTXJWkjOSPGzSNd0RST6S5Olj2O/rR14vSXLOHLd7RZI/aa9PSbJ0DfpcnORL\na16tpEkzoElaa0keAewLPLiqHsBwh/MVk62qW69f/Sq31x6E/XzgY2vTYVWtBK5M8qi12V7S5BjQ\nJN0R2wM/rKqbAarqh1V1BUCShyT5apLTkvznyCNoHpLkzPbz9qmRpCTPTfKeqR0n+UK7QS5Jnpjk\nG0lOT/LJ9qxXklyS5C2t/ewk92ntWyT5cGs7K8kfrmo/s1nFMZyS5G1JvpXkf5P8dmvfLMkxSc5L\n8pkkpyZZmuQw4C5thPGotvuNkvxrG308IcldZijh8cDpVXXrSNtz2n7OSbJX6/eQJP/eju3CJC8a\nWf+zwLNWcx4ldcaAJumOOAHYuYWU9yV5LPzqeaz/DDy9qh4CfAg4tG3zYeAvquq35tJBkm2Bvwae\nUFUPBpYDfzWyyg9b+78Ar2ptbwR+XFX3byN7X5nDfqb3u6pjAFhUVXsBrwDe3Nr+HLiuqvZoNTwE\noKqWATdV1QOraios7Q68t6ruB1wP/OEMZTwKmP6w+s2q6oGtrw+NtD+AIdA9AnhTkh1a+3Lgt2c7\nTkl9WjTpAiQtXFV1Q5KHMASA3wE+kWQZQyjYEzhxeFQgGzFMtW0FbFVVX2u7+HfgKavp5uHAHsD/\ntH3dGfjGyPJPt9+nAX/QXj+B4fmhU3Vel2Tf1exnunvPdAyz9LukvX408K7W5zlJzlrF/i+uqjNm\n2Meo7RmesTjq6Lb/ryW5a/tMAT5XVTcBNyU5GdiLYfTsGmAHJC0oBjRJd0hV3QacApyS5GyGhzSf\nBpxbVY8YXXckTMzkVm4/qr/p1GbAiVX1zFm2u7n9vo1V/zdtdfuZaf3/cwxr0e9sbh55fRsw0xTn\nTfz6c5gy/fl8tZr2Tdt+JC0gTnFKWmtJ7p1k95GmBzI8pPkCYHH7EgFJNk5yv6q6Hrg+yaPb+qPX\nRl0CPDDJnZLszDACBPBN4FFJdmv72jzJvVZT2onAS0bq3Hot9jPjMaym3/8BDmjr7wHcf2TZL9q0\n6Zo4H9htWtsz2v4fzTCN++PWvn+STTM8zP5xwLdb+72AOX1jVFI/DGiS7ogtgCPbRfFnMUwhHlJV\ntwBPB96W5EzgDOCRbZvnAe9NcgbDKNWU/wEuBs4D3g2cDr/6JuJzgaNbH98A7rOauv4O2LpdSH8m\n8Dtrup/VHMNs3scQ6s5rNZwLTAWow4GzRr4kMBf/ATxmWtvPk3wHeD/wgpH2s4CTGYLo3059WYNh\n6vn4NehTUgdSNX1UXJLmR5IlwBeqas8Jl7JOJNkI2Liqfp7knsCXgXu3sLe2+/wM8JqqunAV6xwC\n3FBV75hh2deA/avqurWtQdL88xo0SVp3NgNOblOZAf78joSzZhnDlwVmDWizSbIYeKfhTFp4HEGT\nJEnqjNegSZIkdcaAJkmS1BkDmiRJUmcMaJIkSZ0xoEmSJHXm/wP2gU+2gQ4IfQAAAABJRU5ErkJg\ngg==\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x7f3cb8420190>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sizes = [len(rec.seq) for rec in SeqIO.parse(\"neg_antibp2.fa\", \"fasta\")]\n",
    "print len(sizes), min(sizes), max(sizes)\n",
    "\n",
    "plt.hist(sizes, bins=20)\n",
    "plt.title(\"%i Negative bacteriocin sequences\\nLengths %i to %i\" \\\n",
    "            % (len(sizes),min(sizes),max(sizes)))\n",
    "plt.xlabel(\"Sequence length (bp)\")\n",
    "plt.ylabel(\"Count\")\n",
    "\n",
    "plt.show()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
